



<span id="top"></span>




# <span dir="auto">Search results</span>








- [Content
  pages](/mediawiki/index.php?title=Special:Search&search=SAMtools&fulltext=Search&profile=default "Search in (Main)")
- [Multimedia](/mediawiki/index.php?title=Special:Search&search=SAMtools&fulltext=Search&profile=images "Search for files")
- [Help and Project
  pages](/mediawiki/index.php?title=Special:Search&search=SAMtools&fulltext=Search&profile=help "Search in GMOD, Help")
- [Everything](/mediawiki/index.php?title=Special:Search&search=SAMtools&fulltext=Search&profile=all "Search all of content (including talk pages)")
- [Advanced](/mediawiki/index.php?title=Special:Search&search=SAMtools&fulltext=Search&profile=advanced "Search in custom namespaces")


- Showing below **31** results starting with \#**1**.



**Create the page "<a
href="/mediawiki/index.php?title=SAMtools&amp;action=edit&amp;redlink=1"
class="new" title="SAMtools (page does not exist)">SAMtools</a>" on this
wiki!** See also the search results found.

- 

  [GBrowse](/wiki/GBrowse "GBrowse")

  

  

  ...ncludes a starting VMware image, and uses the example data that
  comes with <span class="searchmatch">SAMtools</span>.

  

  

  11 KB (1,422 words) - 16:47, 10 April 2023

  

- 

  [GBrowse Ubuntu
  HOWTO](/wiki/GBrowse_Ubuntu_HOWTO "GBrowse Ubuntu HOWTO")

  

  

  sudo apt-get install gbrowse gbrowse-calign gbrowse-data
  libbio-<span class="searchmatch">samtools</span>-perl apache2
  libapache2-mod-fcgid libxml-sax-writer-perl libxml-xpathengine-perl
  libyaml-syck-perl <span class="searchmatch">samtools</span>

  

  

  8 KB (799 words) - 20:37, 8 August 2018

  

- 

  [Glossary](/wiki/Glossary "Glossary")

  

  

  ...e Alignment/Map (SAM)\]\]'' format. BAM and SAM are both part of
  \[\[#<span class="searchmatch">SAMtools</span>\|<span class="searchmatch">SAMtools</span>\]\].
  BAM is compressed, binary, indexed format for \[\[Next Generation
  Sequenc ...rmat for \[\[Next Generation Sequencing\]\] data. It is a
  part of
  \[\[#<span class="searchmatch">SAMtools</span>\|<span class="searchmatch">SAMtools</span>\]\].
  \[\[GBrowse\]\] 2 has an adaptor that can read SAM data.

  

  

  14 KB (2,081 words) - 02:50, 14 March 2014

  

- 

  [GBrowse Adaptors](/wiki/GBrowse_Adaptors "GBrowse Adaptors")

  

  

  \| \[http://<span class="searchmatch">samtools</span>.sourceforge.net/
  <span class="searchmatch">SAMtools</span>\]

  

  

  3 KB (443 words) - 16:22, 7 August 2012

  

- 

  [August 2009 GMOD
  Meeting](/wiki/August_2009_GMOD_Meeting "August 2009 GMOD Meeting")

  

  

  ...on next generation sequencing, including
  \[http://<span class="searchmatch">samtools</span>.sourceforge.net
  <span class="searchmatch">SAMtools</span>\], BWA, and
  \[http://maq.sourceforge.net MAQ\].\</div\> ...: Sequence
  alignment/map format (SAM) and
  \[http://<span class="searchmatch">samtools</span>.sourceforge.net
  <span class="searchmatch">SAMtools</span>\]

  

  

  59 KB (8,741 words) - 19:37, 4 September 2013

  

- 

  [GBrowse Tutorial](/wiki/GBrowse_Tutorial "GBrowse Tutorial")

  

  

  ...des a starting \[\[VMware\]\] image, and uses the example data that
  comes with <span class="searchmatch">SAMtools</span>.

  

  

  2 KB (236 words) - 17:10, 18 December 2013

  

- 

  [News/Aug 2009 GMOD Meeting
  Report](/wiki/News/Aug_2009_GMOD_Meeting_Report "News/Aug 2009 GMOD Meeting Report")

  

  

  ...on \[\[August 2009 GMOD Meeting#Visualising NGS Data in GBrowse
  2\|using SAM/<span class="searchmatch">SAMtools</span> in GBrowse\]\].

  

  

  2 KB (301 words) - 19:37, 4 September 2013

  

- 

  [News/GMOD Tutorials! GMOD
  Training!](/wiki/News/GMOD_Tutorials!_GMOD_Training! "News/GMOD Tutorials! GMOD Training!")

  

  

  ...tion Sequence\]\] Data with \[\[GBrowse\]\] and
  \[http://<span class="searchmatch">samtools</span>.sourceforge.net
  <span class="searchmatch">SAMtools</span>\]''' (poster)

  

  

  4 KB (542 words) - 23:11, 9 October 2012

  

- 

  [News/GBrowse NGS Tutorial &
  Poster](/wiki/News/GBrowse_NGS_Tutorial_%26_Poster "News/GBrowse NGS Tutorial & Poster")

  

  

  ...ep instructions for visualizing the example human data that comes
  with the <span class="searchmatch">SAMtools</span> package.

  

  

  1 KB (156 words) - 23:10, 9 October 2012

  

- 

  [News/NGS Analyses in
  Galaxy](/wiki/News/NGS_Analyses_in_Galaxy "News/NGS Analyses in Galaxy")

  

  

  ...ncludes a variety of utilities for SAM/BAM manipulation. Some are
  based on <span class="searchmatch">SAMtools</span> library, some are
  written by \[\[Galaxy\]\] team.

  

  

  2 KB (239 words) - 23:09, 9 October 2012

  

- 

  [Next Generation
  Sequencing](/wiki/Next_Generation_Sequencing "Next Generation Sequencing")

  

  

  ...rowse\]\] 2 to visualize NGS data using the
  \[http://<span class="searchmatch">samtools</span>.sourceforge.net
  <span class="searchmatch">SAMtools</span>\] GBrowse adaptor,
  Bio::DB::Sam. This includes a \[\[VMware\]\] image to work w ...) and
  <span class="searchmatch">SAMtools</span>\|Quest for Standard:
  Sequence alignment/map format (SAM) and
  <span class="searchmatch">SAMtools</span>\]\]

  

  

  3 KB (450 words) - 00:52, 18 November 2010

  

- 

  [JBrowse Tutorial
  2010](/wiki/JBrowse_Tutorial_2010 "JBrowse Tutorial 2010")

  

  

  \* <span class="searchmatch">samtools</span>, and its dependency
  libncurses5-dev \$ wget
  http://sourceforge.net/projects/<span class="searchmatch">samtools</span>/files/<span class="searchmatch">samtools</span>/0.1.7/<span class="searchmatch">samtools</span>-0.1.7a.tar.bz2/download

  

  

  9 KB (1,327 words) - 22:46, 3 October 2012

  

- 

  [GBrowse NGS
  Tutorial](/wiki/GBrowse_NGS_Tutorial "GBrowse NGS Tutorial")

  

  

  ...eration Sequencing\]\] (NGS) data using the
  \[http://<span class="searchmatch">samtools</span>.sourceforge.net
  <span class="searchmatch">SAMtools</span>\] GBrowse adaptor,
  Bio::DB::Sam. ...urceforge.net
  <span class="searchmatch">SAMtools</span>\]. We'll use the example
  human data that comes with <span class="searchmatch">SAMtools</span>
  to do that. First we'll load the reference sequence and some gene
  models.

  

  

  38 KB (5,742 words) - 18:11, 9 October 2012

  

- 

  [ABRF2010 Workshop](/wiki/ABRF2010_Workshop "ABRF2010 Workshop")

  

  

  \*\[http://<span class="searchmatch">samtools</span>.sourceforge.net/
  BAM or SAM\] (for sequence alignment data) ...ompact, fast to read
  way, as part of the
  \[http://<span class="searchmatch">samtools</span>.sourceforge.net/
  <span class="searchmatch">SAMtools</span>\] project.

  

  

  17 KB (2,763 words) - 18:52, 8 October 2012

  

- 

  [September 2010 GMOD
  Meeting](/wiki/September_2010_GMOD_Meeting "September 2010 GMOD Meeting")

  

  

  \*\* SQLite, <span class="searchmatch">SAMtools</span> (NGS) adaptors

  

  

  100 KB (14,725 words) - 17:31, 30 November 2012

  

- 

  [GBrowse 2.0 Install
  HOWTO/Advanced](/wiki/GBrowse_2.0_Install_HOWTO/Advanced "GBrowse 2.0 Install HOWTO/Advanced")

  

  

  ...equires the C <span class="searchmatch">Samtools</span> library.
  See
  http://cpansearch.perl.org/src/LDS/Bio-<span class="searchmatch">SamTools</span>-1.35/README
  for details.

  

  

  23 KB (3,672 words) - 14:44, 10 July 2013

  

- 

  [Arthropod Genomics 2011/Genome Project 101
  Workshop](/wiki/Arthropod_Genomics_2011/Genome_Project_101_Workshop "Arthropod Genomics 2011/Genome Project 101 Workshop")

  

  

  Install <span class="searchmatch">Samtools</span> ....16.tar.bz2
  http://sourceforge.net/projects/<span class="searchmatch">samtools</span>/files/<span class="searchmatch">samtools</span>/0.1.16/<span class="searchmatch">samtools</span>-0.1.16.tar.bz2/download

  

  

  15 KB (2,338 words) - 18:52, 8 October 2012

  

- 

  [News/GMOD in the Cloud 2.0
  Released](/wiki/News/GMOD_in_the_Cloud_2.0_Released "News/GMOD in the Cloud 2.0 Released")

  

  

  \* \[\[GBrowse\]\] 2.54 (with
  <span class="searchmatch">samtools</span>, BigWig and fastCGI support)

  

  

  982 B (152 words) - 19:14, 14 June 2013

  

- 

  [JBrowse Configuration
  Guide](/wiki/JBrowse_Configuration_Guide "JBrowse Configuration Guide")

  

  

  Syntax used to import a indexed fasta(i.e. a fasta file where you run
  \`<span class="searchmatch">samtools</span> faidx yourfile.fa\` which
  outputs yourfile.fa.fai) .... Exactly one of these three options must
  be used. With indexed_fasta, the
  <span class="searchmatch">samtools</span> faidx yourfile.fa must be
  run before hand. With twobit, the twobit file wil

  

  

  185 KB (26,536 words) - 02:48, 30 September 2020

  

- 

  [GBrowse Volvox SAM
  Tutorial](/wiki/GBrowse_Volvox_SAM_Tutorial "GBrowse Volvox SAM Tutorial")

  

  

  ...xy\]\] to run Bowtie and produce a SAM file. In order to use this
  tutorial, <span class="searchmatch">SAMtools</span> and Bio::DB::Sam
  must be installed. This is based on the \[\[GBrowse_NGS_Tut
  <span class="searchmatch">samtools</span> faidx volvox.fa

  

  

  5 KB (853 words) - 00:19, 23 October 2012

  

- 

  [JBrowse Tutorial
  2013](/wiki/JBrowse_Tutorial_2013 "JBrowse Tutorial 2013")

  

  

  \* <span class="searchmatch">samtools</span>, and its dependency
  libncurses5-dev

  

  

  14 KB (1,948 words) - 20:48, 27 August 2013

  

- 

  [GBrowse/tool data](/wiki/GBrowse/tool_data "GBrowse/tool data")

  

  

  ...ncludes a starting VMware image, and uses the example data that
  comes with <span class="searchmatch">SAMtools</span>.

  

  

  8 KB (1,065 words) - 18:35, 10 September 2013

  

- 

  [GBrowse Tutorial
  2011](/wiki/GBrowse_Tutorial_2011 "GBrowse Tutorial 2011")

  

  

  ===Updating <span class="searchmatch">SAMtools</span>=== The version
  of <span class="searchmatch">SAMtools</span> may need to be updated.
  Get the <span class="searchmatch">samtools</span> release:

  

  

  7 KB (932 words) - 22:50, 13 February 2013

  

- 

  [JBrowse Tutorial
  2011](/wiki/JBrowse_Tutorial_2011 "JBrowse Tutorial 2011")

  

  

  \* <span class="searchmatch">samtools</span>, and its dependency
  libncurses5-dev \$ wget
  http://sourceforge.net/projects/<span class="searchmatch">samtools</span>/files/<span class="searchmatch">samtools</span>/0.1.7/<span class="searchmatch">samtools</span>-0.1.7a.tar.bz2

  

  

  9 KB (1,328 words) - 17:58, 14 February 2013

  

- 

  [JBrowse FAQ](/wiki/JBrowse_FAQ "JBrowse FAQ")

  

  

  ...ve run prepare-refseqs.pl on a FASTA file. Instead you can simply
  use the "<span class="searchmatch">samtools</span> faidx" program to
  index your fasta file in a data directory, and set trackL To create an
  Indexed FASTA, install <span class="searchmatch">samtools</span> and
  run

  

  

  65 KB (10,432 words) - 15:23, 16 September 2020

  

- 

  [GMOD Malaysia 2014/JBrowse
  Tutorial](/wiki/GMOD_Malaysia_2014/JBrowse_Tutorial "GMOD Malaysia 2014/JBrowse Tutorial")

  

  

  \* <span class="searchmatch">samtools</span>, and its dependency
  libncurses5-dev

  

  

  14 KB (1,931 words) - 00:01, 20 February 2014

  

- 

  [GMOD Online Training 2014/JBrowse
  Tutorial](/wiki/GMOD_Online_Training_2014/JBrowse_Tutorial "GMOD Online Training 2014/JBrowse Tutorial")

  

  

  \* <span class="searchmatch">samtools</span>, and its dependency
  libncurses5-dev

  

  

  13 KB (1,905 words) - 14:21, 21 May 2014

  

- 

  [JBrowse Desktop](/wiki/JBrowse_Desktop "JBrowse Desktop")

  

  

  ...to use Indexed FASTA (selecting both a fasta and a fai file
  generated by \`<span class="searchmatch">samtools</span> faidx
  file.fasta\`) or 2bit. Even small genomes with normal FASTA will be mu

  

  

  3 KB (528 words) - 22:30, 19 July 2018

  

- 

  [JBrowse 2 Tutorial PAG
  2022](/wiki/JBrowse_2_Tutorial_PAG_2022 "JBrowse 2 Tutorial PAG 2022")

  

  

  .../doc/bgzip.html \<code\>bgzip\</code\>\] from the
  \<code\>\[http://www.htslib.org/
  <span class="searchmatch">Samtools</span>\]\</code\> package. The
  commands to create these files are:
  <span class="searchmatch">samtools</span> faidx yourfile.fa.gz

  

  

  24 KB (4,099 words) - 19:23, 29 January 2022

  

- 

  [JBrowse2 Tutorial PAG
  2023](/wiki/JBrowse2_Tutorial_PAG_2023 "JBrowse2 Tutorial PAG 2023")

  

  

  ...ed bgzip, tabix, <span class="searchmatch">samtools</span> and
  minimap2 via apt: \<code\>sudo apt-get install
  <span class="searchmatch">samtools</span> tabix minimap2\</code\>. \*
  Created a bgzipped and <span class="searchmatch">samtools</span>
  faidx'ed FASTAs file for ''C. elegans'' and ''C. brenneri''.

  

  

  34 KB (5,728 words) - 18:21, 23 January 2023

  

- 

  [JBrowse2 Tutorial PAG
  2024](/wiki/JBrowse2_Tutorial_PAG_2024 "JBrowse2 Tutorial PAG 2024")

  

  

  ...ed bgzip, tabix, <span class="searchmatch">samtools</span> and
  minimap2 via apt: \<code\>sudo apt-get install
  <span class="searchmatch">samtools</span> tabix minimap2\</code\>. \*
  Created a bgzipped and <span class="searchmatch">samtools</span>
  faidx'ed FASTAs file for ''C. elegans'' and ''C. brenneri''.

  

  

  39 KB (6,446 words) - 03:36, 18 January 2024

  



View (previous 50 \| next 50) (<a
href="/mediawiki/index.php?title=Special:Search&amp;limit=20&amp;offset=0&amp;profile=default&amp;search=SAMtools"
class="mw-numlink" title="Show 20 results per page">20</a> \| <a
href="/mediawiki/index.php?title=Special:Search&amp;limit=50&amp;offset=0&amp;profile=default&amp;search=SAMtools"
class="mw-numlink" title="Show 50 results per page">50</a> \| <a
href="/mediawiki/index.php?title=Special:Search&amp;limit=100&amp;offset=0&amp;profile=default&amp;search=SAMtools"
class="mw-numlink" title="Show 100 results per page">100</a> \| <a
href="/mediawiki/index.php?title=Special:Search&amp;limit=250&amp;offset=0&amp;profile=default&amp;search=SAMtools"
class="mw-numlink" title="Show 250 results per page">250</a> \| <a
href="/mediawiki/index.php?title=Special:Search&amp;limit=500&amp;offset=0&amp;profile=default&amp;search=SAMtools"
class="mw-numlink" title="Show 500 results per page">500</a>)








## Navigation menu



### Namespaces

- <span id="ca-nstab-special">[Special
  page](/wiki/Special%3ASearch/SAMtools "This is a special page, you cannot edit the page itself")</span>






### Navigation



- <span id="n-GMOD-Home">[GMOD Home](/wiki/Main_Page)</span>
- <span id="n-Software">[Software](/wiki/GMOD_Components)</span>
- <span id="n-Categories-.2F-Tags">[Categories /
  Tags](/wiki/Categories)</span>
- <span id="n-View-all-pages">[View all
  pages](/wiki/Special:AllPages)</span>




### Documentation



- <span id="n-Overview">[Overview](/wiki/Overview)</span>
- <span id="n-FAQs">[FAQs](/wiki/Category%3AFAQ)</span>
- <span id="n-HOWTOs">[HOWTOs](/wiki/Category%3AHOWTO)</span>
- <span id="n-Glossary">[Glossary](/wiki/Glossary)</span>




### Community



- <span id="n-GMOD-News">[GMOD News](/wiki/GMOD_News)</span>
- <span id="n-Training-.2F-Outreach">[Training /
  Outreach](/wiki/Training_and_Outreach)</span>
- <span id="n-Support">[Support](/wiki/Support)</span>
- <span id="n-GMOD-Promotion">[GMOD
  Promotion](/wiki/GMOD_Promotion)</span>
- <span id="n-Meetings">[Meetings](/wiki/Meetings)</span>
- <span id="n-Calendar">[Calendar](/wiki/Calendar)</span>




### Tools



- <span id="t-specialpages"><a href="/wiki/Special%3ASpecialPages" accesskey="q"
  title="A list of all special pages [q]">Special pages</a></span>








<!-- -->




