<div id="mw-page-base" class="noprint">

</div>

<div id="mw-head-base" class="noprint">

</div>

<div id="content" class="mw-body" role="main">

<span id="top"></span>

<div id="mw-js-message" style="display:none;">

</div>



# <span dir="auto">GBrowse FAQ</span>

<div id="bodyContent">

<div id="siteSub">

From GMOD

</div>

<div id="contentSub">

</div>

<div id="jump-to-nav" class="mw-jump">

Jump to: [navigation](#mw-navigation), [search](#p-search)

</div>

<div id="mw-content-text" class="mw-content-ltr" lang="en" dir="ltr">

<div id="toc" class="toc">

<div id="toctitle">

## Contents

</div>

- [<span class="tocnumber">1</span> <span class="toctext">About this
  FAQ</span>](#About_this_FAQ)
  - [<span class="tocnumber">1.1</span> <span class="toctext">What is
    this FAQ?</span>](#What_is_this_FAQ.3F)
  - [<span class="tocnumber">1.2</span> <span class="toctext">How is it
    maintained?</span>](#How_is_it_maintained.3F)
- [<span class="tocnumber">2</span> <span class="toctext">General
  Questions</span>](#General_Questions)
  - [<span class="tocnumber">2.1</span> <span class="toctext">What is
    GBrowse good for?</span>](#What_is_GBrowse_good_for.3F)
  - [<span class="tocnumber">2.2</span> <span class="toctext">What
    platforms does GBrowse run
    on?</span>](#What_platforms_does_GBrowse_run_on.3F)
  - [<span class="tocnumber">2.3</span> <span class="toctext">How is
    GBrowse distributed?</span>](#How_is_GBrowse_distributed.3F)
  - [<span class="tocnumber">2.4</span> <span class="toctext">What are
    the terms of use for
    GBrowse?</span>](#What_are_the_terms_of_use_for_GBrowse.3F)
  - [<span class="tocnumber">2.5</span> <span class="toctext">I have a
    problem. What do I do?</span>](#I_have_a_problem._What_do_I_do.3F)
- [<span class="tocnumber">3</span>
  <span class="toctext">Problem-solving</span>](#Problem-solving)
  - [<span class="tocnumber">3.1</span> <span class="toctext">Where do I
    download GBrowse?</span>](#Where_do_I_download_GBrowse.3F)
  - [<span class="tocnumber">3.2</span> <span class="toctext">How do I
    install GBrowse?</span>](#How_do_I_install_GBrowse.3F)
  - [<span class="tocnumber">3.3</span> <span class="toctext">Where do I
    find a list of all available
    glyphs?</span>](#Where_do_I_find_a_list_of_all_available_glyphs.3F)
  - [<span class="tocnumber">3.4</span> <span class="toctext">When I
    search, why doesn't GBrowse find my 3-letter gene
    name?</span>](#When_I_search.2C_why_doesn.27t_GBrowse_find_my_3-letter_gene_name.3F)
  - [<span class="tocnumber">3.5</span> <span class="toctext">How do I
    use semantic zooming to hide a track
    completely?</span>](#How_do_I_use_semantic_zooming_to_hide_a_track_completely.3F)
  - [<span class="tocnumber">3.6</span> <span class="toctext">How to add
    an outgoing link to a text on the feature detail page in
    Gbrowse?</span>](#How_to_add_an_outgoing_link_to_a_text_on_the_feature_detail_page_in_Gbrowse.3F)
  - [<span class="tocnumber">3.7</span> <span class="toctext">I have a
    multi-segmented feature (such as a multi-exon transcript). It looks
    fine at low power, but when I zoom in the connecting lines between
    segments disappear.
    Help!</span>](#I_have_a_multi-segmented_feature_.28such_as_a_multi-exon_transcript.29._It_looks_fine_at_low_power.2C_but_when_I_zoom_in_the_connecting_lines_between_segments_disappear._Help.21)
  - [<span class="tocnumber">3.8</span> <span class="toctext">I'm using
    the GFF database adaptor. Is it better to load it using GFF2 or
    GFF3?</span>](#I.27m_using_the_GFF_database_adaptor._Is_it_better_to_load_it_using_GFF2_or_GFF3.3F)
  - [<span class="tocnumber">3.9</span> <span class="toctext">How do I
    pass parameters into functions of
    init_code?</span>](#How_do_I_pass_parameters_into_functions_of_init_code.3F)
  - [<span class="tocnumber">3.10</span> <span class="toctext">How do I
    show circular genomes?</span>](#How_do_I_show_circular_genomes.3F)
  - [<span class="tocnumber">3.11</span> <span class="toctext">There's a
    problem with my overview and detail images - they're the same
    scale</span>](#There.27s_a_problem_with_my_overview_and_detail_images_-_they.27re_the_same_scale)
  - [<span class="tocnumber">3.12</span> <span class="toctext">Can I add
    Popup Balloon Tips to
    GBrowse?</span>](#Can_I_add_Popup_Balloon_Tips_to_GBrowse.3F)
  - [<span class="tocnumber">3.13</span> <span class="toctext">Why is
    GBrowse on Chado so
    slow?</span>](#Why_is_GBrowse_on_Chado_so_slow.3F)
  - [<span class="tocnumber">3.14</span> <span class="toctext">The
    "Bio::Graphics::BrowserConfig=HASH ... Bio/Graphics/Browser.pm line
    587"
    Error</span>](#The_.22Bio::Graphics::BrowserConfig.3DHASH_..._Bio.2FGraphics.2FBrowser.pm_line_587.22_Error)
  - [<span class="tocnumber">3.15</span> <span class="toctext">Can I
    show more than one glyph in the same
    track?</span>](#Can_I_show_more_than_one_glyph_in_the_same_track.3F)
  - [<span class="tocnumber">3.16</span> <span class="toctext">How many
    tracks can be displayed in
    GBrowse?</span>](#How_many_tracks_can_be_displayed_in_GBrowse.3F)

</div>

## <span id="About_this_FAQ" class="mw-headline">About this FAQ</span>

### <span id="What_is_this_FAQ.3F" class="mw-headline">What is this FAQ?</span>

It is the list of Frequently Asked Questions about
[GBrowse](GBrowse.1 "GBrowse").

### <span id="How_is_it_maintained.3F" class="mw-headline">How is it maintained?</span>

It is now maintained as a
<a href="http://en.wikipedia.org/wiki/Wiki" class="extiw"
title="wp:Wiki">Wiki</a> on this site. You can help maintain it by
adding questions and answers.

  

## <span id="General_Questions" class="mw-headline">General Questions</span>

### <span id="What_is_GBrowse_good_for.3F" class="mw-headline">What is GBrowse good for?</span>

GBrowse was designed to view genomes. It displays a graphical
representation of a section of a genome, and shows the positions of
genes and other functional elements. It can be configured to show both
qualitative data such as the splicing structure of a gene, and
quantitative data such as microarray expression levels.

Another good way to get an overview of the features GBrowse offers is to
read the documentation at the [GBrowse](GBrowse.1 "GBrowse") Wiki page.

  

### <span id="What_platforms_does_GBrowse_run_on.3F" class="mw-headline">What platforms does GBrowse run on?</span>

GBrowse is a web-server application that is implemented in the Perl
programming language. It will run on any machine that runs Perl,
including Windows, Macintosh OS X, and most versions of Linux and UNIX.

  

### <span id="How_is_GBrowse_distributed.3F" class="mw-headline">How is GBrowse distributed?</span>

GBrowse is distributed as source code for Macintosh OS X, UNIX and Linux
platforms, and as pre-packaged binaries for Windows machines.

  

### <span id="What_are_the_terms_of_use_for_GBrowse.3F" class="mw-headline">What are the terms of use for GBrowse?</span>

GBrowse is distributed under the Perl Artistic License, which allows for
unrestricted use and distribution, including commercial use and resale.
You may modify and distribute modified versions of GBrowse provided that
you credit the original authors for their contribution.

  

### <span id="I_have_a_problem._What_do_I_do.3F" class="mw-headline">I have a problem. What do I do?</span>

First consult the
<a href="http://old.nabble.com/gmod-gbrowse-f3500.html"
class="external text" rel="nofollow">GBrowse mailing list archive at
Nabble</a>. Your problem may already have been reported and discussed.
If you find no help there, then send email to
gmod-gbrowse@lists.sourceforge.net. If you are pretty certain you have
found a bug, please report it to the <a
href="http://sourceforge.net/tracker/?func=add&amp;group_id=27707&amp;atid=391291"
class="external text" rel="nofollow">bug report tracking system</a> (set
the category to "GBrowse").

## <span id="Problem-solving" class="mw-headline">Problem-solving</span>

### <span id="Where_do_I_download_GBrowse.3F" class="mw-headline">Where do I download GBrowse?</span>

From
<a href="http://sourceforge.net/project/showfiles.php?group_id=27707"
class="external text" rel="nofollow">SourceForge</a>. If you want to
live on the bleeding edge, you may try the development version of
GBrowse. Instructions for accessing the development version of GBrowse
can be found at <a href="Subversion" class="mw-redirect"
title="Subversion">Subversion</a>.

### <span id="How_do_I_install_GBrowse.3F" class="mw-headline">How do I install GBrowse?</span>

After you unpack GBrowse, detailed installation instructions will be
found in the top level directory, in the file INSTALL.

### <span id="Where_do_I_find_a_list_of_all_available_glyphs.3F" class="mw-headline">Where do I find a list of all available glyphs?</span>

There is a list of glyphs at the end of the documentation for
<a href="http://search.cpan.org/perldoc?Bio::Graphics::Glyph"
class="external text" rel="nofollow">Bio::Graphics::Glyph</a> which you
can also see by executing

     perldoc Bio::Graphics::Glyph

from the command line.

### <span id="When_I_search.2C_why_doesn.27t_GBrowse_find_my_3-letter_gene_name.3F" class="mw-headline">When I search, why doesn't GBrowse find my 3-letter gene name?</span>

If you are using the [MySQL](MySQL "MySQL") [GFF](GFF "GFF")
[adaptor](GBrowse_Adaptors "GBrowse Adaptors") and are storing gene
names inside Note attributes, then you may bump up against MySQL's
default four-letter limit on full text searches. To fix this, either:

- Put the gene name in an Alias attribute, e.g. "Alias LEP"

or

- Change MySQL to allow searches on 3-character words.

The latter solution is a multi-step process:

- Open /etc/my.cnf and add the following configuration line to the
  \[mysqld\] section:

<!-- -->

        ft_min_word_len=3

- Restart the mysql server.
- Connect to your database using the mysql command-line client and run
  the command:

<!-- -->

       mysql> repair table fattribute_to_feature quick;

### <span id="How_do_I_use_semantic_zooming_to_hide_a_track_completely.3F" class="mw-headline">How do I use semantic zooming to hide a track completely?</span>

If you wish to turn off a track entirely, you can use the "hide" flag to
hide the track when the display exceeds a certain size:

                [6_frame_translation:50000]
                hide = 1

  

### <span id="How_to_add_an_outgoing_link_to_a_text_on_the_feature_detail_page_in_Gbrowse.3F" class="mw-headline">How to add an outgoing link to a text on the feature detail page in Gbrowse?</span>

Add a line to the appropriate \*.conf file using *link*. For example:

<div class="mw-geshi mw-code mw-content-ltr" dir="ltr">

<div class="perl source-perl">

``` de1
 link = http://www.ncbi.nih.gov/SNP/snp_ref.cgi?rs=$name
```

</div>

</div>

  

### <span id="I_have_a_multi-segmented_feature_.28such_as_a_multi-exon_transcript.29._It_looks_fine_at_low_power.2C_but_when_I_zoom_in_the_connecting_lines_between_segments_disappear._Help.21" class="mw-headline">I have a multi-segmented feature (such as a multi-exon transcript). It looks fine at low power, but when I zoom in the connecting lines between segments disappear. Help!</span>

You need to structure the feature in such a way that it has a single
parent part that spans the whole feature from end to end. When using
[GFF3](GFF3 "GFF3") and a Bio::DB::SeqFeature::Store database (see
[GBrowse Adaptors](GBrowse_Adaptors "GBrowse Adaptors")), that is all
you have to do. For example, using "match" as the feature in track
configuration of the
<a href="GBrowse_Configuration_HOWTO" class="mw-redirect"
title="GBrowse Configuration HOWTO">GBrowse configuration file</a> and
GFF3 like this would work:

                Chr1 . match         1  1000 . . . ID=Hit27
                Chr1 . match_part    1   200 . . . Parent=Hit27
                Chr1 . match_part  500   600 . . . Parent=Hit27
                Chr1 . match_part  900  1000 . . . Parent=Hit27

In [GFF2](GFF2 "GFF2") format, you will need to use an aggregator, in
this case, the "match" aggregator. Example GFF2 will look like this:

                Chr1 . match    1  1000 . . . Hit Hit27
                Chr1 . HSP      1   200 . . . Hit Hit27
                Chr1 . HSP    500   600 . . . Hit Hit27
                Chr1 . HSP    900  1000 . . . Hit Hit27

And you will use "match" (the name of the aggregator, not the name of
the parent feature) as the feature in the track configuration in the
<a href="GBrowse_Configuration_HOWTO" class="mw-redirect"
title="GBrowse Configuration HOWTO">GBrowse config file</a>.

For transcripts, use the "processed_transcript" aggregator and create
features with a main part of "mRNA" and subparts of "CDS", "exon",
and/or various types of UTRs.

### <span id="I.27m_using_the_GFF_database_adaptor._Is_it_better_to_load_it_using_GFF2_or_GFF3.3F" class="mw-headline">I'm using the GFF database adaptor. Is it better to load it using GFF2 or GFF3?</span>

[GFF3](GFF3 "GFF3").

[GFF2](GFF2 "GFF2"), described in the <a
href="http://gmod.svn.sourceforge.net/viewvc/gmod/Generic-Genome-Browser/branches/stable/docs/tutorial/dbgff/tutorial.html?content-type=text%2Fhtml"
class="external text" rel="nofollow">Using GBrowse with Bio::DB::GFF</a>
tutorial, is the older version of the [GFF](GFF "GFF") feature
annotation format. Its main limitation is that it cannot represent
features that have more than one level of nested subparts. For example,
you cannot represent the relationship between a gene, two alternatively
spliced transcripts, and the exons inside the transcripts.
[GFF3](GFF3 "GFF3") corrects this problem as well as a number of other
deficiencies.

### <span id="How_do_I_pass_parameters_into_functions_of_init_code.3F" class="mw-headline">How do I pass parameters into functions of init_code?</span>

You want to define a function in the init_code section which will then
be called within callbacks.

The answer is to do something like this:

               [GENERAL]
               init_code = sub round {
                               my $a = shift;
                               return int($a + 0.5);
                           }

               [TRACKS]
               label    = sub {
                         my $feature = shift;
                         my $score   = $feature->score;
                         return "score = ",round($score);
                         }

Note that you'll need GBrowse version 1.63 or higher for this to work.

  

### <span id="How_do_I_show_circular_genomes.3F" class="mw-headline">How do I show circular genomes?</span>

A patch is currently being developed to display circular genomes. You
can **git clone** the branch code at
<a href="https://github.com/GMOD/GBrowse/tree/nliles_gbrowse_circular"
class="external free"
rel="nofollow">https://github.com/GMOD/GBrowse/tree/nliles_gbrowse_circular</a>.
The patch will display basic glyphs, but has problems showing sequences.
To implement the patch, you must add the tag **region=circular;** to
column 9 of the first line of the gff file.

### <span id="There.27s_a_problem_with_my_overview_and_detail_images_-_they.27re_the_same_scale" class="mw-headline">There's a problem with my overview and detail images - they're the same scale</span>

That usually is a problem with your reference sequences. Either you
haven't set you reference class correctly in your config file (if you
are using [GFF3](GFF3 "GFF3"), it should probably be set to 'Sequence'),
or you don't have reference sequences in your [GFF](GFF "GFF")
file/database. That is, you don't have the lines in your GFF file that
correspond to the chromosomes.

### <span id="Can_I_add_Popup_Balloon_Tips_to_GBrowse.3F" class="mw-headline">Can I add Popup Balloon Tips to GBrowse?</span>

Yes! As of version 1.69, you can populate popup balloons similar to the
ones used by Google Maps. The balloons can contain arbitrary HTML
(including images) and can be set to appear when the user hovers over a
feature or left clicks on it. See [GBrowse
Configuration/Balloons](GBrowse_Configuration/Balloons "GBrowse Configuration/Balloons")
for details.

### <span id="Why_is_GBrowse_on_Chado_so_slow.3F" class="mw-headline">Why is GBrowse on Chado so slow?</span>

While it is convenient to run GBrowse on top of a Chado database that is
being actively edited (so that users can see "live" edits), the Chado
adaptor for GBrowse is fairly slow. This is because Chado is not
designed as a database to drive an interactive display like GBrowse.
Other adaptors like
<a href="http://bioperl.org/wiki/Module:Bio::DB::GFF"
class="external text" rel="nofollow">Bio::DB::GFF</a> and
<a href="http://bioperl.org/wiki/Module:Bio::DB::SeqFeature::Store"
class="external text" rel="nofollow">Bio::DB::SeqFeature::Store</a> run
on highly denormalized but fast to query databases. Chado on the other
hand is highly normalized, which is good for preventing data handling
errors, but bad in general for speed. There are some things that can be
done, like [materializing
views](Materialized_views "Materialized views") to speed some of the
slower queries, but in practice most users find it easier to dump the
contents of their Chado database to GFF3 and and load it into a
<a href="http://bioperl.org/wiki/Module:Bio::DB::SeqFeature::Store"
class="external text" rel="nofollow">Bio::DB::SeqFeature::Store</a>
database. [GMODTools](GMODTools "GMODTools") is a good tool for setting
up periodic dumps of your Chado database to GFF3.

### <span id="The_.22Bio::Graphics::BrowserConfig.3DHASH_..._Bio.2FGraphics.2FBrowser.pm_line_587.22_Error" class="mw-headline">The "Bio::Graphics::BrowserConfig=HASH ... Bio/Graphics/Browser.pm line 587" Error</span>

Users of GBrowse 1.69 will see this message in their Apache error logs:

    Bio::Graphics::BrowserConfig=HASH(0xnnnnnnn) at some_path/Bio/Graphics/Browser.pm line 587

This is a debugging statement that was accidentally left in the GBrowse
1.69 release. You can either ignore it (it does not affect anything), or
install the latest version of GBrowse from
<a href="Subversion" class="mw-redirect"
title="Subversion">Subversion</a>.

### <span id="Can_I_show_more_than_one_glyph_in_the_same_track.3F" class="mw-headline">Can I show more than one glyph in the same track?</span>

Yes. To show multiple glyphs in the same track set the glyph with a Perl
*callback*. For example:

    [newtrack]
    feature = sRNA  sRNA_HL
    glyph    = sub {
                    my $f = shift;
                    my $type = $f->type->method;
                    if ($type eq 'sRNA') {
                         return 'xyplot';
                    }
                    else {
                         return 'processed_transcript';
                    }
                 }

Another example:

    glyph        = sub { my $strand = shift->strand; return $strand >=0 ? 'gene' : 'box' }

For a situation where a attribute wouldn't make sense, you could return
undef.

You can set the glyph based on any information that is available in
callbacks.

From email threads:

- <a
  href="http://gmod.827538.n3.nabble.com/more-than-one-glyph-in-same-track-tp951795p951795.html"
  class="external text" rel="nofollow">More than one glyph in the same
  track</a>, Alaguraj Veluchamy, July 2010.
- <a
  href="http://gmod.827538.n3.nabble.com/Turn-some-Wiggle-tracks-off-when-no-value-in-selected-region-td946842.html#a946982#a946982"
  class="external text" rel="nofollow">Callbacks on Glyphs</a>, Andreas
  Redl, July 2010.

### <span id="How_many_tracks_can_be_displayed_in_GBrowse.3F" class="mw-headline">How many tracks can be displayed in GBrowse?</span>

There is no set maximum number of tracks. As of about GBrowse 2.13,
GBrowse works with more than 1,000 tracks. From this <a
href="http://gmod.827538.n3.nabble.com/http://gmod.827538.n3.nabble.com/Turn-some-Wiggle-tracks-off-when-no-value-in-selected-region-tp946842p946842.html"
class="external text" rel="nofollow">email thread (Kai Xia, 2010/07</a>,
[Lincoln Stein](User%3ALstein "User%3ALstein") said:

<div class="quotebox">

I think you'll find that 20,000 track definitions are going to slow
GBrowse down to the point of unusability. I have made some fixes to
GBrowse2 that allow you to display \> 1000 tracks with reasonable
performance, but I've never tested in the 10,000 track range.

If you combine wiggle data into subtracks, then performance in GBrowse2
should be quite good. There is also an interface that lets you define
metadata for each subtrack and search, filter and sort on the basis of
this metadata (see [Creating and Managing Subtracks with
GBrowse2](Creating_and_Managing_Subtracks_with_GBrowse2 "Creating and Managing Subtracks with GBrowse2")).
I've also just now added an option that lets you hide subtracks that
have no data currently showing.

</div>

</div>

<div class="printfooter">

Retrieved from
"<http://gmod.org/mediawiki/index.php?title=GBrowse_FAQ&oldid=23576>"

</div>

<div id="catlinks" class="catlinks">

<div id="mw-normal-catlinks" class="mw-normal-catlinks">

[Categories](Special%3ACategories "Special%3ACategories"):

- [FAQ](Category%3AFAQ "Category%3AFAQ")
- [GBrowse](Category%3AGBrowse "Category%3AGBrowse")

</div>

</div>

<div class="visualClear">

</div>

</div>

</div>

<div id="mw-navigation">

## Navigation menu

<div id="mw-head">



<div id="left-navigation">

<div id="p-namespaces" class="vectorTabs" role="navigation"
aria-labelledby="p-namespaces-label">

### Namespaces


- <span id="ca-talk"><a
  href="http://gmod.org/mediawiki/index.php?title=Talk:GBrowse_FAQ&amp;action=edit&amp;redlink=1"
  accesskey="t"
  title="Discussion about the content page [t]">Discussion</a></span>

</div>

<div id="p-variants" class="vectorMenu emptyPortlet" role="navigation"
aria-labelledby="p-variants-label">

### 

### Variants[](#)

<div class="menu">

</div>

</div>

</div>





</div>

</div>

</div>

<div id="mw-panel">

<div id="p-logo" role="banner">

<a href="Main_Page"
style="background-image: url(../images/GMOD-cogs.png);"
title="Visit the main page"></a>

</div>

<div id="p-Navigation" class="portal" role="navigation"
aria-labelledby="p-Navigation-label">

### Navigation

<div class="body">

- <span id="n-GMOD-Home">[GMOD Home](Main_Page)</span>
- <span id="n-Software">[Software](GMOD_Components)</span>
- <span id="n-Categories-.2F-Tags">[Categories /
  Tags](Categories)</span>

</div>

</div>

<div id="p-Documentation" class="portal" role="navigation"
aria-labelledby="p-Documentation-label">

### Documentation

<div class="body">

- <span id="n-Overview">[Overview](Overview)</span>
- <span id="n-FAQs">[FAQs](Category%3AFAQ)</span>
- <span id="n-HOWTOs">[HOWTOs](Category%3AHOWTO)</span>
- <span id="n-Glossary">[Glossary](Glossary)</span>

</div>

</div>

<div id="p-Community" class="portal" role="navigation"
aria-labelledby="p-Community-label">

### Community

<div class="body">

- <span id="n-GMOD-News">[GMOD News](GMOD_News)</span>
- <span id="n-Training-.2F-Outreach">[Training /
  Outreach](Training_and_Outreach)</span>
- <span id="n-Support">[Support](Support)</span>
- <span id="n-GMOD-Promotion">[GMOD Promotion](GMOD_Promotion)</span>
- <span id="n-Meetings">[Meetings](Meetings)</span>
- <span id="n-Calendar">[Calendar](Calendar)</span>

</div>

</div>

<div id="p-tb" class="portal" role="navigation"
aria-labelledby="p-tb-label">

### Tools

<div class="body">


- <span id="t-smwbrowselink"><a href="Special%3ABrowse/GBrowse_FAQ" rel="smw-browse">Browse
  properties</a></span>


</div>

</div>

</div>

</div>

<div id="footer" role="contentinfo">

- <span id="footer-info-lastmod">Last updated at 14:48 on 18 May
  2013.</span>
<!-- - <span id="footer-info-viewcount">214,272 page views.</span> -->
- <span id="footer-info-copyright">Content is available under
  <a href="http://www.gnu.org/licenses/fdl-1.3.html" class="external"
  rel="nofollow">a GNU Free Documentation License</a> unless otherwise
  noted.</span>

<!-- -->

- <span id="footer-places-about">[About
  GMOD](GMOD:About "GMOD:About")</span>

<!-- -->






</div>
