<div id="mw-page-base" class="noprint">

</div>

<div id="mw-head-base" class="noprint">

</div>

<div id="content" class="mw-body" role="main">

<span id="top"></span>

<div id="mw-js-message" style="display:none;">

</div>



# <span dir="auto">GMOD News Archives</span>

<div id="bodyContent">

<div id="siteSub">

From GMOD

</div>

<div id="contentSub">

</div>

<div id="jump-to-nav" class="mw-jump">

Jump to: [navigation](#mw-navigation), [search](#p-search)

</div>

<div id="mw-content-text" class="mw-content-ltr" lang="en" dir="ltr">

<div class="plainlinks C"
style="float: right; margin-left: 1em; margin-bottom: 1em; padding-left: 0.5em; padding-right: 0.5em; padding-bottom: 0.5em; border: 1px solid #888888; background-color: #fafafa; line-height: 100%">

<a
href="http://gmod.org/wiki?title=Special%3ANewsChannel&amp;format=rss20&amp;limit=5000&amp;cat1=&amp;cat2=&amp;excat1=&amp;wpSubmitNewsChannelParams=Create+feed"
class="external text" rel="nofollow">RSS</a> <a
href="http://gmod.org/wiki?title=Special%3ANewsChannel&amp;format=rss20&amp;limit=5000&amp;cat1=&amp;cat2=&amp;excat1=&amp;wpSubmitNewsChannelParams=Create+feed"
rel="nofollow" title="RSS feed"><img
src="../mediawiki/images/4/47/RSSIcon16x16.gif" width="16" height="16"
alt="RSS feed" /></a>  
[Recent News Only](GMOD_News "GMOD News")  
[Add a news item](GMOD_News#Adding_a_News_Item "GMOD News")

------------------------------------------------------------------------

<div id="Items" class="switch" style="list-style-type: none">

<table class="RSSMainBody" style="background:inherit;">
<colgroup>
<col style="width: 100%" />
</colgroup>
<tbody>
<tr class="odd">
<td data-valign="top" style="width: 95%"><ul>
<li><a href="News/Prospecting_for_Proposals_for_GSoC_2024">Prospecting
for Proposals for GSoC 2024</a></li>
<li><a href="News/JBrowse2_v1.1.0_Released">JBrowse2 v1.1.0
Released</a></li>
<li><a href="News/Prospecting_for_Proposals_for_GSoC_2021">Prospecting
for Proposals for GSoC 2021</a></li>
<li><a href="Codefest_2020">Codefest 2020</a></li>
<li><a href="Tripal_3.5_Released">Tripal 3.5 Released</a></li>
<li><a href="News/JBrowse_2_release">JBrowse 2 release</a></li>
<li><a href="News/Prospecting_for_Proposals_for_GSoC_2020">Prospecting
for Proposals for GSoC 2020</a></li>
<li><a href="News/Call_for_PAG_Abstracts">Call for PAG
Abstracts</a></li>
<li><a href="News/New_GMOD_Server">New GMOD Server</a></li>
<li><a href="News/GMOD-JBrowse_2016_Survey">GMOD-JBrowse 2016
Survey</a></li>
<li><a href="News/GCC2016">GCC2016</a></li>
<li><a href="News/2016_GMOD_Meeting">2016 GMOD Meeting</a></li>
<li><a href="News/Please_Support_EcoCyc">Please Support EcoCyc</a></li>
<li><a href="News/June_2014_WebApollo_Hackathon">June 2014 WebApollo
Hackathon</a></li>
<li><a
href="News/Applications_Open_for_GMOD_Online_Training">Applications Open
for GMOD Online Training</a></li>
<li><a href="News/Canto_Workshop_at_Biocuration_2014">Canto Workshop at
Biocuration 2014</a></li>
<li><a href="News/Tripal_2.0a_released">Tripal 2.0a released</a></li>
<li><a href="News/Precompiled_Ontologies_in_Chado">Precompiled
Ontologies in Chado</a></li>
<li><a href="News/GCC2014_Registration_is_Open">GCC2014 Registration is
Open</a></li>
<li><a href="News/GMOD_Paper_Cuts,_Feb_10th,_2014">GMOD Paper Cuts, Feb
10th, 2014</a></li>
<li><a href="News/Galaxy_Australasia_Workshop_2014">Galaxy Australasia
Workshop 2014</a></li>
<li><a href="News/GMOD_Paper_Cuts,_Jan_24th,_2014">GMOD Paper Cuts, Jan
24th, 2014</a></li>
<li><a
href="News/Postdoc_Position_in_Computational_and_Plant_Evolutionary_Biology">Postdoc
Position in Computational and Plant Evolutionary Biology</a></li>
<li><a href="News/Prospecting_for_Proposals_for_GSoC_2014">Prospecting
for Proposals for GSoC 2014</a></li>
<li><a href="News/Biocuration_2014">Biocuration 2014</a></li>
<li><a href="News/GMOD_At_PAG_2014">GMOD At PAG 2014</a></li>
<li><a href="News/Announcing_GMOD_Malaysia_2014">Announcing GMOD
Malaysia 2014</a></li>
<li><a href="News/MAKER-P:_an_annotation_pipeline_for_plants">MAKER-P:
an annotation pipeline for plants</a></li>
<li><a href="News/GMOD_Paper_Cuts,_Dec_6th,_2013">GMOD Paper Cuts, Dec
6th, 2013</a></li>
<li><a href="News/Canto,_a_GMOD_tool_for_functional_annotation">Canto, a
GMOD tool for functional annotation</a></li>
<li><a href="News/New_WebApollo_Release_(2013/11/22)">New WebApollo
Release (2013/11/22)</a></li>
<li><a href="News/GMOD_Paper_Cuts,_Nov_22nd,_2013">GMOD Paper Cuts, Nov
22nd, 2013</a></li>
<li><a href="News/GMOD_Paper_Cuts,_Nov_15th,_2013">GMOD Paper Cuts, Nov
15th, 2013</a></li>
<li><a href="News/GMOD_Job_At_QMUL">GMOD Job At QMUL</a></li>
<li><a href="News/GMOD_Paper_Cuts,_Nov_8th,_2013">GMOD Paper Cuts, Nov
8th, 2013</a></li>
<li><a href="News/Open_Position_At_dictyBase">Open Position At
dictyBase</a></li>
<li><a href="News/Tripal_1.1_Publication">Tripal 1.1
Publication</a></li>
<li><a href="News/San_Diego_2014_Registration_Open">San Diego 2014
Registration Open</a></li>
<li><a href="News/GCC2014:_June_30_-_July_2">GCC2014: June 30 - July
2</a></li>
<li><a href="News/Get_the_Full_ImpactStory">Get the Full
ImpactStory</a></li>
<li><a href="News/GMOD_2014_San_Diego:_Save_the_Date">GMOD 2014 San
Diego: Save the Date</a></li>
<li><a href="News/Statistical_Genomics_PostDoc_at_PSU">Statistical
Genomics PostDoc at PSU</a></li>
<li><a href="News/Support_Your_Friendly_Local_GMOD">Support Your
Friendly Local GMOD</a></li>
<li><a href="News/2013_Summer_School_Tutorials_Available">2013 Summer
School Tutorials Available</a></li>
<li><a href="News/JBrowse_1.10.0_released">JBrowse 1.10.0
released</a></li>
<li><a href="News/OpenID_Account_Creation_Fixed">OpenID Account Creation
Fixed</a></li>
<li><a href="News/GMOD&#39;s_GSoC_Going_Great_Guns!">GMOD's GSoC Going
Great Guns!</a></li>
<li><a href="News/GMOD_in_the_Cloud_2.0_Released">GMOD in the Cloud 2.0
Released</a></li>
<li><a href="News/Last_Call_For_Summer_School_Apps">Last Call For Summer
School Apps</a></li>
<li><a href="News/GCC2013_Early_Reg_closes_24_May">GCC2013 Early Reg
closes 24 May</a></li>
<li><a href="News/New_WebApollo_Release">New WebApollo Release</a></li>
</ul></td>
</tr>
</tbody>
</table>

</div>

</div>

The GMOD News Archive contains old news items that have been archived
off of the [GMOD News](GMOD_News "GMOD News") page/feed. The News
includes anything that is relevant to the GMOD user and developer
communities. This includes a wide range of topics, from GMOD meetings
and releases to user success stories and GMOD related publications.

See the [GMOD News](GMOD_News "GMOD News") page for instruction on how
to post a news item.

# <span id="All_News_Items" class="mw-headline">All News Items</span>

<table class="RSSMainBody" style="background:inherit;">
<colgroup>
<col style="width: 100%" />
</colgroup>
<tbody>
<tr class="odd">
<td data-valign="top" style="width: 95%"><div class="rss">
<h3 id="prospecting-for-proposals-for-gsoc-2024"><a
href="News/Prospecting_for_Proposals_for_GSoC_2024">Prospecting for
Proposals for GSoC 2024</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://gmod.org/wiki/GSoC#58;//gmod.org/wiki/GSoC"
title="GSoC">&lt;img alt="GoogleSummer 2016logo.jpg"
src="http://gmod.org/mediawiki/images/thumb/5/5d/GoogleSummer_2016logo.jpg/272px-GoogleSummer_2016logo.jpg.png"
width="272" height="272"
srcset="/mediawiki/images/5/5d/GoogleSummer_2016logo.jpg 1.5x,
/mediawiki/images/5/5d/GoogleSummer_2016logo.jpg 2x" /&gt;</a>
</div>
<p>The Open Genome Informatics group and GMOD has submitted an
application for <a
href="http://gmod.org/wiki/https://summerofcode.withgoogle.com/#58;//summerofcode.withgoogle.com/"
class="external text" rel="nofollow">Google Summer of Code (GSoC)</a>
this year, and we are soliciting project ideas from people and groups
involved in the GMOD project.</p>
<p>GSoC is a global program run by Google to encourage students to get
real-world experience of participating in a software project; students
work on a project over the summer months and receive a stipend from
Google for participating. Thousands of organizations of different sizes
participate in GSoC, with the common factor being that they must produce
open-source code.</p>
<p>Please take a look at the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GSoC#58;//gmod.org/wiki/GSoC"
title="GSoC">GSoC wiki page</a> and think about any projects that might
work well for a GSoC student, and add them to the wiki. If you are
interested in being a mentor, don't hesitate to get in touch with us at
<a href="http://gmod.org/wiki/mailto:rhaw@oicr.on.ca#58;rhaw@oicr.on.ca"
class="external text" rel="nofollow">robin.haw@oicr.on.ca</a> <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">help@gmod.org</a>.</p>
<p>If you have any questions--e.g. what constitutes an appropriate
project; whether your idea is sufficiently GMOD-related--please feel
free to email <a
href="http://gmod.org/wiki/mailto:robin.haw@oicr.on.ca#58;robin.haw@oicr.on.ca"
class="external text" rel="nofollow">rhaw@oicr.on.ca</a> and <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">help@gmod.org</a> for advice!</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2024/02/12</em>
</div>
</div>
<div class="rss">
<h3 id="jbrowse2-v1.1.0-released"><a
href="News/JBrowse2_v1.1.0_Released">JBrowse2 v1.1.0 Released</a></h3>
<div id="toc" class="toc">
<div id="toctitle">
<h2 id="contents">Contents</h2>
</div>
<ul>
<li><span><span class="tocnumber">1</span> <span class="toctext">We're
pleased to announce a new release of JBrowse Web!</span></span>
<ul>
<li><span><span class="tocnumber">1.1</span> <span
class="toctext">Changed callbacks language from JavaScript to
Jexl</span></span></li>
<li><span><span class="tocnumber">1.2</span> <span class="toctext">Fetch
intron and upstream/downstream sequences</span></span></li>
<li><span><span class="tocnumber">1.3</span> <span
class="toctext">Interactive documentation using
Storybook</span></span></li>
<li><span><span class="tocnumber">1.4</span> <span
class="toctext">Enhanced navigation to drawer widget
stack</span></span></li>
<li><span><span class="tocnumber">1.5</span> <span
class="toctext">Downloads</span></span></li>
<li><span><span class="tocnumber">1.6</span> <span class="toctext">1.1.0
(2021-03-29)</span></span>
<ul>
<li><span><span class="tocnumber">1.6.1</span> <span class="toctext">🚀
Enhancement</span></span></li>
<li><span><span class="tocnumber">1.6.2</span> <span class="toctext">🐛
Bug Fix</span></span></li>
<li><span><span class="tocnumber">1.6.3</span> <span class="toctext">📝
Documentation</span></span></li>
<li><span><span class="tocnumber">1.6.4</span> <span class="toctext">🏠
Internal</span></span></li>
<li><span><span class="tocnumber">1.6.5</span> <span
class="toctext">Committers: 6</span></span></li>
</ul></li>
</ul></li>
</ul>
</div>
<h2 id="were-pleased-to-announce-a-new-release-of-jbrowse-web"><span
id="We.27re_pleased_to_announce_a_new_release_of_JBrowse_Web.21"
class="mw-headline">We're pleased to announce a new release of JBrowse
Web!</span></h2>
<p>(Reposted by permission from <a
href="http://gmod.org/wiki/https://jbrowse.org/jb2/blog/2021/03/29/v1.1.0-release#58;//jbrowse.org/jb2/blog/2021/03/29/v1.1.0-release"
class="external free"
rel="nofollow">https://jbrowse.org/jb2/blog/2021/03/29/v1.1.0-release</a></p>
<h3 id="changed-callbacks-language-from-javascript-to-jexl"><span
id="Changed_callbacks_language_from_JavaScript_to_Jexl"
class="mw-headline">Changed callbacks language from JavaScript to
Jexl</span></h3>
<p>To allow users to safely and seamlessly share advanced configurations
in sessions, we now use Jexl to express configuration callbacks. Note
that this is a breaking change, function()-style callbacks will no
longer work.</p>
<p>For details, see the <a
href="http://gmod.org/wiki/https://jbrowse.org/jb2/docs/config_guide#configuration-callbacks#58;//jbrowse.org/jb2/docs/config_guide#configuration-callbacks"
class="external text" rel="nofollow">callbacks section of our
configuration guide</a>.</p>
<h3 id="fetch-intron-and-upstreamdownstream-sequences"><span
id="Fetch_intron_and_upstream.2Fdownstream_sequences"
class="mw-headline">Fetch intron and upstream/downstream
sequences</span></h3>
<p>We also have several other improvements including the ability to get
intron and upstream/downstream sequence in the feature details</p>
<p><a
href="http://gmod.org/wiki/File:Upstream_downstream.png#58;//gmod.org/wiki/File:Upstream_downstream.png"
class="image">&lt;img alt="Upstream downstream.png"
src="http://gmod.org/mediawiki/images/thumb/7/76/Upstream_downstream.png/700px-Upstream_downstream.png"
width="700" height="248"
srcset="/mediawiki/images/thumb/7/76/Upstream_downstream.png/1050px-Upstream_downstream.png
1.5x,
/mediawiki/images/thumb/7/76/Upstream_downstream.png/1400px-Upstream_downstream.png
2x" /&gt;</a></p>
<h3 id="interactive-documentation-using-storybook"><span
id="Interactive_documentation_using_Storybook"
class="mw-headline">Interactive documentation using
Storybook</span></h3>
<p>Another new update is the first release of our interactive Storybook
docs for the embeddable React Linear Genome View. The docs contain live
examples of how the LGV component can be used, along with source-code
examples. The site can be found <a
href="http://gmod.org/wiki/https://jbrowse.org/storybook/lgv/v1.1.0#58;//jbrowse.org/storybook/lgv/v1.1.0"
class="external text" rel="nofollow">here</a>.</p>
<h3 id="enhanced-navigation-to-drawer-widget-stack"><span
id="Enhanced_navigation_to_drawer_widget_stack"
class="mw-headline">Enhanced navigation to drawer widget
stack</span></h3>
<p>We have added a dropdown to enhance navigation between stack of
active widgets. The update also adds a minimize button to allow quick
access to full screen JBrowse web.</p>
<p>See below for demos of the new navigation UI.</p>
<p><a
href="http://gmod.org/wiki/File:Minimize_button_demo.gif#58;//gmod.org/wiki/File:Minimize_button_demo.gif"
class="image">&lt;img alt="Minimize button demo.gif"
src="http://gmod.org/mediawiki/images/0/0f/Minimize_button_demo.gif"
width="480" height="414" /&gt;</a></p>
<p>Demo of using the minimize button in the drawer</p>
<h3 id="downloads"><span id="Downloads"
class="mw-headline">Downloads</span></h3>
<ul>
<li><a
href="http://gmod.org/wiki/https://github.com/GMOD/jbrowse-components/releases/tag/v1.1.0#58;//github.com/GMOD/jbrowse-components/releases/tag/v1.1.0"
class="external text" rel="nofollow">v1.1.0</a></li>
</ul>
<p>To install JBrowse 2 for the web, you can download the link above, or
you can use the JBrowse CLI to automatically download the latest
version. See the JBrowse web quick start for more details.</p>
<h3 id="section"><span id="1.1.0_.282021-03-29.29"
class="mw-headline">1.1.0 (2021-03-29)</span></h3>
<h4 id="enhancement"><span id=".F0.9F.9A.80_Enhancement"
class="mw-headline">🚀 Enhancement</span></h4>
<p>core</p>
<ol>
<li>1846 Improve copy+paste in the data grids for feature details
(@cmdcolin)</li>
<li>1814 Add ability to get promoter sequence and intron sequence for
genes from the feature details panel (@cmdcolin)</li>
<li>1816 Remove some animation effects (@cmdcolin)</li>
<li>1778 Adds dropdown to show drawer widget stack (@teresam856)</li>
<li>1685 Change callbacks language from JavaScript to Jexl
(@peterkxie)</li>
</ol>
<p>Other</p>
<ol>
<li>1831 Add dialog for launching breakpoint split view from variant
feature details (@cmdcolin)</li>
<li>1803 Transcript and gene glyphs can now display implied UTRs, active
by default (@cmdcolin)</li>
<li>1808 Add another heuristic for returning gene features from BigBed
(@cmdcolin)</li>
<li>1774 Add warning dialog in LGV before returning to import form to
prevent accidentally losing the current view (@cmdcolin)</li>
</ol>
<h4 id="bug-fix"><span id=".F0.9F.90.9B_Bug_Fix" class="mw-headline">🐛
Bug Fix</span></h4>
<p>core</p>
<ol>
<li>1811 Check for existence of window more robustly to allow in SSR or
node applications (@elliothershberg)</li>
<li>1793 Fix dotplot rendering outside it's allowed bounds
(@cmdcolin)</li>
<li>1783 Add hic aborting and fix remoteAbort signal propagation
(@cmdcolin)</li>
<li>1723 A few bugfixes (@garrettjstevens)</li>
</ol>
<p>Other</p>
<ol>
<li>1815 Clear tracks when using "Return to import form"
(@cmdcolin)</li>
<li>1819 Standardized sentence casing on drawer widget titles
(@cmdcolin)</li>
<li>1796 Bump generic-filehandle for fixing CORS errors from Chrome
cache pollution (@cmdcolin)</li>
</ol>
<h4 id="documentation"><span id=".F0.9F.93.9D_Documentation"
class="mw-headline">📝 Documentation</span></h4>
<ol>
<li>1824 Add storybook docs page for nextjs usage
(@elliothershberg)</li>
<li>1770 1469 storybook deploy (@elliothershberg)</li>
<li>1807 Update developer guide to cover displays, and highlight working
external plugins (@cmdcolin)</li>
<li>1779 Collaborative release announcement editing (@rbuels)</li>
<li>1791 Add a couple more demos for our live version with MDX
(@cmdcolin)</li>
</ol>
<h4 id="internal"><span id=".F0.9F.8F.A0_Internal"
class="mw-headline">🏠 Internal</span></h4>
<p>Other</p>
<ol>
<li>1820 Create v1.1.0.md, draft of release announcements
(@garrettjstevens)</li>
<li>1823 Add note about previewing changelog to CONTRIBUTING.md
(@garrettjstevens)</li>
</ol>
<p>core</p>
<ol>
<li>1834 Change jbrowse-components monorepo default branch from 'master'
to 'main' (@rbuels)</li>
</ol>
<h4 id="committers-6"><span id="Committers:_6"
class="mw-headline">Committers: 6</span></h4>
<ul>
<li>Colin Diesh (@cmdcolin)</li>
<li>Elliot Hershberg (@elliothershberg)</li>
<li>Garrett Stevens (@garrettjstevens)</li>
<li>Peter Xie (@peterkxie)</li>
<li>Robert Buels (@rbuels)</li>
<li>Teresa Martinez (@teresam856)</li>
</ul>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2021/03/30</em>
</div>
</div>
<div class="rss">
<h3 id="prospecting-for-proposals-for-gsoc-2021"><a
href="News/Prospecting_for_Proposals_for_GSoC_2021">Prospecting for
Proposals for GSoC 2021</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://gmod.org/wiki/GSoC#58;//gmod.org/wiki/GSoC"
title="GSoC">&lt;img alt="GoogleSummer 2016logo.jpg"
src="http://gmod.org/mediawiki/images/thumb/5/5d/GoogleSummer_2016logo.jpg/272px-GoogleSummer_2016logo.jpg.png"
width="272" height="272"
srcset="/mediawiki/images/5/5d/GoogleSummer_2016logo.jpg 1.5x,
/mediawiki/images/5/5d/GoogleSummer_2016logo.jpg 2x" /&gt;</a>
</div>
<p>The Genome Informatics group and GMOD will be submitting an
application for <a
href="http://gmod.org/wiki/https://summerofcode.withgoogle.com/#58;//summerofcode.withgoogle.com/"
class="external text" rel="nofollow">Google Summer of Code (GSoC)</a>
this year, and we are soliciting project ideas from people and groups
involved in the GMOD project.</p>
<p>GSoC is a global program run by Google to encourage students to get
real-world experience of participating in a software project; students
work on a project over the summer months and receive a stipend from
Google for participating. Thousands of organizations of many different
sizes take part in GSoC, with the common factor being that they must
produce open-source code.</p>
<p>Please take a look at the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GSoC#58;//gmod.org/wiki/GSoC"
title="GSoC">GSoC wiki page</a> and think about any projects that might
work well for a GSoC student, and add them to the wiki. If you are
interested in being a mentor, please contact us at <a
href="http://gmod.org/wiki/mailto:robin.haw@oicr.on.ca#58;robin.haw@oicr.on.ca"
class="external text" rel="nofollow">robin.haw@oicr.on.ca</a> <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">help@gmod.org</a>.</p>
<p>If you have any questions--e.g. what constitutes an appropriate
project; whether your idea is sufficiently GMOD-related--please feel
free to email <a
href="http://gmod.org/wiki/mailto:robin.haw@oicr.on.ca#58;robin.haw@oicr.on.ca"
class="external text" rel="nofollow">robin.haw@oicr.on.ca</a> and <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">help@gmod.org</a> for advice!</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2021/02/16</em>
</div>
</div>
<div class="rss">
<h3 id="codefest-2020"><a href="Codefest_2020">Codefest 2020</a></h3>
<div class="center">
<div class="floatnone">
<a
href="http://gmod.org/wiki/File:Palms.jpg#58;//gmod.org/wiki/File:Palms.jpg"
class="image">&lt;img alt="Palms.jpg"
src="http://gmod.org/mediawiki/images/d/df/Palms.jpg" width="1052"
height="218" /&gt;</a>
</div>
</div>
<p><br />
</p>
<strong></strong>
<p><a
href="http://gmod.org/wiki/https://sd.handlery.com/#58;//sd.handlery.com/"
class="external text" rel="nofollow">Handlery Hotel in the Balboa
room</a><br />
(down the street from the Town and Country Hotel)<br />
San Diego, California<br />
January 9-10, 2020 &lt;/strong&gt;</p>
<p>Immediately before <a
href="http://gmod.org/wiki/https://www.intlpag.org/2020/#58;//www.intlpag.org/2020/"
class="external text" rel="nofollow">Plant and Animal Genome XXVIII (PAG
2020)</a></p>
<p>&lt;/center&gt;</p>
<p><strong><a
href="http://gmod.org/wiki/https://docs.google.com/spreadsheets/d/1Kp-fGI0U88PtsbryP1j-BUGA4cduYvC2ojW9Mk0jwxU/edit#gid=0#58;//docs.google.com/spreadsheets/d/1Kp-fGI0U88PtsbryP1j-BUGA4cduYvC2ojW9Mk0jwxU/edit#gid=0"
class="external text" rel="nofollow">Participant list now
available</a></strong></p>
<p><br />
</p>
<p>There will be a GMOD codefest occurring before the Plant and Animal
Genomes meeting in San Diego. The codefest will be at the Town and
Country hotel on January 9 and 10. If you would like to suggest a
problem or project to address, add it to this <a
href="http://gmod.org/wiki/https://docs.google.com/document/d/1_CnUW_W4tNyl7lSlihCwZDKT45VQQxcI3I-VgjnC2Dc/edit?usp=sharing#58;//docs.google.com/document/d/1_CnUW_W4tNyl7lSlihCwZDKT45VQQxcI3I-VgjnC2Dc/edit?usp=sharing"
class="external text" rel="nofollow">Google Doc</a>. The codefest is
open to anyone who'd like to work on any GMOD project (or, better yet,
any combination of GMOD projects), including but not limited to</p>
<div class="floatright">
<a
href="http://gmod.org/wiki/File:EvoHackLaptops2010.jpg#58;//gmod.org/wiki/File:EvoHackLaptops2010.jpg"
class="image" title="GMOD Hackathon">&lt;img alt="GMOD Hackathon"
src="http://gmod.org/mediawiki/images/thumb/7/7f/EvoHackLaptops2010.jpg/200px-EvoHackLaptops2010.jpg"
width="200" height="119"
srcset="/mediawiki/images/thumb/7/7f/EvoHackLaptops2010.jpg/300px-EvoHackLaptops2010.jpg
1.5x, /mediawiki/images/7/7f/EvoHackLaptops2010.jpg 2x" /&gt;</a>
</div>
<ul>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Tripal#58;//gmod.org/wiki/Tripal"
title="Tripal">Tripal</a></li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Chado#58;//gmod.org/wiki/Chado"
class="mw-redirect" title="Chado">Chado</a></li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a> (1 and 2)</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Apollo#58;//gmod.org/wiki/Apollo"
title="Apollo">Apollo</a></li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Galaxy#58;//gmod.org/wiki/Galaxy"
title="Galaxy">Galaxy</a></li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/InterMine#58;//gmod.org/wiki/InterMine"
title="InterMine">InterMine</a></li>
</ul>
<p>We already know that there will be Tripal, Chado, Apollo and JBrowse
developers present.</p>
<h2 id="agenda"><span id="Agenda" class="mw-headline">Agenda</span></h2>
<p>Thursday:</p>
<table class="wikitable">
<tbody>
<tr class="odd">
<td>9:00-10:00</td>
<td>Introduction and working group organization</td>
</tr>
<tr class="even">
<td>10:00-12:00</td>
<td>Working groups</td>
</tr>
<tr class="odd">
<td>12:00-1:30</td>
<td>Lunch (on your own)</td>
</tr>
<tr class="even">
<td>1:30-4:00</td>
<td>Working groups</td>
</tr>
<tr class="odd">
<td>4:00-5:00</td>
<td>Report on progress</td>
</tr>
</tbody>
</table>
<p>Friday:</p>
<table class="wikitable">
<tbody>
<tr class="odd">
<td>9:00-12:00</td>
<td>Working groups</td>
</tr>
<tr class="even">
<td>12:00-1:30</td>
<td>Lunch (on your own)</td>
</tr>
<tr class="odd">
<td>1:30-3:00</td>
<td>Working groups</td>
</tr>
<tr class="even">
<td>3:00-3:45</td>
<td>Report on progress</td>
</tr>
</tbody>
</table>
<ul>
<li>Note: the Tripal steering committee will meet on Friday from 12:00 -
3:00 pm at the Postcard Bistro in the Handlery.</li>
</ul>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2020/01/08</em>
</div>
</div>
<div class="rss">
<h3 id="tripal-3.5-released"><a href="Tripal_3.5_Released">Tripal 3.5
Released</a></h3>
<p>This <a
href="http://gmod.org/wiki/http://tripal.info#58;//tripal.info"
class="external text" rel="nofollow">Tripal</a> release includes major
performance improvements including an improved and blazing fast GFF3
loader. Follow standard Drupal updating procedures to update. Changes in
this version:</p>
<ol>
<li>Performance improvements to the Cross References, Relationships, and
References.</li>
<li>A rewritten GFF3 loader that is extremely fast, even with a fully
populated Tripal site.</li>
<li>A new Sequences fields that combines reference-dervied sequences
with primary sequences, and CDS/proteins for mRNA features and sequences
now have more informative definition lines for FASTA sequences.</li>
<li>Updates to the online documentation</li>
<li>Bug fixes in web services</li>
<li>Bug fixes in the OBO loader for ontologies that won't import.</li>
</ol>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2021/02/08</em>
</div>
</div>
<div class="rss">
<h3 id="jbrowse-2-release"><a href="News/JBrowse_2_release">JBrowse 2
release</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/https://jbrowse.org/jb2/#58;//jbrowse.org/jb2/"
rel="nofollow" title="JBrowse">&lt;img alt="JBrowse"
src="http://gmod.org/mediawiki/images/thumb/5/5a/Jbrowse2_logo.png/200px-Jbrowse2_logo.png"
width="200" height="67"
srcset="/mediawiki/images/thumb/5/5a/Jbrowse2_logo.png/300px-Jbrowse2_logo.png
1.5x,
/mediawiki/images/thumb/5/5a/Jbrowse2_logo.png/400px-Jbrowse2_logo.png
2x" /&gt;</a>
</div>
<p>We are pleased to announce the first public release of <a
href="http://gmod.org/wiki/https://jbrowse.org/jb2/#58;//jbrowse.org/jb2/"
class="external text" rel="nofollow">JBrowse 2</a>!</p>
<p>JBrowse 2 is the successor to JBrowse, but is a completely new
application written with modern web standards and frameworks.</p>
<p>Some of the features new to JBrowse 2 include:</p>
<ul>
<li>New types of views, including circular, synteny, and dotplot
views</li>
<li>Graphical configuration editing</li>
<li>Connections to resources such as UCSC Track Hubs</li>
</ul>
<p>You can see some demos of JBrowse 2 in action <a
href="http://gmod.org/wiki/https://jbrowse.org/jb2/demos/#58;//jbrowse.org/jb2/demos/"
class="external text" rel="nofollow">here</a>. To get started using
JBrowse 2, visit our quickstart guide <a
href="http://gmod.org/wiki/https://jbrowse.org/jb2/docs/quickstart_web/#58;//jbrowse.org/jb2/docs/quickstart_web/"
class="external text" rel="nofollow">here</a>.</p>
<p>We'd love to hear what you think! You can find information about how
to contact us <a
href="http://gmod.org/wiki/https://jbrowse.org/jb2/contact/#58;//jbrowse.org/jb2/contact/"
class="external text" rel="nofollow">here</a>.</p>
<p>The JBrowse Team</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2020/11/09</em>
</div>
</div>
<div class="rss">
<h3 id="prospecting-for-proposals-for-gsoc-2020"><a
href="News/Prospecting_for_Proposals_for_GSoC_2020">Prospecting for
Proposals for GSoC 2020</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://gmod.org/wiki/GSoC#58;//gmod.org/wiki/GSoC"
title="GSoC">&lt;img alt="GoogleSummer 2016logo.jpg"
src="http://gmod.org/mediawiki/images/thumb/5/5d/GoogleSummer_2016logo.jpg/272px-GoogleSummer_2016logo.jpg.png"
width="272" height="272"
srcset="/mediawiki/images/5/5d/GoogleSummer_2016logo.jpg 1.5x,
/mediawiki/images/5/5d/GoogleSummer_2016logo.jpg 2x" /&gt;</a>
</div>
<p>The Genome Informatics group and GMOD will be submitting an
application for <a
href="http://gmod.org/wiki/https://summerofcode.withgoogle.com/#58;//summerofcode.withgoogle.com/"
class="external text" rel="nofollow">Google Summer of Code (GSoC)</a>
this year, and we are soliciting project ideas from people and groups
involved in the GMOD project.</p>
<p>GSoC is a global program run by Google to encourage students to get
real-world experience of participating in a software project; students
work on a project over the summer months and receive a stipend from
Google for participating. Thousands of organizations of many different
sizes take part in GSoC, with the common factor being that they must
produce open-source code.</p>
<p>Please take a look at the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GSoC#58;//gmod.org/wiki/GSoC"
title="GSoC">GSoC wiki page</a> and think about any projects that might
work well for a GSoC student, and add them to the wiki. If you are
interested in being a mentor, please contact us at <a
href="http://gmod.org/wiki/mailto:robin.haw@oicr.on.ca#58;robin.haw@oicr.on.ca"
class="external text" rel="nofollow">robin.haw@oicr.on.ca</a> <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">help@gmod.org</a>.</p>
<p>If you have any questions--e.g. what constitutes an appropriate
project; whether your idea is sufficiently GMOD-related--please feel
free to email <a
href="http://gmod.org/wiki/mailto:robin.haw@oicr.on.ca#58;robin.haw@oicr.on.ca"
class="external text" rel="nofollow">robin.haw@oicr.on.ca</a> and <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">help@gmod.org</a> for advice!</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2020/02/03</em>
</div>
</div>
<div class="rss">
<h3 id="call-for-pag-abstracts"><a
href="News/Call_for_PAG_Abstracts">Call for PAG Abstracts</a></h3>
<h2 id="call-for-pag-abstracts-1"><span id="Call_for_PAG_Abstracts"
class="mw-headline">Call for PAG Abstracts</span></h2>
<p>Time is short!</p>
<p>If you want to attend <a
href="http://gmod.org/wiki/http://www.intlpag.org/#58;//www.intlpag.org/"
class="external text" rel="nofollow">PAG</a> and would like to present
on a topic that would be of interest to the GMOD community, please send
an abstract or at least a descriptive title to <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">help@gmod.org</a>. Types of talks
typically include updates on GMOD software projects, usage stories for
successful sites, proposals for new GMOD projects and descriptions of
plugins for existing GMOD software projects like <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Tripal#58;//gmod.org/wiki/Tripal"
title="Tripal">Tripal</a>, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a> and <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Galaxy#58;//gmod.org/wiki/Galaxy"
title="Galaxy">Galaxy</a>.</p>
<p>Please consider giving a talk and sharing your experience and
ideas!</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2016/11/04</em>
</div>
</div>
<div class="rss">
<h3 id="new-gmod-server"><a href="News/New_GMOD_Server">New GMOD
Server</a></h3>
<h2 id="gmod.org-has-a-new-home"><span id="GMOD.org_has_a_new_home"
class="mw-headline">GMOD.org has a new home</span></h2>
<p>Due to a old server being retired, gmod.org has a new home. In the
course of migrating the server, we also had to update the version of
MediaWiki that is powering the site. If you notice any problems with
gmod.org, please send an email to help at gmod dot org to let us know
what's going on.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2016/09/29</em>
</div>
</div>
<div class="rss">
<h3 id="gmod-jbrowse-2016-survey"><a
href="News/GMOD-JBrowse_2016_Survey">GMOD-JBrowse 2016 Survey</a></h3>
<p>Hello Genome Informaticians,</p>
<p>The following survey is aimed at users (and potential users) of GMOD
genome databases, especially the JBrowse genome browser. It will
directly inform the priorities for renewal of the R01 that funds JBrowse
software development and the GMOD helpdesk.</p>
<p>We know surveys are thankless and dull. Your time in filling out this
one is GREATLY appreciated.</p>
<p><a
href="http://gmod.org/wiki/https://goo.gl/forms/1bKIuAMjGKrS0hUi1#58;//goo.gl/forms/1bKIuAMjGKrS0hUi1"
class="external free"
rel="nofollow">https://goo.gl/forms/1bKIuAMjGKrS0hUi1</a></p>
<p>Thanks &amp; best wishes,</p>
<p>The JBrowse team</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2016/09/23</em>
</div>
</div>
<div class="rss">
<h3 id="gcc2016"><a href="News/GCC2016">GCC2016</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/https://gcc2016.iu.edu/#58;//gcc2016.iu.edu/"
rel="nofollow" title="2016 Galaxy Community Conference">&lt;img
alt="2016 Galaxy Community Conference"
src="http://gmod.org/mediawiki/images/thumb/1/1c/GCC2016Logo.png/250px-GCC2016Logo.png"
width="250" height="167" srcset="/mediawiki/images/1/1c/GCC2016Logo.png
1.5x, /mediawiki/images/1/1c/GCC2016Logo.png 2x" /&gt;</a>
</div>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://gmod.org/wiki/Jun_2016_GMOD_Meeting#58;//gmod.org/wiki/Jun_2016_GMOD_Meeting"
title="Jun 2016 GMOD Meeting">&lt;img alt="GMOD2016ColorsBigLetters
300px.png"
src="http://gmod.org/mediawiki/images/thumb/e/e1/GMOD2016ColorsBigLetters_300px.png/150px-GMOD2016ColorsBigLetters_300px.png"
width="150" height="126"
srcset="/mediawiki/images/thumb/e/e1/GMOD2016ColorsBigLetters_300px.png/225px-GMOD2016ColorsBigLetters_300px.png
1.5x,
/mediawiki/images/thumb/e/e1/GMOD2016ColorsBigLetters_300px.png/300px-GMOD2016ColorsBigLetters_300px.png
2x" /&gt;</a>
</div>
<p><strong>The <a
href="http://gmod.org/wiki/https://gcc2016.iu.edu/#58;//gcc2016.iu.edu/"
class="external text" rel="nofollow">2016 Galaxy Community Conference
(GCC2016)</a> will be held June 25-29, at <a
href="http://gmod.org/wiki/https://gcc2016.iu.edu/location/#58;//gcc2016.iu.edu/location/"
class="external text" rel="nofollow">Indiana University</a> in
Bloomington, Indiana, United states, immediately before the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Jun_2016_GMOD_Meeting#58;//gmod.org/wiki/Jun_2016_GMOD_Meeting"
title="Jun 2016 GMOD Meeting">June 2016 GMOD Meeting</a>, also in
Bloomington.</strong> <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Galaxy#58;//gmod.org/wiki/Galaxy"
title="Galaxy">Galaxy</a> is a GMOD Component which interacts with many
other <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_Components#58;//gmod.org/wiki/GMOD_Components"
title="GMOD Components">GMOD Components</a>, including:</p>
<ul>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Tripal#58;//gmod.org/wiki/Tripal"
title="Tripal">Tripal</a>: A web front end for <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Chado#58;//gmod.org/wiki/Chado"
class="mw-redirect" title="Chado">Chado</a> databases. Galaxy is working
with the Tripal project to make Galaxy be Tripal's analysis engine.</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a>: A client-side genome browser and successor
to the venerable <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a>. <a
href="http://gmod.org/wiki/http://gcc2015.tsl.ac.uk/Lightning/#JBrowse_as_a_Galaxy_Tool#58;//gcc2015.tsl.ac.uk/Lightning/#JBrowse_as_a_Galaxy_Tool"
class="external text" rel="nofollow">JBrowse as a Galaxy Tool</a> was
presented by Eric Rasche at <a
href="http://gmod.org/wiki/http://gcc2015.tsl.ac.uk/#58;//gcc2015.tsl.ac.uk/"
class="external text" rel="nofollow">GCC2015</a>. Ian Holmes, the
JBrowse PI, has put JBrowse-Galaxy integration at "top of the list" for
JBrowse's infrastructure upcoming infrastructure work.</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER#58;//gmod.org/wiki/MAKER"
title="MAKER">MAKER</a>: A genome annotation pipeline that integrates
several gene annotation engines, and combines them to produce annotation
that is better than any individual tool produces. A <a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Documents/Posters?action=AttachFile&amp;do=view&amp;target=ISMB2014_Kandalaft_GenomeAnnotation.pdf#58;//wiki.galaxyproject.org/Documents/Posters?action=AttachFile&amp;do=view&amp;target=ISMB2014_Kandalaft_GenomeAnnotation.pdf"
class="external text" rel="nofollow">MAKER-Galaxy</a> by Agriculture and
Agri-Food Canada was presented at ISMB 2014.</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/InterMine#58;//gmod.org/wiki/InterMine"
title="InterMine">InterMine</a> and <a
href="http://gmod.org/wiki/http://gmod.org/wiki/BioMart#58;//gmod.org/wiki/BioMart"
title="BioMart">BioMart</a>: These are both popular data sources that
are integrated with Galaxy.</li>
</ul>
<p>Galaxy Community Conferences are an opportunity to participate in
presentations, discussions, poster sessions, lightning talks and
breakouts, all about high-throughput biology and the tools that support
it.  The conference also includes <a
href="http://gmod.org/wiki/https://gcc2016.iu.edu/training#58;//gcc2016.iu.edu/training"
class="external text" rel="nofollow">two days of training</a> offering
in-depth topic coverage across several concurrent sessions, and <a
href="http://gmod.org/wiki/https://gcc2016.iu.edu/hacks/#58;//gcc2016.iu.edu/hacks/"
class="external text" rel="nofollow">two days of hackathons</a>.</p>
<p><strong><a
href="http://gmod.org/wiki/https://gcc2016.iu.edu/abstracts/#58;//gcc2016.iu.edu/abstracts/"
class="external text" rel="nofollow">Oral presentation abstract
submission</a> closes April 8; <a
href="http://gmod.org/wiki/https://gcc2016.iu.edu/abstracts/#58;//gcc2016.iu.edu/abstracts/"
class="external text" rel="nofollow">poster and demo abstract
submission</a> close May 20; and <a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/News/GCC2016Scholarships#58;//wiki.galaxyproject.org/News/GCC2016Scholarships"
class="external text" rel="nofollow">scholarship applications</a> close
May 1.</strong></p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2016/04/04</em>
</div>
</div>
<div class="rss">
<h3 id="gmod-meeting"><a href="News/2016_GMOD_Meeting">2016 GMOD
Meeting</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://gmod.org/wiki/Jun_2016_GMOD_Meeting#58;//gmod.org/wiki/Jun_2016_GMOD_Meeting"
title="Jun 2016 GMOD Meeting">&lt;img alt="GMOD2016ColorsBigLetters
300px.png"
src="http://gmod.org/mediawiki/images/thumb/e/e1/GMOD2016ColorsBigLetters_300px.png/260px-GMOD2016ColorsBigLetters_300px.png"
width="260" height="218"
srcset="/mediawiki/images/e/e1/GMOD2016ColorsBigLetters_300px.png 1.5x,
/mediawiki/images/e/e1/GMOD2016ColorsBigLetters_300px.png 2x" /&gt;</a>
</div>
<p><strong>The <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Jun_2016_GMOD_Meeting#58;//gmod.org/wiki/Jun_2016_GMOD_Meeting"
title="Jun 2016 GMOD Meeting">next GMOD Community Meeting</a> will be
held at Indiana University in Bloomington, Indiana, United States, June
30-July 1, directly after the <a
href="http://gmod.org/wiki/https://gcc2016.iu.edu/#58;//gcc2016.iu.edu/"
class="external text" rel="nofollow">2016 Galaxy Community Conference
(GCC2016)</a></strong>. <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Meetings#58;//gmod.org/wiki/Meetings"
title="Meetings">GMOD Meetings</a> are a mix of <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_Membership#Users#58;//gmod.org/wiki/GMOD_Membership#Users"
title="GMOD Membership">user</a> and <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_Membership#Developers#58;//gmod.org/wiki/GMOD_Membership#Developers"
title="GMOD Membership">developer</a> presentations, and are a great
place to find out what is happening in the project, what's coming up,
and what others are doing.</p>
<p><br />
</p>
<p><strong><a
href="http://gmod.org/wiki/https://gmod2016.eventbrite.com#58;//gmod2016.eventbrite.com"
class="external text" rel="nofollow">Please register online at
Eventbrite by June 20th 2016</a>.</strong> Early bird registration ends
May 21.</p>
<p>For those who would like to present a talk or poster, the meeting
registration form includes a section for submitting the presentation
title and abstract.</p>
<p>If you have any suggestions or requests for the meeting, please
contact the <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">GMOD help desk</a>.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2016/04/04</em>
</div>
</div>
<div class="rss">
<h3 id="please-support-ecocyc"><a
href="News/Please_Support_EcoCyc">Please Support EcoCyc</a></h3>
<div class="floatright">
<a href="http://gmod.org/wiki/http://ecocyc.org#58;//ecocyc.org"
rel="nofollow">&lt;img alt="EcoCyc website"
src="http://gmod.org/mediawiki/images/4/44/BioCyc.gif" width="216"
height="60" /&gt;</a>
</div>
<p><a href="http://gmod.org/wiki/http://ecocyc.org#58;//ecocyc.org"
class="external text" rel="nofollow">EcoCyc, the <em>E. coli</em>
information resource</a> and one of the resources offered by the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Pathway_Tools#58;//gmod.org/wiki/Pathway_Tools"
title="Pathway Tools">Pathway Tools</a> group, is in need of letters of
support from the community after receiving a poor grant review, which
could result in a complete loss of funding on July 1st, 2014. If you are
a user of EcoCyc, please consider <a
href="http://gmod.org/wiki/http://brg.ai.sri.com/ptools/ecocyc-letters-of-support.shtml#58;//brg.ai.sri.com/ptools/ecocyc-letters-of-support.shtml"
class="external text" rel="nofollow">writing a short letter in support
of this vital resource</a>. The deadline for letters is <strong>May
26th, 2014</strong>.</p>
<p>From the <a
href="http://gmod.org/wiki/http://brg.ai.sri.com/ptools/ecocyc-letters-of-support.shtml#58;//brg.ai.sri.com/ptools/ecocyc-letters-of-support.shtml"
class="external text" rel="nofollow">Pathway Tools website</a></p>
<blockquote>
<p><strong>ECOCYC FUNDING CRISIS -- DEADLINE MAY 26</strong></p>
<p>EcoCyc received a very unfavorable grant review in February 2014. We
are in discussions with the NIH to resolve this situation.</p>
<p>EcoCyc's usage has steadily increased. We made very strong progress
on our challenging aims from the current grant period, and the project
has produced many publications. EcoCyc received excellent reviews on
previous grant applications. Furthermore, the needs of the prokaryotic
research community for the content and software tools offered by EcoCyc
have never been higher.</p>
<p>In the worst case, we will lose all funding on July 1, 2014 and be
forced to re-apply. Even in the best case, we may receive a crippling
funding cut that causes us to fall behind in its manual literature
curation effort, and requires us to lay off experienced curation staff
until funding can be obtained.</p>
<p>These events could seriously undermine EcoCyc, end the project
altogether, or force us to begin charging usage fees.</p>
<p>To maintain EcoCyc as the free, up to date, and high-quality resource
that you depend on, please tell the NIH what EcoCyc means to your
research. Please click the button below to submit a PDF letter of
support on institutional letterhead, or a short support statement,
explaining the importance of EcoCyc.</p>
<p>We ask all regular users to submit; a short statement will take less
than two minutes of your time. Students and post-docs, please ask your
lab head to submit in addition to your submission.</p>
</blockquote>
<p>More information on what to write and where to submit letters is
available at <a
href="http://gmod.org/wiki/http://brg.ai.sri.com/ptools/ecocyc-letters-of-support.shtml#58;//brg.ai.sri.com/ptools/ecocyc-letters-of-support.shtml"
class="external text" rel="nofollow">the Pathway Tools website</a>.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014/05/21</em>
</div>
</div>
<div class="rss">
<h3 id="june-2014-webapollo-hackathon"><a
href="News/June_2014_WebApollo_Hackathon">June 2014 WebApollo
Hackathon</a></h3>
<p>Berkeley Bioinformatics Open-Source Projects (BBOP) invites you to
join us this summer for a <strong><a
href="http://gmod.org/wiki/http://genomearchitect.org/developers_hackathon_2014#58;//genomearchitect.org/developers_hackathon_2014"
class="external text" rel="nofollow">WebApollo Hackathon</a></strong> at
<a href="http://gmod.org/wiki/http://lbl.gov#58;//lbl.gov"
class="external text" rel="nofollow">Lawrence Berkeley National
Laboratory, California</a>.</p>
<p><strong>When</strong>: Monday, June 2 - 6, 2014</p>
<p><strong>Where</strong>: <a
href="http://gmod.org/wiki/http://www.lbl.gov/LBL-Work/lab-site-map.html#58;//www.lbl.gov/LBL-Work/lab-site-map.html"
class="external text" rel="nofollow">Lawrence Berkeley National
Laboratory. Building 74 (B74)</a>, Room 104. 1 Cyclotron Rd, Berkeley,
CA 94720</p>
<p><strong>What</strong>: five days of intensive, collaborative <a
href="http://gmod.org/wiki/http://gmod.org/wiki/WebApollo#58;//gmod.org/wiki/WebApollo"
title="WebApollo">WebApollo</a> development!</p>
<p>For five days developers will work on new features of interest to
their research communities, improve existing features, and collaborate
on developing features of interest to other colleagues.</p>
<p>Participants should:</p>
<ul>
<li>be able to set up their own WebApollo server before arriving in
Berkeley, including Postgres, Tomcat, etc.</li>
<li>code comfortably in JavaScript (client) and Java (server).</li>
</ul>
<p><strong>What will we do while we are there?</strong> You tell us!
Send your suggestions for feature development, your questions, and any
additional comments to apollo [dash] dev [at] lists [dot] lbl [dot]
gov</p>
<p>Registration is free of charge, but tickets are required. <a
href="http://gmod.org/wiki/https://www.eventbrite.com/e/web-apollo-developers-hackathon-tickets-11263235641#58;//www.eventbrite.com/e/web-apollo-developers-hackathon-tickets-11263235641"
class="external text" rel="nofollow">Register online!</a></p>
<p>Participants are responsible for arranging travel and accommodation
on their own.</p>
<p>More details: <a
href="http://gmod.org/wiki/http://genomearchitect.org/developers_hackathon_2014#58;//genomearchitect.org/developers_hackathon_2014"
class="external text" rel="nofollow">WebApollo Hackathon
information</a></p>
<p><em>via Monica Munoz-Torres</em></p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014-04-22</em>
</div>
</div>
<div class="rss">
<h3 id="applications-open-for-gmod-online-training"><a
href="News/Applications_Open_for_GMOD_Online_Training">Applications Open
for GMOD Online Training</a></h3>
<p>GMOD will be holding its first online training course for those
interested in the set up and use of GMOD components.</p>
<p>The course will be held from <strong>Monday 19th May to Friday 23rd
May 2014</strong>, and will cover core GMOD software, including <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a> and <a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a>, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Galaxy#58;//gmod.org/wiki/Galaxy"
title="Galaxy">Galaxy</a>, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER#58;//gmod.org/wiki/MAKER"
title="MAKER">MAKER</a>, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Tripal#58;//gmod.org/wiki/Tripal"
title="Tripal">Tripal</a>, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/WebApollo#58;//gmod.org/wiki/WebApollo"
title="WebApollo">WebApollo</a>, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Canto#58;//gmod.org/wiki/Canto"
title="Canto">Canto</a>, and the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Chado#58;//gmod.org/wiki/Chado"
class="mw-redirect" title="Chado">Chado</a> database.</p>
<p>If you are interested in attending, please see <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_Online_Training_2014#58;//gmod.org/wiki/GMOD_Online_Training_2014"
title="GMOD Online Training 2014">GMOD Online Training 2014</a> for more
information and to submit your application.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014-03-31</em>
</div>
</div>
<div class="rss">
<h3 id="canto-workshop-at-biocuration-2014"><a
href="News/Canto_Workshop_at_Biocuration_2014">Canto Workshop at
Biocuration 2014</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://gmod.org/wiki/Canto#58;//gmod.org/wiki/Canto"
title="Canto">&lt;img alt="CantoTextLogo.png"
src="http://gmod.org/mediawiki/images/thumb/7/7e/CantoTextLogo.png/240px-CantoTextLogo.png"
width="240" height="59"
srcset="/mediawiki/images/thumb/7/7e/CantoTextLogo.png/360px-CantoTextLogo.png
1.5x,
/mediawiki/images/thumb/7/7e/CantoTextLogo.png/480px-CantoTextLogo.png
2x" /&gt;</a>
</div>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://biocuration2014.events.oicr.on.ca/#58;//biocuration2014.events.oicr.on.ca/"
rel="nofollow">&lt;img alt="ISBLogo.jpg"
src="http://gmod.org/mediawiki/images/8/8d/ISBLogo.jpg" width="240"
height="71" /&gt;</a>
</div>
<p>GMOD will be running a workshop at <a
href="http://gmod.org/wiki/http://biocuration2014.events.oicr.on.ca/#58;//biocuration2014.events.oicr.on.ca/"
class="external text" rel="nofollow">Biocuration 2014</a> to demonstrate
the use of <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Canto#58;//gmod.org/wiki/Canto"
title="Canto">Canto</a>, on Wednesday 9th April in the afternoon. Canto
is a literature curation tool that allows users to create functional
annotations for genes and gene products using <a
href="http://gmod.org/wiki/http://www.obofoundry.org/#58;//www.obofoundry.org/"
class="external text" rel="nofollow">OBO (ontology) terms</a>. Canto
will soon be added to <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Cloud#58;//gmod.org/wiki/Cloud"
title="Cloud">GMOD in the Cloud</a>, and this workshop will show
participants how to get a GMOD in the Cloud instance up and running--it
takes less than ten minutes!--and how to use Canto for literature
curation.</p>
<p>We will have more information closer to the time.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014/03/25</em>
</div>
</div>
<div class="rss">
<h3 id="tripal-2.0a-released"><a href="News/Tripal_2.0a_released">Tripal
2.0a released</a></h3>
<p>The Tripal Development Team is pleased to announce an alpha release
of Tripal 2.0 for Drupal 7. This release is expected to have bugs and
there is some functionality still under development. However, this
release is made to help early adopters of Tripal for Drupal 7. Reports
of bugs or other issues are highly welcomed. Below are available
resources for Tripal 2.0a</p>
<pre><code>   Download Instructions
   New Functionality
   Installation and Tutorial
   Tripal 2.0 API site
   Tripal Mailing List
   Tripal Developer&#39;s Mailing List</code></pre>
<p>The Installation tutorial is still under development but should have
enough information for complete installation of Tripal v2.0a as well as
loading of organisms and features.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014/02/27</em>
</div>
</div>
<div class="rss">
<h3 id="precompiled-ontologies-in-chado"><a
href="News/Precompiled_Ontologies_in_Chado">Precompiled Ontologies in
Chado</a></h3>
<p>Eric Rasche and the <a
href="http://gmod.org/wiki/https://cpt.tamu.edu/#58;//cpt.tamu.edu/"
class="external text" rel="nofollow">Center for Phage Technology at
Texas A&amp;M University</a> are making <a
href="http://gmod.org/wiki/https://cpt.tamu.edu/~docs/chado/#58;//cpt.tamu.edu/~docs/chado/"
class="external text" rel="nofollow">Chado database dumps of precompiled
ontologies publicly available</a> to save other Chado users the time and
hassle of downloading and compiling the ontologies themselves.</p>
<p>Go to the <a
href="http://gmod.org/wiki/https://cpt.tamu.edu/~docs/chado/#58;//cpt.tamu.edu/~docs/chado/"
class="external text" rel="nofollow">download site</a>.</p>
<p>The ontologies currently available are:</p>
<ul>
<li>Chado Feature Properties</li>
<li>Gene Ontology</li>
<li>Plant Ontology</li>
<li>Relationship Ontology</li>
<li>Sequence Ontology</li>
</ul>
<p>These are updated weekly; the workflow is as follows:</p>
<ul>
<li>clone Chado from SVN</li>
<li>build</li>
<li>load ontologies</li>
<li>dump database as SQL</li>
<li>upload to a publicly accessible webserver</li>
</ul>
<p>Please <a
href="http://gmod.org/wiki/mailto:rasche.eric@yandex.ru#58;rasche.eric@yandex.ru"
class="external text" rel="nofollow">contact Eric</a> if you are
interested in having other ontologies added to the dumps, other builds
with different (sub)sets of ontologies, or archived copies of schemas
over time.</p>
<p>Note that none of the Chado-related scripts are installed, and the
GMOD conf files are not created in GMOD_ROOT. For remote access (e.g.,
via Artemis), and tools that do not make use of GMOD_ROOT locally, this
is not a problem.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014/02/20</em>
</div>
</div>
<div class="rss">
<h3 id="gcc2014-registration-is-open"><a
href="News/GCC2014_Registration_is_Open">GCC2014 Registration is
Open</a></h3>
<div class="floatleft">
<a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2014#58;//wiki.galaxyproject.org/Events/GCC2014"
rel="nofollow"
title="2014 Galaxy Community Conference (GCC2014)">&lt;img alt="2014
Galaxy Community Conference (GCC2014)"
src="http://gmod.org/mediawiki/images/thumb/a/a4/GCC2014Logo.png/300px-GCC2014Logo.png"
width="300" height="126"
srcset="/mediawiki/images/thumb/a/a4/GCC2014Logo.png/450px-GCC2014Logo.png
1.5x, /mediawiki/images/thumb/a/a4/GCC2014Logo.png/600px-GCC2014Logo.png
2x" /&gt;</a>
</div>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2014/TrainingDay#58;//wiki.galaxyproject.org/Events/GCC2014/TrainingDay"
rel="nofollow" title="GCC2014 Training Day">&lt;img alt="GCC2014
Training Day"
src="http://gmod.org/mediawiki/images/thumb/4/43/GCC2014TDLogo.png/80px-GCC2014TDLogo.png"
width="80" height="120"
srcset="/mediawiki/images/thumb/4/43/GCC2014TDLogo.png/120px-GCC2014TDLogo.png
1.5x,
/mediawiki/images/thumb/4/43/GCC2014TDLogo.png/160px-GCC2014TDLogo.png
2x" /&gt;</a>
</div>
<p><strong>We are pleased to announce that <a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GCC2014/Register#58;//wiki.galaxyproject.org/Events/GCC2014/Register"
class="external text" rel="nofollow">Early Registration</a> and <a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GCC2014/Abstracts#58;//wiki.galaxyproject.org/Events/GCC2014/Abstracts"
class="external text" rel="nofollow">Talk and Poster Abstract
Submission</a> are now open for the <a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GCC2014#58;//wiki.galaxyproject.org/Events/GCC2014"
class="external text" rel="nofollow">2014 Galaxy Community Conference
(GCC2014)</a>.</strong></p>
<p><a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GCC2014#58;//wiki.galaxyproject.org/Events/GCC2014"
class="external text" rel="nofollow">GCC2014</a> will be held at the <a
href="http://gmod.org/wiki/http://webapps.jhu.edu/jhuniverse/information_about_hopkins/campuses/homewood_campus/#58;//webapps.jhu.edu/jhuniverse/information_about_hopkins/campuses/homewood_campus/"
class="external text" rel="nofollow">Homewood Campus</a> of <a
href="http://gmod.org/wiki/http://jhu.edu#58;//jhu.edu"
class="external text" rel="nofollow">Johns Hopkins University</a>, in <a
href="http://gmod.org/wiki/http://visitors.baltimorecity.gov/#58;//visitors.baltimorecity.gov/"
class="external text" rel="nofollow">Baltimore, Maryland</a>, United
States, from June 30 through July 2, 2014. <a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GCC2014#58;//wiki.galaxyproject.org/Events/GCC2014"
class="external text" rel="nofollow">GCC2014</a> starts with a <a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GCC2014/TrainingDay#58;//wiki.galaxyproject.org/Events/GCC2014/TrainingDay"
class="external text" rel="nofollow">Training Day</a> featuring
<em>five</em> parallel tracks, each with three, two and half hour long
workshops. There are 13 different topics spanning the full Galactic
spectrum of topics. <a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GCC2014/TrainingDay#58;//wiki.galaxyproject.org/Events/GCC2014/TrainingDay"
class="external text" rel="nofollow">Take a look!</a></p>
<p><strong><a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GCC2014/Register#58;//wiki.galaxyproject.org/Events/GCC2014/Register"
class="external text" rel="nofollow">Early registration</a></strong> is
now open. Register early and <em>avoid paying 70% more for regular
registration costs.</em>  Early registration is very affordable, with
combined registration (<a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GCC2014/TrainingDay#58;//wiki.galaxyproject.org/Events/GCC2014/TrainingDay"
class="external text" rel="nofollow">Training Day</a> + main meeting)
starting at $140 for post-docs and students. Registration is capped this
year at 250 participants, <em>and we expect to hit that limit</em>.
Registering early assures you a place at the conference and also a spot
in the Training Day workshops you want to attend.</p>
<p>You can also book affordable conference housing at the same time you
register. See the conference <a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GCC2014/Logistics#58;//wiki.galaxyproject.org/Events/GCC2014/Logistics"
class="external text" rel="nofollow">Logistics page</a> for details on
this and other housing options.</p>
<p><strong><a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GCC2014/Abstracts#58;//wiki.galaxyproject.org/Events/GCC2014/Abstracts"
class="external text" rel="nofollow">Abstract submission</a></strong>
for both oral presentations and posters is also open.  Abstract
submission for oral presentations closes April 4, while poster
submission closes April 25. Poster authors will be notified of
acceptance status within two weeks of submission, while presentation
authors will be notified no later than May2.  <em>Please consider
presenting your work. If you are dealing with big biological data, then
this meeting wants to hear about your work.</em></p>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://www.gigasciencejournal.com/#58;//www.gigasciencejournal.com/"
rel="nofollow" title="GigaScience Journal">&lt;img alt="GigaScience
Journal"
src="http://gmod.org/mediawiki/images/thumb/c/c5/GigaScienceLogo250.png/120px-GigaScienceLogo250.png"
width="120" height="39"
srcset="/mediawiki/images/thumb/c/c5/GigaScienceLogo250.png/180px-GigaScienceLogo250.png
1.5x,
/mediawiki/images/thumb/c/c5/GigaScienceLogo250.png/240px-GigaScienceLogo250.png
2x" /&gt;</a>
</div>
<p>The <em><a
href="http://gmod.org/wiki/http://www.gigasciencejournal.com/#58;//www.gigasciencejournal.com/"
class="external text" rel="nofollow">GigaScience</a></em> "Galaxy: Data
Intensive and Reproducible Research" series announced for the last
conference has published its <a
href="http://gmod.org/wiki/http://www.gigasciencejournal.com/series/Galaxy#58;//www.gigasciencejournal.com/series/Galaxy"
class="external text" rel="nofollow">first papers</a>, <em>and is
continuing to take submissions for this year's meeting and beyond</em>.
BGI is also continuing to cover the article processing charges until the
end of the year, and for more information see their <a
href="http://gmod.org/wiki/http://blogs.biomedcentral.com/gigablog/2014/02/06/rewarding-reproducibility-first-papers-in-our-galaxy-series-utilizing-our-gigagalaxy-platform/#58;//blogs.biomedcentral.com/gigablog/2014/02/06/rewarding-reproducibility-first-papers-in-our-galaxy-series-utilizing-our-gigagalaxy-platform/"
class="external text" rel="nofollow">latest update</a>.</p>
<p>Thanks, and hope to see you in Baltimore!</p>
<p>The http:<a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GCC2014/Organizers#58;//wiki.galaxyproject.org/Events/GCC2014/Organizers"
class="external text" rel="nofollow">GCC2014 Organizing
Committee</a></p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014/02/14</em>
</div>
</div>
<div class="rss">
<h3 id="gmod-paper-cuts-feb-10th-2014"><a
href="News/GMOD_Paper_Cuts,_Feb_10th,_2014">GMOD Paper Cuts, Feb 10th,
2014</a></h3>
<p>__NOFACTBOX__</p>
<p><strong>GMOD Paper Cuts is a periodic selection of choice cuts from
the scientific literature featuring interesting, exciting, or otherwise
eye-catching GMOD-related work.</strong></p>
<p>If you would like a paper to appear in GMOD Paper Cuts, please email
the details to <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">the GMOD helpdesk</a>. Ideally the
paper should be in an open-access publication so that anyone can read
it.</p>
<p>For more GMOD and GMOD-related papers, and to contribute your own
GMOD-related publications, <a
href="http://gmod.org/wiki/http://mnd.ly/WwRe8F#58;//mnd.ly/WwRe8F"
class="external text" rel="nofollow">join our Mendeley group</a>.</p>
<p><br />
</p>
<p><strong>Finding the missing honey bee genes: lessons learned from a
genome upgrade</strong> <sup><span>[1]</span></sup></p>
<blockquote>
<p>The first generation of genome sequence assemblies and annotations
have had a significant impact upon our understanding of the biology of
the sequenced species, the phylogenetic relationships among species, the
study of populations within and across species, and have informed the
biology of humans. As only a few Metazoan genomes are approaching
finished quality (human, mouse, fly and worm), there is room for
improvement of most genome assemblies. The honey bee (Apis mellifera)
genome, published in 2006, was noted for its bimodal GC content
distribution that affected the quality of the assembly in some regions
and for fewer genes in the initial gene set (OGSv1.0) compared to what
would be expected based on other sequenced insect genomes.</p>
<p>Here, we report an improved <a
href="http://hymenopteragenome.org:8080/Amel_4.5/jbrowse/#58;//hymenopteragenome.org:8080/Amel_4.5/jbrowse/"
class="external text" rel="nofollow">honey bee genome assembly
(Amel_4.5)</a> with a new gene annotation set (OGSv3.2), and show that
the honey bee genome contains a number of genes similar to that of other
insect genomes, contrary to what was suggested in OGSv1.0. The new
genome assembly is more contiguous and complete and the new gene set
includes ~5000 more protein-coding genes, 50% more than previously
reported. About 1/6 of the additional genes were due to improvements to
the assembly, and the remaining were inferred based on new RNAseq and
protein data.</p>
<p>Lessons learned from this genome upgrade have important implications
for future genome sequencing projects. Furthermore, the improvements
significantly enhance genomic resources for the honey bee, a key model
for social behavior and essential to global ecology through
pollination.</p>
</blockquote>
<p>Interesting findings from the new assembly of the honey bee genome,
including many more genes than were found in the initial assembly. The
<a
href="http://gmod.org/wiki/http://hymenopteragenome.org#58;//hymenopteragenome.org"
class="external text" rel="nofollow">Hymenoptera Genome Database</a>
uses numerous GMOD resources, including <a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER#58;//gmod.org/wiki/MAKER"
title="MAKER">MAKER</a> for automated genome annotation, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a> and <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a> for sequence browsing, and <a
href="http://gmod.org/wiki/http://gmod.org/wiki/WebApollo#58;//gmod.org/wiki/WebApollo"
title="WebApollo">WebApollo</a> for community genome annotation.</p>
<p><br />
</p>
<p><strong>Highly Specific and Efficient CRISPR/Cas9-Catalyzed
Homology-Directed Repair in Drosophila</strong>
<sup><span>[2]</span></sup></p>
<blockquote>
<p>We and others recently demonstrated that the readily programmable
CRISPR/Cas9 system can be used to edit the Drosophila genome. However,
most applications to date have relied on aberrant DNA repair to
stochastically generate frame-shifting indels and adoption has been
limited by a lack of tools for efficient identification of targeted
events. Here we report optimized tools and techniques for expanded
application of the CRISPR/Cas9 system in Drosophila through
homology-directed repair (HDR) with double-stranded DNA (dsDNA) donor
templates that facilitate complex genome engineering through the precise
incorporation of large DNA sequences including screenable markers. Using
these donors, we demonstrate the replacement of a gene with exogenous
sequences and the generation of a conditional allele. To optimize
efficiency and specificity, we generated transgenic flies that express
Cas9 in the germline, and directly compared HDR and off-target cleavage
rates of different approaches for delivering CRISPR components. We also
investigated HDR efficiency in a mutant background previously
demonstrated to bias DNA repair towards HDR. Finally, we developed a
web-based tool that identifies CRISPR target sites and evaluates their
potential for off-target cleavage using empirically rooted rules.
Overall, we have found that injection of a dsDNA donor and guide
RNA-encoding plasmids into vasa-Cas9 flies yields the highest efficiency
HDR, and that target sites can be selected to avoid off-target
mutations. Efficient and specific CRISPR/Cas9-mediated HDR opens the
door to a broad array of complex genome modifications and greatly
expands the utility of CRISPR technology for Drosophila research.</p>
</blockquote>
<p><strong>CRISPR</strong> is one of the most exciting recent
technological advancements of the past couple of years. This paper
reports new techniques and tools for using the CRISPR/Cas9 system for
complex genome engineering. For more information, see the <a
href="http://gmod.org/wiki/http://flycrispr.molbio.wisc.edu/#58;//flycrispr.molbio.wisc.edu/"
class="external text" rel="nofollow">flyCRISPR website</a>.</p>
<p><br />
</p>
<p><strong>Analysis of Global Gene Expression in <em>Brachypodium
distachyon</em> Reveals Extensive Network Plasticity in Response to
Abiotic Stress</strong> <sup><span>[3]</span></sup></p>
<blockquote>
<p><em>Brachypodium distachyon</em> is a close relative of many
important cereal crops. Abiotic stress tolerance has a significant
impact on productivity of agriculturally important food and feedstock
crops. Analysis of the transcriptome of Brachypodium after chilling,
high-salinity, drought, and heat stresses revealed diverse differential
expression of many transcripts. Weighted Gene Co-Expression Network
Analysis revealed 22 distinct gene modules with specific profiles of
expression under each stress. Promoter analysis implicated short DNA
sequences directly upstream of module members in the regulation of 21 of
22 modules. Functional analysis of module members revealed enrichment in
functional terms for 10 of 22 network modules. Analysis of
condition-specific correlations between differentially expressed gene
pairs revealed extensive plasticity in the expression relationships of
gene pairs. Photosynthesis, cell cycle, and cell wall expression modules
were down-regulated by all abiotic stresses. Modules which were
up-regulated by each abiotic stress fell into diverse and unique gene
ontology GO categories. This study provides genomics resources and
improves our understanding of abiotic stress responses of
Brachypodium.</p>
</blockquote>
<p>Check out the JBrowse-powered <a
href="http://gmod.org/wiki/http://jbrowse.brachypodium.org#58;//jbrowse.brachypodium.org"
class="external text" rel="nofollow">Brachypodium web genome browser</a>
and other resources on the new <a
href="http://gmod.org/wiki/http://brachypodium.org#58;//brachypodium.org"
class="external text" rel="nofollow">Brachypodium website</a>!</p>
<p><br />
</p>
<p><strong>Analyses of Hypomethylated Oil Palm Gene
Space</strong><sup><span>[4]</span></sup></p>
<blockquote>
<p>Demand for palm oil has been increasing by an average of ~8% the past
decade and currently accounts for about 59% of the world's vegetable oil
market. This drives the need to increase palm oil production.
Nevertheless, due to the increasing need for sustainable production, it
is imperative to increase productivity rather than the area cultivated.
Studies on the oil palm genome are essential to help identify genes or
markers that are associated with important processes or traits, such as
flowering, yield and disease resistance. To achieve this, 294,115 and
150,744 sequences from the hypomethylated or gene-rich regions of Elaeis
guineensis and E. oleifera genome were sequenced and assembled into
contigs. An additional 16,427 shot-gun sequences and 176 bacterial
artificial chromosomes (BAC) were also generated to check the quality of
libraries constructed. Comparison of these sequences revealed that
although the methylation-filtered libraries were sequenced at low
coverage, they still tagged at least 66% of the RefSeq supported genes
in the BAC and had a filtration power of at least 2.0. A total 33,752
microsatellites and 40,820 high-quality single nucleotide polymorphism
(SNP) markers were identified. These represent the most comprehensive
collection of microsatellites and SNPs to date and would be an important
resource for genetic mapping and association studies. The gene models
predicted from the assembled contigs were mined for genes of interest,
and 242, 65 and 14 oil palm transcription factors, resistance genes and
miRNAs were identified respectively. Examples of the transcriptional
factors tagged include those associated with floral development and
tissue culture, such as homeodomain proteins, MADS, Squamosa and
Apetala2. The E. guineensis and E. oleifera hypomethylated sequences
provide an important resource to understand the molecular mechanisms
associated with important agronomic traits in oil palm.</p>
</blockquote>
<p>The newly-sequenced <a
href="http://gmod.org/wiki/http://genomsawit.mpob.gov.my/genomsawit/#58;//genomsawit.mpob.gov.my/genomsawit/"
class="external text" rel="nofollow">oil palm genome</a> used the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER#58;//gmod.org/wiki/MAKER"
title="MAKER">MAKER</a> automated annotation pipeline. The oil palm is
one of a number of genomics projects taking off in Malaysia at the
moment. Perfect timing for a <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_Malaysia_2014#58;//gmod.org/wiki/GMOD_Malaysia_2014"
title="GMOD Malaysia 2014">GMOD workshop</a>!</p>
<p><br />
</p>
<p><strong>Production of a reference transcriptome and transcriptomic
database (<a
href="http://gmod.org/wiki/http://cnidarians.bu.edu/EdwardBase/cgi-bin/index.cgi#58;//cnidarians.bu.edu/EdwardBase/cgi-bin/index.cgi"
class="external text" rel="nofollow">EdwardsiellaBase</a>) for the lined
sea anemone, <em>Edwardsiella lineata</em>, a parasitic
cnidarian</strong> <sup><span>[5]</span></sup></p>
<blockquote>
<p>The lined sea anemone <em>Edwardsiella lineata</em> is an informative
model system for evolutionary-developmental studies of parasitism. In
this species, it is possible to compare alternate developmental pathways
leading from a larva to either a free-living polyp or a vermiform
parasite that inhabits the mesoglea of a ctenophore host. Additionally,
E. lineata is confamilial with the model cnidarian <em>Nematostella
vectensis</em>, providing an opportunity for comparative genomic,
molecular and organismal studies.</p>
<p>[...]</p>
<p>The transcriptomic data and database described here provide a
platform for studying the evolutionary developmental genomics of a
derived parasitic life cycle. In addition, these data from E. lineata
will aid in the interpretation of evolutionary novelties in gene
sequence or structure that have been reported for the model cnidarian
<em>N. vectensis</em> (e.g., the split NF-κB locus). Finally, we include
custom computational tools to facilitate the annotation of a
transcriptome based on high-throughput sequencing data obtained from a
“non-model system.”</p>
</blockquote>
<p>Information and resources for the newly-sequenced cnidarian <em>E.
lineata</em>; all genomic data is publicly available at <a
href="http://gmod.org/wiki/http://cnidarians.bu.edu/EdwardBase/cgi-bin/index.cgi#58;//cnidarians.bu.edu/EdwardBase/cgi-bin/index.cgi"
class="external text" rel="nofollow">EdwardsiellaBase</a>, and can be
searched according to contig ID, gene ontology, protein family motif
(Pfam), enzyme commission number, and BLAST. The alignment of the raw
reads to the contigs can also be visualized via <a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a>.</p>
<p><br />
</p>
<p>Happy reading!</p>
<p><br />
</p>
<ol>
<li><span id="cite_note-DOI:10.1186.2F1471-2164-15-86"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1186.2F1471-2164-15-86</code></strong></span></li>
<li><span id="cite_note-DOI:10.1534.2Fgenetics.113.160713"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named
<code>DOI:10.1534.2Fgenetics.113.160713</code></strong></span></li>
<li><span id="cite_note-DOI:10.1371.2Fjournal.pone.0087499"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named
<code>DOI:10.1371.2Fjournal.pone.0087499</code></strong></span></li>
<li><span id="cite_note-DOI:10.1371.2Fjournal.pone.0086728"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named
<code>DOI:10.1371.2Fjournal.pone.0086728</code></strong></span></li>
<li><span id="cite_note-DOI:10.1186.2F1471-2164-15-71"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1186.2F1471-2164-15-71</code></strong></span></li>
</ol>
<div class="smallprint">
Disclaimer: the papers included in this feature are for your
entertainment and edification only. Inclusion does not imply an
endorsement of the material or any association between the authors and
the GMOD project.
</div>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014/02/10</em>
</div>
</div>
<div class="rss">
<h3 id="galaxy-australasia-workshop-2014"><a
href="News/Galaxy_Australasia_Workshop_2014">Galaxy Australasia Workshop
2014</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GAW2014#58;//wiki.galaxyproject.org/Events/GAW2014"
rel="nofollow"
title="1st Galaxy Australasia Workshop 2014 (GAW 2014)">&lt;img alt="1st
Galaxy Australasia Workshop 2014 (GAW 2014)"
src="http://gmod.org/mediawiki/images/thumb/1/16/GAW2014Logo.png/200px-GAW2014Logo.png"
width="200" height="96"
srcset="/mediawiki/images/thumb/1/16/GAW2014Logo.png/300px-GAW2014Logo.png
1.5x, /mediawiki/images/thumb/1/16/GAW2014Logo.png/400px-GAW2014Logo.png
2x" /&gt;</a>
</div>
<p><strong>The <a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GAW2014#58;//wiki.galaxyproject.org/Events/GAW2014"
class="external text" rel="nofollow">1st Galaxy Australasia Workshop
2014 (GAW 2014)</a> will be held in Melbourne, Australia on 24 and 25th
March 2014.</strong></p>
<p>The <a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GAW2014#58;//wiki.galaxyproject.org/Events/GAW2014"
class="external text" rel="nofollow">Galaxy Australasia Workshop</a> is
a great opportunity to participate in two full days of presentations,
discussions, poster sessions, keynotes and lightning talks, all about
ways of using Galaxy for high-throughput biology, imaging and other
scientific applications. The workshop will also include Training
Sessions taught by Galaxy developers and master users. GAW 2014 will run
24 and 25th March, immediately preceding <a
href="http://gmod.org/wiki/http://wp.csiro.au/css/#58;//wp.csiro.au/css/"
class="external text" rel="nofollow">Computational and Simulation
Sciences and eResearch</a> in Melbourne.</p>
<p>GAW 2014 will also include poster session, keynote speakers.</p>
<p><strong>You should attend to:</strong></p>
<ul>
<li>Present your work!</li>
<li>Learn best practices for deploying Galaxy, defining and installing
resources, and managing and moving large datasets.</li>
<li>Network with others in the Galaxy community who are facing similar
challenges and using Galaxy and other tools to address them.</li>
<li>Learn what the Galaxy Project's plans are, and contribute to
Galaxy's future direction.</li>
<li>Learn
<ul>
<li>how to visualize your data in Galaxy and use visualization to guide
your analysis (visual analytics)</li>
<li>how to share, publish, and reuse your analyses with Galaxy</li>
<li>how to perform and enable your users to perform common, yet complex,
analyses using Galaxy</li>
<li>when and how to use Galaxy on the Cloud</li>
</ul></li>
</ul>
<p><strong>Topics will potentially include:</strong></p>
<ul>
<li>Image analysis and processing using Galaxy.</li>
<li>RNAseq/ChIPseq/Variant Calling/RNA Quality Control.</li>
<li>Galaxy on the Research Cloud.</li>
<li>CSIRO galaxy service - partnership between science and IT.</li>
<li>Identifying proteins from mass spec data with Galaxy.</li>
</ul>
<p><strong>Registration</strong></p>
<p><a
href="http://gmod.org/wiki/http://australianbioinformatics.net/gaw-2014#58;//australianbioinformatics.net/gaw-2014"
class="external text" rel="nofollow">Registration is open</a>.
Registration is also <em>free, but space is limited.</em></p>
<p><strong>Call For Abstracts</strong></p>
<p>Participants who wish to give presentations or present posters
(potentially with technical demonstrations) that showcase use of Galaxy
should <a
href="http://gmod.org/wiki/http://australianbioinformatics.net/gaw-2014#58;//australianbioinformatics.net/gaw-2014"
class="external text" rel="nofollow">submit an abstract and brief
one-paragraph bio</a> <strong>by February 15th, 2014</strong>.
Submitters will be notified by February 28th. Speakers, panelists, and
poster presenters will be selected by the program committee based on
relevance to symposium objectives and workshop balance.</p>
<p>Looking forward to seeing you all in Melbourne!</p>
<p><a
href="http://gmod.org/wiki/https://wiki.galaxyproject.org/Events/GAW2014#Organising#58;//wiki.galaxyproject.org/Events/GAW2014#Organising"
class="external text" rel="nofollow">Committee GAW 2014 Organising
Committee</a></p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on yyyy/mm/dd</em>
</div>
</div>
<div class="rss">
<h3 id="gmod-paper-cuts-jan-24th-2014"><a
href="News/GMOD_Paper_Cuts,_Jan_24th,_2014">GMOD Paper Cuts, Jan 24th,
2014</a></h3>
<p>After a break for the festive season, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/PAG#58;//gmod.org/wiki/PAG"
class="mw-redirect" title="PAG">PAG XXII</a>, and the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Jan_2014_GMOD_Meeting#58;//gmod.org/wiki/Jan_2014_GMOD_Meeting"
title="Jan 2014 GMOD Meeting">GMOD community meeting in San Diego</a>,
GMOD Paper Cuts is back with a selection of scholarly publications for
your reading pleasure.</p>
<p>__NOFACTBOX__</p>
<p><strong>GMOD Paper Cuts is a periodic selection of choice cuts from
the scientific literature featuring interesting, exciting, or otherwise
eye-catching GMOD-related work.</strong></p>
<p>If you would like a paper to appear in GMOD Paper Cuts, please email
the details to <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">the GMOD helpdesk</a>. Ideally the
paper should be in an open-access publication so that anyone can read
it.</p>
<p>For more GMOD and GMOD-related papers, and to contribute your own
GMOD-related publications, <a
href="http://gmod.org/wiki/http://mnd.ly/WwRe8F#58;//mnd.ly/WwRe8F"
class="external text" rel="nofollow">join our Mendeley group</a>.</p>
<p><br />
</p>
<p><strong>CoryneBase: <em>Corynebacterium</em> Genomic Resources and
Analysis Tools at Your Fingertips</strong>
<sup><span>[1]</span></sup></p>
<blockquote>
Corynebacteria are used for a wide variety of industrial purposes but
some species are associated with human diseases. With increasing number
of corynebacterial genomes having been sequenced, comparative analysis
of these strains may provide better understanding of their biology,
phylogeny, virulence and taxonomy that may lead to the discoveries of
beneficial industrial strains or contribute to better management of
diseases. [...] <a
href="http://gmod.org/wiki/http://corynebacterium.um.edu.my/#58;//corynebacterium.um.edu.my/"
class="external text" rel="nofollow">CoryneBase</a> offers the access of
a range of Corynebacterium genomic resources as well as analysis tools
for comparative genomics and pathogenomics.
</blockquote>
<p>A new bacterial resource that uses <a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a> for genome visualization. The justification
for using JBrowse is interesting:</p>
<p><em>We chose JBrowse for CoryneBase for the following main reasons:
(1) most of the traditional genome browsers, e.g., GBrowse, are
implemented using the Common Gateway Interface (CGI) protocol—the use of
CGI-based genome browsers will inadvertently incurs unnecessary delays
since the whole-genome browser page need to be reloaded when users
change how the data are displayed; and (2) with the advances in
next-generation sequencing technologies and bioinformatic tools, we
would expect more corynebacterial genomes will be sequenced and
annotated. Therefore a user-friendly genome browser that allows rapid
and seamless browsing of the huge genomic data will be a major advantage
to the research communities.</em></p>
<p><br />
</p>
<p><strong>The Genome Sequence of the Fungal Pathogen <em>Fusarium
virguliforme</em> That Causes Sudden Death Syndrome in Soybean</strong>
<sup><span>[2]</span></sup></p>
<blockquote>
<em>Fusarium virguliforme</em> causes sudden death syndrome (SDS) of
soybean, a disease of serious concern throughout most of the soybean
producing regions of the world. Despite the global importance, little is
known about the pathogenesis mechanisms of <em>F. virguliforme</em>.
Thus, we applied Next-Generation DNA Sequencing to reveal the draft
<em>F. virguliforme</em> genome sequence and identified putative
pathogenicity genes to facilitate discovering the mechanisms used by the
pathogen to cause this disease.
</blockquote>
<p>A new draft sequence of the fungal disease <em>Fusarium
virguliforme</em>. The genome sequence can be <a
href="http://gmod.org/wiki/http://fvgbrowse.agron.iastate.edu/#58;//fvgbrowse.agron.iastate.edu/"
class="external text" rel="nofollow">browsed online using
GBrowse</a>.</p>
<p><br />
</p>
<p><a
href="http://gmod.org/wiki/http://arxiv.org/abs/1401.2286#58;//arxiv.org/abs/1401.2286"
class="external text" rel="nofollow"><strong>Palaeosymbiosis revealed by
genomic fossils of Wolbachia in a strongyloidean
nematode</strong></a></p>
<p>Georgios Koutsovoulos, Benjamin Makepeace, Vincent N. Tanya, Mark
Blaxter; published on arxiv.org</p>
<blockquote>
<p>Wolbachia are common endosymbionts of terrestrial arthropods, and are
also found in nematodes, the animal-parasitic <em>filaria</em>, and the
plant-parasite <em>Radopholus similis</em>. Lateral transfer of
Wolbachia</p>
<p>DNA to the host genome is common. We generated a draft genome
sequence for the strongyloidean nematode parasite <em>Dictyocaulus
viviparus</em>, the cattle lungworm. In the assembly, we identified
nearly 1 Mb of sequence with similarity to Wolbachia. The fragments were
unlikely to derive from a live Wolbachia infection: most were short, and
the genes were disabled through inactivating mutations. Many fragments
were co-assembled with definitively nematode-derived sequence. We found
limited evidence of expression of the Wolbachia-derived genes. The
<em>D. viviparus</em> Wolbachia genes were most similar to filarial
strains, and strains from the host-promiscuous clade F. We conclude that
<em>D. viviparus</em> was infected by Wolbachia in the past. Genome
sequence based surveys are a powerful tool for revealing the genome
archaeology of infection and symbiosis.</p>
</blockquote>
<p>A fascinating study that uses genetic techniques to dig into the
history of a host-parasite relationship. The D. viviparus genome was
annotated using the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER#58;//gmod.org/wiki/MAKER"
title="MAKER">MAKER</a> pipeline.</p>
<p><br />
</p>
<p><strong>Genome of the human hookworm <em>Necator
americanus</em></strong> <sup><span>[3]</span></sup></p>
<blockquote>
<p>The hookworm <em>Necator americanus</em> is the predominant
soil-transmitted human parasite. Adult worms feed on blood in the small
intestine, causing iron-deficiency anemia, malnutrition, growth and
development stunting in children, and severe morbidity and mortality
during pregnancy in women. We report sequencing and assembly of the
<em>N. americanus</em> genome (244 Mb, 19,151 genes). Characterization
of this first hookworm genome sequence identified genes orchestrating
the hookworm's invasion of the human host, genes involved in blood
feeding and development, and genes encoding proteins that represent new
potential drug targets against hookworms. <em>N. americanus</em> has
undergone a considerable and unique expansion of immunomodulator
proteins, some of which we highlight as potential treatments against
inflammatory diseases. We also used a protein microarray to demonstrate
a postgenomic application of the hookworm genome sequence. This genome
provides an invaluable resource to boost ongoing efforts toward
fundamental and applied postgenomic research, including the development
of new methods to control hookworm and human immunological diseases.</p>
</blockquote>
<p>Another somewhat gruesome parasite, the human hookworm, which is
responsible for causing more disease than any other soil-based helminth.
The genome was annotated using <a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER#58;//gmod.org/wiki/MAKER"
title="MAKER">MAKER</a> and can be viewed online at <a
href="http://gmod.org/wiki/http://nematode.net#58;//nematode.net"
class="external text" rel="nofollow">Nematode.net</a> and <a
href="http://gmod.org/wiki/http://wormbase.org#58;//wormbase.org"
class="external text" rel="nofollow">Wormbase</a>.</p>
<p><br />
</p>
<p>Happy reading!</p>
<p><br />
</p>
<ol>
<li><span id="cite_note-DOI:10.1371.2Fjournal.pone.0086318"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named
<code>DOI:10.1371.2Fjournal.pone.0086318</code></strong></span></li>
<li><span id="cite_note-DOI:10.1371.2Fjournal.pone.0081832"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named
<code>DOI:10.1371.2Fjournal.pone.0081832</code></strong></span></li>
<li><span id="cite_note-DOI:10.1038.2Fng.2875"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1038.2Fng.2875</code></strong></span></li>
</ol>
<div class="smallprint">
Disclaimer: the papers included in this feature are for your
entertainment and edification only. Inclusion does not imply an
endorsement of the material or any association between the authors and
the GMOD project.
</div>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014/01/24</em>
</div>
</div>
<div class="rss">
<h3
id="postdoc-position-in-computational-and-plant-evolutionary-biology"><a
href="News/Postdoc_Position_in_Computational_and_Plant_Evolutionary_Biology">Postdoc
Position in Computational and Plant Evolutionary Biology</a></h3>
<p>The project involves construction and analysis of plant gene families
and phylogenies, particularly focusing on the large legume family, and
development of a corresponding database and on-line tools for accessing
and exploring gene families and associated annotations. Analysis of gene
families and genes with known functions for better understanding of key
events in evolution of selected plants and agronomic traits.</p>
<p>The primary research groups are located at <a
href="http://gmod.org/wiki/http://www.iastate.edu/‎#58;//www.iastate.edu/‎"
class="external text" rel="nofollow">Iowa State University (Ames,
IA)</a> and at <a
href="http://gmod.org/wiki/http://www.ncgr.org#58;//www.ncgr.org"
class="external text" rel="nofollow">National Center for Genome
Resources (Santa Fe, NM)</a>. Remote work may be considered. This is a
two-year position, with possibility for extension. Salary will be at
competitive federal rates (&gt;$50k). U.S. citizenship will not be a
requirement for the position.</p>
<p>Applicants must have received Ph.D. in computational biology,
bioinformatics, computer science, plant biology, genetics, or related
discipline involving significant computational experience, statistics,
and genetics.</p>
<p>If interested, please contact <a
href="http://gmod.org/wiki/mailto:steven.cannon@ars.usda.gov#58;steven.cannon@ars.usda.gov"
class="external text" rel="nofollow">Steven Cannon</a>.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014/1/24</em>
</div>
</div>
<div class="rss">
<h3 id="prospecting-for-proposals-for-gsoc-2014"><a
href="News/Prospecting_for_Proposals_for_GSoC_2014">Prospecting for
Proposals for GSoC 2014</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://gmod.org/wiki/GSoC#58;//gmod.org/wiki/GSoC"
title="GSoC">&lt;img alt="Landing-page-gsoc2014.png"
src="http://gmod.org/mediawiki/images/thumb/6/6c/Landing-page-gsoc2014.png/200px-Landing-page-gsoc2014.png"
width="200" height="180"
srcset="/mediawiki/images/thumb/6/6c/Landing-page-gsoc2014.png/300px-Landing-page-gsoc2014.png
1.5x, /mediawiki/images/6/6c/Landing-page-gsoc2014.png 2x" /&gt;</a>
</div>
<p>GMOD is participating in <a
href="http://gmod.org/wiki/http://google-melange.com#58;//google-melange.com"
class="external text" rel="nofollow">Google Summer of Code (GSoC)</a>
this year, and we are soliciting project ideas from people and groups
involved in the GMOD project.</p>
<p>GSoC is a global program run by Google to encourage students to get
real-world experience of participating in a software project; students
work on a project over the summer months and receive a stipend from
Google for participating. Thousands of organizations of many different
sizes take part in GSoC, with the common factor being that they must
produce open-source code.</p>
<p>Please take a look at the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GSoC#58;//gmod.org/wiki/GSoC"
title="GSoC">GSoC wiki page</a> and think about any projects that might
work well for a GSoC student, and add them to the wiki. If you are
interested in being a mentor, please contact us at <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">help@gmod.org</a>.</p>
<p>If you have any questions--e.g. what constitutes an appropriate
project; whether your idea is sufficiently GMOD-related--please feel
free to email <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">help@gmod.org</a> for advice!</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014/01/22</em>
</div>
</div>
<div class="rss">
<h3 id="biocuration-2014"><a href="News/Biocuration_2014">Biocuration
2014</a></h3>
<p><a
href="http://gmod.org/wiki/http://biocuration2014.events.oicr.on.ca/#58;//biocuration2014.events.oicr.on.ca/"
class="external text" rel="nofollow">7th International Biocuration
Conference</a></p>
<p><strong>April 6-9, 2014 -- Toronto, Canada</strong></p>
<p>Abstracts Due: February 10, 2014</p>
<p>Biocuration 2014 will provide a forum for trainees, biocurators,
investigators, clinicians and developers of biological databases to
discuss their work, promote collaboration and foster a sense of
community in this very active and growing area of research. Participants
from academia, government and industry interested in the methods and
tools employed in curation of biological and medical data are encouraged
to attend. While a number of speakers have been invited, please note
that the majority of oral presentations will be drawn from openly
submitted abstracts.</p>
<p>SCOPE:</p>
<p>The proposed 2014 conference sessions and workshops will address the
following challenges in biocuration:</p>
<ul>
<li>Automated Function Prediction (workshop)</li>
<li>Big Data Curation: Dealing with supplementary data (workshop)</li>
<li>Biocreative Text Mining (workshop)</li>
<li>Biological and Clinical Ontologies (session)</li>
<li>Clinical Annotations (session)</li>
<li>Data Integration and Data Sharing (session)</li>
<li>Functional Annotations (session)</li>
<li>Microbial Informatics (session)</li>
<li>Phenotype (workshop)</li>
<li>Social Tools for publishing and curation (workshop)</li>
</ul>
<p>INVITED SPEAKERS:</p>
<ul>
<li>Dr. Tim Hubbard, Wellcome Trust Sanger Institute</li>
<li>Dr. Suzanna Lewis, Lawrence Berkeley National Laboratory</li>
<li>Dr. Patricia Babbitt, California Institute for Quantitative
Biosciences (QB3)</li>
<li>Dr. Lincoln Stein, Ontario Institute for Cancer Research</li>
</ul>
<p>WEBSITE:</p>
<p><a
href="http://gmod.org/wiki/http://biocuration2014.events.oicr.on.ca/#58;//biocuration2014.events.oicr.on.ca/"
class="external text" rel="nofollow">7th International Biocuration
Conference</a></p>
<p><br />
</p>
<p>REGISTRATION:</p>
<p><a
href="http://gmod.org/wiki/http://biocuration2014.events.oicr.on.ca/biocuration-registration-and-abstract-submission#58;//biocuration2014.events.oicr.on.ca/biocuration-registration-and-abstract-submission"
class="external text" rel="nofollow">Online registration is now
open</a></p>
<p><br />
</p>
<p>ABSTRACT SUBMISSION:</p>
<p><a
href="http://gmod.org/wiki/https://www.easychair.org/conferences/?conf=isb2014#58;//www.easychair.org/conferences/?conf=isb2014"
class="external text" rel="nofollow">Submit your abstract for
presentation (poster or talks only)</a></p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014/01/17</em>
</div>
</div>
<div class="rss">
<h3 id="gmod-at-pag-2014"><a href="News/GMOD_At_PAG_2014">GMOD At PAG
2014</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://gmod.org/wiki/PAG#58;//gmod.org/wiki/PAG"
title="PAG">&lt;img alt="Pagxxii.png"
src="http://gmod.org/mediawiki/images/9/98/Pagxxii.png" width="157"
height="54" /&gt;</a>
</div>
<p>For those of you attending <a
href="http://gmod.org/wiki/http://intlpag.org/2014/#58;//intlpag.org/2014/"
class="external text" rel="nofollow">PAG XXII</a>, we have made up <a
href="http://gmod.org/wiki/http://gmod.org/wiki/PAG#58;//gmod.org/wiki/PAG"
class="mw-redirect" title="PAG">a "cheatsheet" of GMOD-related content
to check out</a>.</p>
<p>If you haven't yet signed up for the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Jan_2014_GMOD_Meeting#58;//gmod.org/wiki/Jan_2014_GMOD_Meeting"
title="Jan 2014 GMOD Meeting">Jan 2014 GMOD Meeting</a>, please do so
ASAP!</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014-01-08</em>
</div>
</div>
<div class="rss">
<h3 id="announcing-gmod-malaysia-2014"><a
href="News/Announcing_GMOD_Malaysia_2014">Announcing GMOD Malaysia
2014</a></h3>
<p>We are pleased to announce that GMOD will be running a three day
training course in Kuala Lumpur, Malaysia, on 26-28 February 2014. The
course will be run by experienced GMOD developers and tutors, and will
cover the following tools:</p>
<ul>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Chado#58;//gmod.org/wiki/Chado"
class="mw-redirect" title="Chado">Chado</a> database schema</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a> genome browser</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Galaxy#58;//gmod.org/wiki/Galaxy"
title="Galaxy">Galaxy</a> workflow generator</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Intermine#58;//gmod.org/wiki/Intermine"
class="mw-redirect" title="Intermine">Intermine</a> data warehouse</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a> genome browser</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER#58;//gmod.org/wiki/MAKER"
title="MAKER">MAKER</a> and MAKER-P genome annotation pipeline</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Tripal#58;//gmod.org/wiki/Tripal"
title="Tripal">Tripal</a> website/database interface</li>
</ul>
<p>Date: February 26-28, 2014</p>
<p>Venue: Kuala Lumpur, Malaysia.</p>
<p>Please check the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_Malaysia_2014#58;//gmod.org/wiki/GMOD_Malaysia_2014"
title="GMOD Malaysia 2014">GMOD Malaysia 2014</a> wiki page for updates,
or email the <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">GMOD help desk</a> for more
information.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/12/28</em>
</div>
</div>
<div class="rss">
<h3 id="maker-p-an-annotation-pipeline-for-plants"><a
href="News/MAKER-P:_an_annotation_pipeline_for_plants">MAKER-P: an
annotation pipeline for plants</a></h3>
<p>The popular <a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER#58;//gmod.org/wiki/MAKER"
title="MAKER">MAKER</a> annotation pipeline has been extended and
optimised to allow the annotation of plant species:</p>
<blockquote>
<p>Sequencing diverse plant species of evolutionary, agricultural, and
medicinal interest is becoming routine for even small groups; genome
annotation and analysis is much less so. The MAKER-P pipeline is
designed to make the annotation of novel plant genomes tractable for
small groups with limited bioinformatics experience and resources, and
faster and more transparent for large groups with more experience and
resources. The MAKER-P pipeline generates species-specific repeat
libraries, as well as structural annotations of protein coding genes,
non-coding RNAs, and pseudogenes.<br />
</p>
<p>—Michael Campbell, MAKER development team</p>
</blockquote>
<ul>
<li>Read the publication on MAKER-P: <a
href="http://gmod.org/wiki/http://dx.doi.org/​10.​1104/​pp.​113.​230144#58;//dx.doi.org/​10.​1104/​pp.​113.​230144"
class="external text" rel="nofollow">MAKER-P: a tool-kit for the rapid
creation, management, and quality control of plant genome
annotations</a></li>
</ul>
<ul>
<li>Download and install <a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER#58;//gmod.org/wiki/MAKER"
title="MAKER">MAKER and MAKER-P</a>.</li>
</ul>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/12/16</em>
</div>
</div>
<div class="rss">
<h3 id="gmod-paper-cuts-dec-6th-2013"><a
href="News/GMOD_Paper_Cuts,_Dec_6th,_2013">GMOD Paper Cuts, Dec 6th,
2013</a></h3>
<p>__NOFACTBOX__</p>
<p><strong>GMOD Paper Cuts is a periodic selection of choice cuts from
the scientific literature featuring interesting, exciting, or otherwise
eye-catching GMOD-related work.</strong></p>
<p>If you would like a paper to appear in GMOD Paper Cuts, please email
the details to <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">the GMOD helpdesk</a>. Ideally the
paper should be in an open-access publication so that anyone can read
it.</p>
<p>For more GMOD and GMOD-related papers, and to contribute your own
GMOD-related publications, <a
href="http://gmod.org/wiki/http://mnd.ly/WwRe8F#58;//mnd.ly/WwRe8F"
class="external text" rel="nofollow">join our Mendeley group</a>.</p>
<p><br />
</p>
<p><strong>The king cobra genome reveals dynamic gene evolution and
adaptation in the snake venom system</strong>
<sup><span>[1]</span></sup></p>
<blockquote>
Snakes are limbless predators, and many species use venom to help
overpower relatively large, agile prey. Snake venoms are complex protein
mixtures encoded by several multilocus gene families that function
synergistically to cause incapacitation. To examine venom evolution, we
sequenced and interrogated the genome of a venomous snake, the king
cobra (<em>Ophiophagus hannah</em>), and compared it, together with our
unique transcriptome, microRNA, and proteome datasets from this species,
with data from other vertebrates.
</blockquote>
<p>A fascinating insight into the biology and genome of the king cobra;
the Burmese python has also been recently been sequenced
<sup><span>[2]</span></sup>, with both projects using <a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER#58;//gmod.org/wiki/MAKER"
title="MAKER">MAKER</a> for automated annotation.</p>
<p><br />
</p>
<p><strong>PortEco: a resource for exploring bacterial biology through
high-throughput data and analysis tools</strong>
<sup><span>[3]</span></sup></p>
<blockquote>
<a href="http://gmod.org/wiki/http://porteco.org#58;//porteco.org"
class="external text" rel="nofollow">PortEco</a> aims to collect, curate
and provide data and analysis tools to support basic biological research
in Escherichia coli (and eventually other bacterial systems). PortEco is
implemented as a 'virtual' model organism database that provides a
single unified interface to the user, while integrating information from
a variety of sources.
</blockquote>
<p><a href="http://gmod.org/wiki/http://porteco.org#58;//porteco.org"
class="external text" rel="nofollow">PortEco</a> integrates <em>E.
coli</em> data from numerous sources and uses a wiki-based system to
enable collaborative knowledge management. PortEco is currently testing
a <a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a> installation with the view to migrating over
to JBrowse from their current GBrowse sequence browser.</p>
<p><br />
</p>
<p><strong>The Mouse Genome Database: integration of and access to
knowledge about the laboratory mouse</strong>
<sup><span>[4]</span></sup></p>
<blockquote>
The <a
href="http://gmod.org/wiki/http://www.informatics.jax.org#58;//www.informatics.jax.org"
class="external text" rel="nofollow">Mouse Genome Database (MGD)</a> is
the community model organism database resource for the laboratory mouse,
a premier animal model for the study of genetic and genomic systems
relevant to human biology and disease. MGD maintains a comprehensive
catalog of genes, functional RNAs and other genome features as well as
heritable phenotypes and quantitative trait loci.
</blockquote>
<p>An update on the resources available at <a
href="http://gmod.org/wiki/http://www.informatics.jax.org#58;//www.informatics.jax.org"
class="external text" rel="nofollow">Mouse Genome Database (MGD)</a>,
one of the groups involved in GMOD since the very beginning of the
project.</p>
<p><br />
</p>
<p><strong>NECTAR: a database of codon-centric missense variant
annotations</strong> <sup><span>[5]</span></sup></p>
<blockquote>
<a
href="http://gmod.org/wiki/http://nectarmutation.org#58;//nectarmutation.org"
class="external text" rel="nofollow">NECTAR (Non-synonymous Enriched
Coding muTation ARchive)</a> is a database and web application to
annotate disease-related and functionally important amino acids in human
proteins.
</blockquote>
<p>This stylishly-designed website draws data on disease mutations from
numerous sources, including Ensembl, HGMD-PUBLIC, UniProt, COSMIC, and
ClinVar, and includes <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a>-able disease variants, function annotations,
and paralogue annotations and a DAS server. Sweet!</p>
<p><br />
</p>
<p><strong>Identification of a 31-bp Deletion in the RELN Gene Causing
Lissencephaly with Cerebellar Hypoplasia in Sheep</strong>
<sup><span>[6]</span></sup></p>
<blockquote>
Lissencephaly is an inherited developmental disorder in which neuronal
migration is impaired. A type of lissencephaly associated with
cerebellar hypoplasia (LCH) was diagnosed in a commercial flock of
Spanish Churra sheep. The genotyping of 7 affected animals and 33
controls with the OvineSNP50 BeadChip enabled the localization of the
causative mutation for ovine LCH to a 4.8-Mb interval on sheep
chromosome 4 using genome-wide association and homozygosity mapping.
</blockquote>
<p>Tracking down a sequence variation that causes developmental problems
in sheep: a nice combination of classical genetics (studying families
trees and examining phenotypes) and modern techniques, using resources
provided at <a
href="http://gmod.org/wiki/http://www.livestockgenomics.csiro.au/#58;//www.livestockgenomics.csiro.au/"
class="external text" rel="nofollow">CSIRO's Livestock Genomics
website</a>.</p>
<p><br />
</p>
<p>Happy reading!</p>
<p><br />
</p>
<ol>
<li><span id="cite_note-DOI:10.1073.2Fpnas.1314702110"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1073.2Fpnas.1314702110</code></strong></span></li>
<li><span id="cite_note-DOI:10.1073.2Fpnas.1314475110"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1073.2Fpnas.1314475110</code></strong></span></li>
<li><span id="cite_note-DOI:10.1093.2Fnar.2Fgkt1203"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1093.2Fnar.2Fgkt1203</code></strong></span></li>
<li><span id="cite_note-DOI:10.1093.2Fnar.2Fgkt1225"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1093.2Fnar.2Fgkt1225</code></strong></span></li>
<li><span id="cite_note-DOI:10.1093.2Fnar.2Fgkt1245"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1093.2Fnar.2Fgkt1245</code></strong></span></li>
<li><span id="cite_note-DOI:10.1371.2Fjournal.pone.0081072"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named
<code>DOI:10.1371.2Fjournal.pone.0081072</code></strong></span></li>
</ol>
<div class="smallprint">
Disclaimer: the papers included in this feature are for your
entertainment and edification only. Inclusion does not imply an
endorsement of the material or any association between the authors and
the GMOD project.
</div>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/12/06</em>
</div>
</div>
<div class="rss">
<h3 id="canto-a-gmod-tool-for-functional-annotation"><a
href="News/Canto,_a_GMOD_tool_for_functional_annotation">Canto, a GMOD
tool for functional annotation</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://gmod.org/wiki/Canto#58;//gmod.org/wiki/Canto"
title="Canto">&lt;img alt="CantoTextLogo.png"
src="http://gmod.org/mediawiki/images/thumb/7/7e/CantoTextLogo.png/300px-CantoTextLogo.png"
width="300" height="74"
srcset="/mediawiki/images/thumb/7/7e/CantoTextLogo.png/450px-CantoTextLogo.png
1.5x,
/mediawiki/images/thumb/7/7e/CantoTextLogo.png/600px-CantoTextLogo.png
2x" /&gt;</a>
</div>
<p>We are pleased to announce that <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Canto#58;//gmod.org/wiki/Canto"
title="Canto">Canto</a>, the <a
href="http://gmod.org/wiki/http://pombase.org#58;//pombase.org"
class="external text" rel="nofollow">PomBase</a> community curation
tool, is now a GMOD component.</p>
<p>Canto is an online tool that supports functional gene annotation by
community researchers as well as by professional curators. It provides a
simple interface to gather genetic and molecular data from publications
for inclusion in public biological databases.</p>
<p>Curation in Canto uses OBO ontologies, and the specific set of
ontologies is configurable upon setup. The PomBase Canto instance
currently uses GO, PSI-MOD and FYPO.</p>
<p>Genetic and physical interaction annotation is also supported. Canto
captures all necessary information about each interaction for submission
to BioGRID.</p>
<p>Canto currently supports annotation export in JSON format, either by
individual paper or in bulk (see some <a
href="http://gmod.org/wiki/https://github.com/pombase/pombase-chado/blob/master/data/canto_dump.json#58;//github.com/pombase/pombase-chado/blob/master/data/canto_dump.json"
class="external text" rel="nofollow">example data</a> from PomBase Chado
loading tests).</p>
<p>The JSON data from Canto can be loaded into a relational database
using the Chado schema (as at PomBase) or alternative schemas.</p>
<p>Basic gene association file (GAF) export is also supported for GO
annotations.</p>
<p>More information about Canto:</p>
<ul>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Canto#58;//gmod.org/wiki/Canto"
title="Canto">GMOD wiki page on Canto</a> -- includes downloads and
mailing list</li>
<li><a
href="http://gmod.org/wiki/http://curation.pombase.org/#58;//curation.pombase.org/"
class="external text" rel="nofollow">Canto website at PomBase</a></li>
<li><a
href="http://gmod.org/wiki/https://github.com/pombase/canto#58;//github.com/pombase/canto"
class="external text" rel="nofollow">Canto source code on
GitHub</a></li>
</ul>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/12/02</em>
</div>
</div>
<div class="rss">
<h3 id="new-webapollo-release-20131122"><a
href="News/New_WebApollo_Release_(2013/11/22)">New WebApollo Release
(2013/11/22)</a></h3>
<p>There is a new release of <a
href="http://gmod.org/wiki/http://gmod.org/wiki/WebApollo#58;//gmod.org/wiki/WebApollo"
title="WebApollo">WebApollo</a> available from the <a
href="http://gmod.org/wiki/http://genomearchitect.org/webapollo/releases#58;//genomearchitect.org/webapollo/releases"
class="external text" rel="nofollow">WebApollo release website</a>. The
updated server setup instructions are available on the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/WebApollo_Installation#58;//gmod.org/wiki/WebApollo_Installation"
title="WebApollo Installation">WebApollo Installation</a> page, and the
<a
href="http://gmod.org/wiki/http://genomearchitect.org/webapollo/docs/webapollo_user_guide.pdf#58;//genomearchitect.org/webapollo/docs/webapollo_user_guide.pdf"
class="external text" rel="nofollow">WebApollo user guide</a> has been
updated accordingly.</p>
<p>If you currently have WebApollo instances up, please make sure to
read the updated server setup instructions as some of the configuration
files have changed. Since we now support multiple annotation types, you
might want to upgrade your existing data stores to change transcripts to
mRNAs (the only supported transcript type in previous versions). See the
<a
href="http://gmod.org/wiki/http://gmod.org/wiki/WebApollo_Installation#Upgrading_existing_annotation_data_stores#58;//gmod.org/wiki/WebApollo_Installation#Upgrading_existing_annotation_data_stores"
title="WebApollo Installation">instructions on upgrading your data
store</a>.</p>
<p>You can try the public demo at <a
href="http://gmod.org/wiki/http://genomearchitect.org/WebApolloDemo#58;//genomearchitect.org/WebApolloDemo"
class="external free"
rel="nofollow">http://genomearchitect.org/WebApolloDemo</a>.</p>
<p>Summary of changes:</p>
<ul>
<li>using JBrowse 1.10.9 release</li>
<li>new hybrid store (memory/disk) should use much less memory than the
pure memory store (with a small degrade in performance) - useful for
genomes with many annotations (you can configure which one to use as
best fits your needs)</li>
<li>viewing of annotation info editor for users without write privilege
(cannot modify the data)</li>
<li>FASTA data adapter</li>
<li>different annotation info editor configurations for different
annotation types</li>
<li>database/history merger tool (command line)</li>
<li>data adapters now use iterators when exporting data (improves memory
footprint)</li>
<li>add a configurable option for dumping owner and other meta-data to
the GFF3 adapter</li>
<li>undoing an "add_feature/add_transcript" operation will now warn the
user that proceed will delete the feature</li>
<li>data adapter grouping (see FASTA adapter)</li>
<li>adding/updating PubMed ID will now show the publication title for
confirmation</li>
<li>improved add_transcripts_from_gff3_to_annotations.pl</li>
<li>adding of different annotation types (gene, pseudogene, tRNA, snRNA,
snoRNA, ncRNA, rRNA, miRNA, repeat_region, transposable_element)</li>
<li>adding GO terms now supports searching/autocompletion</li>
<li>annotation info editor now requests data from the server in batch
mode (should improve speed)</li>
</ul>
<p>Fixes:</p>
<ul>
<li>Chado featureprop writeback for generic attributes</li>
<li>add_transcripts_from_gff3_to_annotations.pl now properly loads
status attributes</li>
<li>attributes being lost after undo/redo</li>
<li>redo of a merge_transcript operation after deleting one of the
transcripts *should* work now</li>
<li>properly handle dragged transcripts that contain UTR elements</li>
<li>compressed data sent to the client no longer causes the client to
hang when there are too many annotations</li>
<li>rewrote much of the split_transcript operation to better handle gene
splitting</li>
</ul>
<p>(via Ed Lee on the WebApollo mailing list)</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/11/22</em>
</div>
</div>
<div class="rss">
<h3 id="gmod-paper-cuts-nov-22nd-2013"><a
href="News/GMOD_Paper_Cuts,_Nov_22nd,_2013">GMOD Paper Cuts, Nov 22nd,
2013</a></h3>
<p>__NOFACTBOX__</p>
<p><strong>GMOD Paper Cuts is a periodic selection of choice cuts from
the scientific literature featuring interesting, exciting, or otherwise
eye-catching GMOD-related work.</strong></p>
<p>If you would like a paper to appear in GMOD Paper Cuts, please email
the details to <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">the GMOD helpdesk</a>. Ideally the
paper should be in an open-access publication so that anyone can read
it.</p>
<p>For more GMOD and GMOD-related papers, and to contribute your own
GMOD-related publications, <a
href="http://gmod.org/wiki/http://mnd.ly/WwRe8F#58;//mnd.ly/WwRe8F"
class="external text" rel="nofollow">join our Mendeley group</a>.</p>
<p><br />
</p>
<p><strong>The Genome Database for Rosaceae (GDR): year 10
update</strong> <sup><span>[1]</span></sup></p>
<blockquote>
The <a
href="http://gmod.org/wiki/http://www.rosaceae.org#58;//www.rosaceae.org"
class="external text" rel="nofollow">Genome Database for Rosaceae
(GDR)</a>, the long-standing central repository and data mining resource
for Rosaceae research, has been enhanced with new genomic, genetic and
breeding data, and improved functionality.
</blockquote>
<p>Lots of GMOD goodness in the Genome Database for Rosaceae, which is
built using <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Tripal#58;//gmod.org/wiki/Tripal"
title="Tripal">Tripal</a>, includes <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Pathway_Tools#58;//gmod.org/wiki/Pathway_Tools"
title="Pathway Tools">Pathway Tools</a> data, and has integrated <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a> and <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse_syn#58;//gmod.org/wiki/GBrowse_syn"
title="GBrowse syn">GBrowse syn</a> instances for sequence and synteny
viewing respectively.</p>
<p><br />
</p>
<p><strong>FlyBase 102—advanced approaches to interrogating
FlyBase</strong> <sup><span>[2]</span></sup></p>
<blockquote>
<a href="http://gmod.org/wiki/http://flybase.org#58;//flybase.org"
class="external text" rel="nofollow">FlyBase</a> is the leading website
and database of Drosophila genes and genomes. Whether you are using the
fruit fly <em>Drosophila melanogaster</em> as an experimental system or
wish to understand Drosophila biological knowledge in relation to human
disease or to other model systems, FlyBase can help you successfully
find the information you are looking for.
</blockquote>
<p>A followup to the <a
href="http://gmod.org/wiki/http://nar.oxfordjournals.org/content/40/D1/D706.full#58;//nar.oxfordjournals.org/content/40/D1/D706.full"
class="external text" rel="nofollow">FlyBase 101 paper</a>
<sup><span>[3]</span></sup> with updates on new features and
functionality at FlyBase. FlyBase use the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Chado#58;//gmod.org/wiki/Chado"
class="mw-redirect" title="Chado">Chado</a> database schema and users
can query the database directly.</p>
<p><br />
</p>
<p><strong>PATRIC, the bacterial bioinformatics database and analysis
resource</strong> <sup><span>[4]</span></sup></p>
<blockquote>
<a
href="http://gmod.org/wiki/http://www.patricbrc.org#58;//www.patricbrc.org"
class="external text" rel="nofollow">The Pathosystems Resource
Integration Center (PATRIC)</a> is the all-bacterial Bioinformatics
Resource Center (BRC). [...] PATRIC provides researchers with an online
resource that stores and integrates a variety of data types [e.g.
genomics, transcriptomics, protein–protein interactions (PPIs),
three-dimensional protein structures and sequence typing data] and
associated metadata.
</blockquote>
<p>PATRIC provides an immense data bank on bacteria: over 10,000
bacterial genomes as of Sept. 2013, projections of 15,000 by 2014. This
paper details the resources provided, including their <a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a>-based genome browser, and Disease View,
which integrates data on pathogens, genes, diseases, and geolocation of
disease reports. Very cool!</p>
<p><br />
</p>
<p><strong>Identical sets of methylated and nonmethylated genes in Ciona
intestinalis sperm and muscle cells</strong>
<sup><span>[5]</span></sup></p>
<blockquote>
The discovery of gene body methylation, which refers to DNA methylation
within gene coding region, suggests an as yet unknown role of DNA
methylation at actively transcribed genes. In invertebrates, gene bodies
are the primary targets of DNA methylation, and only a subset of
expressed genes is modified.
</blockquote>
<p>Some interesting results from studies of gene body methylation in the
sea squirt <em>Ciona intestinalis</em> during development.</p>
<p><br />
</p>
<p><strong>Large-Scale Collection and Analysis of Full-Length cDNAs from
<em>Brachypodium distachyon</em> and Integration with Pooideae Sequence
Resources</strong> <sup><span>[6]</span></sup></p>
<blockquote>
A comprehensive collection of full-length cDNAs is essential for correct
structural gene annotation and functional analyses of genes. [...] The
[RIKEN <em>Brachypodium</em> FL cDNA] database represents a “one-stop”
information resource for all genomic information in the Pooideae,
facilitating functional analysis of genes in this model grass plant and
seamless knowledge transfer to the Triticeae crops.
</blockquote>
<p><a
href="http://gmod.org/wiki/http://trifldb.psc.riken.jp/v3/index.pl#58;//trifldb.psc.riken.jp/v3/index.pl"
class="external text" rel="nofollow">TriFLDB, or the Triticeae
full-length CDS database</a>, is a new resource of <em>Brachypodium</em>
full length cDNA clones and other resources; it includes <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a> for exploring the <em>B. distachyon</em>
genome and its annotations further.</p>
<p><br />
</p>
<p>Happy reading!</p>
<p><br />
</p>
<ol>
<li><span id="cite_note-DOI:10.1093.2Fnar.2Fgkt1012"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1093.2Fnar.2Fgkt1012</code></strong></span></li>
<li><span id="cite_note-DOI:10.1093.2Fnar.2Fgkt1092"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1093.2Fnar.2Fgkt1092</code></strong></span></li>
<li><span id="cite_note-DOI:10.1093.2Fnar.2Fgkr1030"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1093.2Fnar.2Fgkr1030</code></strong></span></li>
<li><span id="cite_note-DOI:10.1093.2Fnar.2Fgkt1099"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1093.2Fnar.2Fgkt1099</code></strong></span></li>
<li><span id="cite_note-DOI:10.1186.2F1756-8935-6-38"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1186.2F1756-8935-6-38</code></strong></span></li>
<li><span id="cite_note-DOI:10.1371.2Fjournal.pone.0075265"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named
<code>DOI:10.1371.2Fjournal.pone.0075265</code></strong></span></li>
</ol>
<div class="smallprint">
Disclaimer: the papers included in this feature are for your
entertainment and edification only. Inclusion does not imply an
endorsement of the material or any association between the authors and
the GMOD project.
</div>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/11/22</em>
</div>
</div>
<div class="rss">
<h3 id="gmod-paper-cuts-nov-15th-2013"><a
href="News/GMOD_Paper_Cuts,_Nov_15th,_2013">GMOD Paper Cuts, Nov 15th,
2013</a></h3>
<p>__NOFACTBOX__</p>
<p><strong>GMOD Paper Cuts is a periodic selection of choice cuts from
the scientific literature featuring interesting, exciting, or otherwise
eye-catching GMOD-related work.</strong></p>
<p>If you would like a paper to appear in GMOD Paper Cuts, please email
the details to <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">the GMOD helpdesk</a>. Ideally the
paper should be in an open-access publication so that anyone can read
it.</p>
<p>For more GMOD and GMOD-related papers, and to contribute your own
GMOD-related publications, <a
href="http://gmod.org/wiki/http://mnd.ly/WwRe8F#58;//mnd.ly/WwRe8F"
class="external text" rel="nofollow">join our Mendeley group</a>.</p>
<p><br />
</p>
<p><strong>The Aspergillus genome database: multispecies curation and
incorporation of RNA-Seq data to improve structural gene
annotations</strong> <sup><span>[1]</span></sup></p>
<blockquote>
The Aspergillus Genome Database (<a
href="http://gmod.org/wiki/http://www.aspgd.org#58;//www.aspgd.org"
class="external text" rel="nofollow">AspGD</a>) is a freely available
web-based resource that was designed for Aspergillus researchers and is
also a valuable source of information for the entire fungal research
community.
</blockquote>
<p>A redesigned website and expanded resources for AspGD, which is now
using <a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a> as primary genome browser.</p>
<p><br />
</p>
<p><strong>CottonGen: a genomics, genetics and breeding database for
cotton research</strong> <sup><span>[2]</span></sup></p>
<blockquote>
<a
href="http://gmod.org/wiki/http://www.cottongen.org#58;//www.cottongen.org"
class="external text" rel="nofollow">CottonGen</a> is a curated and
integrated web-based relational database providing access to publicly
available genomic, genetic and breeding data for cotton.
</blockquote>
<p>An update on the <a
href="http://gmod.org/wiki/http://www.cottongen.org#58;//www.cottongen.org"
class="external text" rel="nofollow">CottonGen database</a>, which uses
the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Tripal#58;//gmod.org/wiki/Tripal"
title="Tripal">Tripal</a> framework to make available data from a <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Chado#58;//gmod.org/wiki/Chado"
class="mw-redirect" title="Chado">Chado</a> database, and to integrate
the GMOD tools <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a> and <a
href="http://gmod.org/wiki/http://gmod.org/wiki/CMap#58;//gmod.org/wiki/CMap"
title="CMap">CMap</a>.</p>
<p><br />
</p>
<p><strong>WormBase 2014: new views of curated biology</strong>
<sup><span>[3]</span></sup></p>
<blockquote>
<a
href="http://gmod.org/wiki/http://www.wormbase.org/#58;//www.wormbase.org/"
class="external text" rel="nofollow">WormBase</a> is a highly curated
resource dedicated to supporting research using the model organism
<em>Caenorhabditis elegans</em>.
</blockquote>
<p>Updates and expansions on the WormBase website, which is notable not
only for developing and using a number of GMOD tools, including <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a> and <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Textpresso#58;//gmod.org/wiki/Textpresso"
title="Textpresso">Textpresso</a>, but also for being hosted entirely in
the cloud.</p>
<p><br />
</p>
<p><strong>Gleaning evolutionary insights from the genome sequence of a
probiotic yeast Saccharomyces boulardii.</strong>
<sup><span>[4]</span></sup></p>
<blockquote>
The yeast <em>Saccharomyces boulardii</em> is used worldwide as a
probiotic to alleviate the effects of several gastrointestinal diseases
and control antibiotics-associated diarrhea. While many studies report
the probiotic effects of <em>S. boulardii</em>, no genome information
for this yeast is currently available in the public domain. We report
the 11.4 Mbp draft genome of this probiotic yeast.
</blockquote>
<p>The newly-sequenced <em>Saccharomyces boulardii</em> genome, which
was annotated with using <a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER#58;//gmod.org/wiki/MAKER"
title="MAKER">MAKER</a>.</p>
<p><br />
</p>
<p><strong>DOOR 2.0: presenting operons and their functions through
dynamic and integrated views</strong> <sup><span>[5]</span></sup></p>
<blockquote>
We have recently developed a new version of the DOOR operon database,
DOOR 2.0, which is available online at <a
href="http://gmod.org/wiki/http://csbl.bmb.uga.edu/DOOR/#58;//csbl.bmb.uga.edu/DOOR/"
class="external free" rel="nofollow">http://csbl.bmb.uga.edu/DOOR/</a>
and will be updated on a regular basis. DOOR 2.0 contains genome-scale
operons for 2072 prokaryotes with complete genomes, three times the
number of genomes covered in the previous version published in 2009.
</blockquote>
<p>A bright, lively update to the <a
href="http://gmod.org/wiki/http://csbl.bmb.uga.edu/DOOR/#58;//csbl.bmb.uga.edu/DOOR/"
class="external text" rel="nofollow">Database of prokaryotic
operons</a>, which uses <a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a> for sequence visualization.</p>
<p><br />
</p>
<p>And finally, if you have access to the Wiley Online Library through
your institution, you may like to check out [<a
href="http://gmod.org/wiki/http://onlinelibrary.wiley.com/doi/10.1002/0471250953.bi0915s43/abstract#58;//onlinelibrary.wiley.com/doi/10.1002/0471250953.bi0915s43/abstract"
class="external free"
rel="nofollow">http://onlinelibrary.wiley.com/doi/10.1002/0471250953.bi0915s43/abstract</a>
Cloud Computing with iPlant Atmosphere], from Current Protocols in
Bioinformatics, which details the resources available on the
[www.iplantcollaborative.org/‎ iPlant website] and how to get started
with tools such as <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a>.</p>
<p><br />
</p>
<p>Happy reading!</p>
<p><br />
</p>
<ol>
<li><span id="cite_note-DOI:10.1093.2Fnar.2Fgkt1029"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1093.2Fnar.2Fgkt1029</code></strong></span></li>
<li><span id="cite_note-DOI:10.1093.2Fnar.2Fgkt1064"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1093.2Fnar.2Fgkt1064</code></strong></span></li>
<li><span id="cite_note-DOI:10.1093.2Fnar.2Fgkt1063"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1093.2Fnar.2Fgkt1063</code></strong></span></li>
<li><span id="cite_note-DOI:10.1186.2F1757-4749-5-30"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1186.2F1757-4749-5-30</code></strong></span></li>
<li><span id="cite_note-DOI:10.1093.2Fnar.2Fgkt1048"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1093.2Fnar.2Fgkt1048</code></strong></span></li>
</ol>
<div class="smallprint">
Disclaimer: the papers included in this feature are for your
entertainment and edification only. Inclusion does not imply an
endorsement of the material or any association between the authors and
the GMOD project.
</div>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/11/15</em>
</div>
</div>
<div class="rss">
<h3 id="gmod-job-at-qmul"><a href="News/GMOD_Job_At_QMUL">GMOD Job At
QMUL</a></h3>
<p>There is a <a
href="http://gmod.org/wiki/http://www.jobs.qmul.ac.uk/4313#58;//www.jobs.qmul.ac.uk/4313"
class="external text" rel="nofollow">position as a sysadmin/computing
officer</a> open at the <a
href="http://gmod.org/wiki/http://sbcs.qmul.ac.uk#58;//sbcs.qmul.ac.uk"
class="external text" rel="nofollow">School of Biological &amp; Chemical
Sciences</a> at <a
href="http://gmod.org/wiki/http://www.qmul.ac.uk#58;//www.qmul.ac.uk"
class="external text" rel="nofollow">Queen Mary University of London</a>
(QMUL), with duties including setting up Galaxy and GMOD servers. The
closing date for applications is 21st November 2013; for more details,
see the <a
href="http://gmod.org/wiki/http://www.jobs.qmul.ac.uk/4313#58;//www.jobs.qmul.ac.uk/4313"
class="external text" rel="nofollow">job description</a>.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/11/12</em>
</div>
</div>
<div class="rss">
<h3 id="gmod-paper-cuts-nov-8th-2013"><a
href="News/GMOD_Paper_Cuts,_Nov_8th,_2013">GMOD Paper Cuts, Nov 8th,
2013</a></h3>
<p>__NOFACTBOX__</p>
<p><strong>GMOD Paper Cuts is a periodic selection of choice cuts from
the scientific literature featuring interesting, exciting, or otherwise
eye-catching GMOD-related work.</strong></p>
<p>If you would like a paper to appear in GMOD Paper Cuts, please email
the details to <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">the GMOD helpdesk</a>. Ideally the
paper should be in an open-access publication so that anyone can read
it.</p>
<p>For more GMOD and GMOD-related papers, and to contribute your own
GMOD-related publications, <a
href="http://gmod.org/wiki/http://mnd.ly/WwRe8F#58;//mnd.ly/WwRe8F"
class="external text" rel="nofollow">join our Mendeley group</a>.</p>
<p><br />
</p>
<dl>
<dt>mRNA-Seq and microarray development for the Grooved carpet shell
clam, <em>Ruditapes decussatus</em>: a functional approach to unravel
host-parasite interaction <sup><span>[1]</span></sup></dt>
<dd>
The first characterization of the <em>Ruditapes decussatus</em>
transcriptome; the team responsible have also set up <a
href="http://gmod.org/wiki/http://morse-ccmar.ualg.pt/edge#58;//morse-ccmar.ualg.pt/edge"
class="external text" rel="nofollow">Rdecusdb, the Ruditapes decussatus
database</a>, built with <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Tripal#58;//gmod.org/wiki/Tripal"
title="Tripal">Tripal</a>
</dd>
</dl>
<dl>
<dt>The database of genomic variants: a curated collection of structural
variation in the human genome <sup><span>[2]</span></sup></dt>
<dd>
An update on the <a
href="http://gmod.org/wiki/http://dgv.tcag.ca/#58;//dgv.tcag.ca/"
class="external text" rel="nofollow">Database of Genomic Variants
(DGV)</a>, which provides a publicly accessible, comprehensive curated
catalogue of structural variation (SV) found in the genomes of control
individuals from worldwide populations. The core visualization tool, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a>, has been upgraded with additional functions
to facilitate data analysis and comparison.
</dd>
</dl>
<dl>
<dt>Meta-analysis of 74,046 individuals identifies 11 new susceptibility
loci for Alzheimer's disease <sup><span>[3]</span></sup></dt>
<dd>
A genome-wide association study on humans of European ancestry that
identifies new loci associated with Alzheimer's disease.
</dd>
</dl>
Crossover Heterogeneity in the Absence of Hotspots in <em>Caenorhabditis
elegans</em> <sup><span>[4]</span></sup>
<dl>
<dt>Computational framework to support integration of biomolecular and
clinical data within a translational approach
<sup><span>[5]</span></sup></dt>
<dd>
This article was published earlier this year, and is a good example of
extending the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Chado#58;//gmod.org/wiki/Chado"
class="mw-redirect" title="Chado">Chado</a> schema to fit the needs of
the project.
</dd>
</dl>
<p><br />
</p>
<ol>
<li><span id="cite_note-DOI:10.1186.2F1471-2164-14-741"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1186.2F1471-2164-14-741</code></strong></span></li>
<li><span id="cite_note-DOI:10.1093.2Fnar.2Fgkt958"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1093.2Fnar.2Fgkt958</code></strong></span></li>
<li><span id="cite_note-DOI:10.1038.2Fng.2802"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1038.2Fng.2802</code></strong></span></li>
<li><span id="cite_note-DOI:10.1534.2Fgenetics.113.158857"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named
<code>DOI:10.1534.2Fgenetics.113.158857</code></strong></span></li>
<li><span id="cite_note-DOI:10.1186.2F1471-2105-14-180"><span
class="mw-cite-backlink"><span>↑</span></span> <strong>Cite error:
Invalid <code>&lt;ref&gt;</code> tag; no text was provided for refs
named <code>DOI:10.1186.2F1471-2105-14-180</code></strong></span></li>
</ol>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/11/08</em>
</div>
</div>
<div class="rss">
<h3 id="open-position-at-dictybase"><a
href="News/Open_Position_At_dictyBase">Open Position At
dictyBase</a></h3>
<div class="floatright">
<a href="http://gmod.org/wiki/http://dictybase.org/#58;//dictybase.org/"
rel="nofollow" title="DictyBase">&lt;img alt="DictyBase"
src="http://gmod.org/mediawiki/images/a/a8/DictyMainPageLogo.png"
width="93" height="60" /&gt;</a>
</div>
<p><strong>dictyBase</strong> is hiring. We have an <a
href="http://gmod.org/wiki/http://www.dictybase.org/dictybase_jobs.html#58;//www.dictybase.org/dictybase_jobs.html"
class="external text" rel="nofollow">open position</a> for a
Bioinformatics Software Engineer at <a
href="http://gmod.org/wiki/http://dictybase.org#58;//dictybase.org"
class="external text" rel="nofollow">dictyBase</a>.<br />
See the <a
href="http://gmod.org/wiki/http://www.dictybase.org/dictybase_jobs.html#58;//www.dictybase.org/dictybase_jobs.html"
class="external text" rel="nofollow">official posting</a> for full
details.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/11/04</em>
</div>
</div>
<div class="rss">
<h3 id="tripal-1.1-publication"><a
href="News/Tripal_1.1_Publication">Tripal 1.1 Publication</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://www.ncbi.nlm.nih.gov/pubmed/24163125#58;//www.ncbi.nlm.nih.gov/pubmed/24163125"
rel="nofollow">&lt;img alt="TripalLogo.png"
src="http://gmod.org/mediawiki/images/thumb/0/06/TripalLogo.png/250px-TripalLogo.png"
width="250" height="75" srcset="/mediawiki/images/0/06/TripalLogo.png
1.5x, /mediawiki/images/0/06/TripalLogo.png 2x" /&gt;</a>
</div>
<p>The <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Tripal#58;//gmod.org/wiki/Tripal"
title="Tripal">Tripal</a> team have put out a publication on the 1.1
version of the software. <strong>Tripal v1.1: a standards-based toolkit
for construction of online genetic and genomic databases.</strong> can
be read for free at <a
href="http://gmod.org/wiki/http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3808541/#58;//www.ncbi.nlm.nih.gov/pmc/articles/PMC3808541/"
class="external text" rel="nofollow">PubMedCentral</a> or at the <a
href="http://gmod.org/wiki/http://database.oxfordjournals.org/content/2013/bat075.long#58;//database.oxfordjournals.org/content/2013/bat075.long"
class="external text" rel="nofollow">Database journal</a>.</p>
<p>Here is the abstract:</p>
<blockquote>
<p>Tripal is an open-source freely available toolkit for construction of
online genomic and genetic databases. It aims to facilitate development
of community-driven biological websites by integrating the GMOD Chado
database schema with Drupal, a popular website creation and content
management software. Tripal provides a suite of tools for interaction
with a Chado database and display of content therein. The tools are
designed to be generic to support the various ways in which data may be
stored in Chado. Previous releases of Tripal have supported organisms,
genomic libraries, biological stocks, stock collections and genomic
features, their alignments and annotations. Also, Tripal and its
extension modules provided loaders for commonly used file formats such
as FASTA, GFF, OBO, GAF, BLAST XML, KEGG heir files and InterProScan
XML. Default generic templates were provided for common views of
biological data, which could be customized using an open Application
Programming Interface to change the way data are displayed. Here, we
report additional tools and functionality that are part of release v1.1
of Tripal. These include</p>
<ol>
<li>a new bulk loader that allows a site curator to import data stored
in a custom tab delimited format;</li>
<li>full support of every Chado table for Drupal Views (a powerful tool
allowing site developers to construct novel displays and search
pages);</li>
<li>new modules including 'Feature Map', 'Genetic', 'Publication',
'Project', 'Contact' and the 'Natural Diversity' modules.</li>
</ol>
<p>Tutorials, mailing lists, download and set-up instructions, extension
modules and other documentation can be found at the Tripal website
located at <a
href="http://gmod.org/wiki/http://tripal.info#58;//tripal.info"
class="external free" rel="nofollow">http://tripal.info</a>.</p>
</blockquote>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/10/31</em>
</div>
</div>
<div class="rss">
<h3 id="san-diego-2014-registration-open"><a
href="News/San_Diego_2014_Registration_Open">San Diego 2014 Registration
Open</a></h3>
<p>Registration is now open for the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Jan_2014_GMOD_Meeting#58;//gmod.org/wiki/Jan_2014_GMOD_Meeting"
title="Jan 2014 GMOD Meeting">upcoming GMOD community meeting</a>, to be
held at the <a
href="http://gmod.org/wiki/http://www.bw7seas.com/#58;//www.bw7seas.com/"
class="external text" rel="nofollow">Best Western Seven Seas, San Diego,
CA</a>, on January 16th and 17th 2014.</p>
<p><strong><a
href="http://gmod.org/wiki/https://gmod-sandiego-2014.eventbrite.com/#58;//gmod-sandiego-2014.eventbrite.com/"
class="external text" rel="nofollow">Register online at Eventbrite
now!</a></strong></p>
<p>Early birds can catch the $50 registration worm, which is available
until mid-December; after this, the registration fee will rise to
$70.</p>
<p>If you would like to give a presentation at the meeting, please
submit your abstract with your meeting registration form.</p>
<p>A block of rooms is being held for attendees of the GMOD meeting at a
very good rate; to make reservations in this block, <strong>you must
call</strong> (800) 328-1618 and ask for the GMOD rate. The GMOD rate is
not available through online bookings.</p>
<p>If you have any questions or comments, please feel free to contact
the GMOD helpdesk on help@gmod.org!</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/10/25</em>
</div>
</div>
<div class="rss">
<h3 id="gcc2014-june-30---july-2"><a
href="News/GCC2014:_June_30_-_July_2">GCC2014: June 30 - July 2</a></h3>
<div class="floatleft">
<a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2014#58;//wiki.galaxyproject.org/Events/GCC2014"
rel="nofollow" title="2014 Galaxy Community Conference">&lt;img
alt="2014 Galaxy Community Conference"
src="http://gmod.org/mediawiki/images/thumb/a/a4/GCC2014Logo.png/250px-GCC2014Logo.png"
width="250" height="105"
srcset="/mediawiki/images/thumb/a/a4/GCC2014Logo.png/375px-GCC2014Logo.png
1.5x, /mediawiki/images/thumb/a/a4/GCC2014Logo.png/500px-GCC2014Logo.png
2x" /&gt;</a>
</div>
<p>The <a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2014#58;//wiki.galaxyproject.org/Events/GCC2014"
class="external text" rel="nofollow">2014 Galaxy Community Conference
(GCC2013)</a> will be held June 30 through July 2, at the <a
href="http://gmod.org/wiki/http://webapps.jhu.edu/jhuniverse/information_about_hopkins/campuses/homewood_campus/#58;//webapps.jhu.edu/jhuniverse/information_about_hopkins/campuses/homewood_campus/"
class="external text" rel="nofollow">Homewood Campus</a> of <a
href="http://gmod.org/wiki/http://jhu.edu#58;//jhu.edu"
class="external text" rel="nofollow">Johns Hopkins University</a>], in
<a
href="http://gmod.org/wiki/http://visitors.baltimorecity.gov/#58;//visitors.baltimorecity.gov/"
class="external text" rel="nofollow">Baltimore, Maryland</a>, United
States.</p>
<p>Galaxy Community Conferences are an opportunity to participate in
presentations, discussions, poster sessions, lightning talks and
breakouts, all about high-throughput biology and the tools that support
it.  The conference will also includes a <em>Training Day</em> offering
in-depth topic coverage, across several concurrent sessions.  See <a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC201#58;//wiki.galaxyproject.org/Events/GCC201"
class="external text" rel="nofollow">the GCC2013 web site</a> for an
idea of what happens at a Galaxy Community Conference. The GCC2013 site
includes links to slides and videos for all accepted talks.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2014/10/02</em>
</div>
</div>
<div class="rss">
<h3 id="get-the-full-impactstory"><a
href="News/Get_the_Full_ImpactStory">Get the Full ImpactStory</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://impactstory.org#58;//impactstory.org"
rel="nofollow">&lt;img alt="ImpactstoryLogo.png"
src="http://gmod.org/mediawiki/images/thumb/7/7a/ImpactstoryLogo.png/250px-ImpactstoryLogo.png"
width="250" height="46"
srcset="/mediawiki/images/thumb/7/7a/ImpactstoryLogo.png/375px-ImpactstoryLogo.png
1.5x, /mediawiki/images/7/7a/ImpactstoryLogo.png 2x" /&gt;</a>
</div>
The GMOD wiki now includes <a
href="http://gmod.org/wiki/http://impactstory.org#58;//impactstory.org"
class="external text" rel="nofollow">ImpactStory altmetrics</a> for
publications cited in the page text. See the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Publications#58;//gmod.org/wiki/Publications"
title="Publications">Publications</a> page for an example of ImpactStory
altmetrics.
<p>For those unfamiliar with the term <em>altmetrics,</em> these are
measures beyond simple paper citations. The <a
href="http://gmod.org/wiki/http://altmetrics.org/manifesto/#58;//altmetrics.org/manifesto/"
class="external text" rel="nofollow">Altmetrics Manifesto</a> provides
the following clarification:</p>
<blockquote>
<p>Altmetrics expand our view of what impact looks like, but also of
what’s making the impact. This matters because expressions of
scholarship are becoming more diverse. Articles are increasingly joined
by:</p>
<ul>
<li>The sharing of “raw science” like datasets, code, and experimental
designs</li>
<li>Semantic publishing or “nanopublication,” where the citeable unit is
an argument or passage rather than entire article.</li>
<li>Widespread self-publishing via blogging, microblogging, and comments
or annotations on existing work.</li>
</ul>
</blockquote>
<p><br />
ImpactStory collates metrics from numerous online sources to give a more
complete picture of how an article, data set, piece of software, or
other research product is being used and discussed. More information is
available on the <a
href="http://gmod.org/wiki/http://impactstory.org/#58;//impactstory.org/"
class="external text" rel="nofollow">ImpactStory</a> website.</p>
<p><br />
</p>
<p>We hope to add impact metrics for GMOD software soon.</p>
</div>
<div class="rss">
<h3 id="gmod-2014-san-diego-save-the-date"><a
href="News/GMOD_2014_San_Diego:_Save_the_Date">GMOD 2014 San Diego: Save
the Date</a></h3>
<div class="floatleft">
<a
href="http://gmod.org/wiki/File:2014-gmod-san-diego.png#58;//gmod.org/wiki/File:2014-gmod-san-diego.png"
class="image">&lt;img alt="2014-gmod-san-diego.png"
src="http://gmod.org/mediawiki/images/thumb/e/e1/2014-gmod-san-diego.png/300px-2014-gmod-san-diego.png"
width="300" height="188"
srcset="/mediawiki/images/thumb/e/e1/2014-gmod-san-diego.png/450px-2014-gmod-san-diego.png
1.5x, /mediawiki/images/e/e1/2014-gmod-san-diego.png 2x" /&gt;</a>
</div>
<p>The <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Jan_2014_GMOD_Meeting#58;//gmod.org/wiki/Jan_2014_GMOD_Meeting"
title="Jan 2014 GMOD Meeting">next GMOD Community Meeting</a> will be
held in San Diego, CA, on January 16-17, directly after <a
href="http://gmod.org/wiki/http://intl-pag.org#58;//intl-pag.org"
class="external text" rel="nofollow">PAG XXII</a>. In addition to the
usual talks and updates, we plan to have a poster session, and a
workshop on migrating from GBrowse to JBrowse. If you have items or
ideas for the agenda, please feel free to <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">contact the GMOD helpdesk</a>.</p>
<p>GMOD will also have a big presence at PAG, with a day-long workshop
covering many of the actively-developed GMOD components.</p>
<p>More details on both the GMOD community meeting and GMOD at PAG will
be posted on the wiki closer to the time.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013-09-10</em>
</div>
</div>
<div class="rss">
<h3 id="statistical-genomics-postdoc-at-psu"><a
href="News/Statistical_Genomics_PostDoc_at_PSU">Statistical Genomics
PostDoc at PSU</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://www.huck.psu.edu/center/medical-genomics/#58;//www.huck.psu.edu/center/medical-genomics/"
rel="nofollow">&lt;img alt="HuckInstitutes.png"
src="http://gmod.org/mediawiki/images/thumb/3/35/HuckInstitutes.png/250px-HuckInstitutes.png"
width="250" height="48"
srcset="/mediawiki/images/thumb/3/35/HuckInstitutes.png/375px-HuckInstitutes.png
1.5x, /mediawiki/images/3/35/HuckInstitutes.png 2x" /&gt;</a>
</div>
<p>Are you interested in genomics and do you have skills in
Bioinformatics, Computational Biology and Statistics? The Makova lab in
the Department of Biology at The Pennsylvania State University is
looking to hire a postdoctoral researcher for an NSF-funded project
examining regional variation in mutation rates (see <a
href="http://gmod.org/wiki/http://www.pnas.org//content/early/2013/08/16/1221792110.abstract#58;//www.pnas.org//content/early/2013/08/16/1221792110.abstract"
class="external text" rel="nofollow">our recent publication in PNAS:
Kuruppumullage Don, Ananda, Chiaromonte, Makova 2013)</a>. With new
sequencing technologies, multiple human genomes and their detailed
annotations (e.g., ENCODE) are suddenly accessible to us. This gives us
a terrific opportunity to explore previously inaccessible evolutionary
processes (e.g., mutation) and other biological associations (location
of genes in certain mutation states, as identified by HMMs). Such
knowledge is vital in a clinical setting where disease mutations need to
be evaluated. Our resources and links with medical researchers at
Hershey Medical School and computational biologists from the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Galaxy#58;//gmod.org/wiki/Galaxy"
title="Galaxy">Galaxy</a> team put us in a great position to address
these questions.</p>
<p>Candidates should have experience in bioinformatics, working
knowledge of statistics and should have a broad understanding of
molecular biology and genetics. Familiarity with next-generation
sequencing data analysis is desirable. A PhD is required. You will be
joining an established dynamic group. We are part of the <a
href="http://gmod.org/wiki/http://www.huck.psu.edu/center/medical-genomics#58;//www.huck.psu.edu/center/medical-genomics"
class="external text" rel="nofollow">Center for Medical Genomics</a> and
of the <a
href="http://gmod.org/wiki/http://www.bx.psu.edu/#58;//www.bx.psu.edu/"
class="external text" rel="nofollow">Center for Comparative Genomics and
Bioinformatics</a>. Penn State is a vibrant scientific community with
particular strengths in genomics, bioinformatics and molecular
evolution. Our location, in State College, Pennsylvania, is known for
excellent schools and numerous opportunities for outdoor activities.</p>
<p>The starting date is flexible, with an earlier date preferred. This
position is funded for one year from date of hire, with good possibility
of refunding. Interested applicants should send a pdf with a CV, a
statement of research interests, and contact information for three
referees to <a
href="http://gmod.org/wiki/mailto:kmakova@bx.psu.edu#58;kmakova@bx.psu.edu"
class="external text" rel="nofollow">Kateryna Makova</a>, indicating
postdoc in the subject line. Employment will require successful
completion of background check(s) in accordance with University
policies. Penn State is committed to affirmative action, equal
opportunity, and the diversity of its workforce.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/09/09</em>
</div>
</div>
<div class="rss">
<h3 id="support-your-friendly-local-gmod"><a
href="News/Support_Your_Friendly_Local_GMOD">Support Your Friendly Local
GMOD</a></h3>
<p>The Generic Model Organism Database project is applying for a <a
href="http://gmod.org/wiki/http://grants.nih.gov/grants/guide/pa-files/PAR-11-095.html#58;//grants.nih.gov/grants/guide/pa-files/PAR-11-095.html"
class="external text" rel="nofollow">community resources for genomics
grant</a> from the NIH to continue the work that GMOD does, with a focus
on JBrowse and Chado development, and GMOD coordination (standards,
meetings, and courses).</p>
<p>We need to demonstrate that we support the community and have the
support of the community, and we would like to demonstrate that by
submitting many letters of support from users. We would be very grateful
if you could take a couple of minutes to write such a letter; it should
include information about how you use or plan to use GMOD software.
There are <a
href="http://gmod.org/wiki/User:Scott#Support_letter_templates#58;//gmod.org/wiki/User:Scott#Support_letter_templates"
title="User:Scott">some template paragraphs</a> provided for
inspiration.</p>
<p>The letters should be on institutional letterhead, signed, and then
scanned to a PDF, or you could mail the original to Scott Cain and he
will scan them.</p>
<p>The deadline for getting the letters of support is September 3.
Please <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">contact the GMOD helpdesk</a> or <a
href="http://gmod.org/wiki/mailto:scott@scottcain.net#58;scott@scottcain.net"
class="external text" rel="nofollow">email Scott directly</a> if you're
willing to write a letter or if you need any help.</p>
<p>This grant would help to secure the future of important parts of the
GMOD project, so we would be very grateful for your help!</p>
<p><em>Via <a
href="http://gmod.org/wiki/User:Scott#58;//gmod.org/wiki/User:Scott"
title="User:Scott">Scott Cain</a></em></p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/08/20</em>
</div>
</div>
<div class="rss">
<h3 id="summer-school-tutorials-available"><a
href="News/2013_Summer_School_Tutorials_Available">2013 Summer School
Tutorials Available</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/File:2013-summer-school.png#58;//gmod.org/wiki/File:2013-summer-school.png"
class="image" title="2013 GMOD Summer School">&lt;img alt="2013 GMOD
Summer School"
src="http://gmod.org/mediawiki/images/thumb/f/fc/2013-summer-school.png/300px-2013-summer-school.png"
width="300" height="186"
srcset="/mediawiki/images/thumb/f/fc/2013-summer-school.png/450px-2013-summer-school.png
1.5x,
/mediawiki/images/thumb/f/fc/2013-summer-school.png/600px-2013-summer-school.png
2x" /&gt;</a>
</div>
<p>The tutorials for the material taught at <a
href="http://gmod.org/wiki/http://gmod.org/wiki/2013_GMOD_Summer_School#58;//gmod.org/wiki/2013_GMOD_Summer_School"
title="2013 GMOD Summer School">this year's GMOD summer school</a> are
now all available on the GMOD website. There are updated tutorials for
several components, particularly those where new software releases have
occurred since the last summer school; these include:</p>
<ul>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Galaxy_Tutorial#58;//gmod.org/wiki/Galaxy_Tutorial"
class="mw-redirect" title="Galaxy Tutorial">Galaxy</a></li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse_syn_Tutorial#58;//gmod.org/wiki/GBrowse_syn_Tutorial"
class="mw-redirect" title="GBrowse syn Tutorial">GBrowse syn</a></li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse_Tutorial#58;//gmod.org/wiki/JBrowse_Tutorial"
class="mw-redirect" title="JBrowse Tutorial">JBrowse</a></li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER_Tutorial#58;//gmod.org/wiki/MAKER_Tutorial"
class="mw-redirect" title="MAKER Tutorial">MAKER</a></li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Tripal_Tutorial_v1.1#58;//gmod.org/wiki/Tripal_Tutorial_v1.1"
title="Tripal Tutorial v1.1">Tripal</a></li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/WebApollo_Tutorial#58;//gmod.org/wiki/WebApollo_Tutorial"
class="mw-redirect" title="WebApollo Tutorial">WebApollo</a></li>
</ul>
<p>The full set of GMOD tutorials, featuring all the material from this
year's summer school and more, is available from the <a
href="http://gmod.org/wiki/Category:Tutorials#58;//gmod.org/wiki/Category:Tutorials"
title="Category:Tutorials">Tutorials</a> section of the wiki.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013-08-02</em>
</div>
</div>
<div class="rss">
<h3 id="jbrowse-1.10.0-released"><a
href="News/JBrowse_1.10.0_released">JBrowse 1.10.0 released</a></h3>
<p>I am very pleased to announce the release of JBrowse 1.10.0, a major
new JBrowse release with some big new features, and a *huge* number of
smaller improvements!</p>
<p>It can be downloaded from <a
href="http://gmod.org/wiki/http://jbrowse.org/jbrowse-1-10-0/#58;//jbrowse.org/jbrowse-1-10-0/"
class="external free"
rel="nofollow">http://jbrowse.org/jbrowse-1-10-0/</a>.</p>
<p>As always, see the updated</p>
<p><a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse_Configuration_Guide#58;//gmod.org/wiki/JBrowse_Configuration_Guide"
title="JBrowse Configuration Guide">JBrowse Configuration Guide</a> for
documentation on how to use the new features.</p>
<p><a
href="http://gmod.org/wiki/User:RobertBuels#58;//gmod.org/wiki/User:RobertBuels"
title="User:RobertBuels">RobertBuels</a> (<a
href="http://gmod.org/wiki/User_talk:RobertBuels#58;//gmod.org/wiki/User_talk:RobertBuels"
title="User talk:RobertBuels">talk</a>)</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/07/30</em>
</div>
</div>
<div class="rss">
<h3 id="openid-account-creation-fixed"><a
href="News/OpenID_Account_Creation_Fixed">OpenID Account Creation
Fixed</a></h3>
<p>Update on July 11:</p>
<p>Account creation using OpenID is currently disabled due to problems
with the OpenID plugin. New wiki users wishing to create accounts should
use the 'Log in / create account' link and <a
href="http://gmod.org/wiki/Special:RequestAccount#58;//gmod.org/wiki/Special:RequestAccount"
title="Special:RequestAccount">request an account</a>; your request
should be dealt with within 24-48 hours. Unfortunately, due to spammers
and others creating bogus accounts, we have to had to put protective
measures in place.</p>
<p>Please email <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">the GMOD helpdesk</a> if you have
any problems or want to expedite the creation of your account!</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/07/10</em>
</div>
</div>
<div class="rss">
<h3 id="gmods-gsoc-going-great-guns"><a
href="News/GMOD&#39;s_GSoC_Going_Great_Guns!">GMOD's GSoC Going Great
Guns!</a></h3>
<p>GMOD is lucky enough to have a <a
href="http://gmod.org/wiki/http://www.google-melange.com/#58;//www.google-melange.com/"
class="external text" rel="nofollow">Google Summer of Code</a> student
working with us through the summer. <strong>Flavien Bossiaux</strong> is
a computer science student from the University of Nice, France, and he
is working on a number of improvements to the GMOD website user
experience. He has already created two Mediawiki extensions to allow the
embedding of external content into the GMOD wiki; visit the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Calendar#58;//gmod.org/wiki/Calendar"
title="Calendar">GMOD calendar</a> to see one of them, the <a
href="http://gmod.org/wiki/http://iann.pro#58;//iann.pro"
class="external text" rel="nofollow">iAnn</a> calendar plugin, in
action.</p>
<p>Flavien is recording his thoughts and progress <a
href="http://gmod.org/wiki/http://gmod.org/bflavien#58;//gmod.org/bflavien"
class="external text" rel="nofollow">on his website</a> through the
summer; please check it out! You can also <a
href="http://gmod.org/wiki/https://twitter.com/BFl4vien#58;//twitter.com/BFl4vien"
class="external text" rel="nofollow">follow Flavien on Twitter</a>, and
check out his <a
href="http://gmod.org/wiki/https://github.com/BFlavien#58;//github.com/BFlavien"
class="external text" rel="nofollow">his Github repo</a>.</p>
<p>If you have any thoughts or suggestions on the GSoC project, please
send them along to <a
href="http://gmod.org/wiki/mailto:help@gmod.org#58;help@gmod.org"
class="external text" rel="nofollow">the GMOD helpdesk</a>.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/07/09</em>
</div>
</div>
<div class="rss">
<h3 id="gmod-in-the-cloud-2.0-released"><a
href="News/GMOD_in_the_Cloud_2.0_Released">GMOD in the Cloud 2.0
Released</a></h3>
<p>from Scott Cain:</p>
<div class="floatright">
<a
href="http://gmod.org/wiki/File:GitcLogo.png#58;//gmod.org/wiki/File:GitcLogo.png"
class="image">&lt;img alt="GitcLogo.png"
src="http://gmod.org/mediawiki/images/thumb/6/69/GitcLogo.png/200px-GitcLogo.png"
width="200" height="114"
srcset="/mediawiki/images/thumb/6/69/GitcLogo.png/300px-GitcLogo.png
1.5x, /mediawiki/images/thumb/6/69/GitcLogo.png/400px-GitcLogo.png 2x"
/&gt;</a>
</div>
<p>I am pleased to announce that GMOD in the Cloud 2.0 has finally been
released to the public. This Amazon Web Services (AWS) Amazon Machine
Image (AMI) is public and can be cloned by anybody with an AWS account.
A <a
href="http://gmod.org/wiki/http://cloud.gmod.org/#58;//cloud.gmod.org/"
class="external text" rel="nofollow">demo of GMOD in the Cloud</a> is
running (though it may take a day or two for DNS to update fully).</p>
<p>This AMI has several GMOD software packages installed and configured,
including:</p>
<ul>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Chado#58;//gmod.org/wiki/Chado"
class="mw-redirect" title="Chado">Chado</a> 1.23</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/Tripal#58;//gmod.org/wiki/Tripal"
title="Tripal">Tripal</a> 1.0 (plus a few bug fixes from dev)</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a> 2.54 (with samtools, BigWig and fastCGI
support)</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a> 1.9.5</li>
<li><a
href="http://gmod.org/wiki/http://gmod.org/wiki/WebApollo#58;//gmod.org/wiki/WebApollo"
title="WebApollo">WebApollo</a> (most recent release from May,
2013)</li>
</ul>
<p>For more information about GMOD in the Cloud, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Cloud#58;//gmod.org/wiki/Cloud"
title="Cloud">visit the Cloud page on the wiki</a>, and post questions
about using GMOD in the Cloud to the GMOD developers mailing list,
gmod-devel@lists.sourceforge.net.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/06/14</em>
</div>
</div>
<div class="rss">
<h3 id="last-call-for-summer-school-apps"><a
href="News/Last_Call_For_Summer_School_Apps">Last Call For Summer School
Apps</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/File:2013-summer-school.png#58;//gmod.org/wiki/File:2013-summer-school.png"
class="image" title="2013 GMOD Summer School">&lt;img alt="2013 GMOD
Summer School"
src="http://gmod.org/mediawiki/images/thumb/f/fc/2013-summer-school.png/200px-2013-summer-school.png"
width="200" height="124"
srcset="/mediawiki/images/thumb/f/fc/2013-summer-school.png/300px-2013-summer-school.png
1.5x,
/mediawiki/images/thumb/f/fc/2013-summer-school.png/400px-2013-summer-school.png
2x" /&gt;</a>
</div>
<p>The deadline to apply for the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/2013_GMOD_Summer_School#58;//gmod.org/wiki/2013_GMOD_Summer_School"
title="2013 GMOD Summer School">2013 GMOD Summer School</a> is rapidly
approaching! Please submit your application by June 10th to ensure that
you are in consideration for a place.</p>
<p>The GMOD Summer School is the best way to get to grips with key
components of the GMOD toolkit, including <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Chado#58;//gmod.org/wiki/Chado"
class="mw-redirect" title="Chado">Chado</a>, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GBrowse#58;//gmod.org/wiki/GBrowse"
title="GBrowse">GBrowse</a>, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Galaxy#58;//gmod.org/wiki/Galaxy"
title="Galaxy">Galaxy</a>, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/JBrowse#58;//gmod.org/wiki/JBrowse"
title="JBrowse">JBrowse</a>, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/WebApollo#58;//gmod.org/wiki/WebApollo"
title="WebApollo">WebApollo</a>, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/MAKER#58;//gmod.org/wiki/MAKER"
title="MAKER">MAKER</a>, <a
href="http://gmod.org/wiki/http://gmod.org/wiki/Tripal#58;//gmod.org/wiki/Tripal"
title="Tripal">Tripal</a>, and more. Tool developers will be on hand to
answer questions and help with troubleshooting.</p>
<p>Applications received after June 10th will be kept in case there are
spare places due to accepted applicants not being able to attend.</p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/06/06</em>
</div>
</div>
<div class="rss">
<h3 id="gcc2013-early-reg-closes-24-may"><a
href="News/GCC2013_Early_Reg_closes_24_May">GCC2013 Early Reg closes 24
May</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2013#58;//wiki.galaxyproject.org/Events/GCC2013"
rel="nofollow" title="2013 Galaxy Community Conference">&lt;img
alt="2013 Galaxy Community Conference"
src="http://gmod.org/mediawiki/images/thumb/c/c3/GCC2013Logo.png/200px-GCC2013Logo.png"
width="200" height="133"
srcset="/mediawiki/images/thumb/c/c3/GCC2013Logo.png/300px-GCC2013Logo.png
1.5x, /mediawiki/images/thumb/c/c3/GCC2013Logo.png/400px-GCC2013Logo.png
2x" /&gt;</a>
</div>
<p><a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2013/Register#58;//wiki.galaxyproject.org/Events/GCC2013/Register"
class="external text" rel="nofollow">Early registration</a> for the <a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2013#58;//wiki.galaxyproject.org/Events/GCC2013"
class="external text" rel="nofollow">2013 Galaxy Community Conference
(GCC2013)</a> ends this Friday, 24 May. Early registration <em>saves up
to 75% off</em> regular registration costs, and is downright affordable,
with combined registration (<a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2013/TrainingDay#58;//wiki.galaxyproject.org/Events/GCC2013/TrainingDay"
class="external text" rel="nofollow">Training Day</a> + <a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2013/Program#58;//wiki.galaxyproject.org/Events/GCC2013/Program"
class="external text" rel="nofollow">main meeting</a>) starting at ~ €95
for post-docs and students. Registering early also assures you a spot in
the Training Day workshops you want to attend. Once a Training Day
session becomes full, it will be closed to new registrations.</p>
<p><a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2013#58;//wiki.galaxyproject.org/Events/GCC2013"
class="external text" rel="nofollow">GCC2013</a> is an opportunity to
participate in two full days of presentations, discussions, poster
sessions, keynote, and lightning talks, all about high-throughput
biology and the tools that support it. The conference also includes a
Training Day with in-depth topic coverage of twelve different topics in
15 sessions across 5 tracks.</p>
<p>An <a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2013/Program#58;//wiki.galaxyproject.org/Events/GCC2013/Program"
class="external text" rel="nofollow">early program</a> is available
featuring 25 oral presentations, lightning talks and poster sessions,
and the just added Birds-of-a-Feather gatherings.</p>
<p>If you are a biologist or bioinformatician performing or enabling
high-throughput biological research, then this conference is for
you.</p>
<p>Thanks,</p>
<p><a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2013/Organizers#58;//wiki.galaxyproject.org/Events/GCC2013/Organizers"
class="external text" rel="nofollow">GCC2013 Organizing
Committee</a></p>
<p>PS: And <a
href="http://gmod.org/wiki/http://wiki.galaxyproject.org/Events/GCC2013/Promotion#58;//wiki.galaxyproject.org/Events/GCC2013/Promotion"
class="external text" rel="nofollow">please help get the word
out!</a></p>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/05/20</em>
</div>
</div>
<div class="rss">
<h3 id="new-webapollo-release"><a href="News/New_WebApollo_Release">New
WebApollo Release</a></h3>
<div class="floatright">
<a
href="http://gmod.org/wiki/http://genomearchitect.org/webapollo/releases#58;//genomearchitect.org/webapollo/releases"
rel="nofollow">&lt;img alt="Download the new WebApollo release!"
src="http://gmod.org/mediawiki/images/thumb/4/4a/WebApolloLogo.png/200px-WebApolloLogo.png"
width="200" height="55"
srcset="/mediawiki/images/thumb/4/4a/WebApolloLogo.png/300px-WebApolloLogo.png
1.5x,
/mediawiki/images/thumb/4/4a/WebApolloLogo.png/400px-WebApolloLogo.png
2x" /&gt;</a>
</div>
<p><em><strong>From Ed Lee on the WebApollo mailing
list:</strong></em></p>
<p>We'd like to announce a new WebApollo release.</p>
<p><a
href="http://gmod.org/wiki/http://genomearchitect.org/webapollo/releases#58;//genomearchitect.org/webapollo/releases"
class="external text" rel="nofollow"><strong>Get the newest
release</strong></a> -- <a
href="http://gmod.org/wiki/http://genomearchitect.org/WebApolloDemo#58;//genomearchitect.org/WebApolloDemo"
class="external text" rel="nofollow">Public demo</a> -- <a
href="http://gmod.org/wiki/http://www.gmod.org/wiki/WebApollo_Installation#58;//www.gmod.org/wiki/WebApollo_Installation"
class="external text" rel="nofollow">Server setup instructions</a> -- <a
href="http://gmod.org/wiki/http://genomearchitect.org/webapollo/docs/webapollo_user_guide.pdf#58;//genomearchitect.org/webapollo/docs/webapollo_user_guide.pdf"
class="external text" rel="nofollow">Updated user guide</a></p>
<p>Summary of changes:</p>
<ul>
<li>Fully revamped genomic sequence selection screen
<ul>
<li>sorting by name and length (uses customizable JavaScript
function)</li>
<li>filtering of genomic sequence names</li>
</ul></li>
<li>Annotation info editor
<ul>
<li>allows editing of symbols, comments, and dbxrefs</li>
</ul></li>
<li>Script for bulk loading gene/transcript/exons to annotation
track</li>
<li>Improved login system
<ul>
<li>allows logging in from either genomic region selection screen or
editor</li>
</ul></li>
<li>Configuration for using computed CDS (if available) when first
creating a transcript (rather that using the longest ORF)</li>
<li>Improved memory management</li>
<li>Improved handling of concurrent edits</li>
<li>HTTPS support</li>
<li>Deletion of transcripts now warn users</li>
<li>Accessing editor without logging in hides annotation track</li>
<li>Using "[" and "]" to navigate through subfeatures when a feature in
the annotation track is selected</li>
<li>Improved interface for adding insertions and deletions</li>
<li>Option to hide plus and/or minus strand</li>
<li>Various bugfixes</li>
</ul>
<p><br />
</p>
<div class="newsfooter">
<em>Posted to the <a
href="http://gmod.org/wiki/http://gmod.org/wiki/GMOD_News#58;//gmod.org/wiki/GMOD_News"
title="GMOD News">GMOD News</a> on 2013/05/17</em>
</div>
</div></td>
</tr>
</tbody>
</table>

# <span id="The_Really_Old_News_Archive" class="mw-headline">The Really Old News Archive</span>

Even earlier news items are available at
<a href="http://sourceforge.net/news/?group_id=27707"
class="external text" rel="nofollow">SourceForge</a>.

</div>

<div class="printfooter">

Retrieved from
"<http://gmod.org/mediawiki/index.php?title=GMOD_News_Archives&oldid=24412>"

</div>

<div id="catlinks" class="catlinks">

<div id="mw-normal-catlinks" class="mw-normal-catlinks">

[Categories](Special:Categories "Special:Categories"):

- [GMOD Community](Category:GMOD_Community "Category:GMOD Community")
- <a
  href="http://gmod.org/mediawiki/index.php?title=Category:!FixAfterUpgrade&amp;action=edit&amp;redlink=1"
  class="new"
  title="Category:!FixAfterUpgrade (page does not exist)">!FixAfterUpgrade</a>

</div>

</div>

<div class="visualClear">

</div>

</div>

</div>

<div id="mw-navigation">

## Navigation menu

<div id="mw-head">



<div id="left-navigation">

<div id="p-namespaces" class="vectorTabs" role="navigation"
aria-labelledby="p-namespaces-label">

### Namespaces

- <span id="ca-nstab-main"><a href="GMOD_News_Archives" accesskey="c"
  title="View the content page [c]">Page</a></span>
- <span id="ca-talk"><a
  href="http://gmod.org/mediawiki/index.php?title=Talk:GMOD_News_Archives&amp;action=edit&amp;redlink=1"
  accesskey="t"
  title="Discussion about the content page [t]">Discussion</a></span>

</div>

<div id="p-variants" class="vectorMenu emptyPortlet" role="navigation"
aria-labelledby="p-variants-label">

### 

### Variants[](#)

<div class="menu">

</div>

</div>

</div>

<div id="right-navigation">

<div id="p-views" class="vectorTabs" role="navigation"
aria-labelledby="p-views-label">

### Views

- <span id="ca-view">[Read](GMOD_News_Archives)</span>
- <span id="ca-viewsource"><a
  href="http://gmod.org/mediawiki/index.php?title=GMOD_News_Archives&amp;action=edit"
  accesskey="e" title="This page is protected.
  You can view its source [e]">View source</a></span>
- <span id="ca-history"><a
  href="http://gmod.org/mediawiki/index.php?title=GMOD_News_Archives&amp;action=history"
  accesskey="h" title="Past revisions of this page [h]">View history</a></span>

</div>

<div id="p-cactions" class="vectorMenu emptyPortlet" role="navigation"
aria-labelledby="p-cactions-label">

### Actions[](#)

<div class="menu">

</div>

</div>

<div id="p-search" role="search">

### Search

<div id="simpleSearch">

</div>

</div>

</div>

</div>

<div id="mw-panel">

<div id="p-logo" role="banner">

<a href="Main_Page"
style="background-image: url(../images/GMOD-cogs.png);"
title="Visit the main page"></a>

</div>

<div id="p-Navigation" class="portal" role="navigation"
aria-labelledby="p-Navigation-label">

### Navigation

<div class="body">

- <span id="n-GMOD-Home">[GMOD Home](Main_Page)</span>
- <span id="n-Software">[Software](GMOD_Components)</span>
- <span id="n-Categories-.2F-Tags">[Categories /
  Tags](Categories)</span>
- <span id="n-View-all-pages">[View all pages](Special:AllPages)</span>

</div>

</div>

<div id="p-Documentation" class="portal" role="navigation"
aria-labelledby="p-Documentation-label">

### Documentation

<div class="body">

- <span id="n-Overview">[Overview](Overview)</span>
- <span id="n-FAQs">[FAQs](Category:FAQ)</span>
- <span id="n-HOWTOs">[HOWTOs](Category:HOWTO)</span>
- <span id="n-Glossary">[Glossary](Glossary)</span>

</div>

</div>

<div id="p-Community" class="portal" role="navigation"
aria-labelledby="p-Community-label">

### Community

<div class="body">

- <span id="n-GMOD-News">[GMOD News](GMOD_News)</span>
- <span id="n-Training-.2F-Outreach">[Training /
  Outreach](Training_and_Outreach)</span>
- <span id="n-Support">[Support](Support)</span>
- <span id="n-GMOD-Promotion">[GMOD Promotion](GMOD_Promotion)</span>
- <span id="n-Meetings">[Meetings](Meetings)</span>
- <span id="n-Calendar">[Calendar](Calendar)</span>

</div>

</div>

<div id="p-tb" class="portal" role="navigation"
aria-labelledby="p-tb-label">

### Tools

<div class="body">

- <span id="t-whatlinkshere"><a href="Special:WhatLinksHere/GMOD_News_Archives" accesskey="j"
  title="A list of all wiki pages that link here [j]">What links here</a></span>
- <span id="t-recentchangeslinked"><a href="Special:RecentChangesLinked/GMOD_News_Archives" accesskey="k"
  title="Recent changes in pages linked from this page [k]">Related
  changes</a></span>
- <span id="t-specialpages"><a href="Special:SpecialPages" accesskey="q"
  title="A list of all special pages [q]">Special pages</a></span>
- <span id="t-print"><a
  href="http://gmod.org/mediawiki/index.php?title=GMOD_News_Archives&amp;printable=yes"
  rel="alternate" accesskey="p"
  title="Printable version of this page [p]">Printable version</a></span>
- <span id="t-permalink">[Permanent
  link](http://gmod.org/mediawiki/index.php?title=GMOD_News_Archives&oldid=24412 "Permanent link to this revision of the page")</span>
- <span id="t-info">[Page
  information](http://gmod.org/mediawiki/index.php?title=GMOD_News_Archives&action=info)</span>
- <span id="t-smwbrowselink"><a href="Special:Browse/GMOD_News_Archives" rel="smw-browse">Browse
  properties</a></span>
- <span id="t-pdf">[Print as
  PDF](http://gmod.org/mediawiki/index.php?title=Special:PdfPrint&page=GMOD_News_Archives)</span>

</div>

</div>

</div>

</div>

<div id="footer" role="contentinfo">

- <span id="footer-info-lastmod">Last updated at 11:12 on 22 August
  2013.</span>
- <span id="footer-info-viewcount">515,360 page views.</span>
- <span id="footer-info-copyright">Content is available under
  <a href="http://www.gnu.org/licenses/fdl-1.3.html" class="external"
  rel="nofollow">a GNU Free Documentation License</a> unless otherwise
  noted.</span>

<!-- -->

- <span id="footer-places-about">[About
  GMOD](GMOD:About "GMOD:About")</span>

<!-- -->

- <span id="footer-copyrightico">[<img src="http://www.gnu.org/graphics/gfdl-logo-small.png" width="88"
  height="31" alt="a GNU Free Documentation License" />](http://www.gnu.org/licenses/fdl-1.3.html)</span>
- <span id="footer-poweredbyico">[<img
  src="../mediawiki/skins/common/images/poweredby_mediawiki_88x31.png"
  width="88" height="31" alt="Powered by MediaWiki" />](http://www.mediawiki.org/)
  [<img
  src="../mediawiki/extensions/SemanticMediaWiki/resources/images/smw_button.png"
  width="88" height="31" alt="Powered by Semantic MediaWiki" />](https://www.semantic-mediawiki.org/wiki/Semantic_MediaWiki)</span>

<div style="clear:both">

</div>

</div>
