









<span id="top"></span>







# <span dir="auto">July 2008 GMOD Meeting</span>





From GMOD









Jump to: [navigation](#mw-navigation), [search](#p-search)





<table
style="font-size: 180%; vertical-align: middle; border: 2px solid #A6A6BC; line-height: 120%"
data-cellpadding="10">
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<tbody>
<tr class="odd">
<td>

<a href="File:July2008LogoSmaller.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/3d/July2008LogoSmaller.png" width="200"
height="180" alt="July2008LogoSmaller.png" /></a>

</td>
<td data-valign="center"><strong>GMOD Community Meeting</strong><br />
<br />
July 16-17, 2008<br />
&#10;<p>University of Toronto<br />
Toronto, Ontario, Canada</p></td>
</tr>
</tbody>
</table>

  
The July 2008 [GMOD community meeting](Meetings "Meetings") was held on
July 16-17, 2008 at the University of Toronto, immediately before
<a href="http://www.iscb.org/ismb2008/" class="external text"
rel="nofollow">BOSC and ISMB 2008</a> (also in Toronto), and just a few
days after the [2008 GMOD Summer
School](2008_GMOD_Summer_School "2008 GMOD Summer School"). The meeting
was attended by over 30 people representing more than 20 different
groups.





## Contents



- [<span class="tocnumber">1</span>
  <span class="toctext">Agenda</span>](#Agenda)
  - [<span class="tocnumber">1.1</span> <span class="toctext">July
    16</span>](#July_16)
  - [<span class="tocnumber">1.2</span> <span class="toctext">July
    17</span>](#July_17)
- [<span class="tocnumber">2</span>
  <span class="toctext">Attendees</span>](#Attendees)
- [<span class="tocnumber">3</span> <span class="toctext">GMOD
  Components</span>](#GMOD_Components)
  - [<span class="tocnumber">3.1</span>
    <span class="toctext">Chado</span>](#Chado)
    - [<span class="tocnumber">3.1.1</span>
      <span class="toctext">Companalysis
      Module</span>](#Companalysis_Module)
    - [<span class="tocnumber">3.1.2</span>
      <span class="toctext">Natural Diversity
      Module</span>](#Natural_Diversity_Module)
  - [<span class="tocnumber">3.2</span> <span class="toctext">Community
    Annotation System</span>](#Community_Annotation_System)
    - [<span class="tocnumber">3.2.1</span>
      <span class="toctext">cas-utils 0.1</span>](#cas-utils_0.1)
  - [<span class="tocnumber">3.3</span>
    <span class="toctext">TableEdit</span>](#TableEdit)
  - [<span class="tocnumber">3.4</span>
    <span class="toctext">Apollo</span>](#Apollo)
  - [<span class="tocnumber">3.5</span>
    <span class="toctext">InterMine</span>](#InterMine)
  - [<span class="tocnumber">3.6</span>
    <span class="toctext">CMap</span>](#CMap)
  - [<span class="tocnumber">3.7</span>
    <span class="toctext">GBrowse</span>](#GBrowse)
    - [<span class="tocnumber">3.7.1</span>
      <span class="toctext">GBrowse 1.69</span>](#GBrowse_1.69)
    - [<span class="tocnumber">3.7.2</span>
      <span class="toctext">GBrowse 2</span>](#GBrowse_2)
    - [<span class="tocnumber">3.7.3</span>
      <span class="toctext">GBrowse 3</span>](#GBrowse_3)
      - [<span class="tocnumber">3.7.3.1</span>
        <span class="toctext">Genome Wiki</span>](#Genome_Wiki)
      - [<span class="tocnumber">3.7.3.2</span>
        <span class="toctext">Approximate
        Schedule</span>](#Approximate_Schedule)
    - [<span class="tocnumber">3.7.4</span>
      <span class="toctext">GBrowse Glyphs
      Page</span>](#GBrowse_Glyphs_Page)
  - [<span class="tocnumber">3.8</span> <span class="toctext">Common
    Gene Page</span>](#Common_Gene_Page)
    - [<span class="tocnumber">3.8.1</span> <span class="toctext">Now
      What?</span>](#Now_What.3F)
    - [<span class="tocnumber">3.8.2</span> <span class="toctext">Gene
      Wiki</span>](#Gene_Wiki)
  - [<span class="tocnumber">3.9</span> <span class="toctext">MediaWiki
    Enhancements</span>](#MediaWiki_Enhancements)
    - [<span class="tocnumber">3.9.1</span>
      <span class="toctext">FCKEditor</span>](#FCKEditor)
    - [<span class="tocnumber">3.9.2</span> <span class="toctext">Popup
      Balloons</span>](#Popup_Balloons)
    - [<span class="tocnumber">3.9.3</span>
      <span class="toctext">Collapsible Sections
      Extension</span>](#Collapsible_Sections_Extension)
    - [<span class="tocnumber">3.9.4</span>
      <span class="toctext">Predefined Page Creation
      Extensions</span>](#Predefined_Page_Creation_Extensions)
  - [<span class="tocnumber">3.10</span>
    <span class="toctext">APIs</span>](#APIs)
    - [<span class="tocnumber">3.10.1</span> <span class="toctext">Perl
      based Schema Abstraction Layer for
      Chado</span>](#Perl_based_Schema_Abstraction_Layer_for_Chado)
      - [<span class="tocnumber">3.10.1.1</span>
        <span class="toctext">Discussion</span>](#Discussion)
    - [<span class="tocnumber">3.10.2</span> <span class="toctext">Chado
      Java API</span>](#Chado_Java_API)
- [<span class="tocnumber">4</span> <span class="toctext">GMOD User
  Community</span>](#GMOD_User_Community)
  - [<span class="tocnumber">4.1</span>
    <span class="toctext">SGN</span>](#SGN)
  - [<span class="tocnumber">4.2</span>
    <span class="toctext">WikiMods.org</span>](#WikiMods.org)
  - [<span class="tocnumber">4.3</span>
    <span class="toctext">CellFrame</span>](#CellFrame)
  - [<span class="tocnumber">4.4</span>
    <span class="toctext">Xenbase</span>](#Xenbase)
  - [<span class="tocnumber">4.5</span> <span class="toctext">Centre for
    Molecular and Biomolecular
    Informatics</span>](#Centre_for_Molecular_and_Biomolecular_Informatics)
- [<span class="tocnumber">5</span> <span class="toctext">GMOD
  Project</span>](#GMOD_Project)
  - [<span class="tocnumber">5.1</span> <span class="toctext">Google
    Summer of Code</span>](#Google_Summer_of_Code)
  - [<span class="tocnumber">5.2</span>
    <span class="toctext">Packages</span>](#Packages)
  - [<span class="tocnumber">5.3</span> <span class="toctext">GMOD Help
    Desk</span>](#GMOD_Help_Desk)
    - [<span class="tocnumber">5.3.1</span> <span class="toctext">What's
      Been Done</span>](#What.27s_Been_Done)
    - [<span class="tocnumber">5.3.2</span> <span class="toctext">What's
      Planned</span>](#What.27s_Planned)
      - [<span class="tocnumber">5.3.2.1</span>
        <span class="toctext">GMOD User
        Directory</span>](#GMOD_User_Directory)
      - [<span class="tocnumber">5.3.2.2</span>
        <span class="toctext">User Experience
        Logs</span>](#User_Experience_Logs)
      - [<span class="tocnumber">5.3.2.3</span>
        <span class="toctext">Documentation</span>](#Documentation)
      - [<span class="tocnumber">5.3.2.4</span>
        <span class="toctext">GMOD Logo
        Service</span>](#GMOD_Logo_Service)
      - [<span class="tocnumber">5.3.2.5</span>
        <span class="toctext">Education and
        Outreach</span>](#Education_and_Outreach)
      - [<span class="tocnumber">5.3.2.6</span>
        <span class="toctext">Grants</span>](#Grants)
- [<span class="tocnumber">6</span> <span class="toctext">Agenda
  Proposals</span>](#Agenda_Proposals)



# <span id="Agenda" class="mw-headline">Agenda</span>

## <span id="July_16" class="mw-headline">July 16</span>

| Time | Topic | Who | Presentation |
|----|----|----|----|
| 9:00 | Introductions | [Scott Cain](User%3AScott "User%3AScott") |  |
| 9:30 | The State of GMOD | Scott Cain | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/f/fb/Gmod_meeting_7_2008.ppt"
class="internal" title="Gmod meeting 7 2008.ppt">PPT</a> |
| 10:10 | break |  |  |
| 10:40 | MediaWiki/[TableEdit](TableEdit.1 "TableEdit") Roundtripping | [Jim Hu](User%3AJimHu.1 "User%3AJimHu") |  |
| 11:00 | More MediaWiki enhancements | [Sheldon McKay](User%3AMckays "User%3AMckays") | [Links...](User%3AMckays#Links "User%3AMckays") |
|  | [SGN](Category%3ASGN "Category%3ASGN") Community Annotation | Lukas Mueller | <a
href="http://gmod.org/mediawiki/index.php?title=Special:Upload&amp;wpDestFile=SGNGMOD2008.pdf"
class="new" title="SGNGMOD2008.pdf">PDF</a> |
|  | WikiMods & Chado API | Brad Arshinoff | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/6/65/PerlChadoGMOD2008.pdf"
class="internal" title="PerlChadoGMOD2008.pdf">PDF</a> |
| 11:30 | Lunch |  |  |
| 1:30 | [GMOD Help Desk](GMOD_Help_Desk "GMOD Help Desk") | [Dave Clements](User%3AClements "User%3AClements") | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/1/19/HelpDeskGMOD2008.pdf" class="internal"
title="HelpDeskGMOD2008.pdf">PDF</a> <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/8/88/HelpDeskGMOD2008.ppt" class="internal"
title="HelpDeskGMOD2008.ppt">PPT</a> |
| 2:15 | Rearchitecting [Apollo](Apollo.1 "Apollo") and the need for a database independent Biological API layer | Ed Lee |  |
| 2:50 | break |  |  |
| 3:20 | [InterMine](InterMine "InterMine") and <a href="Chado" class="mw-redirect" title="Chado">Chado</a> | Richard Smith | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/b/b3/IntermineGMOD2008.pdf"
class="internal" title="IntermineGMOD2008.pdf">PDF</a> |
| 3:50 | Show and Tell | "What I did with my Summer" |  |
|  | [CMap](CMap.1 "CMap") | [Ben Faga](User%3AFaga "User%3AFaga") |  |

## <span id="July_17" class="mw-headline">July 17</span>

| Time | Topic | Who | Presentation |
|----|----|----|----|
| 9:00 | New things for [GBrowse](GBrowse.1 "GBrowse") 1.69 | [Sheldon McKay](User%3AMckays "User%3AMckays") |  |
|  | GBrowse 2.0 and Roadmap | [Lincoln Stein](User%3ALstein "User%3ALstein") |  |
| 9:30 | New things for GBrowse 3.0 | Ian Holmes | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/2/23/GBrowse3GMOD2008.pdf" class="internal"
title="GBrowse3GMOD2008.pdf">PDF</a> |
| 10:00 | break |  |  |
| 10:30 | The need for a computable common gene page ([Don Gilbert's proposal](Common_Gene_Page "Common Gene Page")) | Scott Cain, Lincoln Stein | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/8/80/Common_gene_page.ppt" class="internal"
title="Common gene page.ppt">PPT</a> |
| 11:30 | Lunch |  |  |
| 1:30 | More Show and Tell or a mini hackathon or go see Toronto |  |  |
|  | Traits at SGN | Lukas Mueller |  |
|  | CellFrame | Yunchen Gong |  |
|  | Matching Gene Names to Articles at Xenbase | Jeff Bowes |  |
|  | [Django and Chado](Chado_Django_HOWTO "Chado Django HOWTO") - A user interface exploration | Victor de Jager |  |

# <span id="Attendees" class="mw-headline">Attendees</span>





<a href="File:GMOD2008Attendees.JPG" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/3d/GMOD2008Attendees.JPG" width="400"
height="148" alt="GMOD2008Attendees.JPG" /></a>





1.  David Arcoleo -
    <a href="http://beespace.uiuc.edu" class="external text"
    rel="nofollow">BeeSpace</a>, University of Illinois
2.  Brad Arshinoff -
    <a href="http://xanthusbase.org" class="external text"
    rel="nofollow">XanthusBase</a>
3.  Jeff Bowes - <a href="http://xenbase.org" class="external text"
    rel="nofollow">Xenbase</a>
4.  [Robert Buels](User%3ARobertBuels "User%3ARobertBuels") -
    <a href="http://www.sgn.cornell.edu" class="external text"
    rel="nofollow">Sol Genomics Network</a>
    ([SGN](Category%3ASGN "Category%3ASGN"))
5.  [Scott Cain](User%3AScott "User%3AScott") - GMOD
6.  [Dave Clements](User%3AClements "User%3AClements") -
    <a href="http://nescent.org" class="external text"
    rel="nofollow">NESCent</a>, GMOD
7.  Sean Davey - BirdBase, U of Arizona
8.  Victor de Jager
    <a href="http://www.cmbi.ru.nl" class="external text"
    rel="nofollow">University of Nijmegen, The Netherlands &amp; Centre for
    Molecular and Biomolecular Informatics</a>
9.  Mary E Dolan
    <a href="http://www.informatics.jax.org/" class="external text"
    rel="nofollow">Mouse Genome Informatics, The Jackson Laboratory</a>
10. [Ben Faga](User%3AFaga "User%3AFaga") - CSHL
11. Yunchen Gong - <a href="http://utoronto.ca" class="external text"
    rel="nofollow">University of Toronto</a>
12. [Josh Goodman](User%3AJogoodma "User%3AJogoodma") -
    [FlyBase](Category%3AFlyBase "Category%3AFlyBase")
13. [Todd Harris](User%3ATharris "User%3ATharris") -
    <a href="http://www.wormbase.org/" class="external text"
    rel="nofollow">WormBase</a>
14. [Chris Hemmerich](User%3AChemmeri "User%3AChemmeri") - Center for
    Genomics and Bioinformatics, Indiana U.
15. <a href="http://biowiki.org/IanHolmes" class="external text"
    rel="nofollow">Ian Holmes</a> - UC Berkeley
16. [Jim Hu](User%3AJimHu.1 "User%3AJimHu") -
    <a href="http://ecoliwiki.net/" class="external text"
    rel="nofollow">EcoliWiki</a>, Texas A&M
17. Thomas Keane -
    <a href="http://www/sanger.ac.uk" class="external text"
    rel="nofollow">Wellcome Trust Sanger Institute</a>
18. [Ed Lee](User%3AElee "User%3AElee") -
    <a href="http://www.berkeleybop.org" class="external text"
    rel="nofollow">BBOP</a> and [Apollo](Apollo.1 "Apollo")
19. Suzi Lewis - BBOP
20. Margie Manker -
    <a href="http://populargenetics.ca" class="external text"
    rel="nofollow">The Centre for Applied Genomics, Toronto</a>
21. [Sheldon McKay](User%3AMckays "User%3AMckays") -
    <a href="http://modencode.org" class="external text"
    rel="nofollow">modENCODE</a>,
    [WormBase](Category%3AWormBase "Category%3AWormBase")
22. Lukas Mueller
    <a href="http://www.sgn.cornell.edu" class="external text"
    rel="nofollow">Sol Genomics Network</a>
    ([SGN](Category%3ASGN "Category%3ASGN"))
23. Brian O'Connor -
    <a href="http://ucla.edu" class="external text" rel="nofollow">UCLA</a>
24. [Joshua Orvis](User%3AJorvis "User%3AJorvis") -
    <a href="http://www.igs.umaryland.edu" class="external text"
    rel="nofollow">Institute for Genome Sciences</a>, University of
    Maryland
25. Barry Sanders -
    <a href="http://beespace.uiuc.edu" class="external text"
    rel="nofollow">BeeSpace</a>, University of Illinois
26. Stéphanie Sidibe Bocs -
    <a href="http://cirad.fr" class="external text" rel="nofollow">CIRAD</a>
27. [Richard Smith](User%3ARsmith "User%3ARsmith") -
    <a href="http://www.intermine.org" class="external text"
    rel="nofollow">InterMine</a> (and
    [InterMine](InterMine "InterMine"))
28. Kevin Snyder -
    <a href="http://www.xenbase.org" class="external text"
    rel="nofollow">Xenbase</a>; University Of Calgary
29. Jason Stajich - UC Berkeley
30. Haiyan Zhang - [FlyBase](Category%3AFlyBase "Category%3AFlyBase")
31. [Junjun Zhang](User%3AJunjun "User%3AJunjun") -
    <a href="http://www.tcag.ca" class="external text" rel="nofollow">The
    Hospital for Sick Children, Toronto</a>





<a href="File:GMOD2008Discussion.JPG" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/31/GMOD2008Discussion.JPG" width="245"
height="148" alt="GMOD2008Discussion.JPG" /></a>





# <span id="GMOD_Components" class="mw-headline">GMOD Components</span>

This section covers discussion about the [software
components](GMOD_Components "GMOD Components") in GMOD. For a summary of
talks and discussion on how those components are used at particular
databases, see the [GMOD User Community](#GMOD_User_Community) section.

## <span id="Chado" class="mw-headline">Chado</span>

[Scott Cain](User%3AScott "User%3AScott") spoke on
<a href="Chado" class="mw-redirect" title="Chado">Chado</a>.

The GMOD 1.1 release is in the works. There are no schema changes yet.

### <span id="Companalysis_Module" class="mw-headline">Companalysis Module</span>

Joshua Orvis requested better typing / the use of controlled
vocabularies in the [Chado Companalysis
Module](Chado_Companalysis_Module "Chado Companalysis Module") to better
represent scores that are currently in the [analysisfeature
table](Chado_Companalysis_Module#Table:_analysisfeature "Chado Companalysis Module").
Without it there is no way to keep track of what the scores mean. This
issue was also raised by Brett Whitty at the [2008 GMOD Summer
School](2008_GMOD_Summer_School "2008 GMOD Summer School") the week
before.

Also, Joshua (again) proposed the addition of a type_id field to the
analysisfeature table. The use case for this is to allow the distinction
between types of features involved in an analysis. The most direct
examples are 'input_of' and 'created_by' which allow the user to perform
queries of a features role in the analysis. This has been brought up in
previous meetings and in the GMOD mailing list and seems to have had
general approval.

**Action Items**

- get Chris Mungall's input on these issues.

### <span id="Natural_Diversity_Module" class="mw-headline">Natural Diversity Module</span>

[Dave Clements](User%3AClements "User%3AClements") discussed the Chado
Natural Diversity Module. It was developed at
<a href="http://nescent.org" class="external text"
rel="nofollow">NESCent</a> and NCSU to enable
<a href="Chado" class="mw-redirect" title="Chado">Chado</a> to better
support natural diversity studies. This has been laying dormant for a
while and it would be nice to get it in use by more groups so that we
can better generalize it and make it an official Chado module. Lincoln
pointed out there was grant money to do exactly this.

**Action Items:**

- Dave will follow up with Lincoln.

## <span id="Community_Annotation_System" class="mw-headline">Community Annotation System</span>

[Scott Cain](User%3AScott "User%3AScott") also spoke about his work on the
[Community Annotation
System](Community_Annotation_System "Community Annotation System")
(CAS). The next release of CAS, 1.1, will feature

- Switch from Ubuntu to CentOS.
- [GMODWeb](GMODWeb "GMODWeb") will be included.
- cas-utils 0.1

### <span id="cas-utils_0.1" class="mw-headline">cas-utils 0.1</span>

cas-utils is a set of tools that tie together
[GBrowse](GBrowse.1 "GBrowse"), [Apollo](Apollo.1 "Apollo") and
<a href="Chado" class="mw-redirect" title="Chado">Chado</a>. This
includes

- A CGI for selecting a region in GBrowse, extracting the data for that
  region from Apollo and creating an [XML](Category%3AXML "Category%3AXML")
  file and a jnlp (webstart) file for Apollo.
- A CGI for accepting uploads of edited XML files to either be
  immediately loaded into Chado or to be held for validation.
- A configuration Perl module to make modifying the CGI's behavior easy.
- A Module::Build based installer that queries the user for needed setup
  data.

cas-utils is now available for <a
href="http://sourceforge.net/project/showfiles.php?group_id=27707&amp;package_id=284978"
class="external text" rel="nofollow">download</a>.

## <span id="TableEdit" class="mw-headline">TableEdit</span>

[Jim Hu](User%3AJimHu.1 "User%3AJimHu") spoke about progress on
[TableEdit](TableEdit.1 "TableEdit"), currently at release 0.8.

- Now refuses edits until user entered HTML tags are closed, thus
  avoiding nasty side effects.
- Round trip between MediaWiki and
  <a href="Chado" class="mw-redirect" title="Chado">Chado</a> is not yet
  done.
  - Chado to MediaWiki is done, vice versa is not.

## <span id="Apollo" class="mw-headline">Apollo</span>

Ed Lee, lead developer for [Apollo](Apollo.1 "Apollo") spoke about
enhancements to Apollo that have happened since he started working on it
last September:

- undo function
- preferences editor
- Chado adapter enhancements
- Improved graph and GFF3 support

## <span id="InterMine" class="mw-headline">InterMine</span>

Richard Smith spoke about [InterMine](InterMine "InterMine"), a query
optimized data warehouse system for biological data. Has the ability to
create precomputed tables (a la materialized views) at any time (and do
this from the GUI) in response to popular query patterns. Also supports
query templates, which are fill-in-the blank versions of popular
queries.

InterMine is written in [Java](Category%3AJava "Category%3AJava"). It has
one class per Sequence Ontology (SO) term, and use Java class
inheritance for *is_a* relationships. *part_of* relationships are
implemented with Java references and collections.

  

## <span id="CMap" class="mw-headline">CMap</span>

[Ben Faga](User%3AFaga "User%3AFaga") gave a talk on what's new in
[CMap](CMap.1 "CMap"). Some highlights:

- CMap 1.0 came out in March 2008.
- ribbon displays for syntenic blocks
- dotplot displays
- new feature glyphs
- embeddable image generation
- directory guessing for easier installation.

  

## <span id="GBrowse" class="mw-headline">GBrowse</span>

Three talks gave us the [GBrowse](GBrowse.1 "GBrowse") roadmap. Talks
covered the next incremental release (1.69), and the next two major
releases (2 and 3).

### <span id="GBrowse_1.69" class="mw-headline">GBrowse 1.69</span>

[Sheldon McKay](User%3AMckays "User%3AMckays") and [Lincoln
Stein](User%3ALstein "User%3ALstein") spoke about recent enhancements to
[GBrowse](GBrowse.1 "GBrowse"). These features are available in the
current development version ("stable") of GBrowse and will be included
in the upcoming (some would say imminent) 1.69 release of GBrowse.

- Wiggle - Dense quantitative tracks, density can have colored peaks,
  and go below 0.
- Quantitative (BP resolution) data.
- Inline track configuration.
- Design Primers
- Rubberbanding
- Popup windows (with a nice example showing
  [WormBase](Category%3AWormBase "Category%3AWormBase") anatomy cartoons)
- Draggable tracks
- Easy-share tracks
  - DAS server is inside GBrowse. Also a web service.
  - Can now have one GBrowse server share a track with another GBrowse
    server.
  - Data is transferred on the fly, as the user naviages the genome.
  - Can form chains of sharing.
- [Galaxy](Galaxy.1 "Galaxy") Integration.
  - Within Galaxy click on the get data link.
  - Lists data sources including BioMart and WormBase GBrowse.
- Multiple Alignment Format (MAF) and conservation tracks.

### <span id="GBrowse_2" class="mw-headline">GBrowse 2</span>

[Lincoln Stein](User%3ALstein "User%3ALstein") talked about GBrowse 2, the
next major release of GBrowse. This release focuses on performance and
stability. GBrowse 2 will be cluster aware:

- Tracks can be assigned to read data from specific data servers, and
  render tracks using specific render servers.
- Assignment of machines as data and/or render servers is configurable.
  - A server can be a data server or a render server or both,
  - A track may have multiple data and render servers.
  - A single node can serve data and rendering for one or more tracks.
- Tracks loaded with [AJAX](Category%3AAJAX "Category%3AAJAX"). Grayed out
  until they load.
- Turning tracks on and off no longer requires a reload.

Our experience is that the database is usually the bottleneck with
existing GBrowse installations.

- Can also enable editing of feature comments.

### <span id="GBrowse_3" class="mw-headline">GBrowse 3</span>

*GBrowse 3 was renamed [JBrowse](JBrowse.1 "JBrowse") after this
meeting.*

Ian Holmes presented his group's work on [GBrowse
3](JBrowse.1 "JBrowse"), a complete rewrite of
[GBrowse](GBrowse.1 "GBrowse") using a Web 2.0 style interface. Mitch
Skinner has done most of the coding work on this.

Most tracks are now rendered in client using JavaScript. Tracks such as
wiggle tracks can also still be rendered on the server.

GBrowse 3 uses nested containment lists to quickly determine what
features to display. These are 5 to 500x faster than R-trees. The group
is using the <a href="http://modencode.org" class="external text"
rel="nofollow">modENCODE project</a> as a target test audience.

Ian made the observation that when you are asking for guidance on GUIs,
you need large sample sizes. Small sample sizes lead to a large set of
suggestions with very little overlap between users. Large sample sizes
enables you to identify a core set of requests.

#### <span id="Genome_Wiki" class="mw-headline">Genome Wiki</span>

Ian would like to move GBrowse 3 in the direction of being a *genome
wiki*

- Upload tracks and track sharing
- Ability to add comments, ratings, ...
- Requires user management

Genome Wiki is about people sharing tracks, not so much about individual
genes.

#### <span id="Approximate_Schedule" class="mw-headline">Approximate Schedule</span>

- 2008?: A Lightweight AJAX Genome Browser
- 2009?: An AJAX Genome Wiki

They are not currently working on a
<a href="Chado" class="mw-redirect" title="Chado">Chado</a> adaptor.
They hope to do that, but probably not soon.

### <span id="GBrowse_Glyphs_Page" class="mw-headline">GBrowse Glyphs Page</span>

At the [2008 GMOD Summer
School](2008_GMOD_Summer_School "2008 GMOD Summer School") there were
several requests for a GBrowse glyphs page that

- shows what the glyphs look like,
- what track you might use it with,
- links to any other documentation on the glyph.

Lincoln believes that there is already similar documentation in the
GBrowse distribution.

**Action Items:**

- Dave will investigate further.

## <span id="Common_Gene_Page" class="mw-headline">Common Gene Page</span>

[Scott Cain](User%3AScott "User%3AScott") eerily yet effectively channeled
[Don Gilbert](User%3ADongilbert "User%3ADongilbert") on the topic of a
[Common Gene Page](Common_Gene_Page "Common Gene Page").

This not the gene page that people see when they come to your web site.
Rather, it is some minimal set of information about a gene in your
organism, stored in XML format, that can be easily accessed and parsed
by other organizations. It is meant to enable easy sharing of
information about genes between GMOD users.

If you've been around GMOD for a while you know that the concept of a
common gene page is almost as old as GMOD itself. We *might* have
actually moved forward on this at this meeting.

There was discussion on what should be included in the gene page. The
consensus was to keep track of only the minimal amount of information,
See
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/8/80/Common_gene_page.ppt" class="internal"
title="Common gene page.ppt">Scott's presentation</a> for the list we
settled on.

Uniprot [XML](Glossary#XML "Glossary") may be suitable for this.

### <span id="Now_What.3F" class="mw-headline">Now What?</span>

Lincoln proposed a CGI script that has a set of predefined hooks for
populating the XML. This could be a Perl program with methods for
fetching data and then passing it to another routine for placing the
data into an XML format. Each organization would write the classes
called by the hooks to get the data from wherever they keep it. Provides
a framework that can be used across mutliple organizations and that will
always produce structurally identical XML, no matter how it is
originally stored.

Rob Buells from [SGN](Category%3ASGN "Category%3ASGN") produced a prototype
of this program while at the meeting.

**Action Items:**

-  ?

### <span id="Gene_Wiki" class="mw-headline">Gene Wiki</span>

We also discussed the
<a href="http://en.wikipedia.org/wiki/Gene_Wiki" class="extiw"
title="wp:Gene Wiki">Gene Wiki project</a>. This project has created
around 7,000 human gene pages in Wikipedia. Wikipedia asked

- Only *interesting* genes have pages. Interesting was defined as any
  gene with at least one PubMed reference.
- The pages be easy to edit. Moved some nasty tables to the end of the
  page.

Someone might eventually be able to create a MODGeneWiki from GMOD
Common Web Pages.

## <span id="MediaWiki_Enhancements" class="mw-headline">MediaWiki Enhancements</span>

[Sheldon McKay](User%3AMckays "User%3AMckays") spoke about MediaWiki related
work he's been doing for the
<a href="http://www.modencode.org" class="external text"
rel="nofollow">modENCODE project</a>.

### <span id="FCKEditor" class="mw-headline">FCKEditor</span>

<a href="http://mediawiki.fckeditor.net/" class="external text"
rel="nofollow">FCKEditor</a> is a WYSIWIG editor for MediaWiki, but if
you use it off the shelf it becomes hard for your users to use any other
editor, including the default MediaWiki editor, which they may already
be familiar with. Sheldon has extended FCKEditor to make it optional.
Users now see "edit" "rich edit" links and tabs.

**Action Items:**

- Dave will investigate FCKEditor and the modified version for use in
  the GMOD web site.

### <span id="Popup_Balloons" class="mw-headline">Popup Balloons</span>

Sheldon has also created an extension for creating popup balloons in a
MediaWiki Web Site. See [Popup
Balloons](Popup_Balloons "Popup Balloons") for details. This extension
is installed on the GMOD web site.

### <span id="Collapsible_Sections_Extension" class="mw-headline">Collapsible Sections Extension</span>

Does what it says - enables users to collapse and expand sections on
pages in MediaWiki.

### <span id="Predefined_Page_Creation_Extensions" class="mw-headline">Predefined Page Creation Extensions</span>

A set of extensions were created to

- automatically populate pages based on what type of page is being
  created
- generate forms to help users fully populate pages with required
  fields.

These use the Yahoo autocomplettion library.

  

## <span id="APIs" class="mw-headline">APIs</span>

### <span id="Perl_based_Schema_Abstraction_Layer_for_Chado" class="mw-headline">Perl based Schema Abstraction Layer for Chado</span>

Brad Arshinoff from
<a href="http://xanthusbase.org" class="external text"
rel="nofollow">XanthusBase</a>, (soon to be
<a href="http://wikimods.org" class="external text"
rel="nofollow">WikiMods</a>, [see below](#WikiMods.org)) gave a talk
titled *Perl based Schema Abstraction Layer for
<a href="Chado" class="mw-redirect" title="Chado">Chado</a>*. Brad's <a
href="http://gmod.org/mediawiki/index.php?title=Special:Upload&amp;wpDestFile=SGNGMOD2008.pdf"
class="new" title="SGNGMOD2008.pdf">talk</a> gave an overview of a
[Perl](Category%3APerl "Category%3APerl")
[middleware](Category%3AMiddleware "Category%3AMiddleware") package for
<a href="Chado" class="mw-redirect" title="Chado">Chado</a> that was
developed at XanthusBase.

#### <span id="Discussion" class="mw-headline">Discussion</span>

**Q:** [Modware](Modware "Modware") is a Perl-based Chado API that
already exists. Why not use it?

**A:** Thought this would be less work and a lot less
[SQL](Glossary#SQL "Glossary") than Modware. May or may not have worked
out that way.

Eric Just, the developer of Modware, is no longer at
[DictyBase](Category%3ADictyBase "Category%3ADictyBase"). Someone has
replaced him, but we don't know if that person is supporting Modware.

It seems that we have a lot of Perl and Java APIs to Chado, perhaps too
many. What should we do about that? [Lincoln
Stein](User%3ALstein "User%3ALstein") suggested that we document them all
and provide a list of pros and cons for each. That will allow new users
to make the best informed choice about what they want to do.

**Action Items:**

- Dave will create a Chado APIs page.
- Dave will work with Brad to make the middleware available and
  documented.
- Dave will Contact Eric and/or DictyBase about the status of Modware.
  - 2008/08 - Done. Modware is actively being worked on by DictyBase
    staff.

### <span id="Chado_Java_API" class="mw-headline">Chado Java API</span>

Ed Lee presented a talk on the need for a Java interface to to the Chado
schema. He's going to be rewriting the Apollo data model to clearly
define biological concepts and to map well to any of Apollo's potential
data sources, including Chado.

This could be a way to enforce/encourage [Chado Best
Practices](Chado_Best_Practices "Chado Best Practices"). A current
problem for tool developers (such as the Apollo team) is writing code to
work with Chado, when not all Chado users represent the same biological
concepts in similar ways.

Having a cleanly designed, biological level (as opposed to DBMS table
level) API for Java would help organizations follow best practices when
using Chado. It also would make tool development much easier.

# <span id="GMOD_User_Community" class="mw-headline">GMOD User Community</span>

## <span id="SGN" class="mw-headline">SGN</span>



<a href="File:GMOD2008Lukas.JPG" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/d/d6/GMOD2008Lukas.JPG" width="120"
height="148" alt="GMOD2008Lukas.JPG" /></a>



Lukas Mueller from [SGN](Category%3ASGN "Category%3ASGN") spoke about
community annotation at the
<a href="http://www.sgn.cornell.edu/" class="external text"
rel="nofollow">Sol Genomics Network</a>.

- SGN does annotation on genotype and phenotype.
- Have about 60 community annotators.
- ~130 loci have been edited at least once by community members.
- Easy to use interface. Updates go directly to main database.
- Have assigned some entire gne families to people.
- Lukas actively recruits volunteer editors at meetings.

Lukas also takled about SGN's traits (phenotypes) database. SGN uses a
custom database design for their phenotypic data. (They do not use the
[Chado Phenotype
Module](Chado_Phenotype_Module "Chado Phenotype Module"). Suzi Lewis
indicated that her group is working on a new phenotype module for Chado
which will address issues with the current design.)

  

## <span id="WikiMods.org" class="mw-headline">WikiMods.org</span>

Brad Arshinoff from
<a href="http://xanthusbase.org" class="external text"
rel="nofollow">XanthusBase</a>, introduced the
<a href="http://wikimods.org" class="external text"
rel="nofollow">WikiMods</a> web site, a collection of MODS for
prokaryotes with small research scommunites. This will replace the
existing XanthusBase site and add an additional organism in the process.
It is scheduled to launch on July 30 2008 with these sites:

- <a href="http://xanthus.wikimods.org" class="external free"
  rel="nofollow">http://xanthus.wikimods.org</a>
- <a href="http://xeno.wikimods.org" class="external free"
  rel="nofollow">http://xeno.wikimods.org</a>
- <a href="http://demo.wikimods.org" class="external free"
  rel="nofollow">http://demo.wikimods.org</a>

They have migrated Chado from Oracle to
[MySQL](Category%3AMySQL "Category%3AMySQL").

## <span id="CellFrame" class="mw-headline">CellFrame</span>

Yunchen Gong gave a talk about
<a href="http://cellframe.bioknowledge.org/" class="external text"
rel="nofollow">CellFrame</a>, a web site about cell biology and
construction of cell perturbation networks

## <span id="Xenbase" class="mw-headline">Xenbase</span>

Jeff Bowes of <a href="http://xenbase.org" class="external text"
rel="nofollow">Xenbase</a> talked about automatic loading, linking, and
indexing publication abstracts. Xenbase downloads information for every
*Xenopus* related publication. The abstract is then scanned for gene
names/symbols and other controlled vocabulary terms. The publication is
then associated with those terms and genes in Xenbase.

Xenbase has extended the schema to support this indexing scheme and uses
DB2 Net Extender for indexing (but any indexing tool could be used).
Xenbase also scrapes images from each journal they have an agreement
with. They use a Java class for journals, and every journal has its own
subclass.

## <span id="Centre_for_Molecular_and_Biomolecular_Informatics" class="mw-headline">Centre for Molecular and Biomolecular Informatics</span>

Victor de Jager of the University of Nijmegen and the Centre for
Molecular and Biomolecular Informatics, gave a talk on using the Django
web framework with Chado (see [Chado Django
HOWTO](Chado_Django_HOWTO "Chado Django HOWTO") for more). A Django
based web site could be layered on top of the BioObjects proposed by Ed
Lee in his talk.

# <span id="GMOD_Project" class="mw-headline">GMOD Project</span>

## <span id="Google_Summer_of_Code" class="mw-headline">Google Summer of Code</span>

Last year a Google Summer of Code student worked with Lincoln, and
Hilmar Lapp (at NESCent) on a Google Summer of Code project to add
phyogenetic information to GBrowse. Lincoln and Hilmar liked it enough
that they recommend the program. Lincoln cautions that it is a lot of
work to be a mentor in the program.

**Action Items:**

- Dave will investigate further and encourage the GMOD community to
  participate in the program during the summer of 2009.

## <span id="Packages" class="mw-headline">Packages</span>

At the end of the [GMOD Help Desk talk (see below)](#GMOD_Help_Desk),
Dave asked for what else he should be working on. The number one
response was creating GMOD packages that could be installed with Linux
package installers.

Everyone agreed this was an excellent idea, and that it was hard to do,
particularly to keep the packages up to date for all the distributions
you want to support. BioPackages.net would be the place to put them, if
we did this.

Lincoln mentioned that there are 1 year infrastructure grants for this
sort of thing. That would get us where we want to be for a year, but not
after that.

  
**Action Items:**

- No solution or action item was settled on.

## <span id="GMOD_Help_Desk" class="mw-headline">GMOD Help Desk</span>

[Dave Clements](User%3AClements "User%3AClements") gave a talk on his first
10 months at the [GMOD Help Desk](GMOD_Help_Desk "GMOD Help Desk"), and
what he is planning doing in the coming months.

### <span id="What.27s_Been_Done" class="mw-headline">What's Been Done</span>

- [Web
  Site](News/Ten_Recent_Web_Site_Changes "News/Ten Recent Web Site Changes")
- Outreach - Posters, talks, representation, and promotion
- Education
  - <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/4/4a/ChadoWorkshopArthopod2008.pdf"
    class="internal" title="ChadoWorkshopArthopod2008.pdf">Chado
    Workshop</a> with Scott @ Arthropod Genomics, ~35 people
  - [2008 GMOD Summer
    School](2008_GMOD_Summer_School "2008 GMOD Summer School") with
    Scott, Ed, Ben, 25 students for 2 1/2 days
- Components
  - [MAKER](MAKER.1 "MAKER") now part of GMOD.
  - [Phenote](Phenote "Phenote") is on its way.

### <span id="What.27s_Planned" class="mw-headline">What's Planned</span>

#### <span id="GMOD_User_Directory" class="mw-headline">GMOD User Directory</span>

Planning to [TableEdit](TableEdit.1 "TableEdit") to make parts of the
GMOD web site be database driven. Plan on having the same core set of
data and a web page for each user. The core data set will describe what
components they use and how, and be implemented in TableEdit tables.
We'll then be able to use that information to also show which users use
a component on each component page, as well as a complete list of users.

This is a continuation of the *community portal* idea that was started
in the past 10 months. This will help new and existing users get a
handle on who is using which components for what kind of biology.

#### <span id="User_Experience_Logs" class="mw-headline">User Experience Logs</span>

We can't possibly describe or maintain HOWTO pagess for all possible
combinations of operating system (in all their versions), external
software (BioPerl, Java, libgd,... - in all their versions), and GMOD
Components (in all their possible versions and combinations).

However, if we made it easy for GMOD users to record their experiences
installing whatever combination they are using then that might be a
useful approximation. New users would then be able to find several
possible workarounds when they, for eample, can't get libgd to work.
Maybe one of the workarounds will even be for there Linux distribution.

We already have several such logs on the web site.

Dave will create a plan for

- organizing user logs in the web site,
- making it easy to do so, and
- encourage users to do this.

#### <span id="Documentation" class="mw-headline">Documentation</span>

- Components
  - [Chado Documentation
    Reorganization](Chado_Documentation_Reorganization "Chado Documentation Reorganization")
    - Chado API doc
  - [GBrowse](GBrowse.1 "GBrowse") doc, including cookbook and glyphs
    page.
  - [Community Annotation
    System](Community_Annotation_System "Community Annotation System")
  - [TableEdit](TableEdit.1 "TableEdit")
  - Better document (and encourage tighter) integration between
    [Galaxy](Galaxy.1 "Galaxy"), [InterMine](InterMine "InterMine"),
    [BioMart](BioMart "BioMart") and the other GMOD components.
  - Tutorials - screencasts for sophisticad user interfaces, perhaps
    [GBrowse](GBrowse.1 "GBrowse"), [Apollo](Apollo.1 "Apollo") and
    [CMap](CMap.1 "CMap")
- Web site upgrade
  - New MediaWiki
  - TableEdit
  - Better searching
  - New skin.

#### <span id="GMOD_Logo_Service" class="mw-headline">[GMOD Logo Service](GMOD_Logo_Program "GMOD Logo Program")</span>

<a href="http://zfin.org" class="external text"
rel="nofollow">ZFIN's</a> current logo was designed several years ago by
Kari Pape, a student in a University of Oregon design class. Judy
Sprague, ZFIN's manager, worked with the professor and the students to
communicate what ZFIN was all about and at the end of the quarter we had
about 20 designs to pick from, and most of them were spectacularly good.

Many GMOD user databases, web sites, and GMOD components don't have
snazzy logos. [Dave](User%3AClements "User%3AClements") offered to contact
the same department and the local community college as well, and ask if
they would be interested in doing something similar GMOD community. This
time around I would propose that each student or team get a different
database/web site/component.

This was clearly the most popular idea Dave has ever had during his time
at GMOD. I'll investigate ASAP. (*See [GMOD Logo
Program](GMOD_Logo_Program "GMOD Logo Program").*)

#### <span id="Education_and_Outreach" class="mw-headline">[Education and Outreach](Training_and_Outreach "Training and Outreach")</span>

- [2009 Summer
  School](2009_GMOD_Summer_School_-_Americas "2009 GMOD Summer School - Americas")
- [GMOD Course in
  Europe?](2009_GMOD_Summer_School_-_Europe "2009 GMOD Summer School - Europe")
- [Conferences](Training_and_Outreach#Workshops.2C_Presentations_and_Posters "Training and Outreach")
  - [PAG](PAG_2009 "PAG 2009"), Arthropod Genomics, IPlant
- Special emphasis on [comparative
  genomics](Comparative_Genomics "Comparative Genomics") and community
  annotation.

#### <span id="Grants" class="mw-headline">Grants</span>

The Help Desk now offers to review grant proposal prior to submission to
help them fully state how much they can use GMOD components, and thus
avoid reinventing the wheel for their project.

We will also start suggesting that grants that propose using GMOD
components also include a limited amount of funding for GMOD in the
grant. This could either be core project funding, funding for existing
components or funding for new components to become part of GMOD.

  

  

# <span id="Agenda_Proposals" class="mw-headline">Agenda Proposals</span>

If you have something you want to be on the agenda at this meeting
please add it below.

- The AJAX-GBrowse project, as demo'd at
  <a href="http://genome.biowiki.org/" class="external free"
  rel="nofollow">http://genome.biowiki.org/</a> - Ian Holmes
- [InterMine](InterMine "InterMine") and
  <a href="Chado" class="mw-redirect" title="Chado">Chado</a> Richard
  Smith ([Scott](User%3AScott "User%3AScott") 14:04, 6 June 2008 (EDT))
- Advances in the [Common Gene
  Page](Common_Gene_Page "Common Gene Page") effort (see also here an
  old page at blog.gmod.org:
  <a href="http://blog.gmod.org/common_gene_pages" class="external free"
  rel="nofollow">http://blog.gmod.org/common_gene_pages</a> , as well as
  Don Gilbert's page on the topic on his server:
  <a href="http://eugenes.org/gmod/gene-report-examples/"
  class="external free"
  rel="nofollow">http://eugenes.org/gmod/gene-report-examples/</a> )
  ([Scott](User%3AScott "User%3AScott") 14:04, 6 June 2008 (EDT))
- TableEdit round trip/integration progress and plans
  --[JimHu](User%3AJimHu.1 "User%3AJimHu") 18:35, 6 June 2008 (EDT)
- Java Chado data model API with higher level, user friendly
  "Biological" layer [Ed](User%3AElee "User%3AElee")
- GBrowse 1.69 show and tell [Sheldon McKay](User%3AMckays "User%3AMckays")
- MediaWiki enhancements [Sheldon McKay](User%3AMckays "User%3AMckays")

<!-- -->

- [GMOD Help Desk](GMOD_Help_Desk "GMOD Help Desk") - [Dave
  Clements](User%3AClements "User%3AClements")
  - priorities for 2008-2009
  - Evaluation - Has the Help Desk been helpful? -
    [Dave](User%3AClements "User%3AClements") and [Don
    Gilbert](User%3ADongilbert "User%3ADongilbert")
  - Grant review service
  - Funding GMOD
  - GMOD and the Google Summer of Code in 2009?
  - The Chado Natural Diversity module
  - [Galaxy](Galaxy.1 "Galaxy") Integration. [Galaxy](Galaxy.1 "Galaxy")
    already integrates with [BioMart](BioMart "BioMart"), and the
    current (May 2008) development version of
    [Galaxy](Galaxy.1 "Galaxy") integrates with the current (May 2008,
    1.69 Beta, e.g. "stable") development version of
    [GBrowse](GBrowse.1 "GBrowse"). Once this goes to production in both
    Galaxy and GBrowse, should Galaxy work on integrating with other
    GMOD components such as
    <a href="Chado" class="mw-redirect" title="Chado">Chado</a> or
    [InterMine](InterMine "InterMine")?





Retrieved from
"<http://gmod.org/mediawiki/index.php?title=July_2008_GMOD_Meeting&oldid=21983>"







[Categories](Special%3ACategories "Special%3ACategories"):

- [Meetings](Category%3AMeetings "Category%3AMeetings")
- [AJAX](Category%3AAJAX "Category%3AAJAX")
- [Analysis](Category%3AAnalysis "Category%3AAnalysis")
- [Apollo](Category%3AApollo "Category%3AApollo")
- [Java](Category%3AJava "Category%3AJava")
- [GBrowse](Category%3AGBrowse "Category%3AGBrowse")
- [InterMine](Category%3AInterMine "Category%3AInterMine")
- [Javascript](Category%3AJavascript "Category%3AJavascript")
- [Middleware](Category%3AMiddleware "Category%3AMiddleware")
- [SGN](Category%3ASGN "Category%3ASGN")
- [XML](Category%3AXML "Category%3AXML")
- [JBrowse](Category%3AJBrowse "Category%3AJBrowse")
- [GBrowse 2](Category%3AGBrowse_2 "Category%3AGBrowse 2")
- [TableEdit](Category%3ATableEdit "Category%3ATableEdit")















## Navigation menu









### Namespaces


- <span id="ca-talk"><a
  href="http://gmod.org/mediawiki/index.php?title=Talk:July_2008_GMOD_Meeting&amp;action=edit&amp;redlink=1"
  accesskey="t"
  title="Discussion about the content page [t]">Discussion</a></span>





### 

### Variants[](#)























<a href="Main_Page"
style="background-image: url(../images/GMOD-cogs.png);"
title="Visit the main page"></a>





### Navigation



- <span id="n-GMOD-Home">[GMOD Home](Main_Page)</span>
- <span id="n-Software">[Software](GMOD_Components)</span>
- <span id="n-Categories-.2F-Tags">[Categories /
  Tags](Categories)</span>







### Documentation



- <span id="n-Overview">[Overview](Overview)</span>
- <span id="n-FAQs">[FAQs](Category%3AFAQ)</span>
- <span id="n-HOWTOs">[HOWTOs](Category%3AHOWTO)</span>
- <span id="n-Glossary">[Glossary](Glossary)</span>







### Community



- <span id="n-GMOD-News">[GMOD News](GMOD_News)</span>
- <span id="n-Training-.2F-Outreach">[Training /
  Outreach](Training_and_Outreach)</span>
- <span id="n-Support">[Support](Support)</span>
- <span id="n-GMOD-Promotion">[GMOD Promotion](GMOD_Promotion)</span>
- <span id="n-Meetings">[Meetings](Meetings)</span>
- <span id="n-Calendar">[Calendar](Calendar)</span>







### Tools




- <span id="t-smwbrowselink"><a href="Special%3ABrowse/July_2008_GMOD_Meeting" rel="smw-browse">Browse
  properties</a></span>












- <span id="footer-info-lastmod">Last updated at 18:53 on 8 October
  2012.</span>
<!-- - <span id="footer-info-viewcount">316,533 page views.</span> -->
- <span id="footer-info-copyright">Content is available under
  <a href="http://www.gnu.org/licenses/fdl-1.3.html" class="external"
  rel="nofollow">a GNU Free Documentation License</a> unless otherwise
  noted.</span>

<!-- -->

- <span id="footer-places-about">[About
  GMOD](GMOD%3AAbout "GMOD%3AAbout")</span>

<!-- -->







