



<span id="top"></span>




# <span dir="auto">Property:Has description</span>









This is a property of type
[Text](Special%253ATypes/Text "Special%253ATypes/Text").

  
<span id="SMWResults"></span>



## Pages using the property "Has description"

Showing 21 pages using this property.

<table style="width: 100%; ">
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="header">
<th class="smwpropname"><h3 id="b">B</h3></th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td class="smwpropname"><a href="BLAST_Graphic_Viewer.1"
title="BLAST Graphic Viewer">BLAST Graphic Viewer</a> <span
class="smwbrowse"><a href="Special%253ABrowse/BLAST-20Graphic-20Viewer"
title="Special%253ABrowse/BLAST-20Graphic-20Viewer">+</a></span></td>
<td class="smwprops">BLAST Graphic Viewer provides a graphical
visualization of the regions contained within HSPs relative to the query
sequence. BLAST Graphic Viewer is also known as blastGraphic.  </td>
</tr>
<tr class="even">
<td class="smwpropname"><a href="BioMart"
title="BioMart">BioMart</a> <span class="smwbrowse"><a
href="Special%253ABrowse/BioMart"
title="Special%253ABrowse/BioMart">+</a></span></td>
<td class="smwprops">Links to BioMart installations in the wild  ,
BioMart is a freely available, open source, federated database system
that provides unified access to disparate, geographically distributed
data sources. It is designed to be data agnostic and platform
independent, such that existing databases can easily be incorporated
into the BioMart framework.  </td>
</tr>
<tr class="odd">
<td class="smwpropname"><h3 id="c">C</h3></td>
<td></td>
</tr>
<tr class="even">
<td class="smwpropname"><a href="CMap.1" title="CMap">CMap</a> <span
class="smwbrowse"><a href="Special%253ABrowse/CMap"
title="Special%253ABrowse/CMap">+</a></span></td>
<td class="smwprops">Comparative mapping resource for grasses
(plant)  ,
A publicly accessible legume resource that will integrate genetic and
molecular data from multiple legume species and enable cross-species
comparisons  ,
The Honey Bee genome database  ,
<a href="Special%253APageProperty/CMap%253A%253AHas_description"
title="Special:PageProperty/CMap::Has description">…</a></td>
</tr>
<tr class="odd">
<td class="smwpropname"><a href="Canto" title="Canto">Canto</a> <span
class="smwbrowse"><a href="Special%253ABrowse/Canto"
title="Special%253ABrowse/Canto">+</a></span></td>
<td class="smwprops">Curate GO annotations for proteins using UniProtKB
identifiers  ,
Canto is an online tool that supports functional gene annotation by
community researchers as well as by professional curators. It provides a
simple interface to gather genetic and molecular data from publications
for inclusion in public biological databases. Canto employs OBO
ontologies to ensure a consistent vocabulary is used to describe gene
and gene product functionality and location. Genetic and physical
interaction annotation is also supported. Canto captures all necessary
information about each interaction for submission to BioGRID. Canto
currently supports annotation export in JSON format, either by
individual paper or in bulk. The JSON data from Canto can be loaded into
a relational database using the Chado schema (as at PomBase) or
alternative schemas.  , Curate GO annotations for
S. pombe  </td>
</tr>
<tr class="even">
<td class="smwpropname"><h3 id="d">D</h3></td>
<td></td>
</tr>
<tr class="odd">
<td class="smwpropname"><a href="DIYA" title="DIYA">DIYA</a> <span
class="smwbrowse"><a href="Special%253ABrowse/DIYA"
title="Special%253ABrowse/DIYA">+</a></span></td>
<td class="smwprops">DIYA (Do-It-Yourself Annotator) is a modular and
configurable open source pipeline framework, written in Perl, used for
the rapid annotation of microbial genome sequences. The software is
currently used to take nucleotide sequence contigs as input, either in
the form of complete genomes or the result of shotgun sequencing, and
produce an annotated sequence. === Status === Active development of DIYA
is currently on hiatus, pending the hiring of additional people to work
on it. [[User%253AStewarta|Andrew Stewart]], one of DIYA's developers also
has plans to convert the DIYA modules into a series of [[Galaxy]]
modules.  </td>
</tr>
<tr class="even">
<td class="smwpropname"><h3 id="g">G</h3></td>
<td></td>
</tr>
<tr class="odd">
<td class="smwpropname"><a href="GBrowse.1"
title="GBrowse">GBrowse</a> <span class="smwbrowse"><a
href="Special%253ABrowse/GBrowse"
title="Special%253ABrowse/GBrowse">+</a></span></td>
<td class="smwprops">GBrowse is a combination of database and
interactive web pages for manipulating and displaying annotations on
genomes. Features include: * Simultaneous bird's eye and detailed views
of the genome. * Scroll, zoom, center. * Use a variety of [[GBrowse
Configuration HOWTO#Glyphs and Glyph Options|premade glyphs]] or create
your own. * Attach arbitrary URLs to any annotation. * Order and
appearance of tracks are customizable by administrator and end-user. *
Search by annotation ID, name, or comment. * Supports third party
annotation using [[GFF]] formats. * Settings persist across sessions. *
DNA and [[GFF]] dumps. * Connectivity to different databases, including
[[BioSQL]] and [[Chado]]. * Multi-language support. * Third-party
feature loading. * Customizable [[GBrowse Plugins|plug-in]] architecture
(e.g. run [[wp:BLAST|BLAST]], dump &amp; import many formats, find
oligonucleotides, [[PrimerDesigner.pm|design primers]], create
restriction maps, edit features) Note that the information on this page
refers to GBrowse 2; GBrowse 1.x is recommended only for applications
where legacy browser support is required and a single database is
used.  </td>
</tr>
<tr class="even">
<td class="smwpropname"><a href="GBrowse_syn.1"
title="GBrowse syn">GBrowse syn</a> <span class="smwbrowse"><a
href="Special%253ABrowse/GBrowse-20syn"
title="Special%253ABrowse/GBrowse-20syn">+</a></span></td>
<td class="smwprops">GBrowse_syn, or the Generic Synteny Browser, is a
[[GBrowse]]-based [[synteny]] browser designed to display multiple
genomes, with a central reference species compared to two or more
additional species. It can be used to view multiple sequence alignment
data, synteny or co-linearity data from other sources against genome
annotations provided by GBrowse. GBrowse_syn is included with the
standard GBrowse package (version 1.69 and later).  </td>
</tr>
<tr class="odd">
<td class="smwpropname"><a href="GO_Graphic_Viewer.1"
title="GO Graphic Viewer">GO Graphic Viewer</a> <span
class="smwbrowse"><a href="Special%253ABrowse/GO-20Graphic-20Viewer"
title="Special%253ABrowse/GO-20Graphic-20Viewer">+</a></span></td>
<td class="smwprops">The GO Graphic Viewer module (Bio%253A%253AGMOD::GO::View)
generates a graphic that displays the parent and child relationships of
a selected GO term. It also provides the visualization for the result
from the GO::TermFinder Perl module created by the Stanford Microarray
Database (SMD). This module is useful when analyzing experimental or
computational results that produce a set of gene products that may have
a common function or process. This distribution also includes two
examples of its use in web browser-based user interfaces (goView.pl and
goTermFinder.pl).  </td>
</tr>
<tr class="even">
<td class="smwpropname"><a href="GO%253A%253ATermFinder.1"
title="GO::TermFinder">GO::TermFinder</a> <span class="smwbrowse"><a
href="Special%253ABrowse/GO%253A%253ATermFinder"
title="Special%253ABrowse/GO%253A%253ATermFinder">+</a></span></td>
<td class="smwprops">GO TermFinder takes a list of genes as input, and
determines whether there are enriched GO terms for that list. These
modules provide an object oriented set of libraries for dealing with
files produced by the [http://www.geneontology.org/ Gene Ontology
project]. In particular, the GO TermFinder object allows a client
program to provide a list of genes, from which it will determine
p-values for the annotation associated with that list of genes. It does
this using the hypergeometric distribution. In addition, a corrected
p-value is calculated, to correct for multiple hypothesis testing. This
distribution also contains a version of Shuai Weng's GO-View module (and
associated classes), such that you can graphically display the output of
the GO TermFinder. Some simple clients of these modules are provided in
the examples directory of the download.  </td>
</tr>
<tr class="odd">
<td class="smwpropname"><a href="Galaxy.1"
title="Galaxy">Galaxy</a> <span class="smwbrowse"><a
href="Special%253ABrowse/Galaxy"
title="Special%253ABrowse/Galaxy">+</a></span></td>
<td class="smwprops">Fully supported, publicly accessible platforms for
using Galaxy  ,
[https://galaxyproject.org/ Galaxy] is an open, web-based platform for
''accessible, reproducible, and transparent'' computational biomedical
research. * ''Accessibility:'' Galaxy enables users without programming
experience to easily specify parameters and run tools and workflows. *
''Reproducibility:'' Galaxy captures all information necessary so that
any user can repeat and understand a complete computational analysis. *
''Transparency:'' Galaxy enables users to share and publish analyses via
the web and create Pages--interactive, web-based documents that describe
a complete analysis. Galaxy is open source for all organizations and is
[https://galaxyproject.org/use/ available on a wide variety of platforms
and publicly accessible websites]. Galaxy servers make analysis tools,
genomic data, tutorial demonstrations, persistent workspaces, and
publication services available to any scientist that has access to the
Internet. Local Galaxy servers can be set up by [http://getgalaxy.org/
downloading the Galaxy application] and customizing it to meet
particular needs.
[[File:GCC2019Logo.png|left|250px|link=https://galaxyproject.org/events/gcc2019/
2019 Galaxy Community Conference]] === 2019 Galaxy Community Conference
=== The [https://galaxyproject.org/events/gcc2019 2019 Galaxy Community
Conference (GCC2019)] will be held 1-6 July, in Freiburg Germany.
[https://galaxyproejct.org/gcc2 Galaxy Community Conferences] are an
opportunity to participate in presentations, discussions, poster
sessions, lightning talks and breakouts, all about high-throughput
biology and the tools that support it.  The 2019 conference includes
training that offers in-depth topic coverage across several concurrent
sessions, and a CollaborationFest.  </td>
</tr>
<tr class="even">
<td class="smwpropname"><h3 id="i">I</h3></td>
<td></td>
</tr>
<tr class="odd">
<td class="smwpropname"><a href="InterMine"
title="InterMine">InterMine</a> <span class="smwbrowse"><a
href="Special%253ABrowse/InterMine"
title="Special%253ABrowse/InterMine">+</a></span></td>
<td class="smwprops">InterMine should always be checked out in source
code form  ,
InterMine makes it easy to integrate multiple data sources into a single
data warehouse. It has a core data model based on the Sequence Ontology
and supports several biological data formats, just configure which
organisms or data files are required. It is easy to extend the data
model and integrate your own data, Java and Perl APIs and an XML format
to help import custom data. Currently supported formats include Chado,
GFF3, FASTA, GO and gene association files, UniProt XML, PSI XML
(protein interactions), InParanoid orthologs, Ensembl, UniProt, and many
others. A web application allows creation of custom queries, includes
template queries (web forms to run 'canned' queries) and can upload and
operate on lists of data. It is possible to configure/create widgets to
analyse lists with graphs and enrichment statistics. An admin user can
publish new template queries, change report pages and create public
lists at any time without any programming. Many aspects of the web app
can be configured and branded.  </td>
</tr>
<tr class="even">
<td class="smwpropname"><h3 id="j">J</h3></td>
<td></td>
</tr>
<tr class="odd">
<td class="smwpropname"><a href="JBrowse.1"
title="JBrowse">JBrowse</a> <span class="smwbrowse"><a
href="Special%253ABrowse/JBrowse"
title="Special%253ABrowse/JBrowse">+</a></span></td>
<td class="smwprops">Browse the genome of Ötzi the ice man  ,
JBrowse is a genome browser with a fully dynamic AJAX interface, being
developed as the eventual successor to GBrowse. It is very fast and
scales well to large datasets. JBrowse is javascript-based and does
almost all of its work directly in the user's web browser, with minimal
requirements for the server. ===Features=== *Fast, smooth scrolling and
zooming. Explore your genome with unparalleled speed. *Scales easily to
multi-gigabase genomes and deep-coverage sequencing. *Supports GFF3,
BED, FASTA, Wiggle, BigWig, BAM, VCF (with tabix), REST, and more. BAM,
BigWig, and VCF data are displayed directly from the compressed binary
file with no conversion needed. *Very light server resource
requirements. In fact, JBrowse has no back-end server code, just tools
for formatting data files to be read directly over HTTP. Serve huge
datasets from a single low-cost cloud instance.  </td>
</tr>
<tr class="even">
<td class="smwpropname"><a href="JBrowse2"
title="JBrowse2">JBrowse2</a> <span class="smwbrowse"><a
href="Special%253ABrowse/JBrowse2"
title="Special%253ABrowse/JBrowse2">+</a></span></td>
<td class="smwprops">JBrowse 2 is a multi-modal genome browser that can
view data in linear, circular, dotplot, and syntenic modes, in order to
better serve the needs of structural variant visualization and
comparative genomics. It is coded using React and follows in the
footsteps of JBrowse 1 in being highly pluggable. Users can write
plugins to create custom data adapters, track types, and even entirely
new view types. See https://jbrowse.org/jb2/ for more info  </td>
</tr>
<tr class="odd">
<td class="smwpropname"><h3 id="m">M</h3></td>
<td></td>
</tr>
<tr class="even">
<td class="smwpropname"><a href="MAKER.1" title="MAKER">MAKER</a> <span
class="smwbrowse"><a href="Special%253ABrowse/MAKER"
title="Special%253ABrowse/MAKER">+</a></span></td>
<td class="smwprops">online version of the MAKER tool  ,
The ant genomes were annotated using MAKER.  ,
MAKER is an easy-to-configure, portable genome
[[:Category%253AAnnotation|annotation]] pipeline. MAKER allows smaller
''eukaryotic'' genome projects and ''prokaryotic'' genome projects to
annotate their genomes and to create genome databases. MAKER identifies
repeats, aligns ESTs and proteins to a genome, produces ''ab initio''
gene predictions and automatically synthesizes these data into gene
annotations with evidence-based quality values. MAKER is also easily
trainable: outputs of preliminary runs can be used to automatically
retrain its gene prediction algorithm, producing higher quality
gene-models on subsequent runs. MAKER's inputs are minimal. Its outputs
are in [[GFF3]] or [[Glossary#FASTA|FASTA]] format, and can be directly
loaded into [[Chado]], [[GBrowse]], [[JBrowse]], [[WebApollo]] or
[[Apollo]]. For those annotating novel plant genomes, the
[http://www.yandell-lab.org/software/maker-p.html MAKER-P pipeline] is
designed to make the annotation tractable for small groups with limited
bioinformatics experience and resources, and faster and more transparent
for large groups with more experience and resources. The MAKER-P
pipeline generates species-specific repeat libraries, as well as
structural annotations of protein coding genes, non-coding RNAs, and
pseudogenes. The [http://www.yandell-lab.org/software/maker-p.html
MAKER-P website] has more information on the additional protocols for
plant genome annotation. Additional MAKER options and capabilities
include: * Map old annotation sets on to new assemblies. * Merge
multiple legacy annotation sets into a consensus set of annotations. *
Update existing annotations to take new evidence into account. * Tag
pre-existing gene models with evidence alignments and quality control
metrics to assist in downstream manual curation. * Use GFF3 pass-through
to include both evidence alignments and predicted gene models from
algorithms not natively supported by MAKER. * MAKER is &lt;abbr
title="message-passing interface"&gt;MPI&lt;/abbr&gt;-capable for rapid
parallelization across computer clusters. * You can also easily
integrate raw InterProScan results into MAKER, which will identify
protein domains; add GO functional categories; and help assign putative
gene functions to genome annotations. This data then becomes accessible
as part of the GFF3 output and can be loaded into a [[Chado]] database,
[[JBrowse]], [[GBrowse]], [[WebApollo]] or [[Apollo]]. MAKER comes with
sample data for testing purposes. See the &lt;tt&gt;/data&lt;/tt&gt;
directory in the download.  , <a
href="Special%253APageProperty/MAKER%253A%253AHas_description"
title="Special:PageProperty/MAKER::Has description">…</a></td>
</tr>
<tr class="odd">
<td class="smwpropname"><h3 id="p">P</h3></td>
<td></td>
</tr>
<tr class="even">
<td class="smwpropname"><a href="Pathway_Tools.1"
title="Pathway Tools">Pathway Tools</a> <span class="smwbrowse"><a
href="Special%253ABrowse/Pathway-20Tools"
title="Special%253ABrowse/Pathway-20Tools">+</a></span></td>
<td class="smwprops">list of PGDBs hosted elsewhere  ,
Pathway Tools is a comprehensive symbolic systems biology software
system that supports several use cases in bioinformatics and systems
biology: *Development of organism-specific databases called
Pathway/Genome Databases (PGDBs) that integrate many bioinformatics
datatypes, from genomes to pathways to regulatory networks. *Development
of metabolic-flux models using flux-balance analysis *Scientific
visualization, web publishing, and dissemination of those
organism-specific databases, including: **Automatic display of metabolic
pathways and full metabolic networks; generation of metabolic map
diagram and of metabolic map poster
([http://bioinformatics.ai.sri.com/posters/ecoli-metab.pdf example]).
**Genome browser; comparative genome browser; generation of genome
poster ([http://bioinformatics.ai.sri.com/posters/ecoli-genome.pdf
example]). **Display of operons, regulons, and full transcriptional
regulatory networks *Analysis of omics datasets, including painting
omics data on to diagrams of the full metabolic network, enrichment
analysis. *Computational inferences including prediction of metabolic
pathways, prediction of metabolic pathway hole fillers, prediction of
operons *Comparative analyses of model-organism databases. *Analysis of
biological networks: **Interactively tracing metabolites through the
metabolic network **Finding dead-end metabolites in metabolic networks
**Identifying choke points (potential drug targets) in metabolic
networks Pathway Tools has four components: *PathoLogic: Creates a new
PGDB containing the predicted metabolic pathways of an organism, given a
Genbank entry as input. *Pathway/Genome Navigator: Supports query,
visualization, and analysis of PGDBs. The Navigator powers the BioCyc
web site at BioCyc.org. *MetaFlux: Supports development of metabolic
flux models from Pathway/Genome Databases. *Pathway/Genome Editors:
Provide interactive editing capabilities for PGDBs.  </td>
</tr>
<tr class="odd">
<td class="smwpropname"><h3 id="t">T</h3></td>
<td></td>
</tr>
<tr class="even">
<td class="smwpropname"><a href="Textpresso"
title="Textpresso">Textpresso</a> <span class="smwbrowse"><a
href="Special%253ABrowse/Textpresso"
title="Special%253ABrowse/Textpresso">+</a></span></td>
<td class="smwprops">Textpresso is an information extracting and
processing (text mining) package for biological literature whose
capabilities go far beyond that of a simple keyword search engine. The
two key elements are the collection of the full text of scientific
articles split into individual sentences, and the implementation of
semantic categories, for which a database of articles and individual
sentences can be searched. The source of the full text articles are
PDFs, and additional bibliographical information that is obtained from
other citation databases can be processed as well.
[http://ilex.caltech.edu/trac/alere/ Alere] is a package of scripts that
can be used to construct a corpus (retrieve articles) for use with
'''Textpresso'''. Textpresso is supported by a grant from the National
Human Genome Research Institute at the US National Institutes of Health
# HG004090.  </td>
</tr>
<tr class="odd">
<td class="smwpropname"><a href="Tripal.1"
title="Tripal">Tripal</a> <span class="smwbrowse"><a
href="Special%253ABrowse/Tripal"
title="Special%253ABrowse/Tripal">+</a></span></td>
<td class="smwprops">Tripal is a web front end for [[Chado]] databases,
based on the [http://www.drupal.org Drupal] content management system.
Tripal leverages many of the features of Drupal--ability to search and
edit content, detail pages for different types of content, customizable
page layouts, and so on--to provide users with an intuitive, flexible
interface to a database. Features and advantages of Tripal: * Allows
rapid development of an online genomic database. * Site administrators
can add non-biological content to their sites and allow editing of
content by non-technical users. * Integrates directly with GMOD
[[Chado]]. * Provides visualizations for Chado "modules" such as [[Chado
Sequence Module|features]], [[Chado Companalysis Module|analyses]],
[[Chado Library Module|libraries]], [[Chado Stock Module|stocks]],
[[Chado Modules|and others]]. * Provides an interface for easy
editing/updating of data in Chado. * Provides full text and categorical
search capabilities for Chado content. * Can integrate GMOD tools such
as [[CMap]] and [[GBrowse]]. * Integrates with powerful Drupal features
like [http://drupal.org/project/views Drupal views] and
[http://drupal.org/project/panels panels] * Provides fully customizable
PHP-based templates to allow users to change the look and feel of their
installation * Web-based management of Chado (i.e. installation of Chado
v1.11 and loading of ontologies) * Loaders for GFF3 and FASTA format
files. * Provides an API to allow for creation of custom extensions by
anyone. What is Tripal capable of? Check out the
[http://tripal.gmod.oicr.on.ca/1.1/ Tripal demo server]. ==== Tripal
Homepage ==== Get the most up-to-date information about Tripal
[http://www.tripal.info here].  </td>
</tr>
<tr class="even">
<td class="smwpropname"><h3 id="w">W</h3></td>
<td></td>
</tr>
<tr class="odd">
<td class="smwpropname"><a href="WebApollo.1"
title="WebApollo">WebApollo</a> <span class="smwbrowse"><a
href="Special%253ABrowse/WebApollo"
title="Special%253ABrowse/WebApollo">+</a></span></td>
<td class="smwprops">WebApollo is a browser-based tool for visualization
and editing of sequence annotations. It is designed for distributed
community annotation efforts, where numerous people may be working on
the same sequences in geographically different locations; real-time
updating keeps all users in sync during the editing process. The
features of WebApollo include: *History tracking, including browsing of
an annotation's edit history and full undo/redo functions *Real time
updating: edits in one client are instantly pushed to all other clients
*Convenient management of user login, authentication, and edit
permissions *Two-stage curation process: edit within a temporary
workspace, then publish to a curated database *Ability to add comments,
either chosen from a pre-defined set of comments or as freeform text.
*Ability to add dbxrefs [database crossreferences] -- e.g. for GO
functional annotation *Can set start of translation for a transcript or
let server determine automatically *Flagging of non-canonical splice
sites in curated annotations *Edge matching to selected feature:
matching edges across annotations and evidence tracks are highlighted
*Option to color transcript CDS by reading frame *Loading of data
directly from GFF3, BigWig, and BAM files, both remotely and from user's
local machine. *Configurable heat map rendering of BigWig data
*Per-session track configuration to set annotation colors, height, and
other properties *Export of annotation tracks as GFF3 and optionally
other formats *Search by sequence residues using server-side interface
to BLAT or other sequence search programs The public demo can be
accessed at: [http://genomearchitect.org/WebApolloDemo
http://genomearchitect.org/WebApolloDemo] (''Apis mellifera''
genome).  </td>
</tr>
<tr class="even">
<td class="smwpropname"><a href="WebApollo2"
title="WebApollo2">WebApollo2</a> <span class="smwbrowse"><a
href="Special%253ABrowse/WebApollo2"
title="Special%253ABrowse/WebApollo2">+</a></span></td>
<td class="smwprops">WebApollo2 is a browser-based tool for
visualization and editing of sequence annotations. It is designed for
distributed community annotation efforts, where numerous people may be
working on the same sequences in geographically different locations;
real-time updating keeps all users in sync during the editing process.
The features of WebApollo include: *History tracking, including browsing
of an annotation's edit history and full undo/redo functions *Real time
updating: edits in one client are instantly pushed to all other clients
*Convenient management of user login, authentication, and edit
permissions *Two-stage curation process: edit within a temporary
workspace, then publish to a curated database *Ability to add comments,
either chosen from a pre-defined set of comments or as freeform text.
*Ability to add dbxrefs [database crossreferences] -- e.g. for GO
functional annotation *Can set start of translation for a transcript or
let server determine automatically *Flagging of non-canonical splice
sites in curated annotations *Edge matching to selected feature:
matching edges across annotations and evidence tracks are highlighted
*Option to color transcript CDS by reading frame *Loading of data
directly from GFF3, BigWig, and BAM files, both remotely and from user's
local machine. *Configurable heat map rendering of BigWig data
*Per-session track configuration to set annotation colors, height, and
other properties *Export of annotation tracks as GFF3 and optionally
other formats *Search by sequence residues using server-side interface
to BLAT or other sequence search programs The public demo can be
accessed at: [http://genomearchitect.org/WebApolloDemo
http://genomearchitect.org/WebApolloDemo] (''Apis mellifera''
genome).  , Username: demo@demo.com,
Password: demo  </td>
</tr>
<tr class="odd">
<td class="smwpropname"><a href="WebGBrowse.1"
title="WebGBrowse">WebGBrowse</a> <span class="smwbrowse"><a
href="Special%253ABrowse/WebGBrowse"
title="Special%253ABrowse/WebGBrowse">+</a></span></td>
<td class="smwprops">The [[GBrowse|Generic Genome Browser (GBrowse)]] is
one of the most widely used visualization tools for sequence display
characterized by a rich set of utilities helpful in presenting a variety
of genome features pertaining to a reference sequence. Despite the tool
being meant for easy use by the biologists, the administration of
GBrowse can become tedious, particularly for those without adequate
computational resources. Moreover, there is no exhaustive, accurate,
"GBrowse user's perspective" based reference available for the various
glyphs useful in building the feature tracks on GBrowse, thus adding to
the complexities involved in configuration. WebGBrowse guides the users
through the process of configuring the [[GFF3]] datasets for GBrowse
display. It presents the user with a "Glyph Library" comprising of
around 40 different glyphs to choose from. It generates the
configuration data by collecting the user's input through an organized,
step-by-step process guided by the configurable parameter information
available in the glyph library. A summary information on all the
available glyphs can be viewed at the glyph library page. A
pre-installed GBrowse instance allows the users to preview/load their
datasets with desired configuration in GBrowse without requiring them to
prepare any computer hardware, install any software or learn
sophisticated GBrowse configuration rules for feature display. In other
words, WebGBrowse addresses most of the intricacies involved in GBrowse
administration, so that even a computer non-savvy biologist can
configure his GFF dataset without worrying about GBrowse software
installation or GBrowse configuration semantics. WebGBrowse can be
[http://webgbrowse.cgb.indiana.edu/software.html downloaded] and
installed locally, or you can use the
[http://webgbrowse.cgb.indiana.edu/ public WebGBrowse server at
CGB].  </td>
</tr>
<tr class="even">
<td class="smwpropname"><h3 id="x">X</h3></td>
<td></td>
</tr>
<tr class="odd">
<td class="smwpropname"><a href="XORT.1" title="XORT">XORT</a> <span
class="smwbrowse"><a href="Special%253ABrowse/XORT"
title="Special%253ABrowse/XORT">+</a></span></td>
<td class="smwprops">XORT is a utility written in Perl that can be used
to read to and write from a [[Glossary#Relational|relational]]
[[Glossary#Schema|schema]] using [[Glossary#XML|XML]] as an interchange
format. In GMOD XORT is frequently used to transfer data to and from
[[Chado]]. XORT’s three major elements are an XML-database schema
mapping specification, an XORT language for both data retrieval and data
construction, and a collection of tools that are based on the
specification to facilitate the mapping and data exchange. The mapping
specification specifies the schema requirement, mapping, operation, and
object reference mechanism; the XORT language is used to construct the
dump specification; the tool collections comprise the data validation,
data input, data output and data debug system. Because it originates
from the [[Chado]] database, we denote the corresponding XML "Chado
XML."  </td>
</tr>
</tbody>
</table>




</span>

|  |  |
|----|----|
| <span class="smw-highlighter" data-type="1" state="inline" data-title="Property"><span class="smwbuiltin">[Has type](Property%253AHas_type "Property:Has type")</span><span class="smwttcontent">This property is a special property in this wiki.</span></span> | [Text](Special%253ATypes/Text "Special%253ATypes/Text") <span class="smwsearch">[+](Special%253ASearchByProperty/Has-20type/Text "Special%253ASearchByProperty/Has-20type/Text")</span> |






## Navigation menu









### Navigation



- <span id="n-GMOD-Home">[GMOD Home](Main_Page)</span>
- <span id="n-Software">[Software](GMOD_Components)</span>
- <span id="n-Categories-.2F-Tags">[Categories /
  Tags](Categories)</span>




### Documentation



- <span id="n-Overview">[Overview](Overview)</span>
- <span id="n-FAQs">[FAQs](Category%253AFAQ)</span>
- <span id="n-HOWTOs">[HOWTOs](Category%253AHOWTO)</span>
- <span id="n-Glossary">[Glossary](Glossary)</span>




### Community



- <span id="n-GMOD-News">[GMOD News](GMOD_News)</span>
- <span id="n-Training-.2F-Outreach">[Training /
  Outreach](Training_and_Outreach)</span>
- <span id="n-Support">[Support](Support)</span>
- <span id="n-GMOD-Promotion">[GMOD Promotion](GMOD_Promotion)</span>
- <span id="n-Meetings">[Meetings](Meetings)</span>
- <span id="n-Calendar">[Calendar](Calendar)</span>




### Tools

- <span id="t-smwbrowselink"><a href="Special%253ABrowse/Property%253AHas_description"
  rel="smw-browse">Browse properties</a></span>



- <span id="footer-info-lastmod">Last updated at 23:14 on 1 October
  2013.</span>
<!-- - <span id="footer-info-viewcount">12,125 page views.</span> -->
- <span id="footer-info-copyright">Content is available under
  <a href="http://www.gnu.org/licenses/fdl-1.3.html" class="external"
  rel="nofollow">a GNU Free Documentation License</a> unless otherwise
  noted.</span>

<!-- -->



<!-- -->




