



<span id="top"></span>




# <span dir="auto">September 2010 GMOD Meeting</span>









<table style="vertical-align: middle; border: 2px solid #A6A6BC;"
data-cellpadding="10">
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<tbody>
<tr class="odd">
<td>

<a href="File:Sept2010MtgLogo300.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/4/40/Sept2010MtgLogo300.png/250px-Sept2010MtgLogo300.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/4/40/Sept2010MtgLogo300.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/4/40/Sept2010MtgLogo300.png 2x"
width="250" height="213" alt="Sept2010MtgLogo300.png" /></a>

</td>
<td style="font-size: 180%; line-height: 120%"><strong>September 2010
GMOD Meeting</strong><br />
13-14 September 2010<br />
Cambridge, UK<br />
<br />
&#10;

<a href="File:Sept2010Group.jpg" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/30/Sept2010Group.jpg" width="300" height="81"
alt="Sept2010Group.jpg" /></a>

</td>
<td style="text-align: center;"><a href="GMOD_Europe_2010"
title="Part of GMOD Europe 2010"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/d/d6/GMOD2010Europe300.png/200px-GMOD2010Europe300.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/d/d6/GMOD2010Europe300.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/d/d6/GMOD2010Europe300.png 2x"
width="200" height="171" alt="Part of GMOD Europe 2010" /></a>

<a href="GMOD_Europe_2010" title="GMOD Europe 2010">Part of GMOD Europe
2010</a>
</td>
</tr>
</tbody>
</table>

  

This [GMOD community meeting](Meetings "Meetings") was held 13-14
September 2010, in Cambridge, UK, as part of [GMOD Europe
2010](GMOD_Europe_2010 "GMOD Europe 2010"), which also included
[Satellite
Meetings](Satellite_Meetings_-_GMOD_Europe_2010 "Satellite Meetings - GMOD Europe 2010"),
an [InterMine
Workshop](InterMine_Workshop_-_GMOD_Europe_2010 "InterMine Workshop - GMOD Europe 2010"),
and a [BioMart
Workshop](BioMart_Workshop_-_GMOD_Europe_2010 "BioMart Workshop - GMOD Europe 2010").
The meeting was
[sponsored](#Sponsor:_Cambridge_Computational_Biology_Institute) and
hosted by the <a href="http://www.ccbi.cam.ac.uk/" class="external text"
rel="nofollow">Cambridge Computational Biology Institute</a> at the
University of Cambridge.

[GMOD Meetings](Meetings "Meetings") are a mix of
[user](GMOD_Membership#Users "GMOD Membership") and
[developer](GMOD_Membership#Developers "GMOD Membership") presentations,
and are a great place to find out what is happening in the project,
what's coming up, and what others are doing. The [January 2010 GMOD
Meeting](January_2010_GMOD_Meeting "January 2010 GMOD Meeting") was the
previous event. The next meeting is likely to be held in spring 2011.


## Contents



- [<span class="tocnumber">1</span>
  <span class="toctext">Agenda</span>](#Agenda)
  - [<span class="tocnumber">1.1</span> <span class="toctext">Monday, 13
    September</span>](#Monday.2C_13_September)
  - [<span class="tocnumber">1.2</span> <span class="toctext">Tuesday,
    14 September</span>](#Tuesday.2C_14_September)
  - [<span class="tocnumber">1.3</span> <span class="toctext">Wednesday
    & Thursday, 15-16
    September</span>](#Wednesday_.26_Thursday.2C_15-16_September)
- [<span class="tocnumber">2</span>
  <span class="toctext">Presentations</span>](#Presentations)
  - [<span class="tocnumber">2.1</span> <span class="toctext">By
    Topic</span>](#By_Topic)
  - [<span class="tocnumber">2.2</span> <span class="toctext">The State
    of GMOD</span>](#The_State_of_GMOD)
    - [<span class="tocnumber">2.2.1</span> <span class="toctext">What's
      New</span>](#What.27s_New)
    - [<span class="tocnumber">2.2.2</span> <span class="toctext">Next
      GMOD Meeting?</span>](#Next_GMOD_Meeting.3F)
  - [<span class="tocnumber">2.3</span> <span class="toctext">Help Desk
    Update</span>](#Help_Desk_Update)
    - [<span class="tocnumber">2.3.1</span>
      <span class="toctext">Mailing List
      Archives</span>](#Mailing_List_Archives)
    - [<span class="tocnumber">2.3.2</span> <span class="toctext">GMOD
      Membership Requirements</span>](#GMOD_Membership_Requirements)
    - [<span class="tocnumber">2.3.3</span> <span class="toctext">GMOD
      Promotion</span>](#GMOD_Promotion)
    - [<span class="tocnumber">2.3.4</span> <span class="toctext">GMOD
      Logo Program</span>](#GMOD_Logo_Program)
    - [<span class="tocnumber">2.3.5</span> <span class="toctext">2010
      GMOD Community Survey</span>](#2010_GMOD_Community_Survey)
    - [<span class="tocnumber">2.3.6</span>
      <span class="toctext">Events</span>](#Events)
    - [<span class="tocnumber">2.3.7</span>
      <span class="toctext">JBrowse
      Development</span>](#JBrowse_Development)
    - [<span class="tocnumber">2.3.8</span>
      <span class="toctext">JBrowse Grant
      Proposal</span>](#JBrowse_Grant_Proposal)
    - [<span class="tocnumber">2.3.9</span> <span class="toctext">New
      Components</span>](#New_Components)
    - [<span class="tocnumber">2.3.10</span> <span class="toctext">Some
      Interesting Documents</span>](#Some_Interesting_Documents)
    - [<span class="tocnumber">2.3.11</span> <span class="toctext">GMOD
      on the Web</span>](#GMOD_on_the_Web)
  - [<span class="tocnumber">2.4</span> <span class="toctext">The Open
    Microscopy Environment: Open Informatics for Biological
    Imaging</span>](#The_Open_Microscopy_Environment:_Open_Informatics_for_Biological_Imaging)
    - [<span class="tocnumber">2.4.1</span> <span class="toctext">Data
      Standards and Formats</span>](#Data_Standards_and_Formats)
    - [<span class="tocnumber">2.4.2</span>
      <span class="toctext">Imaging Culture and
      Sustainability</span>](#Imaging_Culture_and_Sustainability)
  - [<span class="tocnumber">2.5</span> <span class="toctext">PSICQUIC:
    The PSI Common QUery
    Interface</span>](#PSICQUIC:_The_PSI_Common_QUery_Interface)
    - [<span class="tocnumber">2.5.1</span>
      <span class="toctext">PSICQUIC Web
      Services</span>](#PSICQUIC_Web_Services)
    - [<span class="tocnumber">2.5.2</span>
      <span class="toctext">Formats</span>](#Formats)
    - [<span class="tocnumber">2.5.3</span>
      <span class="toctext">MIQL</span>](#MIQL)
    - [<span class="tocnumber">2.5.4</span>
      <span class="toctext">Creating a PSICQUIC
      Service</span>](#Creating_a_PSICQUIC_Service)
    - [<span class="tocnumber">2.5.5</span>
      <span class="toctext">PSICQUIC
      Applications</span>](#PSICQUIC_Applications)
    - [<span class="tocnumber">2.5.6</span>
      <span class="toctext">PSICQUIC
      Development</span>](#PSICQUIC_Development)
  - [<span class="tocnumber">2.6</span> <span class="toctext">MolGenIS
    and XGAP</span>](#MolGenIS_and_XGAP)
    - [<span class="tocnumber">2.6.1</span>
      <span class="toctext">MolGenIS</span>](#MolGenIS)
    - [<span class="tocnumber">2.6.2</span>
      <span class="toctext">XGAP</span>](#XGAP)
    - [<span class="tocnumber">2.6.3</span> <span class="toctext">GMOD
      Link Ideas</span>](#GMOD_Link_Ideas)
    - [<span class="tocnumber">2.6.4</span>
      <span class="toctext">OntoCAT</span>](#OntoCAT)
  - [<span class="tocnumber">2.7</span> <span class="toctext">The GMOD
    Chado Natural Diversity
    Module</span>](#The_GMOD_Chado_Natural_Diversity_Module)
    - [<span class="tocnumber">2.7.1</span>
      <span class="toctext">Motivation</span>](#Motivation)
    - [<span class="tocnumber">2.7.2</span>
      <span class="toctext">Developmental
      History</span>](#Developmental_History)
    - [<span class="tocnumber">2.7.3</span>
      <span class="toctext">Schema</span>](#Schema)
    - [<span class="tocnumber">2.7.4</span> <span class="toctext">CV
      Terms and APIs</span>](#CV_Terms_and_APIs)
  - [<span class="tocnumber">2.8</span> <span class="toctext">Cosmic
    GBrowse: Visualising cancer mutations in genomic
    context</span>](#Cosmic_GBrowse:_Visualising_cancer_mutations_in_genomic_context)
    - [<span class="tocnumber">2.8.1</span>
      <span class="toctext">Data</span>](#Data)
    - [<span class="tocnumber">2.8.2</span>
      <span class="toctext">Configuration and
      Setup</span>](#Configuration_and_Setup)
    - [<span class="tocnumber">2.8.3</span>
      <span class="toctext">Display</span>](#Display)
    - [<span class="tocnumber">2.8.4</span> <span class="toctext">Future
      Development</span>](#Future_Development)
  - [<span class="tocnumber">2.9</span> <span class="toctext">GMOD
    Projects at the Center for Genomics and
    Bioinformatics</span>](#GMOD_Projects_at_the_Center_for_Genomics_and_Bioinformatics)
    - [<span class="tocnumber">2.9.1</span> <span class="toctext">A
      Simple Web Interface for Configuring GBrowse:
      WebGBrowse</span>](#A_Simple_Web_Interface_for_Configuring_GBrowse:_WebGBrowse)
    - [<span class="tocnumber">2.9.2</span>
      <span class="toctext">Web-based Bioinformatics Pipelines for
      Biologists:
      ISGA</span>](#Web-based_Bioinformatics_Pipelines_for_Biologists:_ISGA)
      - [<span class="tocnumber">2.9.2.1</span>
        <span class="toctext">Why develop ISGA as a separate
        package?</span>](#Why_develop_ISGA_as_a_separate_package.3F)
      - [<span class="tocnumber">2.9.2.2</span>
        <span class="toctext">Present and
        Future</span>](#Present_and_Future)
  - [<span class="tocnumber">2.10</span> <span class="toctext">GMOD RPC
    API: The almost RESTful GMOD
    API</span>](#GMOD_RPC_API:_The_almost_RESTful_GMOD_API)
    - [<span class="tocnumber">2.10.1</span> <span class="toctext">Perl
      Implementation</span>](#Perl_Implementation)
      - [<span class="tocnumber">2.10.1.1</span>
        <span class="toctext">Goals</span>](#Goals)
    - [<span class="tocnumber">2.10.2</span> <span class="toctext">How
      to participate</span>](#How_to_participate)
    - [<span class="tocnumber">2.10.3</span>
      <span class="toctext">Discussion</span>](#Discussion)
  - [<span class="tocnumber">2.11</span> <span class="toctext">Overview
    of current resources and update on DAS Meeting Cambridge
    2010</span>](#Overview_of_current_resources_and_update_on_DAS_Meeting_Cambridge_2010)
    - [<span class="tocnumber">2.11.1</span> <span class="toctext">Some
      DAS 1.5/1.6 Commands</span>](#Some_DAS_1.5.2F1.6_Commands)
    - [<span class="tocnumber">2.11.2</span> <span class="toctext">Why
      use DAS 1.6 over 1.5?</span>](#Why_use_DAS_1.6_over_1.5.3F)
    - [<span class="tocnumber">2.11.3</span> <span class="toctext">Easy
      DAS</span>](#Easy_DAS)
    - [<span class="tocnumber">2.11.4</span> <span class="toctext">1.6E
      Proposals</span>](#1.6E_Proposals)
    - [<span class="tocnumber">2.11.5</span> <span class="toctext">DAS
      Clients</span>](#DAS_Clients)
  - [<span class="tocnumber">2.12</span>
    <span class="toctext">InterMine: new Mines and new
    features</span>](#InterMine:_new_Mines_and_new_features)
    - [<span class="tocnumber">2.12.1</span> <span class="toctext">Mines
      for MODs</span>](#Mines_for_MODs)
    - [<span class="tocnumber">2.12.2</span>
      <span class="toctext">metabolicMine</span>](#metabolicMine)
    - [<span class="tocnumber">2.12.3</span>
      <span class="toctext">InterMine 0.94
      Release</span>](#InterMine_0.94_Release)
  - [<span class="tocnumber">2.13</span>
    <span class="toctext">Literature Curation in
    GMOD</span>](#Literature_Curation_in_GMOD)
  - [<span class="tocnumber">2.14</span> <span class="toctext">Towards a
    GO Annotation Tool: Curation Accelerator
    Software</span>](#Towards_a_GO_Annotation_Tool:_Curation_Accelerator_Software)
    - [<span class="tocnumber">2.14.1</span> <span class="toctext">The
      Challenge</span>](#The_Challenge)
    - [<span class="tocnumber">2.14.2</span> <span class="toctext">The
      Tool</span>](#The_Tool)
  - [<span class="tocnumber">2.15</span>
    <span class="toctext">BioPivot</span>](#BioPivot)
    - [<span class="tocnumber">2.15.1</span> <span class="toctext">Deep
      Zoom Technology</span>](#Deep_Zoom_Technology)
  - [<span class="tocnumber">2.16</span> <span class="toctext">CRAWL
    (Chado RESTful Access Web-service
    Layer)</span>](#CRAWL_.28Chado_RESTful_Access_Web-service_Layer.29)
    - [<span class="tocnumber">2.16.1</span>
      <span class="toctext">Challenges</span>](#Challenges)
    - [<span class="tocnumber">2.16.2</span>
      <span class="toctext">Solutions</span>](#Solutions)
    - [<span class="tocnumber">2.16.3</span>
      <span class="toctext">Collaborative Interfaces and Rich Internet
      Applications</span>](#Collaborative_Interfaces_and_Rich_Internet_Applications)
    - [<span class="tocnumber">2.16.4</span> <span class="toctext">The
      way of CRAWL</span>](#The_way_of_CRAWL)
    - [<span class="tocnumber">2.16.5</span> <span class="toctext">GMOD
      Rest</span>](#GMOD_Rest)
  - [<span class="tocnumber">2.17</span> <span class="toctext">Lessons
    from the Apache
    Community</span>](#Lessons_from_the_Apache_Community)
  - [<span class="tocnumber">2.18</span> <span class="toctext">Lightning
    Talks</span>](#Lightning_Talks)
    - [<span class="tocnumber">2.18.1</span>
      <span class="toctext">GBrowse 2 and JBrowse
      Features</span>](#GBrowse_2_and_JBrowse_Features)
    - [<span class="tocnumber">2.18.2</span> <span class="toctext">CpG
      Island and STR Annotator
      Plugins</span>](#CpG_Island_and_STR_Annotator_Plugins)
    - [<span class="tocnumber">2.18.3</span> <span class="toctext">GMOD
      Sustainability and
      Organization</span>](#GMOD_Sustainability_and_Organization)
- [<span class="tocnumber">3</span>
  <span class="toctext">Participants</span>](#Participants)
- [<span class="tocnumber">4</span>
  <span class="toctext">Logistics</span>](#Logistics)
- [<span class="tocnumber">5</span> <span class="toctext">Sponsor:
  Cambridge Computational Biology
  Institute</span>](#Sponsor:_Cambridge_Computational_Biology_Institute)
- [<span class="tocnumber">6</span>
  <span class="toctext">Feedback</span>](#Feedback)
- [<span class="tocnumber">7</span> <span class="toctext">The Next
  Meeting</span>](#The_Next_Meeting)



# <span id="Agenda" class="mw-headline">Agenda</span>


<a href="http://gre.lifesci.dundee.ac.uk/staff/jason_swedlow.html"
rel="nofollow" title="Jason Swedlow"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/8/80/JasonSwedlow.jpg/150px-JasonSwedlow.jpg"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/8/80/JasonSwedlow.jpg/225px-JasonSwedlow.jpg 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/8/80/JasonSwedlow.jpg/300px-JasonSwedlow.jpg 2x"
width="150" height="123" alt="Jason Swedlow" /></a>


Professor Jason Swedlow


*The <a href="http://www.openmicroscopy.org/" class="external text"
rel="nofollow">Open Microscopy Environment</a>: Open Informatics for
Biological Imaging*

- Professor,
  <a href="http://gre.lifesci.dundee.ac.uk/staff/jason_swedlow.html"
  class="external text" rel="nofollow">Wellcome Trust Centre for Gene
  Regulation and Expression</a>,
  <a href="http://www.dundee.ac.uk/" class="external text"
  rel="nofollow">University of Dundee</a>
- Principal Investigator,
  <a href="http://www.openmicroscopy.org/" class="external text"
  rel="nofollow">Open Microscopy Environment (OME)</a>



The meeting's guest speaker was Prof Jason Swedlow, who discussed his
work with with the
<a href="http://www.openmicroscopy.org/" class="external text"
rel="nofollow">Open Microscopy Environment (OME)</a>, an open
international consortium that develops and releases data specifications
and management tools for biological imaging. OME metadata enables image
sharing, analysis, and integration with other data types.

Dr Swedlow is a Professor at the
<a href="http://gre.lifesci.dundee.ac.uk/staff/jason_swedlow.html"
class="external text" rel="nofollow">Wellcome Trust Centre for Gene
Regulation and Expression</a> and the
<a href="http://www.dundee.ac.uk/" class="external text"
rel="nofollow">University of Dundee</a>. Jason's research focuses on
mechanisms and regulation of chromosome segregation during mitotic cell
division.



## <span id="Monday.2C_13_September" class="mw-headline">Monday, 13 September</span>

| Time | Topic | Presenter(s) | Links |
|----|----|----|----|
| 09:15 | Introductions | [Scott Cain](User%3AScott "User%3AScott") |  |
| 10:00 | The State of GMOD | [Scott Cain](User%3AScott "User%3AScott") | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/7/78/TheStateofGMODSept2010.pdf"
class="internal" title="TheStateofGMODSept2010.pdf">PDF</a>, [Summary](#The_State_of_GMOD) |
| 10:30 | Break |  |  |
| 11:00 | [Help Desk](GMOD_Help_Desk "GMOD Help Desk") Update | [Dave Clements](User%3AClements "User%3AClements") | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/b/bf/HelpDeskSept2010.pdf" class="internal"
title="HelpDeskSept2010.pdf">PDF</a>, <a
href="ftp://ftp.gmod.org/pub/gmod/Meetings/2010/GMODSeptember/HelpDeskSept2010.ppt"
class="external text" rel="nofollow">PPT</a>, [Summary](#Help_Desk_Update) |
| 11:30 | Keynote: The <a href="http://www.openmicroscopy.org/" class="external text"
rel="nofollow">Open Microscopy Environment</a>: Open Informatics for Biological Imaging | Jason Swedlow | <a
href="http://gmod.org/mediawiki/index.php?title=Special:Upload&amp;wpDestFile=OpenMicroscopySept2010.pdf"
class="new" title="OpenMicroscopySept2010.pdf">PDF</a>, <a
href="http://gmod.org/mediawiki/index.php?title=Special:Upload&amp;wpDestFile=OpenMicroscopySept2010.ppt"
class="new" title="OpenMicroscopySept2010.ppt">PPT</a>, [Summary](#The_Open_Microscopy_Environment:_Open_Informatics_for_Biological_Imaging) |
| 12:30 | Catered Lunch |  |  |
| 13:45 | <a href="http://psicquic.googlecode.com" class="external text"
rel="nofollow">PSICQUIC</a>: The PSI Common QUery Interface | [Bruno Aranda](User%3ABaranda "User%3ABaranda") | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/d/d2/PSICQUICSept2010.pdf" class="internal"
title="PSICQUICSept2010.pdf">PDF</a>, [Summary](#PSICQUIC:_The_PSI_Common_QUery_Interface) |
| 14:15 | [MolGenIS](MolGenIS "MolGenIS") and <a href="http://www.xgap.org" class="external text"
rel="nofollow">XGAP</a> | Morris Swertz | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/d/df/MolGenISXGAPSept2010.pdf"
class="internal" title="MolGenISXGAPSept2010.pdf">PDF</a>, [Summary](#MolGenIS_and_XGAP) |
| 14:45 | The GMOD Chado Natural Diversity Module | Bob MacCallum | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/0/02/ChadoNDSept2010.pdf" class="internal"
title="ChadoNDSept2010.pdf">PDF</a>, <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/c/cb/ChadoNDSept2010.ppt" class="internal"
title="ChadoNDSept2010.ppt">PPT</a>, <a
href="https://docs.google.com/present/edit?id=0ASHJk7ao3qFUZGo2ODJydl8xMjNmZGh3eDVnNg&amp;hl=en"
class="external text" rel="nofollow">gdoc</a>, [Summary](#The_GMOD_Chado_Natural_Diversity_Module) |
| 15:15 | Break |  |  |
| 15:45 | Cosmic [GBrowse](GBrowse.1 "GBrowse"): Visualising cancer mutations in genomic context | David Beare | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/6/6c/CosmicGBrowseSept2010.pdf"
class="internal" title="CosmicGBrowseSept2010.pdf">PDF</a>, <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/0/0e/CosMicGBrowseSept2010.ppt"
class="internal" title="CosMicGBrowseSept2010.ppt">PPT</a>, [Summary](#Cosmic_GBrowse:_Visualising_cancer_mutations_in_genomic_context) |
| 16:15 | GMOD Projects at the Center for Genomics and Bioinformatics | [Chris Hemmerich](User%3AChemmeri "User%3AChemmeri") | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/d/d1/ISGAWebGBrowseSept2010.pdf"
class="internal" title="ISGAWebGBrowseSept2010.pdf">PDF</a>, <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/3f/ISGAWebGBrowseSept2010.pptx"
class="internal" title="ISGAWebGBrowseSept2010.pptx">PPT</a>, [Summary](#GMOD_Projects_at_the_Center_for_Genomics_and_Bioinformatics) |

## <span id="Tuesday.2C_14_September" class="mw-headline">Tuesday, 14 September</span>

| Time | Topic | Presenter(s) | Links |
|----|----|----|----|
| 09:15 | [GMOD RPC API](GMOD_RPC_API "GMOD RPC API"): The almost RESTful GMOD API | [Josh Goodman](User%3AJogoodma "User%3AJogoodma") | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/e/ee/GMOD_RPC_Sept_2010.pdf"
class="internal" title="GMOD RPC Sept 2010.pdf">PDF</a>, [Summary](#GMOD_RPC_API:_The_almost_RESTful_GMOD_API) |
| 09:45 | Overview of current resources and update on <a href="DAS" class="mw-redirect" title="DAS">DAS</a> Meeting Cambridge 2010 | Jonathan Warren | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/e/e6/DAS_Sept2010.pdf" class="internal"
title="DAS Sept2010.pdf">PDF</a>, <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/e/e7/DAS_Sept2010.pptx" class="internal"
title="DAS Sept2010.pptx">PPT</a>, [Summary](#Overview_of_current_resources_and_update_on_DAS_Meeting_Cambridge_2010) |
| 10:15 | [InterMine](InterMine "InterMine"): new Mines and new features | [Richard Smith](User%3ARsmith "User%3ARsmith") | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/1/1c/InterMine_Sept2010.pdf"
class="internal" title="InterMine Sept2010.pdf">PDF</a>, [Summary](#InterMine:_new_Mines_and_new_features) |
| 10:40 | Break |  |  |
| 11:00 | Literature Curation in GMOD | [Daniel Renfro](User%3ADanielRenfro "User%3ADanielRenfro") | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/6/62/Literature_Curation_Sept2010.pdf"
class="internal" title="Literature Curation Sept2010.pdf">PDF</a>, <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/4/4d/Literature_Curation_Sept2010.ppt"
class="internal" title="Literature Curation Sept2010.ppt">PPT</a>, [Summary](#Literature_Curation_in_GMOD) |
| 11:30 | Towards a GO Annotation Tool: Curation Accelerator Software | Helen Field | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/7/70/GO_Curation_Sept2010.pdf"
class="internal" title="GO Curation Sept2010.pdf">PDF</a>, <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/5/58/GO_Curation_Sept2010.key"
class="internal" title="GO Curation Sept2010.key">KEY</a>, [Summary](#Towards_a_GO_Annotation_Tool:_Curation_Accelerator_Software) |
| 12:00 | BioPivot: Applying Microsoft Live Labs Pivot to Problems in Bioinformatics | Steve Taylor | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/a/ae/BioPivot_Sept2010.pdf"
class="internal" title="BioPivot Sept2010.pdf">PDF</a>, <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/a/a2/BioPivot_Sept2010.ppt"
class="internal" title="BioPivot Sept2010.ppt">PPT</a>, [Summary](#BioPivot) |
| 12:30 | Catered Lunch |  |  |
| 13:45 | CRAWL (<a href="Chado" class="mw-redirect" title="Chado">Chado</a> RESTful Access Web-service Layer) | [Giles Verlarde](User%3ABuggy "User%3ABuggy") | <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/3d/CRAWL_Sept2010.pdf" class="internal"
title="CRAWL Sept2010.pdf">PDF</a>, [Summary](#CRAWL_.28Chado_RESTful_Access_Web-service_Layer.29_-_a_programmatic_interface_for_querying_pathogen_genomics_data) |
| 14:15 | Lessons the GMOD community can glean from the <a
href="http://www.slideshare.net/bosc2010/gardler-bosc2010-communitydevelopmentattheasf"
class="external text" rel="nofollow">Apache Software Foundation</a> |  | [Summary](#Lessons_from_the_Apache_Community) |
| 14:45 | Lightning talks |  | [Summary](#Lightning_Talks) |
|  | [GBrowse 2 and JBrowse Features](#GBrowse_2_and_JBrowse_Features) | [Scott Cain](User%3AScott "User%3AScott") | [Summary](#GBrowse_2_and_JBrowse_Features) |
|  | [CpG Island and STR Annotator Plugins](#CpG_Island_and_STR_Annotator_Plugins) | Joan Pontius | [Summary](#CpG_Island_and_STR_Annotator_Plugins) |
|  | [GMOD Sustainability and Organization](#GMOD_Sustainability_and_Organization) | [Dave Clements](User%3AClements "User%3AClements") | [Summary](#GMOD_Sustainability_and_Organization) |
| 15:15 | Break |  |  |

## <span id="Wednesday_.26_Thursday.2C_15-16_September" class="mw-headline">Wednesday & Thursday, 15-16 September</span>

[GMOD Europe 2010](GMOD_Europe_2010 "GMOD Europe 2010") continued after
the GMOD meeting, starting with the [Satellite
Meetings](Satellite_Meetings_-_GMOD_Europe_2010 "Satellite Meetings - GMOD Europe 2010")
(topics were [Post Reference Genome
Tools](Post_Reference_Genome_Tools "Post Reference Genome Tools") and
[Community
Annotation](Community_Annotation_-_September_2010_Satellite "Community Annotation - September 2010 Satellite"))
and the [InterMine
Workshop](InterMine_Workshop_-_GMOD_Europe_2010 "InterMine Workshop - GMOD Europe 2010"),
and finishing with the [BioMart
Workshop](BioMart_Workshop_-_GMOD_Europe_2010 "BioMart Workshop - GMOD Europe 2010").
See [GMOD Europe 2010](GMOD_Europe_2010#Events "GMOD Europe 2010") for a
complete schedule.

# <span id="Presentations" class="mw-headline">Presentations</span>

Presentations spanned two days and covered a wide variety of topics.

## <span id="By_Topic" class="mw-headline">By Topic</span>

The talks can be roughly categorized:

<table>
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<tbody>
<tr class="odd">
<td><table class="wikitable">
<colgroup>
<col style="width: 100%" />
</colgroup>
<tbody>
<tr class="odd">
<td>GMOD Project
<ul>
<li><a href="#The_State_of_GMOD">The State of GMOD</a></li>
<li><a href="#Help_Desk_Update">Help Desk Update</a></li>
<li><a href="#Lessons_from_the_Apache_Community">Lessons from the Apache
Community</a></li>
<li><a href="#GMOD_Sustainability_and_Organization">GMOD Sustainability
and Organization</a></li>
</ul></td>
</tr>
<tr class="even">
<td>Components
<ul>
<li><a href="#What.27s_New">What's New</a></li>
<li><a href="#JBrowse_Development">JBrowse Development</a> and <a
href="#JBrowse_Grant_Proposal">JBrowse Grant Proposal</a></li>
<li><a href="#New_Components">New Components</a></li>
<li><a href="#The_GMOD_Chado_Natural_Diversity_Module">The GMOD Chado
Natural Diversity Module</a></li>
<li><a
href="#Cosmic_GBrowse:_Visualising_cancer_mutations_in_genomic_context">Cosmic
GBrowse: Visualising cancer mutations in genomic context</a></li>
<li><a
href="#GMOD_Projects_at_the_Center_for_Genomics_and_Bioinformatics">GMOD
Projects at the Center for Genomics and Bioinformatics</a></li>
<li><a href="#GMOD_RPC_API:_The_almost_RESTful_GMOD_API">GMOD RPC API:
The almost RESTful GMOD API</a></li>
<li><a href="#InterMine:_new_Mines_and_new_features">InterMine: new
Mines and new features</a></li>
<li><a href="#GBrowse_2_and_JBrowse_Features">GBrowse 2 and JBrowse
Features</a></li>
</ul></td>
</tr>
<tr class="odd">
<td>Extending GMOD
<ul>
<li><a href="#New_Components">New Components</a></li>
<li><a href="#PSICQUIC:_The_PSI_Common_QUery_Interface">PSICQUIC: The
PSI Common QUery Interface</a></li>
<li><a href="#MolGenIS_and_XGAP">MolGenIS and XGAP</a></li>
<li><a href="#The_GMOD_Chado_Natural_Diversity_Module">The GMOD Chado
Natural Diversity Module</a></li>
<li><a
href="#GMOD_Projects_at_the_Center_for_Genomics_and_Bioinformatics">GMOD
Projects at the Center for Genomics and Bioinformatics</a></li>
<li><a href="#GMOD_RPC_API:_The_almost_RESTful_GMOD_API">GMOD RPC API:
The almost RESTful GMOD API</a></li>
<li><a
href="#Towards_a_GO_Annotation_Tool:_Curation_Accelerator_Software">Towards
a GO Annotation Tool: Curation Accelerator Software</a></li>
<li><a href="#BioPivot">BioPivot</a></li>
<li><a href="#CRAWL_.28Chado_RESTful_Access_Web-service_Layer.29">CRAWL
(Chado RESTful Access Web-service Layer)</a></li>
<li><a href="#CpG_Island_and_STR_Annotator_Plugins">CpG Island and STR
Annotator Plugins</a></li>
<li><em>Also see the <a href="Post_Reference_Genome_Tools"
title="Post Reference Genome Tools">Post Reference Genome Tools</a>
Satellite Meeting.</em></li>
</ul></td>
</tr>
</tbody>
</table></td>
<td data-valign="top"><table class="wikitable">
<colgroup>
<col style="width: 100%" />
</colgroup>
<tbody>
<tr class="odd">
<td>Data Types
<ul>
<li><a
href="#The_Open_Microscopy_Environment:_Open_Informatics_for_Biological_Imaging">The
Open Microscopy Environment: Open Informatics for Biological Imaging</a>
(Images)</li>
<li><a href="#PSICQUIC:_The_PSI_Common_QUery_Interface">PSICQUIC: The
PSI Common QUery Interface</a> (Molecular Interactions)</li>
<li><a
href="#Cosmic_GBrowse:_Visualising_cancer_mutations_in_genomic_context">Cosmic
GBrowse: Visualising cancer mutations in genomic context</a>
(Mutations)</li>
<li><a href="#Literature_Curation_in_GMOD">Literature Curation in
GMOD</a> (Literature)</li>
<li><a
href="#Towards_a_GO_Annotation_Tool:_Curation_Accelerator_Software">Towards
a GO Annotation Tool: Curation Accelerator Software</a> (GO)</li>
<li><a href="#CpG_Island_and_STR_Annotator_Plugins">CpG Island and STR
Annotator Plugins</a> (CpG islands and short tandem repeats)</li>
</ul></td>
</tr>
<tr class="even">
<td>Community Annotation
<ul>
<li><a href="#Literature_Curation_in_GMOD">Literature Curation in
GMOD</a></li>
<li><a
href="#Towards_a_GO_Annotation_Tool:_Curation_Accelerator_Software">Towards
a GO Annotation Tool: Curation Accelerator Software</a></li>
<li><a href="#CRAWL_.28Chado_RESTful_Access_Web-service_Layer.29">CRAWL
(Chado RESTful Access Web-service Layer)</a></li>
<li><em>See also the <a
href="Community_Annotation_-_September_2010_Satellite"
title="Community Annotation - September 2010 Satellite">Community
Annotation</a> Satellite Meeting.</em></li>
</ul></td>
</tr>
<tr class="odd">
<td>Visualization
<ul>
<li><a
href="#Cosmic_GBrowse:_Visualising_cancer_mutations_in_genomic_context">Cosmic
GBrowse: Visualising cancer mutations in genomic context</a></li>
<li><a href="#BioPivot">BioPivot</a></li>
<li><a href="#GBrowse_2_and_JBrowse_Features">GBrowse 2 and JBrowse
Features</a></li>
<li><a href="#Some_Interesting_Documents">Some Interesting
Documents</a></li>
<li><a href="#CpG_Island_and_STR_Annotator_Plugins">CpG Island and STR
Annotator Plugins</a></li>
</ul></td>
</tr>
<tr class="even">
<td>Web Services and Distributed Resources
<ul>
<li><a href="#PSICQUIC:_The_PSI_Common_QUery_Interface">PSICQUIC: The
PSI Common QUery Interface</a></li>
<li><a href="#GMOD_RPC_API:_The_almost_RESTful_GMOD_API">GMOD RPC API:
The almost RESTful GMOD API</a></li>
<li><a
href="#Overview_of_current_resources_and_update_on_DAS_Meeting_Cambridge_2010">Overview
of current resources and update on DAS Meeting Cambridge 2010</a></li>
<li><a href="#CRAWL_.28Chado_RESTful_Access_Web-service_Layer.29">CRAWL
(Chado RESTful Access Web-service Layer)</a></li>
</ul></td>
</tr>
</tbody>
</table></td>
</tr>
</tbody>
</table>

## <span id="The_State_of_GMOD" class="mw-headline">The State of GMOD</span>



[<img src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/c/c0/ScottSept2010.jpg" width="100"
height="135" alt="Scott Cain" />](User%3AScott "Scott Cain")



[Scott Cain](User%3AScott "User%3AScott"),
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/7/78/TheStateofGMODSept2010.pdf"
class="internal" title="TheStateofGMODSept2010.pdf">PDF</a>

GMOD is:

- A set of interoperable open-source **software** components for
  visualizing, annotating, and managing biological data.
- An active **community** of developers and users asking diverse
  questions, and facing common challenges, with their biological data.

These two things are equally important.

GMOD is used by

- hundreds of organizations
- large and small
- corporate and academic
- all over the world
- across the tree of life

### <span id="What.27s_New" class="mw-headline">What's New</span>

[GBrowse](GBrowse.1 "GBrowse")

- Releases
  - 1.70, 2.14
- Features
  - Rubberband region selection
  - Drag and drop track ordering
  - Collapsible tracks
  - Popup balloons
  - Allele/gentotype frequency
  - Geolocation popups
  - Circular genome support (1.71)
  - Asynchronous updates (2.0)
  - User authentication
  - Multiple server support (2.0)
  - SQLite, SAMtools (NGS) adaptors

[JBrowse](JBrowse.1 "JBrowse")

- GMOD's 2nd Generation Genome Browser
- It's *fast*
- Completely new genome browser implementation:
  - Client side rendering
  - Heavy use of [AJAX](Glossary#AJAX "Glossary")
  - Uses [JSON](Glossary#JSON "Glossary") and <a
    href="http://bioinformatics.oxfordjournals.org/cgi/content/short/23/11/1386"
    class="external text" rel="nofollow">Nested Containment Lists</a>

[GBrowse_syn](GBrowse_syn.1 "GBrowse syn")

- [GBrowse](GBrowse.1 "GBrowse") based comparative genomics viewer
- Shows a reference sequence compared to 2+ others
- Can also show any GBrowse-based annotations
- Syntenic blocks do not have to be colinear
- Can also show duplications

<a href="Chado" class="mw-redirect" title="Chado">Chado</a>

- Chado is the GMOD [schema](Glossary#Schema "Glossary"); it is modular
  and extensible, allowing the addition of new data types “easily.”
  Covered data types in ontologies, organisms, sequence features,
  genotypes, phenotypes, libraries, stocks, microarrays, with natural
  diversity recently being rolled into the schema (but not yet
  released).
- 1.0 Release solidified the Chado that most people were already using
  from source.
- 1.1 Introduced support for GBrowse to use full text searching and
  “summary statistics” (i.e., feature density plots). Version 0.30 of
  <a href="http://search.cpan.org/perldoc?Bio::DB::Das::Chado"
  class="external text" rel="nofollow">Bio::DB::Das::Chado</a> is needed
  for these functions.

[Tripal](Tripal.1 "Tripal")

- New (2009) web front end for
  <a href="Chado" class="mw-redirect" title="Chado">Chado</a> databases
- Set of Drupal modules
- Modules approximately correspond to Chado modules
- Easy to create new modules
- Includes user authentication, job management, curation support

[TableEdit](TableEdit.1 "TableEdit")

- A MediaWiki extension (MediaWiki software used at Wikipedia, GMOD.org)
- Provides graphical user interface (GUI) to wiki tables
- Can also provide GUI to database tables
- Work in progress to use this with
  <a href="Chado" class="mw-redirect" title="Chado">Chado</a>
- Potential to give wiki access to a Chado database
- See <a href="http://ecoliwiki.net" class="external free"
  rel="nofollow">http://ecoliwiki.net</a>

[BioMart](BioMart "BioMart")

- BioMart is a query-oriented data management system
- Provides a web based query interface
- Strong data federation
- [BioMart
  Workshop](BioMart_Workshop_-_GMOD_Europe_2010 "BioMart Workshop - GMOD Europe 2010")
  on Thursday.

[InterMine](InterMine "InterMine")

- InterMine is a query-oriented data management system
- Provides a web based query interface
- Very flexible queries and query optimization
- [InterMine
  Workshop](InterMine_Workshop_-_GMOD_Europe_2010 "InterMine Workshop - GMOD Europe 2010")
  on Wednesday

[MAKER](MAKER.1 "MAKER")

- Genome annotation pipeline for creating gene models
- Output can be loaded into [GBrowse](GBrowse.1 "GBrowse"),
  [Apollo](Apollo.1 "Apollo"),
  <a href="Chado" class="mw-redirect" title="Chado">Chado</a>, …
- Incorporates
  - SNAP, RepeatMasker, exonerate, BLAST, Augustus, FGENESH, GeneMark,
    MPI
- Other capabilities
- Map existing annotation onto new assemblies
- Merge multiple legacy annotation sets into a consensus set
- Update existing annotations with new evidence
- Integrate raw InterProScan results
- MAKER Online in beta

[Apollo](Apollo.1 "Apollo")

- Java-based GUI application for browsing and annotating genomic
  sequences
- Can be installed via WebStart (i.e., by clicking on a link)
- Can read/write to
  <a href="Chado" class="mw-redirect" title="Chado">Chado</a>,
  [GFF3](GFF3 "GFF3"), GenBank, GAME XML

### <span id="Next_GMOD_Meeting.3F" class="mw-headline">Next [GMOD Meeting](Meetings "Meetings")?</span>

[<img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1b/GMODAmericas2011Logo.jpg/100px-GMODAmericas2011Logo.jpg.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1b/GMODAmericas2011Logo.jpg/150px-GMODAmericas2011Logo.jpg.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1b/GMODAmericas2011Logo.jpg/200px-GMODAmericas2011Logo.jpg.png 2x"
width="100" height="100" alt="March 2011 GMOD Meeting" />](GMOD_Americas_2011 "March 2011 GMOD Meeting")

- Next Spring Sometime:
- ABRF: Association of Biomolecular Resource Facilities
  - Feb. 19-22, San Antonio, TX
- Biology of Genomes
  - May 10-14, Cold Spring Harbor Lab, NY
- Suggestions?

**Note: The next meeting is the [March 2011 GMOD
Meeting](March_2011_GMOD_Meeting "March 2011 GMOD Meeting") being held
at <a href="http://nescent.org" class="external text"
rel="nofollow">NESCent</a>, in Durham, North Carolina, United States, as
part of [GMOD Americas 2011](GMOD_Americas_2011 "GMOD Americas 2011").**

## <span id="Help_Desk_Update" class="mw-headline">[Help Desk](GMOD_Help_Desk "GMOD Help Desk") Update</span>

[Dave Clements](User%3AClements "User%3AClements"),
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/b/bf/HelpDeskSept2010.pdf" class="internal"
title="HelpDeskSept2010.pdf">PDF</a>, <a
href="ftp://ftp.gmod.org/pub/gmod/Meetings/2010/GMODSeptember/HelpDeskSept2010.ppt"
class="external text" rel="nofollow">PPT</a>

### <span id="Mailing_List_Archives" class="mw-headline">Mailing List Archives</span>

[GMOD Mailing Lists](GMOD_Mailing_Lists "GMOD Mailing Lists") are all
over. Many are hosted at SourceForge, but several are elsewhere (EBI,
Bluehost, Berkeley, ...). Some don't have public archives and those that
do are spread around The lists at SourceForge have searchable archives
but the search interface is frustrating.

Since May/June 2010, all emails to GMOD mailing lists have been archived
in a single
<a href="http://gmod.827538.n3.nabble.com/" class="external text"
rel="nofollow">searchable hierarchy at Nabble</a>. Nabble has a
functional search capability and you can now search all lists, or just a
single list.

### <span id="GMOD_Membership_Requirements" class="mw-headline">[GMOD Membership](GMOD_Membership "GMOD Membership") Requirements</span>

[GMOD's requirements for software to join
GMOD](GMOD_Membership "GMOD Membership") were codified in February 2010,
following the [January 2010 GMOD
Meeting](January_2010_GMOD_Meeting "January 2010 GMOD Meeting"). These
requirements were in use before February, but were inconsistently
applied.

Version 1 Requirements:

- Meets a common need
- Useful over time
- Configurable and Extensible
- Open source license for all users
- Interoperable with existing [GMOD
  Components](GMOD_Components "GMOD Components")
- Commitment of support

For next version, want to add:

- Support [mailing list](GMOD_Mailing_Lists "GMOD Mailing Lists") that
  is publicly archived
- Publicly accessible code repository

Discussion favored these additions. The issue of incompatible open
source licenses also came up. GMOD currently requires any
<a href="http://www.opensource.org/licenses" class="external text"
rel="nofollow">OSI approved license</a>. However, some of those licenses
are not compatible with each other, meaning they such components can't
be bundled together.

### <span id="GMOD_Promotion" class="mw-headline">[GMOD Promotion](GMOD_Promotion "GMOD Promotion")</span>



[<img src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/c/c2/PoweredByGMOD150.png" width="150"
height="71" alt="Powered by GMOD" />](Powered_by_GMOD "Powered by GMOD")



Help spread the word about GMOD components and the GMOD project.

Why?

- Increased visibility leads to
  → Increased adoption, which leads to
  → more projects contributing back
- Increased adoption & development leads to
  → increased funding

How?

- [Cite](GMOD_Promotion#Citing_GMOD "GMOD Promotion") GMOD, GMOD
  Components in your papers, presentations, grants
- [Powered by GMOD](Powered_by_GMOD "Powered by GMOD") icons
- [Speakers](GMOD_Promotion#Speakers "GMOD Promotion") at [your
  event](Calendar#GMOD_Event_List "Calendar"); not just
  [Scott](User%3AScott "User%3AScott") and
  [Dave](User%3AClements "User%3AClements"). PIs and developers are also
  available.
- [Graphics &
  slides](GMOD_Promotion#Presentation_Material "GMOD Promotion") for
  your presentations, posters
- [Presentation and event
  promotion](GMOD_Promotion#Presentation_and_Event_Promotion "GMOD Promotion")
- [Brochures](GMOD_Promotion#Brochures "GMOD Promotion") (GMOD project,
  events)
- [Bling!](GMOD_Promotion#Mugs.2C_T-Shirts.2C_Buttons.2C_... "GMOD Promotion")

The [GMOD Promotion](GMOD_Promotion "GMOD Promotion") page launched in
July 2010.

### <span id="GMOD_Logo_Program" class="mw-headline">[GMOD Logo Program](GMOD_Logo_Program "GMOD Logo Program")</span>



|  |  |  |  |  |  |  |  |  |
|----|----|----|----|----|----|----|----|----|
| <a href="File:ApolloLogo.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1b/ApolloLogo.png/100px-ApolloLogo.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1b/ApolloLogo.png/150px-ApolloLogo.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1b/ApolloLogo.png/200px-ApolloLogo.png 2x"
width="100" height="36" alt="ApolloLogo.png" /></a> | <a href="File:GBrowseLogo.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/0/04/GBrowseLogo.png/110px-GBrowseLogo.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/0/04/GBrowseLogo.png/165px-GBrowseLogo.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/0/04/GBrowseLogo.png/220px-GBrowseLogo.png 2x"
width="110" height="26" alt="GBrowseLogo.png" /></a> | <a href="File:RedFlyLogo.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1d/RedFlyLogo.png/50px-RedFlyLogo.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1d/RedFlyLogo.png/75px-RedFlyLogo.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1d/RedFlyLogo.png/100px-RedFlyLogo.png 2x"
width="50" height="35" alt="RedFlyLogo.png" /></a> | <a href="File:MetabolicMineLogoWide.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/b/b6/MetabolicMineLogoWide.png/130px-MetabolicMineLogoWide.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/b/b6/MetabolicMineLogoWide.png/195px-MetabolicMineLogoWide.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/b/b6/MetabolicMineLogoWide.png/260px-MetabolicMineLogoWide.png 2x"
width="130" height="28" alt="MetabolicMineLogoWide.png" /></a> | <a href="File:CMapLogo.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/3/33/CMapLogo.png/40px-CMapLogo.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/3/33/CMapLogo.png/60px-CMapLogo.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/3/33/CMapLogo.png/80px-CMapLogo.png 2x"
width="40" height="46" alt="CMapLogo.png" /></a> | <a href="File:PhenoteLogo_Medium.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/3/3d/PhenoteLogo_Medium.png/100px-PhenoteLogo_Medium.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/3/3d/PhenoteLogo_Medium.png/150px-PhenoteLogo_Medium.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/3/3d/PhenoteLogo_Medium.png/200px-PhenoteLogo_Medium.png 2x"
width="100" height="19" alt="PhenoteLogo Medium.png" /></a> | <a href="File:PomBaseLogo.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/5/5e/PomBaseLogo.png/100px-PomBaseLogo.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/5/5e/PomBaseLogo.png/150px-PomBaseLogo.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/5/5e/PomBaseLogo.png/200px-PomBaseLogo.png 2x"
width="100" height="27" alt="PomBaseLogo.png" /></a> | <a href="File:GBrowse_syn_logo.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/4/44/GBrowse_syn_logo.png/100px-GBrowse_syn_logo.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/4/44/GBrowse_syn_logo.png/150px-GBrowse_syn_logo.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/4/44/GBrowse_syn_logo.png/200px-GBrowse_syn_logo.png 2x"
width="100" height="32" alt="GBrowse syn logo.png" /></a> | <a href="File:InterMineLogo.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/13/InterMineLogo.png/120px-InterMineLogo.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/13/InterMineLogo.png/180px-InterMineLogo.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/13/InterMineLogo.png/240px-InterMineLogo.png 2x"
width="120" height="15" alt="InterMineLogo.png" /></a> |



Nine projects got new logos in the [Spring 2010 Logo
Program](Spring_2010_Logo_Program "Spring 2010 Logo Program"). Logos
were done by John Aikman's Spring 2010 Advanced Design class at
Linn-Benton Community College, Albany, Oregon, United States. Each
project worked with 2-3 students during the quarter to produce the
selected logos.

We might do this again in 2011.

### <span id="2010_GMOD_Community_Survey" class="mw-headline">2010 GMOD Community Survey</span>

The [2008 GMOD Community
Survey](2008_GMOD_Community_Survey "2008 GMOD Community Survey") covered
components and project wide topics. The [2009 GMOD Community
Survey](2009_GMOD_Community_Survey "2009 GMOD Community Survey") focused
on genome and comparative genomics browsing. The 2010 GMOD Community
Survey will cover components and project wide topics. We may use it to
produce a *GMOD Project publication*.

These surveys help guide the project and also show potential and current
GMOD users what the larger community is doing.

Look for the 2010 survey in October.

### <span id="Events" class="mw-headline">Events</span>

[Satellite
Meetings](Satellite_Meetings_-_GMOD_Europe_2010 "Satellite Meetings - GMOD Europe 2010")!

The satellites at the [January 2010 GMOD
Meeting](January_2010_GMOD_Meeting "January 2010 GMOD Meeting") were
such a success that we decided to do them again. Satellites are *birds
of a feather* discussions where participants with a common interest
discuss that topic. The satellites at this meeting were:

- [Post Reference Genome
  Tools](Post_Reference_Genome_Tools "Post Reference Genome Tools")
- [Community
  Annotation](Community_Annotation_-_September_2010_Satellite "Community Annotation - September 2010 Satellite")

See the satellite meeting pages for summaries of the discussion.

<a href="GMOD_Summer_School" class="mw-redirect"
title="GMOD Summer School">GMOD Summer School</a>

In 2010 we held our [4th summer
school](2010_GMOD_Summer_School_-_Americas "2010 GMOD Summer School - Americas")
in May at NESCent, in Durham, North Carolina, US. We had 62 applicants
for 25 slots.

The 2011 course will likely be at NESCent again. However, starting in
2011, summer school expenses will no longer be covered by a grant (see
below). This means that we will start charging tuition, and that we will
also start seeking sponsors.

Summer school sessions become online tutorials that include starting and
ending VMware images, step by step instructions, and example datasets.

Other [Upcoming Events of Note](Calendar "Calendar")

- <a href="http://hinv.jp/biocuration2010/" class="external text"
  rel="nofollow">Biocuration 2010</a>
  October, Tokyo, Japan
- <a href="http://bioinformatics.ai.sri.com/ptools10/"
  class="external text" rel="nofollow">Pathway Tools Workshop</a>
  October, Menlo Park, California, US
- [GMOD Evo Hackathon](GMOD_Evo_Hackathon "GMOD Evo Hackathon")
  November, Durham, North Carolina, US
- Computational and Comparative Genomics
  November, Cold Spring Harbor, New York, US
- [Plant and Animal Genome](PAG_2011 "PAG 2011")
  January, San Diego, California, US
- Workshop on Molecular Evolution
  January, Cesky Krumlov, Czech Republic
- [Galaxy](Galaxy.1 "Galaxy") Developers Conference
  2011, Europe

### <span id="JBrowse_Development" class="mw-headline">[JBrowse](JBrowse.1 "JBrowse") Development</span>

1.1 just released

- Scalability: very large data sets, including NGS reads, human EST/SNP
  tracks
- Extensibility: custom tracks
- Backward incompatible [JSON](Glossary#JSON "Glossary") format

1.2 Release (December 2010)

- improved NGS display (paired-end reads, possibly read-to-genome
  alignments)
- reduced memory usage for NGS
- minor UI enhancements including y-axis labels for wiggle tracks

### <span id="JBrowse_Grant_Proposal" class="mw-headline">[JBrowse](JBrowse.1 "JBrowse") Grant Proposal</span>

Sent proposal in this summer; if approved will start around February
2011.

[GBrowse](GBrowse.1 "GBrowse") → [JBrowse](JBrowse.1 "JBrowse")

- JBrowse concepts have proven themselves
- Scalable to coming data set sizes
- GBrowse development will wind down during the grant.

New Features  

- JBrowse ecosystem on par with what GBrowse has
- <a href="DAS" class="mw-redirect" title="DAS">DAS</a> and web services
  support
- Scalability and NGS
- Large numbers of tracks
- Community annotation (upload/publish, tagging, comment, …)
- Mobile device support?

GBrowse → JBrowse Migration Support

- Migration Scripts: Config files, data (data is easy)
- Simultaneous GBrowse and JBrowse support
- JBrowse running on top of GBrowse config and data

### <span id="New_Components" class="mw-headline">New Components</span>

[ISGA](ISGA "ISGA") - [Chris Hemmerich](User%3AChemmeri "User%3AChemmeri")
*et al.* at Indiana U.

- Bioinformatics pipeline service software built on
  <a href="Ergatis" class="mw-redirect" title="Ergatis">Ergatis</a>
- Newest GMOD component

[WebGBrowse](WebGBrowse.1 "WebGBrowse") - Ram Podicheti *et al.* at
Indiana U.

- Hosted [GBrowse](GBrowse.1 "GBrowse") and
  [GUI](Glossary#GUI "Glossary") for GBrowse configuration
- Nominated and approved, almost in.

SOBA - Ginger Fan *et al.* U of Utah

- [GFF3](GFF3 "GFF3") file analysis and reporting
- Tabular and graphical reports
- Nominated and approved, code being refactored

<a href="GMOD-DBSF" class="mw-redirect" title="GMOD-DBSF">GMOD-DBSF</a>,
[genes4all](Genes4all "Genes4all"), … - Alexie Papanicolaou at CSIRO

- Drupal based toolkit for building organism web sites
- Submitted for publication; not yet nominated

### <span id="Some_Interesting_Documents" class="mw-headline">Some Interesting Documents</span>



<a href="File:Urgi.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/2/23/Urgi.png/100px-Urgi.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/2/23/Urgi.png/150px-Urgi.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/2/23/Urgi.png/200px-Urgi.png 2x"
width="100" height="68" alt="Urgi.png" /></a>


<a href="File:MaizeGDB.gif" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/7/7f/MaizeGDB.gif" width="265" height="60"
alt="MaizeGDB.gif" /></a>


<a href="http://database.oxfordjournals.org/" rel="nofollow"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/3/3d/DatabaseJournal.jpg/265px-DatabaseJournal.jpg.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/3/3d/DatabaseJournal.jpg/398px-DatabaseJournal.jpg.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/3/3d/DatabaseJournal.jpg/530px-DatabaseJournal.jpg.png 2x"
width="265" height="32" alt="DatabaseJournal.jpg" /></a>



[File:How to load chado to biomart.pdf](File:How_to_load_chado_to_biomart.pdf "File:How to load chado to biomart.pdf")  
AO Keliet, J Amselem, S Derozie, and D Steinbach, all @
<a href="http://urgi.versailles.inra.fr/" class="external text"
rel="nofollow">INRA URGI</a>

**<a href="http://database.oxfordjournals.org/cgi/content/abstract/baq007"
class="external text" rel="nofollow">Choosing a genome browser for a
Model Organism Database: surveying the Maize community</a>**

TZ Sen, LC Harper, ML Schaeffer, CM. Andorf, TE Seigfried, DA Campbell,
and CJ. Lawrence

How and why <a href="http://maizegdb.org/" class="external text"
rel="nofollow">MaizeGDB</a> picked [GBrowse](GBrowse.1 "GBrowse")

Appeared in
*<a href="http://database.oxfordjournals.org/" class="external text"
rel="nofollow">Database: The Journal of Biological Databases and
Curation</a>*

<a href="http://www.nature.com/nmeth/journal/v7/n3s/index.html"
class="external text" rel="nofollow"><em>Nature Methods</em> Supplement
on visualizing biological data</a>, March 2010

- <a
  href="http://www.nature.com/nmeth/journal/v7/n3s/abs/nmeth.f.301.html"
  class="external text" rel="nofollow">Visualizing biological data - now
  and in the future</a>, SI O'Donoghue, *et al.*
- <a href="http://www.nature.com/nmeth/journal/v7/n3s/abs/nmeth.1422.html"
  class="external text" rel="nofollow">Visualizing genomes: techniques and
  challenges</a>, CB Nielsen, *et al.*
- <a href="http://www.nature.com/nmeth/journal/v7/n3s/abs/nmeth.1434.html"
  class="external text" rel="nofollow">Visualization of multiple
  alignments, phylogenies and gene family evolution</a>, JB Proctor, *et
  al.*
- <a href="http://www.nature.com/nmeth/journal/v7/n3s/abs/nmeth.1431.html"
  class="external text" rel="nofollow">Visualization of image data from
  cells to organisms</a>, T Walker, *et al.*
- <a href="http://www.nature.com/nmeth/journal/v7/n3s/abs/nmeth.1427.html"
  class="external text" rel="nofollow">Visualization of macromolecular
  structures</a>, SI O'Donoghue, *et al.*
- <a href="http://www.nature.com/nmeth/journal/v7/n3s/abs/nmeth.1436.html"
  class="external text" rel="nofollow">Visualization of omics data for
  systems biology</a>, N Gehlenborg, *et al.*

### <span id="GMOD_on_the_Web" class="mw-headline">GMOD on the Web</span>

GMOD.org

- Moving from CSHL to OICR, real soon now
- MediaWiki upgrade
- Probably lots of new extensions
- Maybe a modified skin
- Look into adding
  - User log section
  - Scrapbook for contributed code
  - Membership directory ([TableEdit](TableEdit.1 "TableEdit") based)
  - Semi-automated publication listing/linking

Should GMOD have a social presence?

GMOD already has [mailing
lists](GMOD_Mailing_Lists "GMOD Mailing Lists"), wiki, [GMOD
News](GMOD_News "GMOD News") (RSS), and IRC. Should GMOD have a presence
in social media as well? If so, what should the goals be? Outreach?
Community building or forums? Social bookmarking? Which tools should we
use: <a href="http://twitter.com" class="external text"
rel="nofollow">Twitter</a>,
<a href="http://facebook.com" class="external text"
rel="nofollow">Facebook</a>,
<a href="http://connotea.org" class="external text"
rel="nofollow">Connotea</a>,
<a href="http://stumbleupon.com" class="external text"
rel="nofollow">StumbleUpon</a>,
<a href="http://technolrati.com" class="external text"
rel="nofollow">Technorati</a>,
<a href="http://network.nature.com/" class="external text"
rel="nofollow">Nature Network</a>…

<a href="http://www.biocurator.org/publications.shtml"
class="external text" rel="nofollow">ISB uses Connotea</a> to bookmark
<a href="http://www.connotea.org/tag/biocuration" class="external text"
rel="nofollow">biocuration</a>,
<a href="http://www.connotea.org/tag/text%20mining"
class="external text" rel="nofollow">text mining</a>, and
<a href="http://www.connotea.org/tag/semantic%20annotation"
class="external text" rel="nofollow">semantic annotation</a> papers. (I
have since found out that ISB also <a
href="http://www.linkedin.com/groups?about=&amp;gid=1851280&amp;trk=anet_ug_grppro"
class="external text" rel="nofollow">uses LinkedIn</a> for social
networking.)

This generated some discussions and some conclusions:

- Community bookmarking may be worthwhile.
- If you can automatically tweet page updates and news items, do it.
- Don't manually post stuff to twitter
- Don't build community through Facebook. There are better time
  investments.

## <span id="The_Open_Microscopy_Environment:_Open_Informatics_for_Biological_Imaging" class="mw-headline">The <a href="http://www.openmicroscopy.org/" class="external text"
rel="nofollow">Open Microscopy Environment</a>: Open Informatics for Biological Imaging</span>



<a href="http://gre.lifesci.dundee.ac.uk/staff/jason_swedlow.html"
rel="nofollow" title="Jason Swedlow"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/8/80/JasonSwedlow.jpg/100px-JasonSwedlow.jpg"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/8/80/JasonSwedlow.jpg/150px-JasonSwedlow.jpg 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/8/80/JasonSwedlow.jpg/200px-JasonSwedlow.jpg 2x"
width="100" height="82" alt="Jason Swedlow" /></a>



Jason Swedlow, <a
href="http://gmod.org/mediawiki/index.php?title=Special:Upload&amp;wpDestFile=OpenMicroscopySept2010.pdf"
class="new" title="OpenMicroscopySept2010.pdf">PDF</a>, <a
href="http://gmod.org/mediawiki/index.php?title=Special:Upload&amp;wpDestFile=OpenMicroscopySept2010.ppt"
class="new" title="OpenMicroscopySept2010.ppt">PPT</a>

Images are pretty pictures *and* measurements.

The Open Microscopy Environment (OME) was founded in 2000, by people who
had a strong need, but weren't necessarily image metadata experts. OME
has grown considerably over the years by embracing collaborators.

OME software is released primarily under the GPL. and to a lesser
extent, LGPL licenses. The commercial company Glencoe Software was
founded in 2005. Glencoe is a GPL based company that provides
customization.

As a project OME is very public in building and publishing project
roadmaps, and in making its bug tracking system be publicly accessible.
The OME community meets annually in Paris in the Spring (which
undoubtedly encourages collaboration).They also care a great deal about
quality. OME software includes automatic reporting of exceptions, and
their bug tracking system is publicly accessible.

They use Hudson for continuous integration All remote stuff done through
ICE form ZeroC. Needed to talk to everything in use out there, Ruby to
Matlab. Use ICE for job distribution.

### <span id="Data_Standards_and_Formats" class="mw-headline">Data Standards and Formats</span>

OME also produces, and its software is based on, an open data model for
biological images, called
<a href="http://ome-xml.org" class="external text"
rel="nofollow">OME-XML</a>. This format can be dropped into the header
of a TIFF file.

<a href="http://bio-formats.org" class="external text"
rel="nofollow">Bio-Formats.org</a> is another OME effort. Bio-Formats
provides tools to convert ~ 95 image formats into a single common format
that standardized tools (include OME tools) can work with. OME is trying
to establish Bio-Formats the default library for reading biological
image data.

Metadata matters. Researchers need computation access to image data in
the real world. OME uses structured annotations that can be attached to
any part of image data.

Genome community knows what it's baseline is. It has a basic framework,
the location on the genome. With an image it's not at all clear what the
baseline is.

### <span id="Imaging_Culture_and_Sustainability" class="mw-headline">Imaging Culture and Sustainability</span>

Most biologists view image output as a result, not as shared data. Most
images that are created are not useful and are destined to sit on DVDs,
CDs, floppies, ... until they are finally thrown away when the media can
no longer be read, or the researcher moves.

How do we make sure that useful data sticks around? Images take up a lot
of storage, relative to many data types. Images in publications are of
high quality, but the resolution is relatively low. (And research
indicates that at least 25% of published articles have images that have
been manipulated in some way, although usually just cleaned up.) OME
started as trying to solve a local, lab at a time problem.

*More to come ...*

## <span id="PSICQUIC:_The_PSI_Common_QUery_Interface" class="mw-headline"><a href="http://psicquic.googlecode.com" class="external text"
rel="nofollow">PSICQUIC</a>: The PSI Common QUery Interface</span>



[<img src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/6/65/Sept2010Bruno.jpg" width="100"
height="106" alt="Bruno Baranda" />](User%3ABaranda "Bruno Baranda")



[Bruno Aranda](User%3ABaranda "User%3ABaranda"),
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/d/d2/PSICQUICSept2010.pdf" class="internal"
title="PSICQUICSept2010.pdf">PDF</a>

The *<a href="http://www.psidev.info/" class="external text"
rel="nofollow">Proteomics Standards Initiative (PSI)</a> Common Query
Interface* (\[PSICQUIC, pronounced like "psychic" - most of the time)
standardizes access to molecular interaction data. PSICQUIC is a web
service specification based on PSI standards. Resources that implement
PSICQUIC are listed in a public registry. There are currently more than
*14 million* binary interactions from at least 12 different resources
(<a href="http://www.ebi.ac.uk/intact/" class="external text"
rel="nofollow">IntAct</a>,
<a href="http://reactome.org" class="external text"
rel="nofollow">Reactome</a>,
<a href="http://www.ebi.ac.uk/chembldb/" class="external text"
rel="nofollow">ChEMBL</a>, ...) available using PSICQUIC. This
widespread adoption allows client programs that speak PSICQUIC to
uniformly access all this no matter where it is located.

PSI talked for many years about standards and formats and how to share
data. They 2002-2006 thinking about standards. They found it was very
complicated to agree on something. but that it has been easy to
implement. Most PSICQUIC implementation came out of 3 biohackathons.

### <span id="PSICQUIC_Web_Services" class="mw-headline">PSICQUIC Web Services</span>

Methods

Several methods are supported:

- `getByInteraction` - Retrieves interactions by using an interaction
  AC.
- `getByInteractionList` - Retrieves interactions by using a list of
  interaction AC.
- `getByInteractor` - Retrieves interactions by using a participant
  identifier.
- `getByInteractorList` - Retrieves interactions by using a list of
  participant identifiers.
- `getByQuery` - Retrieves interactions by using a Molecular Interaction
  Query Language (MIQL) query (full text searches)
- `getVersion` - Returns the version of the web service implementation.
- `getSupportedDbAcs` - Returns the supported database identifiers
- `getSupportedReturnTypes` - Returns the list of available format types
  for the results.

A limited number of interactions can be fetched. It is possible to
retrieve large datasets using pagination. Most methods have two
additional parameters:

- First result: Index for the first result to retrieve.
- Max results: Number of interactions returned per query.

IMX Consortium and UniProt identifiers are currently being used. Don't
have the one single identifier yet.

SOAP and REST

As PSICQUIC is a Web Service, you can access the data:

- Via SOAP
  - A WSDL file exists, and it is the same for all the databases.
  - IntAct has developed a Java client, but any other language can be
    used.
  - The SoapUI client uses this.
  - However, SOAP's future in PSICQUIC is uncertain and may go away in
    the future.
- Via REST
  - Retrieving data directly by using a URL
  - Easy to access and data can be obtained just using an internet
    browser.
  - Effective for scripting.

### <span id="Formats" class="mw-headline">Formats</span>

PSICQUIC has two standard formats: PSI-MI XML and PSI-MI TAB. The XML is
more complete, and therefore more verbose. PSI-MI TAB is a tabular
format.

- Try these queries at IntAct:
  <a
  href="http://www.ebi.ac.uk/Tools/webservices/psicquic/intact/webservices/current/search/query/species:rat"
  class="external free"
  rel="nofollow">http://www.ebi.ac.uk/Tools/webservices/psicquic/intact/webservices/current/search/query/species:rat</a>
  <a
  href="http://www.ebi.ac.uk/Tools/webservices/psicquic/intact/webservices/current/search/query/brca2?format=xml25"
  class="external free"
  rel="nofollow">http://www.ebi.ac.uk/Tools/webservices/psicquic/intact/webservices/current/search/query/brca2?format=xml25</a>

Other formats are in progress:

- BioPAX (<a
  href="http://www.ebi.ac.uk/Tools/webservices/psicquic/intact/webservices/current/search/query/brca2?format=biopax"
  class="external text" rel="nofollow">IntAct example</a>)
- rdf-xml (<a
  href="http://www.ebi.ac.uk/Tools/webservices/psicquic/intact/webservices/current/search/query/brca2?format=rdf-xml"
  class="external text" rel="nofollow">IntAct example</a>)
- rdf-n3 (<a
  href="http://www.ebi.ac.uk/Tools/webservices/psicquic/intact/webservices/current/search/query/brca2?format=rdf-n3"
  class="external text" rel="nofollow">IntAct example</a>)
- rdf-n3-triple (<a
  href="http://www.ebi.ac.uk/Tools/webservices/psicquic/intact/webservices/current/search/query/brca2?format=rdf-n3-triple"
  class="external text" rel="nofollow">IntAct example</a>)
- rdf-turtle (<a
  href="http://www.ebi.ac.uk/Tools/webservices/psicquic/intact/webservices/current/search/query/brca2?format=rdf-n3-turtle"
  class="external text" rel="nofollow">IntAct example</a>)

As these formats are works in progress, some of these links may fail.

PSICQUIC Registry

The <a
href="http://www.ebi.ac.uk/Tools/webservices/psicquic/registry/registry?action=STATUS"
class="external text" rel="nofollow">PSICQUIC registry</a> contains a
list of the PSICQUIC services available from different providers. It is
a web service itself, and it can be accessed remotely using REST.
Information can be found about the services, such as the URLs to use,
number of interactions provided, versioning, *etc.* The registry
classifies the different services with tags from a PSI ontology.
Querying by tags is a work in progress. Instructions on using the
registry are at
<a href="http://code.google.com/p/psicquic/wiki/Registry"
class="external text" rel="nofollow">Google Code</a>.

### <span id="MIQL" class="mw-headline">MIQL</span>

PSICQUIC also defines the Molecular Interactions Query Language (MIQL).
MIQL allows more powerful and flexible queries and is the default query
syntax for PSIQCUIC. Designed for fast and effective searches on PSI-MI
TAB files. All fields (columns) can be searched with specific queries.
MIQL is a consensus between the different databases, so you should be
able to use the same query across different repositories.

The MIQL syntax is based on the
<a href="http://lucene.apache.org/java/docs/queryparsersyntax.html"
class="external text" rel="nofollow">Lucene syntax</a>. A query is
broken into terms and operators:

- Terms: single words or phrases (group of words surrounded by quotes).
  E.g. `brca2 AND “pull down”`
- Fields: search in specific columns. E.g. `brca2 AND species:human`
- Term modifiers: wildcard searches, fuzzy searches, proximity and range
  searches. E.g. `brc*`
- Operands: OR (or space), AND, NOT, +, -. E.g.
  `brca2 AND rpa1 / brca2 NOT mouse / +brca2 –mouse –expansion:spoke`
- Grouping and field grouping: `brca2 AND (mouse "in vitro")`

### <span id="Creating_a_PSICQUIC_Service" class="mw-headline">Creating a PSICQUIC Service</span>

Simplest recipe to implement PSICQUIC

- Ingredients:
  - PSI-MITAB compliant file.
  - Subversion: to get the source code.
  - Maven: to run the scripts and start the service.
- Steps:
  - Generate the MITAB compliant file.
  - Get the
    <a href="http://code.google.com/p/psicquic/" class="external text"
    rel="nofollow">Reference Implementation (RI)</a>
  - Run the script to index the file.
  - Start the service with the script provided .

### <span id="PSICQUIC_Applications" class="mw-headline">PSICQUIC Applications</span>

PSICQUIC is already implemented in several existing applications,
including Cytoscape 2.7.x, PSICQUIC View, Envision2, and PSICQUIC Client
for Android.

**There is not currently anything in the GMOD suite that uses PSICQUIC.
Should there be?**

### <span id="PSICQUIC_Development" class="mw-headline">PSICQUIC Development</span>

- Smart PSICQUICs: Identification and removal of redundancy
  - Merger and Cluster PSICQUIC services
- PSICQUIC 2.0
  - Overcome the current limitations and many fancy features:
    - Queries using CV terms not possible in the reference
      implementation (it is possible in IntAct).
    - PSI-MI XML is created from the MITAB, so no *n-ary* interactions.
  - New features:
    - Redundancy detection mechanism. ROG/RIG ids by default.
    - Built from PSI-MI XML, so complex data available.

A GMOD component?

Flybase is using Chado Interaction format. Ecoli has lots of
interaction. Can we have a
<a href="Chado" class="mw-redirect" title="Chado">Chado</a> service that
talks PSICQUIC?

Following the talk a couple of possible actions arised:

- Exporting from Chado to MITAB, so we can just create PSICQUIC services
  from any Chado-based application.
- Creating a component / adding interaction information to existing
  components.

Bruno is unfamiliar with Chado, but if someone wants to give it a shot,
he is more than willing to help and participate. All information about
PSICQUIC can be found at
<a href="http://psicquic.googlecode.com" class="external text"
rel="nofollow">Google Code</a>.

And some <a href="http://code.google.com/p/psimi/wiki/PsimiTabFormat"
class="external text" rel="nofollow">basic information about the MITAB
format</a> may help.

## <span id="MolGenIS_and_XGAP" class="mw-headline">[MolGenIS](MolGenIS "MolGenIS") and <a href="http://www.xgap.org" class="external text"
rel="nofollow">XGAP</a></span>



<a href="File:Sept2010Morris.jpg" class="image"
title="Morris Swertz"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/7/74/Sept2010Morris.jpg" width="100"
height="100" alt="Morris Swertz" /></a>



Morris Swertz,
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/d/df/MolGenISXGAPSept2010.pdf"
class="internal" title="MolGenISXGAPSept2010.pdf">PDF</a>

[MolGenIS](MolGenIS "MolGenIS") is a flexible bioinformatics application
toolkit for data management and interfacing. XGAP is an *•eXtensible
Genotype And Phenotype* system that was generated with MolGenIS to store
and visualize xQTL and GWAS data.

One aim of this talk is to explore possible links between MolGenIS and
GMOD: <a href="Chado" class="mw-redirect" title="Chado">Chado</a>,
<a href="DAS" class="mw-redirect" title="DAS">DAS</a>,
[BioMart](BioMart "BioMart"), [InterMine](InterMine "InterMine"),
[GBrowse](GBrowse.1 "GBrowse"), ...?

### <span id="MolGenIS" class="mw-headline">[MolGenIS](MolGenIS "MolGenIS")</span>

MolGenIS has been used to generate systems for many different types of
applications and datatypes. MolGenIS based systems and users include
GEN2PHEN, XGAP, UMCG, FIMM, Sysgenet, and many others.

MolGenIS is a system generator. It addresses the recurring issue of
generating custom databases for each new application that comes along.
The traditional approach requires database design, backend (server)
coding, [API](Glossary#API "Glossary") development, and user interface
coding, all of which is bioinformatician intensive. This approach does
not have reusability and interoperability as a natural byproduct of
development. With MolGenIS system developers provide a system definition
which MolGenIS then uses to automatically instantiate a system that
implements the definition. Writing a system definition requires learning
new skills, but is still much less time intensive then creating a system
from scratch.

MolGenIS includes built in support for many features:

- database generation
- server code generation
- User interface generation, including edit interfaces and audit trails
- Import/Export to Excel
- R interoperability
- workflow ready web services using REST, SOAP and RDF
- UML documentation of underlying models

MolGenIS also comes with extensive documentation, including a
development manual.

Generated systems can also be customized. The user interface can be
extended with plugins implemented in as a Java class, and a layout
definition. Similarly, plugins can be added to the server side by
defining a Java class.

The database backend currently uses a custom [object-relational mapping
(ORM)](Glossary#Object-Relational_Mapping "Glossary"). Hibernate was
considered six years ago, but was lacking key features. The long term
hope is to migrate to a standard ORM such as Hibernate.

### <span id="XGAP" class="mw-headline">XGAP</span>

XGAP (eXtensible Genotype And Phenotype) was developed for xQTL and GWAS
data.

The data is logically in a series of matrices with a different matrix
for each datatype (e.g., genotype, microarray, LC/MS, ...). The initial
idea was to create a database table for each datatype, but this would
have led to a proliferation of structurally similar database tables, and
would require schema changes with the addition of each new type in the
future. (Imagine Chado's feature table split into gene, ssr, snp, exon,
etc. tables.)

XGAP addresses this by embracing a generic matrix model: any trait X any
subject. All matrices are stored in a common database table where each
row corresponds to a single element in a matrix. Schema changes are not
required to add new matrices or new columns to existing matrices. This
is all done by adding matrix and column definitions to definition tables
in the database.

FuGE (*Functional Genomics Experiment*) is a standard model for this
type of information. XGAP builds on top of this.

### <span id="GMOD_Link_Ideas" class="mw-headline">GMOD Link Ideas</span>

- <a href="Chado" class="mw-redirect" title="Chado">Chado</a>
  - XGAP harmonization towards Chado?
  - MolGenIS 4 Chado? Did BioSQL a few years ago.

<!-- -->

- [GBrowse](GBrowse.1 "GBrowse") and
  <a href="DAS" class="mw-redirect" title="DAS">DAS</a>
  - Have XGAP data projected on genome browser?
  - Serve XGAP data as custom tracks?

<!-- -->

- [BioMart](BioMart "BioMart") / [InterMine](InterMine "InterMine")
  - Consume BioMARTdata to auto-annotate experimental data?
  - Export XGAP experiments into MART/MINE query environments?

### <span id="OntoCAT" class="mw-headline"><a href="http://www.ontocat.org" class="external text"
rel="nofollow">OntoCAT</a></span>

## <span id="The_GMOD_Chado_Natural_Diversity_Module" class="mw-headline">The GMOD Chado Natural Diversity Module</span>



<a href="File:Sept2010Bob.jpg" class="image" title="Bob MacCallum"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/8/86/Sept2010Bob.jpg" width="100" height="112"
alt="Bob MacCallum" /></a>



Bob MacCallum,
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/0/02/ChadoNDSept2010.pdf" class="internal"
title="ChadoNDSept2010.pdf">PDF</a>,
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/c/cb/ChadoNDSept2010.ppt" class="internal"
title="ChadoNDSept2010.ppt">PPT</a>, <a
href="https://docs.google.com/present/edit?id=0ASHJk7ao3qFUZGo2ODJydl8xMjNmZGh3eDVnNg&amp;hl=en"
class="external text" rel="nofollow">gdoc</a>

### <span id="Motivation" class="mw-headline">Motivation</span>

- Manage phenotypic and genotypic data for both field collected and
  captive bred organisms
- Store collection site information for growing "next gen"-based
  variation data
- Leverage existing/future Chado modules, GMOD tools and know-how

### <span id="Developmental_History" class="mw-headline">Developmental History</span>

- 2007
  - Early version:
  - HeliconiusDB @ <a href="http://nescent.org" class="external text"
    rel="nofollow">NESCent (National Evolutionary Synthesis Center)</a>
  - Inspired by
    <a href="http://www.maizegenetics.net/gdpdm/" class="external text"
    rel="nofollow">GDPDM (The Genomic Diversity and Phenotype Data
    Model)</a>
- 2009-2010
  - Reincarnation spearheaded by:
    - Sook Jung @ Washington State University,
      <a href="http://www.rosaceae.org/" class="external text"
      rel="nofollow">GDR (Genome Database for <em>Rosaceae</em>)</a>
  - <a href="Chado_Natural_Diversity_Module_Working_Group"
    class="external text" rel="nofollow">GMOD working group</a> formed
- August 2010
  - Natural Diversity module merged into Chado svn trunk

### <span id="Schema" class="mw-headline">Schema</span>

Makes use to the pre-existing stock module. Adds support for Experiment,
Geolocation, and Genotype and Phenotype (reusing some existing tables),
The talk walked through how three specific use cases would be
implemented:

- Cross experiment
- Field collection
- Phenotype assay

### <span id="CV_Terms_and_APIs" class="mw-headline">CV Terms and APIs</span>

Schema is very flexible. nd_experiment.type and nd_experiment_stock.type
are key. There are several ways to do the same thing. The working group
is hoping to agree on core CV terms to aid API development. VectorBase
is planning a simplified API that abstract the module's tables into:

- stocks
- experiments, for which we propose at least three subclasses:
  - field collections
  - phenotyping experiments
  - genotyping experiments
- projects
- protocols

## <span id="Cosmic_GBrowse:_Visualising_cancer_mutations_in_genomic_context" class="mw-headline">Cosmic [GBrowse](GBrowse.1 "GBrowse"): Visualising cancer mutations in genomic context</span>



<a href="File:Sept2010Dave.jpg" class="image" title="Dave Beare"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/4/47/Sept2010Dave.jpg" width="100" height="101"
alt="Dave Beare" /></a>



David Beare,
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/6/6c/CosmicGBrowseSept2010.pdf"
class="internal" title="CosmicGBrowseSept2010.pdf">PDF</a>,
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/0/0e/CosMicGBrowseSept2010.ppt"
class="internal" title="CosMicGBrowseSept2010.ppt">PPT</a>

The Cancer Genome Project (CGP) started in 2000.
<a href="http://www.sanger.ac.uk/genetics/CGP/cosmic/"
class="external text" rel="nofollow">COSMIC, the Catalogue Of Somatic
Mutations In Cancer</a> was launched on 4 February 2004. COSMIC is a
website and backing Oracle database. COSMIC mutation data comes from
several sources.

1.  Three curators who read and annotate publications.
2.  Other database(s) e.g. TP53 (IARC), International Agency for
    Research on Cancer
3.  Sequencing/mutation detection

The project is planning on launching
<a href="http://www.sanger.ac.uk/fgb2/gbrowse/cosmic"
class="external text" rel="nofollow">COSMIC GBrowse</a> on 22 September
2010.

[GBrowse](GBrowse.1 "GBrowse") and CGP

|  |  |
|----|----|
| Q. | How could we visualise the data deluge from next generation sequencing? |
| A. | [GBrowse](GBrowse.1 "GBrowse"). (See <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/0/05/Jan2010GBrowseCancerGenome.pdf"
class="internal" title="Jan2010GBrowseCancerGenome.pdf">[Keiran Raine's
presentation</a> at the [January 2010 GMOD Meeting](January_2010_GMOD_Meeting "January 2010 GMOD Meeting").) A near instant solution to the problem (days/weeks, rather than months/years for an in house solution). Looked at lots of options. GBrowse looked like the clear winner - it's configurable and meets needs. |
| Q. | COSMIC was designed to be gene centric but what about sequencing whole cancer genomes and visualising mutations in genomic context? |
| A. | Gbrowse. Again! |

### <span id="Data" class="mw-headline">Data</span>

- Reference
  - Reference genome (GRCh37) + cytogenetic bands
  - Ensembl annotations (e! 58)
  - Cosmic Transcripts

<!-- -->

- Cosmic
  - Mutations (substitutions, insertions/deletions)
  - Rearrangements
  - Copy Number Profiles
    - analysis of SNP6 microarray data over 800 cell lines
    -  % samples which have copy number features (amplification,
      homozygous deletion, LOH, change)

### <span id="Configuration_and_Setup" class="mw-headline">Configuration and Setup</span>

- Hardware
  - 5 Virtual Machines \[Debian Linux, 2G RAM) \]
  - dev + master + renderfarm slaves (2) +
    [PostgreSQL](PostgreSQL "PostgreSQL"). The Master talks to the two
    slaves, both of which talk to the reference and mutations databases.

<!-- -->

- Software
  - apache 2.2.9
  - mod_fastcgi 2.4.6
  - GBrowse 2.13 (perl 5.10.0 + [BioPerl](BioPerl "BioPerl") 1.61 +
    <a href="http://search.cpan.org/perldoc?Bio::Graphics"
    class="external text" rel="nofollow">Bio::Graphics</a> 2.11\]
    *Note:' 'There was significant renderfarm development between 2.13
    and 2.14*

<!-- -->

- Databases
  - PostgreSQL
    - 2 databases: ‘Reference’ and ‘Cosmic’
  - scripts to query/format/populate these databases

<!-- -->

- Configuration
  - cosmic css/theme
  - perl callbacks: glyphs, colours, hyperlinks, popups/tooltips

### <span id="Display" class="mw-headline">Display</span>

COSMIC GBrowse shows:

- genes, COSMIC transcripts, non-coding RNA
- breakpoints with lightning (!) and detailed popups
- Copy number change, with color, and links to
  <a href="http://www.sanger.ac.uk/cgi-bin/genetics/CGP/conan/search.cgi"
  class="external text" rel="nofollow">CONAN</a>.
- LOH, with color
- Mutations density
- Mutation details (intronic, nonsense, missense, Silent, Non-coding,
  frameshift, in frame, complex, deletion, insertion), with colors and
  shapes, provide a key and detailed popups
- See <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/6/6c/CosmicGBrowseSept2010.pdf"
  class="internal" title="CosmicGBrowseSept2010.pdf">slides</a> for
  screenshots.

### <span id="Future_Development" class="mw-headline">Future Development</span>

At COSMIC

- Embed cosmic GBrowse in some cosmic web pages - replace old and slow
  drawing code and extend functionality.
- Current version is a summarised view of whole cosmic dataset. We need
  to be able to display subsets of data. How can we display all
  mutations for a specific sample or group of samples, or from a
  specific tissue or tumour type? oo many for a static list of data
  sources, but there is a neat trick ..
  - Define data source in the URL, eg sample COLO-829:
    <a href="http://www.sanger.ac.uk/fgb2/gbrowse/sample_COLO-829"
    class="external free"
    rel="nofollow">http://www.sanger.ac.uk/fgb2/gbrowse/sample_COLO-829</a>

<!-- -->

- GBrowse.conf ... (need at least 2.09)
  - [GBrowse 2.0 HOWTO, "Using Pipes in the GBrowse.conf Data Source
    Name"](GBrowse_2.0_HOWTO#Using_Pipes_in_the_GBrowse.conf_Data_Source_Name "GBrowse 2.0 HOWTO").



        [=~sample_.+]
        description = Cosmic Database v48 (sample filtered)
        path           = /gbrowse/bin/source_config.pl -sample $1 |

            # path points to a script which generates the config
            # sample name ‘COLO-829’ is passed to the script from regular expression
            # track configuration generated for data source  COLO-829  …

        [Mutations]
        remote feature = http://…/cosmic_export.cgi?sample=COLO-829

            # cgi script returns COLO-829 mutation data from COSMIC



GBrowse Developement

- remote feature - perl callbacks cannot be used until Safe::World is
  fixed
- `init_code` - perl callbacks defined with `init_code` not accessible
  from slaves
- BAM/SAM read sorting by similarity to reference
- GC plots can give \>100% values

CGP

CGP committed to using GBrowse as its internal browser for next gen
sequencing data, and an external browser for COSMIC data (genomic view
of mutations, breakpoints and copy number data). COSMIC GBrowse to be
released soon (22/9/2010?). CGP is also involved in GBrowse development.
A new developer has been recruited, but details are still being
discussed.

## <span id="GMOD_Projects_at_the_Center_for_Genomics_and_Bioinformatics" class="mw-headline">GMOD Projects at the Center for Genomics and Bioinformatics</span>



<a href="File:Sept2010Chris.jpg" class="image"
title="Chris Hemmerich"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/32/Sept2010Chris.jpg" width="100"
height="108" alt="Chris Hemmerich" /></a>



[Chris Hemmerich](User%3AChemmeri "User%3AChemmeri"),
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/d/d1/ISGAWebGBrowseSept2010.pdf"
class="internal" title="ISGAWebGBrowseSept2010.pdf">PDF</a>,
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/3f/ISGAWebGBrowseSept2010.pptx"
class="internal" title="ISGAWebGBrowseSept2010.pptx">PPT</a>

### <span id="A_Simple_Web_Interface_for_Configuring_GBrowse:_WebGBrowse" class="mw-headline">A Simple Web Interface for Configuring [GBrowse](GBrowse.1 "GBrowse"): [WebGBrowse](WebGBrowse.1 "WebGBrowse")</span>

(By Ram Podicheti, as channeled by Chris)

[WebGBrowse](WebGBrowse.1 "WebGBrowse") is a web interface for
configuring GBrowse installations. You can upload [GFF](GFF "GFF") files
and optionally upload an existing [GBrowse](GBrowse.1 "GBrowse") config
file to use as starting point. From there, you can add, edit, and remove
new tracks using web forms. WebGBrowse comes with extensive help
embedded in the forms and includes a tutorial. Users can preview their
changes at any point in GBrowse. WebGBrowse makes GBrowse more feasible
for small projects who can figure out configuration, but don't have the
resources to setup their own server.

WebGBrowse can be
<a href="http://webgbrowse.cgb.indiana.edu/" class="external text"
rel="nofollow">downloaded</a> and locally installed. There is a [mailing
list](WebGBrowse.1#Mailing_List "WebGBrowse") for support, feature
requests, and contributions. We want to help you help us add support for
more features. WebGBrowse has passed the [nomination
process](GMOD_Membership "GMOD Membership") and is now a pending GMOD
component. Waiting only migration of development environment to a public
repository.

WebGBrowse has support GBrowse 2 for quite a while. It does not support
callbacks yet (and this is hard due to security considerations),

### <span id="Web-based_Bioinformatics_Pipelines_for_Biologists:_ISGA" class="mw-headline">Web-based Bioinformatics Pipelines for Biologists: [ISGA](ISGA "ISGA")</span>

(By Chris, Aaron Buechlein, Ram, Jeong-Hyeon Choi, and Boshu Liu as
channeled by Chris)

[ISGA](ISGA "ISGA") is a workflow management system that can meet the
needs of a small sequencing center. It supports flexible pipeline
definition for new pipelines, and for incorporating new programs as
components. ISGA supports distributed computing environments, if you
have a potential need to grow beyond local computing resources. ISGA was
created at CGB to minimize CGB staff involvement in running pipelines.
ISGA frees up staff resources for building new pipelines.

ISGA is built on top of
<a href="Ergatis" class="mw-redirect" title="Ergatis">Ergatis</a>.
Ergatis is developed and support by the Institute for Genome Sciences,
U. Maryland. Ergatis enables building pipelines from existing programs,
supports distributed computing environments, and has robust monitoring
of pipeline execution. Ergatis comes with 10+ readily available
pipelines, and there are more available in the community. There are
currently 220 tool/component definitions that come with Ergatis, and
again, there are more in the community. Components and pipelines are
defined in [XML](Glossary#XML "Glossary"). XML/BSML is the common data
exchange format. XML/BSML is optional, but recommended for reusable
components. Includes conversion tools for FASTA, GFF,
<a href="Chado" class="mw-redirect" title="Chado">Chado</a>, etc... This
isolates format changes from other programs. Ergatis runs on Condor out
of the box.

Ergatis's interface assumes that a computationally savvy biologist will
be using it. In practice, this can lead to the informatics staff being
the practical interface between biologists and Ergatis. CGB had several
goals when developing ISGA:

- Wanted to support single-lab biologists that are self-sufficient but
  have limited bioinformatics resources and that embrace tools that
  don’t require extensive training
- Ability for biologists to run pre-configured pipelines quickly
- Option to customize specific tools in a pipeline
- An interface that encourages exploration:
  - Remove complexity and information biologists don’t need
  - Inline help
  - Immediately detect errors and allow biologists to correct them
  - Return output in useful formats
  - Simple tools for visualizing and searching large result sets

<table>
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<tbody>
<tr class="odd">
<td>

<a href="File:ErgatisBoxes.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/8/8a/ErgatisBoxes.png" class="thumbimage"
width="295" height="243" /></a>

Ergatis and the bioinformatician


</td>
<td>

<a href="File:ISGABoxes.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/b/b3/ISGABoxes.png" class="thumbimage"
width="409" height="255" /></a>

ISGA and the bioinformatician


</td>
</tr>
</tbody>
</table>

ISGA does this and several other things too: First, it simplifies
pipelines by hiding housekeeping components and by grouping components
into clusters representing processes. ISGA supports customization. Users
can disable components, replace components with pre-computed data, and
edit scientifically-active program parameters. It also provides help and
validation for all forms, and incorporates visualization and analysis
tools. In addition ISGA support the concepts of users and data privacy,
and users can upload and download data,

#### <span id="Why_develop_ISGA_as_a_separate_package.3F" class="mw-headline">Why develop ISGA as a separate package?</span>

[ISGA](ISGA "ISGA") only re-implements the web interface of
<a href="Ergatis" class="mw-redirect" title="Ergatis">Ergatis</a>.
Ergatis libraries, component definitions, and method of running and
monitoring pipelines is used by ISGA as-is. ISGA adds and removes
Ergatis features such as accessing component information and building
pipelines from components. ISGA biologist users need to be given limited
functionality for simplicity and security. Ergatis bioinformatician
users need full functionality and a complex interface to work
efficiently. *A hybrid ISGA/Ergatis interface wouldn’t serve anyone.*

#### <span id="Present_and_Future" class="mw-headline">Present and Future</span>

ISGA at Indiana has run over 100 pipelines, and has more than 60 users.
There are two external sites evaluating their own ISGA installation that
CBG knows of.

Recent developments in ISGA include

- Celera assembly pipeline
  - Ability to accept parameters with pipeline inputs
  - Ability to iterate components over a list of pipeline inputs
  - Conversion scripts for Hawkeye visualization
- Installation instructions :shame
- isga-users@lists.sourceforge.net
- Administration improvements
  - Online configuration
  - User classes and pipeline quotas

And there is more in the works:

- Pipelines
  - SHORE SNP Calling (ISGA)
  - Gene clustering over Microbial phylogenies (Ergatis)
  - Transcriptome annotation pipeline (Ergatis)
  - Methyl-seq (Ergatis)
- Features
  - Pipeline reproducibility and provenance
  - User groups and sharing
  - Modular pipeline and toolbox installation
    - ISGA pipelines as standalone Ergatis templates
  - ISGA pipeline over Amazon EC2 via CLoVR

CloVR

- Cloud Resources through CloVR
  - Execute Ergatis Pipelines over an SGE instance hosted on Amazon EC2
    machine images
  - CloVR manages creation and shutdown of cloud images as part of
    pipeline
  - Upload input as part of pipeline or access data hosted at Amazon
  - Results are retrieved to local machine
  - Ergatis assumes a shared filesystem, so some modification is
    required to manage file transfers
- Using CloVR with ISGA
  - ISGA/Ergatis pipelines can be ported to ISGA/CloVR
  - ISGA installation communicates with local Ergatis and CloVR
  - EC2 presents challenges for billing customers

## <span id="GMOD_RPC_API:_The_almost_RESTful_GMOD_API" class="mw-headline">[GMOD RPC API](GMOD_RPC_API "GMOD RPC API"): The almost RESTful GMOD API</span>



<a href="File:Sept2010Josh.jpg" class="image" title="Josh Goodman"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/4/48/Sept2010Josh.jpg" width="100" height="96"
alt="Josh Goodman" /></a>



[Josh Goodman](User%3AJogoodma "User%3AJogoodma"),
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/e/ee/GMOD_RPC_Sept_2010.pdf"
class="internal" title="GMOD RPC Sept 2010.pdf">PDF</a>

Josh started with this scenario:



Fetch me all genes annotated with GO:0003677 (DNA Binding) from *D.
melanogaster, C. elegans, T. castaneum,* and *B. mori*. Then fetch the
current ID, symbol and list of orthologs for each.



We currently do this with a mixture of file downloads,
[SQL](Glossary#SQL "Glossary") calls to different DB systems, a
patchwork of parsing scripts, and screen scraping. Instead, we should be
doing:

    $ curl http://flybase.org/gmodrpc/v1.1/ontology/gene/GO:0003677
    $ curl http://wormbase.org/gmodrpc/v1.1/ontology/gene/GO:0003677

This idea was motivated by a discussion at the [July 2008 GMOD
Meeting](July_2008_GMOD_Meeting "July 2008 GMOD Meeting") where a simple
request, like the one above, required screen scraping. This work uses
the REST protocol to gather information. REST is an alternative or
successor to CORBA, a heavyweight protocol for sharing information, and
SOAP, a more recent, but still too heavy for our purposes protocol for
doing the same.

The [GMOD RPC API](GMOD_RPC_API "GMOD RPC API") proposal supports a
number of information services:

- Organisms
- Full text search
- Location
- Gene ontology
- Orthology
  - Gene
  - Organism
- Fetch common gene page

In an ideal world each MOD would provide these services.

The idea is to provide top level classes. FlyBase will provide a
specific
<a href="Chado" class="mw-redirect" title="Chado">Chado</a>/Perl based
implementation. However, the proposal is trying to be agnostic in terms
of what data types are expected. Josh is working on the Perl
implementation. Others are working on PHP ([Jim
Hu](User%3AJimHu.1 "User%3AJimHu")) and Java implementations.

### <span id="Perl_Implementation" class="mw-headline">Perl Implementation</span>

- Strict [MVC](Glossary#MVC "Glossary") separation.
  - <a href="http://search.cpan.org/perldoc?Moose" class="external text"
    rel="nofollow">Moose</a> used for the *model*
    - Moose is much better than Perl 5 objects.
    - GMOD RPC API will provide base code and utility functions. You
      extend base class of each service to implement based on your
      environment.
  - <a href="http://search.cpan.org/perldoc?Template::Toolkit"
    class="external text" rel="nofollow">Template::Toolkit</a> for the
    *view*
  - <a href="http://perldancer.org/" class="external text"
    rel="nofollow">Perl’s Dancer</a> for the *controller*
    - Simple and clean with minimal dependencies. Perl implementation of
      Ruby's Sinatra.
    - Easy to install. Decided against Catalyst because of installation
      and dependencies. Want something simple to get this off the
      ground.
    - Can be run under CGI, PSGI (Plack), and FastCGI on a variety of
      web servers (Apache, Nginx and lighttpd)
- <a href="http://search.cpan.org/perldoc?Log::Log4perl"
  class="external text" rel="nofollow">Log::Log4perl</a> for logging
- Standard <a href="http://search.cpan.org/perldoc?Test::More"
  class="external text" rel="nofollow">Test::More</a> unit tests

#### <span id="Goals" class="mw-headline">Goals</span>

- Short term
  - Alpha release by end of October 2010
  - Beta release by end of December 2010
- Long term
  - <a href="DAS" class="mw-redirect" title="DAS">DAS</a> tie in
  - Validation for XML formats
  - Java, PHP and Python APIs
  - Evaluate additional API features

### <span id="How_to_participate" class="mw-headline">How to participate</span>

- Subscribe to [gmod‐devel](GMOD_Mailing_Lists "GMOD Mailing Lists")
- <a href="GMOD_REST_API" class="mw-redirect" title="GMOD REST API">GMOD
  REST API</a>
- <a href="SVN" class="mw-redirect" title="SVN">SVN</a> code repos
  - <a href="http://gmod.svn.sourceforge.net/viewvc/gmod/GMOD‐RPC‐perl/"
    class="external free"
    rel="nofollow">http://gmod.svn.sourceforge.net/viewvc/gmod/GMOD‐RPC‐perl/</a>
  - <a href="http://gmod.svn.sourceforge.net/viewvc/gmod/GMOD‐RPC‐java/"
    class="external free"
    rel="nofollow">http://gmod.svn.sourceforge.net/viewvc/gmod/GMOD‐RPC‐java/</a>
  - <a href="http://gmod.svn.sourceforge.net/viewvc/gmod/GMOD‐RPC‐php/"
    class="external free"
    rel="nofollow">http://gmod.svn.sourceforge.net/viewvc/gmod/GMOD‐RPC‐php/</a>

### <span id="Discussion" class="mw-headline">Discussion</span>

Plan is to keep old version APIs around. That is to keep the old URLs
accessible by having constant and stable URLs.

There is a mechanism to query what services are available - returned as
an XML list of services from above list (organism, ...).

Queries can use taxonomy id, or just genus and species. Can get sequence
by asking for a gene and SO terms

Can this REST interface be made a standard feature in a GBrowse
implementation? That's ideally what we should shoot for. This is where
this should go.

Is there a way to pull this in whole, instead of at a retail level? Not
currently, but FlyBase puts gene reports into XML. Then use XSLT to
generate the web pages, and then save XML in Lucene for ful text
searching.

## <span id="Overview_of_current_resources_and_update_on_DAS_Meeting_Cambridge_2010" class="mw-headline">Overview of current resources and update on <a href="DAS" class="mw-redirect" title="DAS">DAS</a> Meeting Cambridge 2010</span>



<a href="File:Sept2010Jonathan.jpg" class="image"
title="Jonathan Warren"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/6/6a/Sept2010Jonathan.jpg" width="100"
height="79" alt="Jonathan Warren" /></a>



Jonathan Warren,
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/e/e6/DAS_Sept2010.pdf" class="internal"
title="DAS Sept2010.pdf">PDF</a>,
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/e/e7/DAS_Sept2010.pptx" class="internal"
title="DAS Sept2010.pptx">PPT</a>

<a href="DAS" class="mw-redirect" title="DAS">DAS</a> stands for
*Distributed Annotation System.* It is based on HTTP and XML. From a
user perspective, you run a client program (e.g.,
[GBrowse](GBrowse.1 "GBrowse"), [Apollo](Apollo.1 "Apollo"), Ensembl,
...) pick a coordinate system, connect to a DAS registry to get a DAS
server list, and then request a region of interest from the reference
and many annotations from the DAS servers.

<a href="File:DASSept2010.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/8/87/DASSept2010.png" width="509" height="272"
alt="DASSept2010.png" /></a>

### <span id="Some_DAS_1.5.2F1.6_Commands" class="mw-headline">Some DAS 1.5/1.6 Commands</span>

- Sources
- Features - used by genomic clients
- Sequence
- Types
- Stylesheet - give tips to the client about how the data should be
  displayed
- Structure
- Alignment
- Interaction (has largely been superseded by PSICQUIC).

### <span id="Why_use_DAS_1.6_over_1.5.3F" class="mw-headline">Why use DAS <a
href="http://www.ebi.ac.uk/~aj/1.6_draft7/documents/spec.html#coordinates#coordinates"
class="external text" rel="nofollow">1.6</a> over 1.5?</span>

- Clarification of the way DAS is being used - should promote
  interoperability
- Represent features with more than two levels 1.6
  - Represent Genes → Transcript → Exons
  - [GFF3](GFF3 "GFF3") will be a supported format (Adapters for servers
    and databases).
  - MyDAS server will support this without the need for a database
- Reliably relate feature types to a more structured ontology
  - cvId attributes in the xml for SO: or ECO ids - use of these may
    become mandatory in a future specification.
- Already have servers that produce 1.6 data.
- Lots of clarification in the standard makes it easier for to write DAS
  client code
- Cleaner ontology support. Hope is people can more easily use the DAS
  registry.
- Sources documents advantages:
  - have coordinate systems which mean you are mapping annotations to
    the correct genomes/sequences.
  - Smoother running of Ensembl and other DAS clients.
  - You can automatically load many DAS sources to the DAS registry
    using your sources document and the registry should keep in sync
    with new additions/deletions/alterations.
  - MyDAS and Proserver support the use of sources and all other 1.6
    specification commands and responses.

Registry has a new search interface, backed by lucene. It is very quick.
JSDAS is a JavaScript library based on JQuery and registry web service.
It makes makes developing registry interfaces a whole lot easier. JSDAS
is also usable with any sources document not just the registry.

Some example queries  

- keywords parameter to sources cmds e.g.
  - <a href="http://www.dasregistry.org/das/sources?keywords=zebrafish"
    class="external free"
    rel="nofollow">http://www.dasregistry.org/das/sources?keywords=zebrafish</a>
- keywords parameter to coordinatesystem command
  - <a
    href="http://www.dasregistry.org/das/coordinatesystem?keywords=gorilla"
    class="external free"
    rel="nofollow">http://www.dasregistry.org/das/coordinatesystem?keywords=gorilla</a>
- added total, start, end attributes to coordinatesystem request
  response if rows specified
  - <a
    href="http://www.dasregistry.org/das/coordinatesystem?keywords=gene&amp;rows=1-10"
    class="external free"
    rel="nofollow">http://www.dasregistry.org/das/coordinatesystem?keywords=gene&amp;rows=1-10</a>

### <span id="Easy_DAS" class="mw-headline">Easy DAS</span>

Easy DAS is a DAS server hosted at the EBI. Easy DAS removes the need to
have your own servers and databases. It accepts file uploades in a
number of formats. You can use EBI as your sever.

### <span id="1.6E_Proposals" class="mw-headline"><a href="http://www.biodas.org/wiki/DAS1.6E" class="external text"
rel="nofollow">1.6E Proposals</a></span>

- Can specifiy what formats your server can give to clients.
- Can make up your own format
  - Ask that you write up a description of your format on DAS wiki.
- DAS Writeback (implemented, Create Read, Update Delete)
  - Apollo could use thse.
- Longer genomic alignments

### <span id="DAS_Clients" class="mw-headline">DAS Clients</span>

New viewers:

- <a href="http://www.biodalliance.org/" class="external text"
  rel="nofollow">Dalliance</a> Thomas Down
  (<a href="http://www.biodalliance.org/human/ncbi36/"
  class="external text" rel="nofollow">example</a>)
- <a href="http://www.broadinstitute.org/igv/" class="external text"
  rel="nofollow">IGV</a> Broad Institute
- <a href="http://code.google.com/p/karyodas/" class="external text"
  rel="nofollow">Karyodas</a> (Decipher, mykaryoview)
- [Apollo](Apollo.1 "Apollo") - new DAS DataAdapter release soon.
  (Written by Jonathon about 6 months ago. Needs to be finished and sent
  to [Ed Lee](User%3AElee "User%3AElee").)
- [JBrowse](JBrowse.1 "JBrowse")? - in current grant proposal.

Other DAS Clients:

- Ensembl uses DAS to pull in genomic, gene and protein annotations. It
  also provides data via DAS.
- [GBrowse](GBrowse.1 "GBrowse") is a generic genome browser, and is
  both a consumer and provider of DAS.
- [IGB](IGB "IGB") is a desktop application for viewing genomic data.
- SPICE is an application for projecting protein annotations onto 3D
  structures.
- Dasty2 is a web-based viewer for protein annotations
- Jalview is a multiple alignment editor.
- PeppeR is a graphical viewer for 3D electron microscopy data.
- DASMI is an integration portal for protein interaction data.
- DASher is a Java-based viewer for protein annotations.
- EpiC presents structure-function summaries for antibody design.
- STRAP is a STRucture-based sequence Alignment Program.

Jonathan doesn't think it's hard to write a DAS adaptor for existing
databases.

## <span id="InterMine:_new_Mines_and_new_features" class="mw-headline">[InterMine](InterMine "InterMine"): new Mines and new features</span>



<a href="File:Sept2010Richard.jpg" class="image"
title="Richard Smith"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/4/47/Sept2010Richard.jpg" width="100"
height="80" alt="Richard Smith" /></a>



[Richard Smith](User%3ARsmith "User%3ARsmith"),
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/1/1c/InterMine_Sept2010.pdf"
class="internal" title="InterMine Sept2010.pdf">PDF</a>

[InterMine](InterMine "InterMine") is a query-optimised data warehouse
system. It is a means to integrate all your existing databases into a
single coherent database. InterMine is written in Java and uses an
object-based data model. It is implemented on top of
[PostgreSQL](PostgreSQL "PostgreSQL"), is free and open source (LGPL).

InterMine comes with tools for loading data from popular data sources
such as UniProt, Ensembl,
<a href="Chado" class="mw-redirect" title="Chado">Chado</a>, PSI,
Inparaoid, and many more. InterMine also provides Java and Perl
[APIs](Glossary#API "Glossary") for loading custom data sources, FASTA,
XML, and [GFF3](GFF3 "GFF3"). If there are conflicts in data from one or
more sources, you can write rules to prioritize one data point over
another.

InterMine's web interface works for any data model, includes advanced
functionality for bench biologist. It is highly conﬁgurable from within
the web interface. The web interface includes:

- QueryBuilder - custom queries, advanced
- Template queries - pre-deﬁned queries
- Report pages - conﬁgurable, templates (GO enrichment, publication
  enrichment, ...)
- Lists - upload, use in queries, analysis widgets
- Export & API - exports in several formats and has a RESTful query API
- MyMine - users can create an account and save their lists and queries.

### <span id="Mines_for_MODs" class="mw-headline">Mines for MODs</span>

This project is creating *mines* at several large MODs. FlyMine was the
original application for InterMine. [RGD](Category%3ARGD "Category%3ARGD")'s
mine has been released. [SGD](Category%3ASGD "Category%3ASGD") and
<a href="http://zfin.org" class="external text"
rel="nofollow">ZFIN's</a> mines are in beta. There is a common interface
across all of the mines. Each organization supports their with mine with
a 0.5 FTE.

InterMine runs alongside the existing web site. It contains MOD data
plus data from other sources. The InterMine team is working on better
support for embedding of InterMine, and new features for MODs. SGD is
using InterMine for searching, replacing older technology. This project
is providing lots of input for InterMine development!

These mines can currently be integrated through exporting of InterMine
lists from one mine to another. Richard gave an example of:

1.  Identify a list of (141) upregulated genes in heart at FlyMine.
2.  Export the orthologues to RGD
3.  Identify the list of those genes that are related to cardiovascular
    disease (40).
4.  Export the orthologues back to FlyMine
5.  Look for relevant publications, export sequences, ...

They are working on enabling this type of interoperability by fetching
results dynamically between mines. This would also allow people to
create mashups.

### <span id="metabolicMine" class="mw-headline">metabolicMine</span>



<a href="File:MetabolicMineLogoWide.png" class="image"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/b/b6/MetabolicMineLogoWide.png/200px-MetabolicMineLogoWide.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/b/b6/MetabolicMineLogoWide.png/300px-MetabolicMineLogoWide.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/b/b6/MetabolicMineLogoWide.png/400px-MetabolicMineLogoWide.png 2x"
width="200" height="43" alt="MetabolicMineLogoWide.png" /></a>



This brand new project will contain data on metabolic diseases such as
diabetes and obesity. InterMine fits quite well with this kind of
research. Currently researcher look at 10+ web sites for their gene of
interest. The idea is to put all that data in one place. This is being
done for human, mouse and rat, and from lots of data sources (SNPs,
HapMap, expression, ENCODE, ...) many of which parsers already exist
for. Users will be able to prioritise candidate genes, compare lists,
ﬁnd common attributes, and upload data.

### <span id="InterMine_0.94_Release" class="mw-headline">InterMine 0.94 Release</span>

Search has been greatly expanded. The existing Templates and
QueryBuilder interfaces are structured searches. 0.94 adds full text
searching of the whole database, using Lucene. For FlyMine it takes 45
minutes to create a 2.5 GB index ﬁle, using parallel fetching and
precomputes. Each object is a document with attributes as ﬁelds and
related data (e.g. GO, pathways) included as well.

The 0.94 release also adds support for *faceted* search. *Facets* are
different aspects of the current result that can be used to group and
filter them. Some common facets might be organism or feature (Sequence
Ontology) type. For organism you could include or exclude subsets of
data based on what their organism is. You can have a facet on any
property, and you may filter with multiple facets. Displaying all this
information is *hard*. InterMine's faceted search uses BOBO from
LinkedIn, and integrates with InterMine's lists and templates.

InterMine 0.94 also includes

- cleaner template management and implementation
- significant performance improvements
- a GUI to simplify installation and data source management
- [Galaxy](Galaxy.1 "Galaxy") integration - any query you run, you can
  export to Galaxy.
- Improved Perl web service API
- Automatic SO to model generation
- Upload list of genome regions
- CytoscapeWeb plugin

## <span id="Literature_Curation_in_GMOD" class="mw-headline">Literature Curation in GMOD</span>

[Daniel Renfro](User%3ADanielRenfro "User%3ADanielRenfro"),
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/6/62/Literature_Curation_Sept2010.pdf"
class="internal" title="Literature Curation Sept2010.pdf">PDF</a>,
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/4/4d/Literature_Curation_Sept2010.ppt"
class="internal" title="Literature Curation Sept2010.ppt">PPT</a>

The goal of literature curation is to get information from papers into a
format where they can be searched, grouped, reasoned with, and
displayed. Curation is often done by PIs, students, the occasional
bioinformaticist, and paid curators. This is comparatively time
consuming, and there are lots of papers, but few people. It is also
relatively costly and depends heavily on the expertise of the curators.

GMOD provides a number of tools to help with curation:

- [PubSearch](PubSearch "PubSearch")
- [Textpresso](Textpresso "Textpresso")
- [Chado Publication
  Module](Chado_Publication_Module "Chado Publication Module")
- [TableEdit](TableEdit.1 "TableEdit")
- [Tripal](Tripal.1 "Tripal")
- [Pathway Tools](Pathway_Tools.1 "Pathway Tools")
- [Apollo](Apollo.1 "Apollo") (or [Artemis](Artemis "Artemis"))
- [MAKER](MAKER.1 "MAKER")

Some of these support automated annotation while others support manual
annotation.

Daniel closed the talk with some details on distributed / community
annotation. EcoliWiki curates PubMed entries and then notifies authors
for further refinement. CACAO, a related project, teaches undergraduates
how to do annotation, and then puts them in teams, have them compete,
and then rewards them with food and drink (and grades). Uniprot has an
option to submit suggestions and comments. The RFAM database is a wiki.
PubMed has authors tag things. Finally the BeeSpace project did some
work with fly and their difficult gene names and did pretty good.

See the [summary of the Community Annotation Satellite
Meeting](Community_Annotation_-_September_2010_Satellite "Community Annotation - September 2010 Satellite")
for more on this topic.

## <span id="Towards_a_GO_Annotation_Tool:_Curation_Accelerator_Software" class="mw-headline">Towards a GO Annotation Tool: Curation Accelerator Software</span>



<a href="File:Sept2010Helen.jpg" class="image" title="Helen Field"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/5/54/Sept2010Helen.jpg" width="100" height="96"
alt="Helen Field" /></a>



Helen Field, <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/7/70/GO_Curation_Sept2010.pdf"
class="internal" title="GO Curation Sept2010.pdf">PDF</a>,
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/5/58/GO_Curation_Sept2010.key"
class="internal" title="GO Curation Sept2010.key">Keynote</a>

### <span id="The_Challenge" class="mw-headline">The Challenge</span>

[FlyBase](Category%3AFlyBase "Category%3AFlyBase") (and other) literature
curators read papers, extract information per paper, recording new
constructs, phenotypes, and allele construction. FlyBase also does gene
ontology (GO) annotation and wants to make this easy for curators or the
community to do. The goal is to avoid having people do things that
computers could do in order to save curation time and make it easy for a
beginner or community annotator.

Each gene should have up to 3 GO terms associated with it, one for each
of GO's 3 areas: molecular function, biological process, and cellular
component. Each GO term has a standard definition designed to have the
same biological meaning across the entire living world, i.e. for all
disciplines, in any organism, in any database. GO ontology terms are
designed to standardise the biology.

Creating GO annotation is a common problem in the GMOD community. There
is a file standard called GO Annotation Format (GAF) for GO annotations.
This format is tabular and is currently being revised to add two new
columns (for 17 columns total). We wanted a tool that can create GAF
entries in standard GO format or in FlyBase format.

Existing GO curation tools -- usually have a Perl or Java front end and
directly input annotation to a database. FlyBase curators edit **text**
files. The curation process at FlyBase is independent of database
access. This means the curation and database interfaces can be developed
independently, with each doing their own job well.

### <span id="The_Tool" class="mw-headline">The Tool</span>

FlyBase's strategy is to create a curator-designed GUI, that lends
itself to both speed and accuracy. Using a modular design, FlyBase have
designed a GO "clever editor" of "GO lines" (text lines) that uses an
XML conﬁguration file to say how many textual components per line item
and what is the allowed content of each sub-line component. An API will
permit customisation of inputs. Programming is in Java. A proof of
concept desktop GO tool has been created. The basic function of the GO
tool is to create, edit and house existing "GO lines" - i.e. lines in
GAF files. The tool shows a GAF line as it is being built. The focus is
on reducing mouse-work, tool switches, ‘manual’ editing, and textual
errors. The latter are avoided by using autocomplete.

There is one window, with buttons for many tools: the next adaptation
will be for phenotype/anatomy Controlled Vocabulary annotation. This
tool may ultimately be translated -- to a web service. The tool
currently runs on Mac OS X (testing) 10.5 and 10.6 preAlpha_v002. It can
use as little as 800MB of RAM (testing for less - but depends on inputs
for autocomplete). It accepts flat file input but can also talk to
[PostgreSQL](PostgreSQL "PostgreSQL"). Plans for the future include
<a href="Chado" class="mw-redirect" title="Chado">Chado</a> queries, and
XML conﬁguration for the clever editor, APIs in collaboration, and a JSP
or J2EE/SOAP web interface.

See the
[talk](File:GO_Curation_Sept2010.pdf "File:GO Curation Sept2010.pdf")
for many screenshots of the prototype.

## <span id="BioPivot" class="mw-headline">BioPivot</span>



<a href="File:Sept2010Steve.jpg" class="image"
title="Stephen Taylor"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/d/df/Sept2010Steve.jpg" width="100" height="97"
alt="Stephen Taylor" /></a>



Applying Microsoft Live Labs Pivot to Problems in Bioinformatics

Steve Taylor, <a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/a/ae/BioPivot_Sept2010.pdf"
class="internal" title="BioPivot Sept2010.pdf">PDF</a>,
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/a/a2/BioPivot_Sept2010.ppt"
class="internal" title="BioPivot Sept2010.ppt">PPT</a>

Steve is interested in visualization of large numbers of genome regions,
and in querying and filtering properties of genome regions. This talk
covers his work on the BioPivot tool, based on Microsoft's Pivot tool.
He'd also like to have an open discussion of other applications of this
technology.

Steve is the head of the
<a href="http://www.cbrg.ox.ac.uk/" class="external text"
rel="nofollow">Computational Biology Research Group (CBRG)</a> at the
University of Oxford. CBRG provides core services for research at
Oxford. They use lots of next generation sequencing for mainly human and
mouse projects. They visualize all this in
[GBrowse](GBrowse.1 "GBrowse"). CBRG has over 50 different GBrowse
databases showing time series, arrays, ChIP-Seq, and RNA-Seq, histone
modification data, cis and trans Interaction data, PCR amplified
regions, and exome sequencing and SNP detection.

There are challenges with this much data. For example slight problems
experimental conditions or antibodies can cause peak finding algorithms
to find a lot of false positives. These algorithms have lots of
parameters and you must pick a cutoff, and you have to eyeball lots of
peaks to do this. Which of my peaks overlap with genes, exons,
promoters, CpG islands, etc.

The traditional approach is to make a spreadsheet of data with links to
[GBrowse](GBrowse.1 "GBrowse")/UCSC regions of interest, click/filter
various parameters, and add data to spreadsheet after each new analysis.
This is **slow, boring**, and **error-prone.**

### <span id="Deep_Zoom_Technology" class="mw-headline">Deep Zoom Technology</span>

Recent technologies that use *deep zooming* can help us here. Steve
included pointers to

- <a
  href="http://www.ted.com/talks/blaise_aguera_y_arcas_demos_photosynth.html"
  class="external text" rel="nofollow">Blaise Aguera y Arcas's TED 2007
  talk</a> on Photosynth. This is "Google Maps on steroids."
- <a href="http://www.seadragon.com/showcase/" class="external text"
  rel="nofollow">Seadragon/Photosynth showcase</a>
- <a href="http://www.getpivot.com/" class="external text"
  rel="nofollow">Microsoft Live Labs’ Pivot</a>, which what BioPivot
  uses.

Bioinformatics has very few compelling interfaces. Wouldn't it be cool
to use this in bioinformatics. Take thousands of regions of interest of
genomes and view and filter seamlessly on metadata.

Steve then did a BioPivot demo showing over 6000 peaks found in a
ChIP-Seq experiment. (Editor's note: There aren't any screen shots of
the demo in the slides so this is hard to imagine. My notes during the
demo simply said "Holy cow.") *Facets* were shown on the side, and you
could filter the peaks based on their facets, or metadata.

Here are a couple of examples that can be viewed and manipulated with
the <a href="http://silverlight.net/" class="external text"
rel="nofollow">Silverlight</a> plug-in that can be used on Windows PCs
and Mac OS X (though not with Chrome just yet). Eventually this will
work on Linux in
<a href="http://www.mono-project.com/Moonlight" class="external text"
rel="nofollow">Moonlight</a>.

- <a href="http://www.cbrg.ox.ac.uk/data/biopivot/example/small/"
  class="external text" rel="nofollow">Small example</a>
- <a href="http://www.cbrg.ox.ac.uk/data/biopivot/example/atrx/"
  class="external text" rel="nofollow">Large 6000 peak example</a>

These facets came from column 9 of the [GFF3](GFF3 "GFF3") files for the
genomes. BioPivot includes tools to create Pivot metadata from GFF3, and
to add annotation to a GFF3 file for things like nearest gene, exons,
introns, intergenic, intragenic, and TSS/TES up and down stream regions.

BioPivot uses Pivot, but other *Zoomable User Interfaces (ZUIs)* are
available. <a href="http://openzoom.org" class="external text"
rel="nofollow">OpenZoom</a> is one such package that includes an SDK for
Flash, Flex, and AIR, as well as APIs.

There are a number of ways they would like to extend BioPivot:

- RNA-Seq parsers e.g. cufflinks, DESeq
- Get feedback from the community
- What else can we do with this tech?

See the
<a href="http://www.cbrg.ox.ac.uk/data/biopivot" class="external text"
rel="nofollow">BioPivot page at CBRG</a> for more.

## <span id="CRAWL_.28Chado_RESTful_Access_Web-service_Layer.29" class="mw-headline">CRAWL (<a href="Chado" class="mw-redirect" title="Chado">Chado</a> RESTful Access Web-service Layer)</span>



<a href="File:Sept2010Giles.jpg" class="image"
title="Giles Verlarde"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/7/70/Sept2010Giles.jpg" width="100"
height="120" alt="Giles Verlarde" /></a>



A programmatic interface for querying pathogen genomics data

[Giles Verlarde](User%3ABuggy "User%3ABuggy"),
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/3d/CRAWL_Sept2010.pdf" class="internal"
title="CRAWL Sept2010.pdf">PDF</a>

  
*Editor's note: None of these summaries do the talks justice. However,
this summary is especially short of the mark. To really appreciate the
talk, download the
<a href="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/3/3d/CRAWL_Sept2010.pdf" class="internal"
title="CRAWL Sept2010.pdf">slides</a>.*

**Caveat:** *Vaporware Alert!* This talk is about work that is currently
in the rapid prototypes, proof of concepts, and requirements gathering
stage. Terms and conditions apply. This is by no means a standard (yet),
but it might be useful to others.

Giles works at GeneDB, a pathogen database with 45 organisms as of a
couple months ago, and getting new ones all the time. GeneDB has a focus
on web services and includes a lot of
<a href="Chado" class="mw-redirect" title="Chado">Chado</a> related
code. It also include cross-organism computed data such as orthologues
and domains.



<a href="Chado" class="mw-redirect" title="Chado">Chado</a> is a
relational database [schema](Glossary#Schema "Glossary") that underlies
many GMOD installations. It is **capable of representing many of the
general classes** of data frequently encountered in modern biology such
as *sequence, sequence comparisons, phenotypes, genotypes, ontologies,
publications, and phylogeny.* It has been designed to handle **complex
representations of biological knowledge** and should be considered one
of the most sophisticated relational schemas currently available in
molecular biology. The **price of this capability** is that the **new
user must spend some time becoming familiar** with its fundamentals.

- A database for very deep curation
- An integrated database
- A database that is generic enough to use for any organism



GeneDB's web front end uses Hibernate mappings to Chado, DAO caches, and
Lucene search. It has weekly data updates (that take over a day to run).
The code is quite meaty.

GeneDB collaborates with EupathDB an umbrella functional genomics
integrative resource. It's annotation team is at Sanger, SBRI, and the
University of Georgia. GeneDB also integrates and shares data from
TriTrypDB and PlasmoDB.

### <span id="Challenges" class="mw-headline">Challenges</span>

There are several challenges here:

- EupathDB
  - Need to know what has changed
  - Need to be able to get the data
- Remote annotation
  - DB-Artemis via VPN
  - Consistency
  - Current setup works for power users, but need to build Rich Internet
    Applications to broadly enable this.
- Sanger
  - Need to exploit our own data as well!
  - Chado-complexity - [SQL](Glossary#SQL "Glossary") is hard.

The solutions to these challenges must all support rapid prototyping
where it is quick to implement new queries, and it runs directly on top
of the database (pure SQL), and there is no time to rebuild caches.
Solutions also need to be lightweight and not tax the existing website.

### <span id="Solutions" class="mw-headline">Solutions</span>

*CRAWL* uses Python, Jython, CherryPy, and Ropy to address these
challenges. CherryPy is a multi-threaded web app server in which it is
simple to reuse controller classes in different contexts. Code is built
as a library first, implementing the *model* and *controller* parts of
the MVC architecture. There is a conscious choice to ignore the *view*
part of MVC early on. This enforced a decoupling of the data layer from
the view. It also lends itself to building a command line wrapper
application, and to supporting unit testing.

*The essential purpose of CRAWL is mask the complexity of the SQL as
much as possible, and allow you to get on with data analysis
development.*

EupathDB queries GeneDB daily using a "what's new" web service
implemented with CRAWL. This gathers recent annotation changes which
result in links back to GeneDB in EupathDB web sites.

The command line wrapper is used locally at Sanger to extract all sorts
of information. This capability could be used by non-local
bioinformaticians as well.

### <span id="Collaborative_Interfaces_and_Rich_Internet_Applications" class="mw-headline">Collaborative Interfaces and Rich Internet Applications</span>

Rich Internet Applications (RIAs) use [AJAX](Glossary#AJAX "Glossary")
techniques such as data refreshing without page reloads, and
autocompletes of text boxes. Google Maps and Flickr are well known
examples of RIAs. These fundamentally depend on web services.

Giles then showed several screenshots from what some folks call
*Web-[Artemis](Artemis "Artemis")*, a genome annotation editor
implemented in a web browser. This has several potential uses. Genomics
visualisation and community annotation are two of them. Since it's web
based, annotators won’t have to use a desktop application via a VPN.
This is the kind of application we are moving towards. Whole point is
that people can build applications without having to use Giles. It is
possible (and Giles showed an example, SNP-Mashup) to use things like
Web-Artemis in a mashup, embedding it as widget with other independently
written widgets, integrating software that integrates data.

### <span id="The_way_of_CRAWL" class="mw-headline">The way of CRAWL</span>

- Built as a library first
  - Deployed as
    - Standalone Web services app
    - Command line app
- Used for
  - Collaborating with EupathDB
  - Query multi-organism data sets in house without going through WS
  - Building RIAs and (s)mashups

### <span id="GMOD_Rest" class="mw-headline">GMOD Rest</span>

CRAWL currently use the Ropy Python REST framework. It speaks Chado. Is
it time to implement the GMOD REST interface? GeneDB’s Chado may have
little differences. Any work must be tested on other databases.

## <span id="Lessons_from_the_Apache_Community" class="mw-headline">Lessons from the <a
href="http://www.slideshare.net/bosc2010/gardler-bosc2010-communitydevelopmentattheasf"
class="external text" rel="nofollow">Apache Community</a></span>

There was a <a
href="http://www.slideshare.net/bosc2010/gardler-bosc2010-communitydevelopmentattheasf"
class="external text" rel="nofollow">thought-provoking talk by Ross
Gardler at BOSC this year on "Community Development at the Apache
Software Foundation"</a>. There might be some lessons we can apply to
the GMOD community.

- All technical discussions are made in public.
- You have to have a community before a project.
- If you contribute to a project, you get a vote.
- You can approve new projects, and volunteer to work on it.
- You can veto, but you have to say why.
- Make components generic enough so they could be used outside their
  domain.
- Jakarta managed all the Java projects under Apache. Switched to flat
  model. Jakarta no longer exists. Every project is now at the same
  organizational level.

## <span id="Lightning_Talks" class="mw-headline">Lightning Talks</span>

### <span id="GBrowse_2_and_JBrowse_Features" class="mw-headline">[GBrowse](GBrowse.1 "GBrowse") 2 and [JBrowse](JBrowse.1 "JBrowse") Features</span>

[Scott Cain](User%3AScott "User%3AScott")

Scott talked about new features in [GBrowse](GBrowse.1 "GBrowse") 2 and
[JBrowse](JBrowse.1 "JBrowse"). Some topics covered:

- Really fine grained control of downloads.
- Early uploading of files and sharing.
- Can have user accounts.
- Chado GBrowse now supports full text searching.
  - feature table gets a new column, other tables get a new column, and
    there's a new materialized view, with a new column.

### <span id="CpG_Island_and_STR_Annotator_Plugins" class="mw-headline">CpG Island and STR Annotator Plugins</span>



<a href="File:Sept2010Joan.jpg" class="image" title="Joan Pontius"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/7/76/Sept2010Joan.jpg" width="100" height="113"
alt="Joan Pontius" /></a>



Joan Pontius

Two [GBrowse](GBrowse.1 "GBrowse") plugins were presented that were
designed by Joey Bullard, an undergraduate student at RIT, and a summer
student at the Laboratory of Genomic Diversity in Frederick Maryland
under the direction of Joan Pontius. CpG islands in mammalian genomes
are often an indication of a well conserved region, such as those of
genes and promoter regions. ShortTandemRepeats are useful in experiments
for mapping phenotypes to a genetic locus.

[`CpGIslandAnnotator.pm`](File:CpGIslandAnnotator.pm "File:CpGIslandAnnotator.pm")  
Finds total number of CG dimers in a window (default size of 400 bp) and
also calculates the expected number of CG dimers based on the basepair
composition of the window. A "segments" glyph is used to display the CpG
islands, with the height of the glyph representing the number of CG
dimers, and the color representing the ratio of observed counts to
expected.

<!-- -->

[`STRAnnotator.pm`](File:STRAnnotator.pm "File:STRAnnotator.pm")  
A plugin that uses the segments glyph to show Short Tandem Repeats,
2mer-5mers occurring in tandem a minimum of 5 times. The user can
configure the number of tandem repeats.

Also presented was a CGI script that generates and displays in real
time, the scatterplot of a GWA study SNP. The CGI script uses an offset
file, that has, for each SNP, the offset in the Affymetrix or Illumina
file of the genotype calls and fluorescence values of the SNPs. The CGI
takes as an argument the SNP id, it then looks up the offset, retrieves
the data and generates the scatterplot as a png file.

The CGI can be used in [GBrowse](GBrowse.1 "GBrowse") to display the
scatterplots for individual SNPs as a mouseover. By embedding the call
to the CGI inside image tags (`<img>`), and having the `<img>` tags
inside a balloon click, the scatterplot can be displayed to the GBrowse
display instantly (less than a second) and without generating a
temporary flat file.

### <span id="GMOD_Sustainability_and_Organization" class="mw-headline">GMOD Sustainability and Organization</span>

[Dave Clements](User%3AClements "User%3AClements")

The GMOD project has no central organization *per se*, and has no legal
status whatsoever. The GMOD *core*, such as it is, is not particularly
well funded. It currently consists of [Scott
Cain](User%3AScott "User%3AScott"), the GMOD Project Coordinator, and [Dave
Clements](User%3AClements "User%3AClements"), the [GMOD Help
Desk](GMOD_Help_Desk "GMOD Help Desk"). Scott's funding comes entirely
from WormBase, while Dave is only 60% funded (via the JBrowse/GBrowse
grant) to work on the Help Desk. This is not particularly good ground to
stand on for the long term.

NSF is moving towards funding software infrastructure projects such as
GMOD:

- <a href="http://www.nsf.gov/pubs/2010/nsf10551/nsf10551.htm"
  class="external text" rel="nofollow">Software Infrastructure for
  Sustained Innovation (SI<sup>2</sup>)</a>, specifically the Scientific
  Software Integration (SSI) and Scientific Software Innovation
  Institutes (S2I2) awards.

What can we do to take advantage of these opportunities, and what other
opportunities are there?

Anup Mahurkar suggested also pursuing funding from NIGMS. Joan Pontius
wondered if there might be support for a *Google Genomes* project?

Steve Taylor pointed out that In Britain, the
<a href="http://www.bbsrc.ac.uk/" class="external text"
rel="nofollow">BBSRC</a> has a <a
href="http://www.bbsrc.ac.uk/funding/opportunities/2010/2010-bioinformatics-biological-resources-fund.aspx"
class="external text" rel="nofollow">program for funding</a> this sort
of thing. Being the BBSRC, it must have some sort of British focus.
(Subsequent followup also identified
<a href="http://www.euussciencetechnology.eu/" class="external text"
rel="nofollow">BILAT-USA and Link2US</a> as a possible sources of
general European funding.)

[Josh Goodman](User%3AJogoodma "User%3AJogoodma") suggested having a *hosted
GMOD* service as a possible way to support this. This comes up every
year and would be a possible way to initially get funding and then
generate revenue from support fees.

We also discussed if GMOD events should strive to become larger, and if
the GMOD organization should aim to become more "official". [Daniel
Renfro](User%3ADanielRenfro "User%3ADanielRenfro") stressed that going for
larger events should not be done at the expense of the relatively small
(~40-60 people), informal and networking-intensive GMOD Meetings that we
currently have. [Scott Cain](User%3AScott "User%3AScott") posed the question
of becoming a sister organization to the International Society for
Biocuration. That organization is the natural complement to GMOD, and
they have (larger) annual meetings and are a legal entity.

[Chris Hemmerich](User%3AChemmeri "User%3AChemmeri") raised the possibility
of having more frequent hackathons. This question was spawned by the
upcoming [GMOD Evo Hackathon](GMOD_Evo_Hackathon "GMOD Evo Hackathon")
(November 2010). That hackathon is unique event, given its extensive
organizational and financial support from NESCent. The 2007 Hackathon
and future hackathons are unlikely to have that degree of support. That
said, GMOD is quite willing to organize future hackathons on particular
topics, given that they need to be done with probably minimal
organizational support and no financial support.

Jerven Bolleman suggested teaming up with user interface specialists and
setting up testing suites and doing integration testing. Giles Velarde
suggested education and training as a *core* deliverable. These are all
areas that span GMOD and would benefit from more central support, and
the need for for training funds is increasing, given that funds for this
were greatly reduced in the upcoming Help Desk grant.

Finally, Jerven suggested extending GMOD past its core genomic strength
as another growth area.

# <span id="Participants" class="mw-headline">Participants</span>

| Participant | Affilliation(s) | URL |
|----|----|----|
| [Scott Cain](User%3AScott "User%3AScott") | OICR | <a href="http://gmod.org/" class="external free"
rel="nofollow">http://gmod.org/</a> |
| [Dave Clements](User%3AClements "User%3AClements") | NESCent, GMOD | <a href="http://nescent.org" class="external free"
rel="nofollow">http://nescent.org</a> <a href="http://gmod.org" class="external free"
rel="nofollow">http://gmod.org</a> |
| [Josh Goodman](User%3AJogoodma "User%3AJogoodma") | FlyBase - Indiana University | <a href="http://flybase.org" class="external free"
rel="nofollow">http://flybase.org</a> |
| [Richard Smith](User%3ARsmith "User%3ARsmith") | Cambridge University | <a href="http://www.intermine.org" class="external free"
rel="nofollow">http://www.intermine.org</a> |
| Anup Mahurkar | Institute for Genome Sciences University of Maryland School of Medicine |  |
| joan pontius | SAIC-NCI-FREDERICK Laboratory of Genomic Diversity | <a href="http://lgd.abcc.ncifcrf.gov/cgi-bin/gbrowse/cat/"
class="external free"
rel="nofollow">http://lgd.abcc.ncifcrf.gov/cgi-bin/gbrowse/cat/</a> |
| Christelle Robert | The Roslin Institute The University of Edinburgh |  |
| Matthew Eldridge | Cancer Research UK - Cambridge Research Institute |  |
| Fengyuan Hu | Department of Genetics, University of Cambridge |  |
| [Daniel Renfro](User%3ADanielRenfro "User%3ADanielRenfro") | EcoliWiki, SubtilisWiki, Hu lab - Texas A&M University | <a href="http://ecoliwiki.net" class="external text"
rel="nofollow">EcoliWiki</a>, <a href="http://subtiliswiki.net" class="external text"
rel="nofollow">SubtilisWiki</a>, <a href="http://gowiki.tamu.edu" class="external text"
rel="nofollow">GONUTS</a> |
| Ellen Adlem | Cambridge University Cambridge Institue of Medical Research | <a href="http://www.t1dbase.org" class="external free"
rel="nofollow">http://www.t1dbase.org</a> |
| Kerstin Koch | KWS Saat AG Bioinformatics Grimsehlstr. |  |
| Oliver Burren | Cambridge University | <a href="http://www.t1dbase.org" class="external free"
rel="nofollow">http://www.t1dbase.org</a> |
| Chris Jiggins | University of Cambridge | <a href="http://heliconius.zoo.cam.ac.uk/" class="external free"
rel="nofollow">http://heliconius.zoo.cam.ac.uk/</a> |
| Jason Swedlow | Wellcome Trust Centre for Gene Regulation and Expression, University of Dundee, The Open Microscopy Environment (OME) | <a href="http://gre.lifesci.dundee.ac.uk/staff/jason_swedlow.html"
class="external free"
rel="nofollow">http://gre.lifesci.dundee.ac.uk/staff/jason_swedlow.html</a>, <a href="http://www.openmicroscopy.org/" class="external free"
rel="nofollow">http://www.openmicroscopy.org/</a> |
| Dave Beare | Cancer Genome Project, Wellcome Trust Sanger Institute | <a href="http://www.sanger.ac.uk/research/projects/cancergenome.html"
class="external free"
rel="nofollow">http://www.sanger.ac.uk/research/projects/cancergenome.html</a> |
| seth redmond | Imperial College / Vectorbase |  |
| [Chris Hemmerich](User%3AChemmeri "User%3AChemmeri") |  | <a href="http://cgb.indiana.edu" class="external free"
rel="nofollow">http://cgb.indiana.edu</a> |
| Emmanuel Quevillon | Institut Pasteur | <a
href="http://www.pasteur.fr/ip/easysite/go/03b-00000m-0q8/recherche/logiciels-et-banques-de-donnees"
class="external free"
rel="nofollow">http://www.pasteur.fr/ip/easysite/go/03b-00000m-0q8/recherche/logiciels-et-banques-de-donnees</a> |
| Bob MacCallum | VectorBase Imperial College London | <a href="http://www.vectorbase.org" class="external free"
rel="nofollow">http://www.vectorbase.org</a> |
| Ewan Mollison | Tun Abdul Razak Research Centre, Hertford | <a href="http://www.tarrc.co.uk" class="external free"
rel="nofollow">http://www.tarrc.co.uk</a> |
| Jen Harrow | Wellcome Trust Sanger Institute |  |
| Gos Micklem | University of Cambridge | <a href="http://www.sysbiol.cam.ac.uk/index.php?page=dr-gos-micklem"
class="external free"
rel="nofollow">http://www.sysbiol.cam.ac.uk/index.php?page=dr-gos-micklem</a> |
| Malcolm Hinsley | Wellcome Trust Sanger Institute |  |
| Gemma Barson | Wellcome Trust Sanger Institute | <a href="http://www.sanger.ac.uk/" class="external free"
rel="nofollow">http://www.sanger.ac.uk/</a> |
| Brett Whitty | Michigan State University | <a href="http://buell-lab.plantbiology.msu.edu" class="external free"
rel="nofollow">http://buell-lab.plantbiology.msu.edu</a>, <a href="http://solanaceae.plantbiology.msu.edu" class="external free"
rel="nofollow">http://solanaceae.plantbiology.msu.edu</a>, <a href="http://potatogenome.net" class="external free"
rel="nofollow">http://potatogenome.net</a> |
| Morris Swertz | Genomics Coordination Center, University Medical Center Groningen EMBL - European Bioinformatics Institute | <a href="http://www.molgenis.org" class="external free"
rel="nofollow">http://www.molgenis.org</a> |
| Jerven Bolleman | UniProt Swiss-Prot |  |
| Alex Kalderimis | InterMine, Cambridge University | <a href="http://www.intermine.org" class="external free"
rel="nofollow">http://www.intermine.org</a>, <a href="http://www.flymine.org" class="external free"
rel="nofollow">http://www.flymine.org</a> |
| Oksana Riba Grognuz | Swiss Institute of Bioinformatics (SIB) Department of Ecology and Evolution, University of Lausanne |  |
| Dr Helen Imogen Field | FlyBase Dept Genetics University of Cambridge | <a href="http://www.gen.cam.ac.uk/research/flybase.html"
class="external free"
rel="nofollow">http://www.gen.cam.ac.uk/research/flybase.html</a> |
| Kim Rutherford | Cambridge Systems Biology Centre | <a href="http://www.pombase.org/" class="external free"
rel="nofollow">http://www.pombase.org/</a> |
| Robert Wilson | National Institute for Medical Research, London |  |
| Gerd Anders | Public research institute: Max-Delbrueck-Centrum Berlin (MDC), Researcher and database developer | <a
href="http://www.mdc-berlin.de/en/research/core_facilities/cf_massspectromety_bimsb/teammember/index.html"
class="external free"
rel="nofollow">http://www.mdc-berlin.de/en/research/core_facilities/cf_massspectromety_bimsb/teammember/index.html</a> <a
href="http://www.mdc-berlin.de/en/research/core_facilities/cf_bioinformatic/teammember/index.html"
class="external free"
rel="nofollow">http://www.mdc-berlin.de/en/research/core_facilities/cf_bioinformatic/teammember/index.html</a> |
| Joeri van der Velde | University of Groningen, GBIC UMGC, dept. of Genetics Genomics Coordination Center |  |
| Jonathan Warren | The Sanger Institute | <a href="http://www.dasregistry.org" class="external free"
rel="nofollow">http://www.dasregistry.org</a> |
| Stephen Taylor | CBRG, Oxford University | <a href="http://www.cbrg.ox.ac.uk/" class="external free"
rel="nofollow">http://www.cbrg.ox.ac.uk/</a> |
| [Bruno Aranda](User%3ABaranda "User%3ABaranda") | EMBL-EBI | <a href="http://www.ebi.ac.uk/intact" class="external free"
rel="nofollow">http://www.ebi.ac.uk/intact</a>, <a href="http://psicquic.googlecode.com" class="external free"
rel="nofollow">http://psicquic.googlecode.com</a> |
| Mahmut Uludag | European Bioinformatics Institute |  |
| [Giles Velarde](User%3ABuggy "User%3ABuggy") | The Sanger Centre | <a href="http://www.genedb.org" class="external free"
rel="nofollow">http://www.genedb.org</a>, <a href="http://www.sanger.ac.uk" class="external free"
rel="nofollow">http://www.sanger.ac.uk</a> |
| Andy Jenkinson | European Bioinformatics Institute |  |
| Kevin Howe | Wellcome Trust Sanger Institute |  |

# <span id="Logistics" class="mw-headline">Logistics</span>

This meeting was held in the <a
href="http://www.cam.ac.uk/map/v4/drawmap.cgi?mp=down;xx=183;yy=216;mt=c;tl=Department%20of%20Genetics%3B%20Biffen%20Lecture%20Theatre"
class="external text" rel="nofollow">Biffen Lecture Theatre</a>, in the
Department of Genetics on the University of Cambridge campus.

Wireless

Thanks to Ian Clark, the Biffen Lecture Theatre had wireless. Dave Judge
requested accounts for all attendees for the duration of [GMOD Europe
2010](GMOD_Europe_2010 "GMOD Europe 2010").

Power

<a href="File:Sept2010Power.jpg" class="image"
title="The amazing 15-plug strip"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/d/d3/Sept2010Power.jpg" width="300" height="68"
alt="The amazing 15-plug strip" /></a>

The Biffen Lecture Theatre did not have power outlets throughout the
room. To help us through the days, Gos Micklem secured a *15-socket*
extension strip which was placed at the back of the room.

Transportation and Lodging

See the
[Transportation](GMOD_Europe_2010#Transportation "GMOD Europe 2010") and
[Lodging](GMOD_Europe_2010#Lodging "GMOD Europe 2010") sections on the
[GMOD Europe 2010](GMOD_Europe_2010 "GMOD Europe 2010") pages for
details.

Registration

The GMOD Meeting had a registration fee (£50 early, £65 late) to cover
catered lunches,
[coffee/tea](GMOD_Promotion#Mugs.2C_T-Shirts.2C_Buttons.2C_... "GMOD Promotion")
breaks, and other expenses.

# <span id="Sponsor:_Cambridge_Computational_Biology_Institute" class="mw-headline">Sponsor: <a href="http://www.ccbi.cam.ac.uk/" class="external text"
rel="nofollow">Cambridge Computational Biology Institute</a></span>



<a href="http://www.ccbi.cam.ac.uk/" rel="nofollow"
title="Cambridge Computational Biology Institute"><img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/a/ae/CCBILogo.png/200px-CCBILogo.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/a/ae/CCBILogo.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/a/ae/CCBILogo.png 2x"
width="200" height="85"
alt="Cambridge Computational Biology Institute" /></a>



The September 2010 GMOD Meeting was sponsored by the
<a href="http://www.ccbi.cam.ac.uk/" class="external text"
rel="nofollow">Cambridge Computational Biology Institute</a>, which is
hosting the meeting and is also the home of
[InterMine](InterMine "InterMine"). The CCBI is "set up to bring
together the unique strengths of Cambridge in medicine, biology,
mathematics and the physical sciences. Its aim is to create a centre of
excellence in research and teaching and to promote collaborations both
within the Cambridge area and beyond."

Please thank Gos Miclem, Shelley Lawson, and [Richard
Smith](User%3ARsmith "User%3ARsmith") for hosting the event. We could not
have done this without their support, effort and time.

  

# <span id="Feedback" class="mw-headline">Feedback</span>

Attendees were asked to provide feedback at the end of the meeting.

**Q: Would you recommend [GMOD meetings](Meetings "Meetings") to
others**

| Yes | Maybe | No  |
|-----|-------|-----|
| 94% | 6%    | 0%  |

"Maybe" Response:

- if they are already aware of most of the tools out there - or if more
  introductory sessions are to be provided.

  
**Q: Please rate the meeting(s) using the following scale: 1 (not at
all) to 3 (reasonably) to 5 (exceptionally).**

|                                         | 1   | 2   | 3   | 4   | 5   |
|-----------------------------------------|-----|-----|-----|-----|-----|
| How useful was the meeting?             | 0%  | 6%  | 13% | 63% | 19% |
| Was the meeting well run and organized? | 0%  | 0%  | 25% | 25% | 50% |

  
**Q: Was the meeting what you expected?**

| No  | Somewhat | Yes | Other |
|-----|----------|-----|-------|
| 13% | 20%      | 60% | 7%    |

Longer responses:

- Smaller and less formal than expected - but have usually attended
  biology meetings.
- More or less. I was expecting a bigger turnout from experienced UK
  GMOD developers. This group was mostly fairly new to GMOD, which is
  good to see from another point of view.
- Almost yes, was attended more around gmod tools dev and future dev
- No, better more interesting
- Every GMOD meeting is different. It was very useful to talk with the
  attendees from the big European facilities.
- The meeting was not quite what I was expecting in that I would have
  liked more introductory sessions before the specific jargons of each
  talk - but having said that I learnt a lot, so I'm not complaining ;-)

  
**Q: Which [presentations and sessions](#Presentations) at this meeting
were the most useful or interesting?**

- All were informative
- Jason Swedlow was an inspiration! it is good to learn about
  collaborative (successful) efforts.
- DAS, all the web service talks, GBrowse updates and InterMine.
- GMOD and PSICQUIC GMOD RPC (aka REST API) CRAWL
- The State of GMOD, GMOD RPC (aka REST API), BioPivot: Applying
  Microsoft Live Labs Pivot to Problems in Bioinformatics
- Keynote: The Open Microscopy Environment
- They were all pretty good.
- GMOD RPC, InterMine, Literature Curation in GMOD, Update from the Help
  Desk, The State of GMOD
- DAS, InterMine
- the presentations that included demo's and code
- Update from the Help Desk
- The keynote was extremely interesting. The presentations on web
  services were useful and I hope to see work continue on this. Finally,
  the talks on visualization and web interfaces were useful.
- the keynote speakers
- GBrowse talks + BioPivot + InterMine

**Q: Do you have suggestions for improving GMOD meetings in the
future?**

- Cambridgshire is a good location for these due to the large
  bioinformatics community - lecture theatre was ok - food not so good.
- It's not your fault, but the savoury food was a bit nasty - it had
  been out of the fridge for an hour, but it would have been better if
  the 'hot' items had been out of the oven for an hour (or less).
- I liked the idea of engaging in a community. It would be good to have
  a central location, publication as it took this meeting to get my head
  around what GMOD was about (it is not really obvious from the
  website). It would be a nice service for GMOD to use BioPivot to
  create an OSI bioinformatics 'one-stop-shop': using as images the
  output screen shots. Sorting perhaps by data type (stuff for
  microarrays, stuff to annotate genomes, stuff to integrate
  annotation).
- Great job as always!
- I liked the open discussion format of the [satellite
  meetings](Satellite_Meetings_-_GMOD_Europe_2010 "Satellite Meetings - GMOD Europe 2010").
- It's a very friendly and open forum for discussing work.
- Thanks for the meeting.
- As always, the mix of GMOD developers and users as well as prospective
  users and external developers makes for a lot of interesting
  discussion.
- Please can you have at least the next meeting venue decided each time:
  this will determine who can come. If you had this decided before the
  meeting you might then also be able to set the agenda - and encourage
  the right audience to show up. eg \[a GMOD veteran\] felt things were
  too low-level this time, but for me as a newbie the agenda was a
  brilliant introduction to the community and the tools that are around,
  and how they can enhance bioinformatics services... So advance
  planning would really keep the ship steering well. Thanks, I enjoyed
  meeting everyone and learned a lot - and hope to contribute

  

# <span id="The_Next_Meeting" class="mw-headline">The Next Meeting</span>



[<img
src="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1b/GMODAmericas2011Logo.jpg/100px-GMODAmericas2011Logo.jpg.png"
srcset="https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1b/GMODAmericas2011Logo.jpg/150px-GMODAmericas2011Logo.jpg.png 1.5x, https://raw.githubusercontent.com/GMOD/gmod.github.io/main/mediawiki/images/thumb/1/1b/GMODAmericas2011Logo.jpg/200px-GMODAmericas2011Logo.jpg.png 2x"
width="100" height="100" alt="March 2011 GMOD Meeting" />](March_2011_GMOD_Meeting "March 2011 GMOD Meeting")



The next [GMOD Meeting](Meetings "Meetings") will be held in [March 2011
in Durham, North
Carolina](March_2011_GMOD_Meeting "March 2011 GMOD Meeting") at
<a href="http://www.nescent.org/" class="external text"
rel="nofollow">NESCent</a>, as part of [GMOD Americas
2011](GMOD_Americas_2011 "GMOD Americas 2011"). This event will also
include [Satellite
Meetings](Satellite_Meetings_-_GMOD_Americas_2011 "Satellite Meetings - GMOD Americas 2011")
and the [2011 GMOD Spring
Training](2011_GMOD_Spring_Training "2011 GMOD Spring Training") [GMOD
School](GMOD_Schools "GMOD Schools").




[Categories](Special%3ACategories "Special%3ACategories"):

- [Meetings](Category%3AMeetings "Category%3AMeetings")
- [AJAX](Category%3AAJAX "Category%3AAJAX")
- [Community
  Annotation](Category%3ACommunity_Annotation "Category%3ACommunity Annotation")
- [Chado](Category%3AChado "Category%3AChado")
- [DAS](Category%3ADAS "Category%3ADAS")
- [Ergatis](Category%3AErgatis "Category%3AErgatis")
- [GBrowse](Category%3AGBrowse "Category%3AGBrowse")
- [GMOD Web Site](Category%3AGMOD_Web_Site "Category%3AGMOD Web Site")
- [InterMine](Category%3AInterMine "Category%3AInterMine")
- [JBrowse](Category%3AJBrowse "Category%3AJBrowse")
- [Natural
  Diversity](Category%3ANatural_Diversity "Category%3ANatural Diversity")
- <a
  href="http://gmod.org/mediawiki/index.php?title=Category%3AREST&amp;action=edit&amp;redlink=1"
  class="new" title="Category%3AREST (page does not exist)">REST</a>






## Navigation menu



### Namespaces


- <span id="ca-talk"><a
  href="http://gmod.org/mediawiki/index.php?title=Talk:September_2010_GMOD_Meeting&amp;action=edit&amp;redlink=1"
  accesskey="t"
  title="Discussion about the content page [t]">Discussion</a></span>


### 

### Variants[](#)








<a href="Main_Page"
style="background-image: url(../images/GMOD-cogs.png);"
title="Visit the main page"></a>


### Navigation



- <span id="n-GMOD-Home">[GMOD Home](Main_Page)</span>
- <span id="n-Software">[Software](GMOD_Components)</span>
- <span id="n-Categories-.2F-Tags">[Categories /
  Tags](Categories)</span>




### Documentation



- <span id="n-Overview">[Overview](Overview)</span>
- <span id="n-FAQs">[FAQs](Category%3AFAQ)</span>
- <span id="n-HOWTOs">[HOWTOs](Category%3AHOWTO)</span>
- <span id="n-Glossary">[Glossary](Glossary)</span>




### Community



- <span id="n-GMOD-News">[GMOD News](GMOD_News)</span>
- <span id="n-Training-.2F-Outreach">[Training /
  Outreach](Training_and_Outreach)</span>
- <span id="n-Support">[Support](Support)</span>
- <span id="n-GMOD-Promotion">[GMOD Promotion](GMOD_Promotion)</span>
- <span id="n-Meetings">[Meetings](Meetings)</span>
- <span id="n-Calendar">[Calendar](Calendar)</span>




### Tools

- <span id="t-smwbrowselink"><a href="Special%3ABrowse/September_2010_GMOD_Meeting"
  rel="smw-browse">Browse properties</a></span>



- <span id="footer-info-lastmod">Last updated at 17:31 on 30 November
  2012.</span>
<!-- - <span id="footer-info-viewcount">645,605 page views.</span> -->
- <span id="footer-info-copyright">Content is available under
  <a href="http://www.gnu.org/licenses/fdl-1.3.html" class="external"
  rel="nofollow">a GNU Free Documentation License</a> unless otherwise
  noted.</span>

<!-- -->

- <span id="footer-places-about">[About
  GMOD](GMOD%3AAbout "GMOD%3AAbout")</span>

<!-- -->




