<div id="mw-page-base" class="noprint">

</div>

<div id="mw-head-base" class="noprint">

</div>

<div id="content" class="mw-body" role="main">

<span id="top"></span>

<div id="mw-js-message" style="display:none;">

</div>



# <span dir="auto">News/GMOD Paper Cuts, Jan 24th, 2014</span>

<div id="bodyContent">

<div id="siteSub">

From GMOD

</div>

<div id="contentSub">

</div>

<div id="jump-to-nav" class="mw-jump">

Jump to: [navigation](#mw-navigation), [search](#p-search)

</div>

<div id="mw-content-text" class="mw-content-ltr" lang="en" dir="ltr">

After a break for the festive season,
<a href="../PAG" class="mw-redirect" title="PAG">PAG XXII</a>, and the
[GMOD community meeting in San
Diego](../Jan_2014_GMOD_Meeting "Jan 2014 GMOD Meeting"), GMOD Paper
Cuts is back with a selection of scholarly publications for your reading
pleasure.

  

**GMOD Paper Cuts is a periodic selection of choice cuts from the
scientific literature featuring interesting, exciting, or otherwise
eye-catching GMOD-related work.**

If you would like a paper to appear in GMOD Paper Cuts, please email the
details to
<a href="mailto:help@gmod.org" class="external text" rel="nofollow">the
GMOD helpdesk</a>. Ideally the paper should be in an open-access
publication so that anyone can read it.

For more GMOD and GMOD-related papers, and to contribute your own
GMOD-related publications,
<a href="http://mnd.ly/WwRe8F" class="external text" rel="nofollow">join
our Mendeley group</a>.

  
**CoryneBase: *Corynebacterium* Genomic Resources and Analysis Tools at
Your Fingertips**
<sup>[\[1\]](#cite_note-DOI:10.1371.2Fjournal.pone.0086318-1)</sup>

> Corynebacteria are used for a wide variety of industrial purposes but
> some species are associated with human diseases. With increasing
> number of corynebacterial genomes having been sequenced, comparative
> analysis of these strains may provide better understanding of their
> biology, phylogeny, virulence and taxonomy that may lead to the
> discoveries of beneficial industrial strains or contribute to better
> management of diseases. \[...\]
> <a href="http://corynebacterium.um.edu.my/" class="external text"
> rel="nofollow">CoryneBase</a> offers the access of a range of
> Corynebacterium genomic resources as well as analysis tools for
> comparative genomics and pathogenomics.

A new bacterial resource that uses [JBrowse](../JBrowse.1 "JBrowse") for
genome visualization. The justification for using JBrowse is
interesting:

*We chose JBrowse for CoryneBase for the following main reasons: (1)
most of the traditional genome browsers, e.g., GBrowse, are implemented
using the Common Gateway Interface (CGI) protocol—the use of CGI-based
genome browsers will inadvertently incurs unnecessary delays since the
whole-genome browser page need to be reloaded when users change how the
data are displayed; and (2) with the advances in next-generation
sequencing technologies and bioinformatic tools, we would expect more
corynebacterial genomes will be sequenced and annotated. Therefore a
user-friendly genome browser that allows rapid and seamless browsing of
the huge genomic data will be a major advantage to the research
communities.*

  

**The Genome Sequence of the Fungal Pathogen *Fusarium virguliforme*
That Causes Sudden Death Syndrome in Soybean**
<sup>[\[2\]](#cite_note-DOI:10.1371.2Fjournal.pone.0081832-2)</sup>

> *Fusarium virguliforme* causes sudden death syndrome (SDS) of soybean,
> a disease of serious concern throughout most of the soybean producing
> regions of the world. Despite the global importance, little is known
> about the pathogenesis mechanisms of *F. virguliforme*. Thus, we
> applied Next-Generation DNA Sequencing to reveal the draft *F.
> virguliforme* genome sequence and identified putative pathogenicity
> genes to facilitate discovering the mechanisms used by the pathogen to
> cause this disease.

A new draft sequence of the fungal disease *Fusarium virguliforme*. The
genome sequence can be
<a href="http://fvgbrowse.agron.iastate.edu/" class="external text"
rel="nofollow">browsed online using GBrowse</a>.

  

<a href="http://arxiv.org/abs/1401.2286" class="external text"
rel="nofollow"><strong>Palaeosymbiosis revealed by genomic fossils of
Wolbachia in a strongyloidean nematode</strong></a>

Georgios Koutsovoulos, Benjamin Makepeace, Vincent N. Tanya, Mark
Blaxter; published on arxiv.org

> Wolbachia are common endosymbionts of terrestrial arthropods, and are
> also found in nematodes, the animal-parasitic *filaria*, and the
> plant-parasite *Radopholus similis*. Lateral transfer of Wolbachia
>
> DNA to the host genome is common. We generated a draft genome sequence
> for the strongyloidean nematode parasite *Dictyocaulus viviparus*, the
> cattle lungworm. In the assembly, we identified nearly 1 Mb of
> sequence with similarity to Wolbachia. The fragments were unlikely to
> derive from a live Wolbachia infection: most were short, and the genes
> were disabled through inactivating mutations. Many fragments were
> co-assembled with definitively nematode-derived sequence. We found
> limited evidence of expression of the Wolbachia-derived genes. The *D.
> viviparus* Wolbachia genes were most similar to filarial strains, and
> strains from the host-promiscuous clade F. We conclude that *D.
> viviparus* was infected by Wolbachia in the past. Genome sequence
> based surveys are a powerful tool for revealing the genome archaeology
> of infection and symbiosis.

A fascinating study that uses genetic techniques to dig into the history
of a host-parasite relationship. The D. viviparus genome was annotated
using the [MAKER](../MAKER.1 "MAKER") pipeline.

  
**Genome of the human hookworm *Necator americanus***
<sup>[\[3\]](#cite_note-DOI:10.1038.2Fng.2875-3)</sup>

> The hookworm *Necator americanus* is the predominant soil-transmitted
> human parasite. Adult worms feed on blood in the small intestine,
> causing iron-deficiency anemia, malnutrition, growth and development
> stunting in children, and severe morbidity and mortality during
> pregnancy in women. We report sequencing and assembly of the *N.
> americanus* genome (244 Mb, 19,151 genes). Characterization of this
> first hookworm genome sequence identified genes orchestrating the
> hookworm's invasion of the human host, genes involved in blood feeding
> and development, and genes encoding proteins that represent new
> potential drug targets against hookworms. *N. americanus* has
> undergone a considerable and unique expansion of immunomodulator
> proteins, some of which we highlight as potential treatments against
> inflammatory diseases. We also used a protein microarray to
> demonstrate a postgenomic application of the hookworm genome sequence.
> This genome provides an invaluable resource to boost ongoing efforts
> toward fundamental and applied postgenomic research, including the
> development of new methods to control hookworm and human immunological
> diseases.

Another somewhat gruesome parasite, the human hookworm, which is
responsible for causing more disease than any other soil-based helminth.
The genome was annotated using [MAKER](../MAKER.1 "MAKER") and can be
viewed online at <a href="http://nematode.net" class="external text"
rel="nofollow">Nematode.net</a> and
<a href="http://wormbase.org" class="external text"
rel="nofollow">Wormbase</a>.

  
Happy reading!

  

1.  <span id="cite_note-DOI:10.1371.2Fjournal.pone.0086318"><span class="mw-cite-backlink">[↑](#cite_ref-DOI:10.1371.2Fjournal.pone.0086318_0)</span>
    **Cite error: Invalid `<ref>` tag; no text was provided for refs
    named `DOI:10.1371.2Fjournal.pone.0086318`**</span>
2.  <span id="cite_note-DOI:10.1371.2Fjournal.pone.0081832"><span class="mw-cite-backlink">[↑](#cite_ref-DOI:10.1371.2Fjournal.pone.0081832_0)</span>
    **Cite error: Invalid `<ref>` tag; no text was provided for refs
    named `DOI:10.1371.2Fjournal.pone.0081832`**</span>
3.  <span id="cite_note-DOI:10.1038.2Fng.2875"><span class="mw-cite-backlink">[↑](#cite_ref-DOI:10.1038.2Fng.2875_0)</span>
    **Cite error: Invalid `<ref>` tag; no text was provided for refs
    named `DOI:10.1038.2Fng.2875`**</span>

<div class="smallprint">

Disclaimer: the papers included in this feature are for your
entertainment and edification only. Inclusion does not imply an
endorsement of the material or any association between the authors and
the GMOD project.

</div>

  

<div class="newsfooter">

*Posted to the [GMOD News](../GMOD_News "GMOD News") on 2014/01/24*

</div>

</div>

<div class="printfooter">

Retrieved from
"<http://gmod.org/mediawiki/index.php?title=News/GMOD_Paper_Cuts,_Jan_24th,_2014&oldid=25210>"

</div>

<div id="catlinks" class="catlinks">

<div id="mw-normal-catlinks" class="mw-normal-catlinks">

[Categories](../Special:Categories "Special:Categories"):

- [News Items](../Category:News_Items "Category:News Items")
- [Paper Cuts](../Category:Paper_Cuts "Category:Paper Cuts")

</div>

</div>

<div class="visualClear">

</div>

</div>

</div>

<div id="mw-navigation">

## Navigation menu

<div id="mw-head">



<div id="left-navigation">

<div id="p-namespaces" class="vectorTabs" role="navigation"
aria-labelledby="p-namespaces-label">

### Namespaces


- <span id="ca-talk"><a
  href="http://gmod.org/mediawiki/index.php?title=Talk:News/GMOD_Paper_Cuts,_Jan_24th,_2014&amp;action=edit&amp;redlink=1"
  accesskey="t"
  title="Discussion about the content page [t]">Discussion</a></span>

</div>

<div id="p-variants" class="vectorMenu emptyPortlet" role="navigation"
aria-labelledby="p-variants-label">

### 

### Variants[](#)

<div class="menu">

</div>

</div>

</div>





</div>

</div>

</div>

<div id="mw-panel">

<div id="p-logo" role="banner">

<a href="../Main_Page"
style="background-image: url(../../images/GMOD-cogs.png);"
title="Visit the main page"></a>

</div>

<div id="p-Navigation" class="portal" role="navigation"
aria-labelledby="p-Navigation-label">

### Navigation

<div class="body">

- <span id="n-GMOD-Home">[GMOD Home](../Main_Page)</span>
- <span id="n-Software">[Software](../GMOD_Components)</span>
- <span id="n-Categories-.2F-Tags">[Categories /
  Tags](../Categories)</span>
- <span id="n-View-all-pages">[View all
  pages](../Special:AllPages)</span>

</div>

</div>

<div id="p-Documentation" class="portal" role="navigation"
aria-labelledby="p-Documentation-label">

### Documentation

<div class="body">

- <span id="n-Overview">[Overview](../Overview)</span>
- <span id="n-FAQs">[FAQs](../Category:FAQ)</span>
- <span id="n-HOWTOs">[HOWTOs](../Category:HOWTO)</span>
- <span id="n-Glossary">[Glossary](../Glossary)</span>

</div>

</div>

<div id="p-Community" class="portal" role="navigation"
aria-labelledby="p-Community-label">

### Community

<div class="body">

- <span id="n-GMOD-News">[GMOD News](../GMOD_News)</span>
- <span id="n-Training-.2F-Outreach">[Training /
  Outreach](../Training_and_Outreach)</span>
- <span id="n-Support">[Support](../Support)</span>
- <span id="n-GMOD-Promotion">[GMOD Promotion](../GMOD_Promotion)</span>
- <span id="n-Meetings">[Meetings](../Meetings)</span>
- <span id="n-Calendar">[Calendar](../Calendar)</span>

</div>

</div>

<div id="p-tb" class="portal" role="navigation"
aria-labelledby="p-tb-label">

### Tools

<div class="body">


- <span id="t-smwbrowselink"><a href="../Special:Browse/News-2FGMOD_Paper_Cuts,_Jan_24th,_2014"
  rel="smw-browse">Browse properties</a></span>


</div>

</div>

</div>

</div>

<div id="footer" role="contentinfo">

- <span id="footer-info-lastmod">Last updated at 22:34 on 26 January
  2014.</span>
<!-- - <span id="footer-info-viewcount">14,753 page views.</span> -->
- <span id="footer-info-copyright">Content is available under
  <a href="http://www.gnu.org/licenses/fdl-1.3.html" class="external"
  rel="nofollow">a GNU Free Documentation License</a> unless otherwise
  noted.</span>

<!-- -->

- <span id="footer-places-about">[About
  GMOD](../GMOD:About "GMOD:About")</span>

<!-- -->






</div>
