



<span id="top"></span>




# <span dir="auto">News/Software Developer - Phytozome</span>









**Software Developer 3/4 - Phytozome - 74226**  
**Organization: GN-Genomics**

The DOE Joint Genome Institute (JGI) in Walnut Creek, CA (a division of
the Lawrence Berkeley National Lab) has an exciting opportunity
available for an experienced Software Developer / Engineer to support
the Phytozome group. The Phytozome group in the JGI Plant Genomics
Program is responsible for the annotation and analysis of plant genomes,
and for providing both internal and external users with visualization,
querying and data access capabilities to facilitate progress in
fundamental, biofuel and climate-related Plant science. The Phytozome
website and associated databases and analysis tools provide JGI’s
collaborators and the broader plant biology community with intuitive and
powerful interfaces to access and explore plant genomic data. As the JGI
anticipates a massive increase in RNA-Seq and resequencing output, we
have an opportunity for an experienced developer to help us enhance and
expand Phytozome to host even more plant genomes and their annotations,
and to develop new databases and interfaces to support users’
exploration of plant genomic diversity and gene expression data.

Under broad supervision, the ideal candidate will develop, implement,
test, integrate and deploy new and/or existing software components of
the Phytozome system for the visualization and querying of plant genomic
and related data. They should demonstrate detailed knowledge and strong
technical ability in web user interface design concepts and
implementation, data modeling, and database schema design, as well as
extensive familiarity with standard genomic sequence data types and
analysis methods.

The selected candidate will be hired at the Software Developer 3 or 4
classification level depending upon the candidate's level of experience
and qualifications. Position reports to the Phytozome Group Lead.

<span style="font-size: larger;">**Specific Job Duties**</span>

**Essential**

- Work as a member of the Phytozome development team to create, maintain
  and extend the suite of visualization, backend, and data management
  tools that support the public interface to the JGI’s Plant Genomics
  program.
- Create and maintain database-backed web applications for the display
  and querying of genomic data.
- Evaluate the suitability of third party open source tools (e.g. GMOD)
  versus in-house development to provide desired functionality to
  Phytozome.
- Optimize the Phytozome user experience by constantly monitoring and
  improving backend data retrieval performance (as determined by
  benchmarking) and front-end usability (as determined by user surveys
  and best practices).
- Follow best practices for software version control, documentation,
  testing and release engineering.

**Marginal**

- Work with Phytozome/Metazome internal and external user community and
  Scientific Advisory Board to prioritize new feature requests, and
  provide accurate estimates of implementation timelines.
- Assist in the preparation of manuscripts, tutorials and conference
  presentations based on Phytozome.
- Work with JGI’s Systems group to anticipate hardware needs for
  increasing data set sizes and expanded functionality.

**Essential duties for the Software Developer 4** - In addition to the
job duties described above, the Software Developer 4 will:

- Operate at a higher level of independence and responsibility when
  carrying out assignments.
- Will serve as a technical lead and key contributor called upon to
  solve highly technical, complex problems.

<span style="font-size: larger;">**Key Success Factors**</span>

**Essential** -

- B.S. in technical or scientific field (or equivalent experience) and a
  minimum of five years programming experience.
- Minimum five years experience and demonstrated success designing and
  implementing bioinformatics analysis systems and software in a
  scientific or technical environment, e.g., University, National
  Laboratory, corporate R&D.
- Familiarity with standard genomic sequence data repositories (e.g,
  GenBank, SwissProt) and functional classification databases and
  ontologies (e.g., KEGG, Panther, Metacyc, GO).
- Solid experience programming with object-oriented Perl and Java.
- Advanced knowledge of SQL and relational database design.
- High degree of comfort developing on, and for, linux systems.
- Demonstrated experience developing database-backed applications and
  experience with web technologies including: Apache, Tomcat, CGI
  programming, JDBC, JAXB, GWT, AJAX.
- Ability to effectively learn and apply new technical concepts to
  troubleshoot and solve problems and challenges of a diverse scope.
- Effective interpersonal communication and presentation skills.
- Effective time management, organization, and planning skills.
- Ability to work effectively in a diverse team environment.

**Desirable** -

- M.S./Ph.D. in Molecular Biology, Genetics or related field.
- Mathematical sophistication to understand algorithmic concepts behind
  sequence alignment, gene family clustering, and phylogenetic tree
  building.
- Experience with the management and visualization of gene expression
  and genomic diversity data.
- Solid understanding of statistics.
- Experience developing software that includes or interacts with GMOD
  and bioperl components.
- Working knowledge of Python and PHP.

**Essential Qualifications for the Software Developer 4** - In addition
to the qualifications listed above, the Software Developer 4 requires:

- A minimum of eight years related experience.
- Demonstrated ability to solve highly complex problems and work
  independently on end-to-end implementation and maintenance of software
  projects through multiple releases.

**How To Apply:** <a
href="https://lbl.taleo.net/careersection/2/jobdetail.ftl?lang=en&amp;job=74226"
class="external text" rel="nofollow">Apply directly online</a> and
follow the instructions to complete the application process.

Berkeley Lab is an affirmative action/equal opportunity employer
committed to the development of a diverse workforce.

  



*Posted to the [GMOD News](../GMOD_News "GMOD News") on 2012/05/16*






[Category](../Special%3ACategories "Special%3ACategories"):

- [News Items](../Category%3ANews_Items "Category%3ANews Items")






## Navigation menu







<a href="../Main_Page"
style="background-image: url(../../images/GMOD-cogs.png);"
title="Visit the main page"></a>


### Navigation



- <span id="n-GMOD-Home">[GMOD Home](../Main_Page)</span>
- <span id="n-Software">[Software](../GMOD_Components)</span>
- <span id="n-Categories-.2F-Tags">[Categories /
  Tags](../Categories)</span>
- <span id="n-View-all-pages">[View all
  pages](../Special:AllPages)</span>




### Documentation



- <span id="n-Overview">[Overview](../Overview)</span>
- <span id="n-FAQs">[FAQs](../Category%3AFAQ)</span>
- <span id="n-HOWTOs">[HOWTOs](../Category%3AHOWTO)</span>
- <span id="n-Glossary">[Glossary](../Glossary)</span>




### Community



- <span id="n-GMOD-News">[GMOD News](../GMOD_News)</span>
- <span id="n-Training-.2F-Outreach">[Training /
  Outreach](../Training_and_Outreach)</span>
- <span id="n-Support">[Support](../Support)</span>
- <span id="n-GMOD-Promotion">[GMOD Promotion](../GMOD_Promotion)</span>
- <span id="n-Meetings">[Meetings](../Meetings)</span>
- <span id="n-Calendar">[Calendar](../Calendar)</span>




### Tools

- <span id="t-smwbrowselink"><a href="../Special%253ABrowse/News-2FSoftware_Developer_-2D_Phytozome"
  rel="smw-browse">Browse properties</a></span>



- <span id="footer-info-lastmod">Last updated at 19:16 on 16 May
  2012.</span>
<!-- - <span id="footer-info-viewcount">10,078 page views.</span> -->
- <span id="footer-info-copyright">Content is available under
  <a href="http://www.gnu.org/licenses/fdl-1.3.html" class="external"
  rel="nofollow">a GNU Free Documentation License</a> unless otherwise
  noted.</span>

<!-- -->



<!-- -->




